{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed to load file C:\\Users\\mikoo\\AI\\osmo\\cosmetics-07-00003-v2.pdf with error: RetryError[<Future at 0x2d02e4317f0 state=finished raised UnboundLocalError>]. Skipping...\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[Document(id_='ada63541-a30f-404a-995c-a2a3f90efe59', embedding=None, metadata={'page_label': '1', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='A Principal Odor Map Uni\\x00es Diverse Tasks in Human Olfactory Perception\\nBrian K. Lee1†, Emily J. Mayhew2†, Benjamin Sanchez-Lengeling1, Jennifer N. Wei1, Wesley W.\\nQian1,3, Kelsie Li\\x00le4, Ma\\x00hew Andres4, Britney B. Nguyen4, Theresa Moloy4, Jane K. Parker5,\\nRichard C. Gerkin1,6, Joel D. Mainland4,7,*, Alexander B. Wiltschko1,*\\n1Google Research, Brain Team; Cambridge, MA, USA.\\n2Depa\\x00ment of Food Science and Human Nutrition, Michigan State University; East Lansing, MI, USA.\\n3Depa\\x00ment of Computer Science, University of Illinois; Urbana-Champaign, IL, USA.\\n4Monell Chemical Senses Center; Philadelphia, PA, USA.\\n5Depa\\x00ment of Food and Nutritional Sciences, University of Reading; Reading, Berkshire, UK.\\n6School of Life Sciences, Arizona State University; Tempe, AZ, USA.\\n7Depa\\x00ment of Neuroscience, University of Pennsylvania; Philadelphia, PA, USA.\\n†These authors contributed equally.\\n*Corresponding authors. Email: jmainland@monell.org, alex.bw@googlemail.com\\nAbstract\\nMapping molecular structure to odor perception is a key challenge in olfaction. Here, we use\\ngraph neural networks (GNN) to generate a Principal Odor Map (POM) that preserves\\nperceptual relationships and enables odor quality prediction for novel odorants. The model is\\nas reliable as a human in describing odor quality: on a prospective validation set of 400 novel\\nodorants, the model-generated odor pro\\x00le more closely matched the trained panel mean\\n(n=15) than did the median panelist. Applying simple, interpretable, theoretically-rooted\\ntransformations, the POM outpe\\x00ormed chemoinformatic models on several other odor\\nprediction tasks, indicating that the POM successfully encoded a generalized map of\\nstructure-odor relationships. This approach broadly enables odor prediction and paves the\\nway toward digitizing odors.\\n(119 words)\\nOne-Sentence Summary\\nAn odor map achieves human-level odor description pe\\x00ormance and generalizes to diverse\\nodor-prediction tasks.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='09cf0cef-bcf9-44b1-bbfb-8b85c052377d', embedding=None, metadata={'page_label': '2', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Introduction\\nA fundamental problem in neuroscience is mapping the physical prope\\x00ies of a stimulus to\\nperceptual characteristics. In vision, wavelength maps to color; in audition, frequency maps\\nto pitch. By contrast, the mapping from chemical structures to olfactory percepts is poorly\\nunderstood. Detailed and modality-speci\\x00c maps like the CIE color space (1), and Fourier\\nspace (2) led to a be\\x00er understanding of visual and auditory coding. Similarly, to be\\x00er\\nunderstand olfactory coding, olfaction needs a be\\x00er map.\\nPitch increases monotonically with frequency; in contrast, the relationship between odor\\npercept and odorant structure is riddled with discontinuities, exempli\\x00ed by Sell’s triplets (3),\\ntrios of molecules in which the structurally similar pair is not the perceptually similar pair (Fig.\\n1A). These discontinuities in the structure-odor relationship suggest that standard\\nchemoinformatic representations of molecules—functional group counts, physical prope\\x00ies,\\nmolecular \\x00ngerprints, etc.— used in recent odor modeling work (4–6) are inadequate to map\\nodor space.\\nResults\\nTo generate odor-relevant representations of molecules, we constructed a Message Passing\\nNeural Network (MPNN) (7), a speci\\x00c type of graph neural network (GNN) (8), to map\\nchemical structures to odor percepts. Each molecule is represented as a graph, with each\\natom described by its valence, degree, hydrogen count, hybridization, formal charge, and\\natomic number. Each bond is described by its degree, aromaticity, and whether it is in a ring.\\nUnlike traditional \\x00ngerprinting techniques (9), which assign equal weight to all molecular\\nfragments within a set bond radius, a GNN can optimize fragment weights for odor-speci\\x00c\\napplications. Neural networks have unlocked predictive modeling breakthroughs in diverse\\nperceptual domains (e.g., natural images (10), faces (11), and sounds (12)) and naturally\\nproduce intermediate representations of their input data that are functionally\\nhigh-dimensional, data-driven maps. We use the \\x00nal layer of the GNN (hencefo\\x00h, “our\\nmodel”) to directly predict odor qualities, and the penultimate layer of the model as a\\nprincipal odor map (POM). The POM 1) faithfully represents known perceptual hierarchies and\\ndistances, 2) extends to novel odorants, 3) is robust to discontinuities in structure-odor\\ndistances, and 4) generalizes to other olfactory tasks.\\nTo train the model, we curated a reference dataset of approximately 5000 molecules, each\\ndescribed by multiple odor labels (e.g. creamy, grassy), by combining the Goodscents (13)\\nand Le\\x00ngwell (14) (GS/LF) \\x00avor and fragrance databases (Fig. 1B). The model (Fig. 1C)\\nachieved strong cross-validation predictive pe\\x00ormance of AUROC=0.89 (15).\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='4455f25c-ec2a-4a7c-9c89-ed88d5c92f71', embedding=None, metadata={'page_label': '3', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Fig. 1. POM preserves the structure of odor perceptual space. (A) Example triplet of molecules in which the\\nstructurally similar pair is not the perceptually similar pair. (B) The GNN was trained on a curated dataset of\\n~5000 semantically labeled molecules drawn from GoodScents (13) and Le\\x00ngwell (14) \\x00avor and fragrance\\ndatabases; one square represents 100 molecules; three example training set molecules and their odor\\ndescriptions are shown: 2-methyl-2-hexenoic acid (top), 2,5-dimethyl-3-thioisovalerylfuran (middle),\\n1-methyl-3-hexenyl acetate (bo\\x00om). (C) Schematic illustrating the process of training a GNN to generate the\\nPOM. (D-F) Odorants plo\\x00ed by the \\x00rst and second principal components (PC) of their (D) perceptual labels\\nfrom GS/LF training dataset (138 labels), (E) cFP structural \\x00ngerprints (radius 4, 2048-bit), and (F) POM\\ncoordinates (256 dimensions). Areas dense with molecules having the broad category labels \\x00oral, meaty, or\\nalcoholic are shaded; areas dense with narrow category labels are outlined. The POM recapitulates the true\\nperceptual map, but the FP map does not; note that only relative (not absolute) coordinates ma\\x00er.\\nTo test how well the POM represents known perceptual relationships, we compared both the\\nPOM and a map built with standard chemoinformatic features - Morgan \\x00ngerprints (FP) -  to\\nempirical perceptual space (Fig. 1D-F). We measured the \\x00delity of the maps in representing\\ntrue relative perceptual distances, (e.g. two molecules that smell of jasmine should be nearer\\nto each other than to a beefy molecule) and hierarchies (e.g. jasmine and lavender are\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='e16d4cbd-0231-48d3-9855-0476f1d257a5', embedding=None, metadata={'page_label': '4', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='subtypes of the \\x00oral odor family). The POM be\\x00er represents relative distances: distances in\\nthe perceptual map (Fig. 1D) are more signi\\x00cantly correlated to distances in the POM\\n(R=0.73, Fig. S1A) than to distances in the FP map (R=-0.12, p <0.001, Fig. S1B). The POM be\\x00er\\nrepresents perceptual hierarchies: molecules with a shared odor label have signi\\x00cantly\\ntighter cluster density (CD) in the POM (CD = 0.51± 0.19) than in the FP map (CD = 0.68 ± 0.23,\\np <0.001, Fig. S2), where smaller CD values denote more dense clusters.\\nTo test if the model extends to novel odorants, we designed a prospective validation\\nchallenge (16) in which we benchmarked model predictive pe\\x00ormance against individual\\nhuman raters. In olfaction, no reliable instrumental method of measuring odor perception\\nexists, and trained human sensory panels are the gold standard for odor characterization (17).\\nLike other sensory modalities, odor perception is variable across individuals (18, 19), but\\ngroup-averaged odor ratings have been shown to be stable across repeated measurements\\n(20) and represent our best avenue to establish the ground-truth odor character for novel\\nodorants. We trained a coho\\x00 of subjects to describe their perception of odorants using the\\nRate-All-That-Apply method (RATA) and a 55-word odor lexicon. During training sessions,\\neach term in the lexicon was paired with visual and odor references (Table S1; Fig. S3). Only\\nsubjects that met pe\\x00ormance standards on the pretest of 20 common odorants (Data S2;\\nindividual test-retest correlation R > 0.35; reasonable label selection for common odorants)\\nwere invited to join the panel.\\nTo avoid trivial test cases, we applied the following selection criteria for the set of 400 novel\\nodorants: 1) molecules must be structurally distinct from each other (Fig. S4), 2) molecules\\nshould cover the widest gamut of odor labels (Data S1), and 3) molecules must be structurally\\nor perceptually distinct from any training example (e.g. Fig. 1A, Data S1). Our prospective\\nvalidation set consists of 55-odor label RATA data for 400 novel, intensity-balanced odorants\\ngenerated by our coho\\x00 of ≥15 panelists (2 replicates). Summary statistics and correlation\\nstructure of the human perceptual data is presented in Fig. S5-7. Our panel’s mean ratings\\nwere highly stable (panel test-retest: R = 0.80, n = 15; Fig. S8) and more consistent than the\\nDREAM coho\\x00’s ratings (6) (Fig. S9-10).\\nOf the 400 molecules characterized, 80 were dropped from the \\x00nal prospective validation\\nset due to low intensity (42) (Fig. S11), redundancy (1), mistaken inclusion (1), or with con\\x00rmed\\nor potential contamination (26) (Data S1). Model pe\\x00ormance was evaluated on the remaining\\n320 molecules without model retraining.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='77a03fc9-786a-4552-b69d-f5314913f946', embedding=None, metadata={'page_label': '5', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Fig. 2: GNN model displays human-level odor description pe\\x00ormance. (A) GNN model label predictions,\\n(B) random forest (RF) model label predictions, (C) panel mean ratings with standard error bars, and (D)\\nindividual panelist ratings, averaged over 2 replicates, for the molecule\\n2,3-dihydrobenzofuran-5-carboxaldehyde. In panels A-C, the top 5 ranked descriptors are in orange (GNN),\\npurple (RF), or green (panel). Descriptors in panels A-D are ordered by panel mean ratings. Panels A, B, and D\\nare annotated with the Pearson correlation coe\\x00cient of their data to the panel mean rating shown in panel C.\\nPanel D includes panelist/panel correlation coe\\x00cients for the panelist that best matches the panel mean and\\nfor the panelist with the median match. (E) Cumulative density plot showing the distribution of correlations\\nbetween human panelists and the panel mean (in green) and between the GNN, RF, and GNN shu\\x00ed model\\npredictions and the panel mean on a per molecule basis. Curves shi\\x00ed to the right are more strongly correlated\\nto the panel mean. (F) Di\\x00erence in the median correlation to the panel mean relative to the median human\\nsubject’s correlation to the panel mean for models trained using k-nearest neighbor (KNN) and RF, trained on\\ncFPs or Mordred features, and the GNN model. Only the GNN model has a median correlation to the panel mean\\nthat is higher than that of the median panelist.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='a9bb81ef-c236-4666-9c14-aa270ea8025d', embedding=None, metadata={'page_label': '6', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\"To measure the model's pe\\x00ormance, we compared the concordance of its normalized\\npredictions with the normalized panel mean rating (Fig. 2A and 2C). While there is\\nconsiderable variation across molecules in the ability of both individual raters and the model\\nto match the panel mean ratings, the model output comes closer to the panel mean than\\ndoes the median panelist for 53% of molecules (Fig. 2E and 2F). The model’s superiority at the\\ntask is even more impressive given that panelists are able to smell each odorant as they rate\\nit, while the model’s predictions are based solely on nominal molecular structure.\\nAs a baseline comparison, we trained a cFP-based random forest (RF) model, the previous\\nstate-of-the-a\\x00 (6), on the same dataset (Fig. 2B). This baseline model surpassed the median\\npanelist for only 41% of molecules, showing that our GNN model’s pe\\x00ormance increase\\ncomes not only from the volume and quality of the data, but impo\\x00antly from the model\\narchitecture.\\nThe GNN model shows human-level pe\\x00ormance in aggregate, but how does it pe\\x00orm\\nacross perceptual and chemical classes? When we disaggregate pe\\x00ormance by odor label,\\nthe model is within the distribution of human raters for all labels except musk and surpasses\\nthe median panelist for 32/55 labels (58%, Fig. 3A). This per-label view suppo\\x00s the view that\\nthe GNN model is superior to the previous state of the a\\x00 model trained on the same data\\n(paired 2-tailed t-test p=1.0e-7).\\nPredictive pe\\x00ormance for a given label depends on the complexity of the structure-odor\\nmapping for that label, so it is unsurprising that it pe\\x00orms best for labels like garlic and \\x00shy\\nthat have clear structural determinants (sulfur-containing for garlic; amines for \\x00shy), and\\nworst for the label musk, which includes at least 5 distinct structural classes (macrocyclic,\\npolycyclic, nitro, steroid-type, and straight-chain) (21, 22). In contrast, a panelist’s\\npe\\x00ormance for a given label depends on their familiarity with the label in the context of\\nsmell; consequently, we see strong panelist-panel agreement for labels describing common\\nfood smells like nu\\x00y, garlic, and cheesy and weak agreement for labels like musk and hay.\\nWeak agreement for musk may also be due to genetic variability in perception, a\\nwell-documented phenomenon (23).\\nModel pe\\x00ormance also depends on the number of training examples for a given label; with\\nenough examples, models can learn even complex structure-percept relationships. In general,\\nour model’s pe\\x00ormance is high for labels with many training examples (e.g, fruity, sweet,\\n\\x00oral) (Fig. 3B), but pe\\x00ormance for labels with few training examples can be either high (e.g.,\\n\\x00shy, camphoreous, cooling) or low (e.g, ozone, sharp, fermented). In other words, collecting\\nmore training data raises the \\x00oor for model pe\\x00ormance. Likewise, model pe\\x00ormance is\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint \", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='f8953176-3f59-4eeb-8972-1c08db87cfa3', embedding=None, metadata={'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='bounded above by panel test-retest correlation (Fig. S13). When we disaggregate by chemical\\nclasses (e.g. esters, phenols, amines), both panelist and model pe\\x00ormance is relatively\\nuniform (Fig. 3C), with sulfur-containing molecules showing strongest pe\\x00ormance from\\npanelists and the model (R = 0.52).\\nChemical materials are impure - a fact too o\\x00en unaccounted for in olfactory research(24). To\\nmeasure the contribution of impurities to the odor percept of our stimuli, we applied a gas\\nchromatography-mass spectrometry (GC-MS and gas chromatography-olfactometry\\n(GC-O) quality control (QC) procedure to 50 stimuli (Data S1). This QC procedure matches an\\nodor percept to its causal molecule, allowing us to identify stimuli for which the primary odor\\ncharacter was not due to the nominal compound. Our QC led to diverse conclusions: the\\nnominal compound caused the odor (12/50), the nominal compound and contaminants\\ncontribute to the odor (16/50), contaminants caused the odor (18/50), or the cause of the\\nodor could not be determined (4/50) (Fig. 3D). In some cases, while we purchased a novel\\nodorant, the dominant odorant was not novel; for example, the stimulus\\n4,5-dimethyl-1,3-thiazol-2-amine was described by the panel as bu\\x00ery, sweet, and dairy, but\\nthis odor percept was a\\x00ributed through QC to the contaminant diacetyl, a well-known\\nbu\\x00ery odorant. In another case, the purchased odorant, isobornyl methylacrylate, was\\ndescribed by the panel and the model as both piney and \\x00oral; however, through QC we\\ndetermined that the nominal compound was \\x00oral only and that the piney aroma was due to\\nthe closely related compound, borneol, which was detected as a contaminant in the sample.\\nBased on QC results, we removed 26 molecules known or suspected to have high degrees of\\nodorous contamination (Data S1).\\nThe prevalence of odorous contamination that we found demonstrates that it is not safe to\\nassume that the odor percept of a purchased chemical is due to the nominal compound. The\\nFlavor & Fragrance (F&F) industry is motivated to minimize odorous contaminants for\\ncommercially valued odorants, but there is no such incentive for non-F&F commodity\\nchemicals. We stress the need for caution and diligence in expanding odor stimulus space.\\nImplications of each QC result on model pe\\x00ormance are unique (Data S1). In some cases,\\nthe model pe\\x00ormed well despite the presence of odorous contaminants. We estimate that,\\nif these contaminants were removed from the rated samples, model pe\\x00ormance improves\\nin 6 of 50 scenarios, degrades in another 6 of 50 scenarios, remains neutral in 21 of 50\\nscenarios, and cannot be determined in 17 of 50 scenarios.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='b3dda87d-415b-45b1-8792-762b9cd8c8a9', embedding=None, metadata={'page_label': '8', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Fig. 3. Model pe\\x00ormance is robust across structural and perceptual classes. (A) Correlation of GNN (in\\norange) and RF (in purple) model predictions and panelist ratings (in gray) to the panel mean for each of the 55\\nodor labels. (B) GNN model correlation to panel mean for each of the 55 odor labels plo\\x00ed against the number\\nof molecules in the training data for which the label applies. Circle size is propo\\x00ional to the number of test set\\nmolecules for which the label applies. Selected data points are annotated. (C) Mean correlation of GNN (in\\norange) and RF (in purple) model predictions and panelist ratings (in gray) to the panel mean for molecules\\nbelonging to 12 common chemical classes. (D) Categorization of gas chromatography-olfactometry quality\\ncontrol results for 50 validation set stimuli.\\nTo test if the model is robust to discontinuities in structure-odor distances, we designed an\\nadditional challenge in which 41 new triplets (example in Fig. 4A) were constructed and\\nvalidated by the panel (as in Fig. 1A). In each triplet, the anchor molecule is a known odorant,\\nand is matched with one structurally similar and one structurally dissimilar novel odorant, and\\nin which the more structurally dissimilar odorant is predicted to be the more perceptually\\nsimilar of the two to the anchor. Our trained panelists were presented with the three\\nodorants as a set and rated the perceptual distance between each of the molecules in the\\ntriplet (Fig. 4B). Con\\x00rming the model’s predictions -- counterintuitive under simpler\\nstructural models of odor -- our panelists generally rated the structurally dissimilar molecules\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='a261bef9-68a0-44b9-9b61-1834d1b510ce', embedding=None, metadata={'page_label': '9', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='as being more perceptually similar to an anchor molecule than the anchor’s structural\\nneighbor (p < 2.2e-16, Fig. 4C). This signi\\x00cant result is fu\\x00her evidence that the POM\\novercomes discontinuities in the structure-odor relationship.\\nFig. 4. POM solves discontinuities in structure-odor mapping. (A) Example triplet of molecules identi\\x00ed by\\nthe GNN model in which the structurally similar pair is not the perceptually similar pair. We used the model to\\nselect 41 such triplets. (B) Diagram of the psychophysical task in which panelists rated perceptual distances\\nbetween molecules in predicted triplets. (C) Mean perceptual distance rating for molecules that are structurally\\ndissimilar (le\\x00) or structurally similar (right) to the same anchor molecule. Lines connect each pair of molecules\\ncompared to the same anchor molecule; line color corresponds to the relative di\\x00erence in perceptual similarity.\\nPerceptual similarity followed model predictions rather than structural similarity.\\nA reliable structure-odor map allows us to explore odor space at scale. We compiled a list of\\n~500,000 potential odorants whose empirical prope\\x00ies are currently unknown to science or\\nindustry; most have never been synthesized before. Because a molecule’s coordinates in the\\nPOM are directly computable from the model, we can plot these potential odorants in the\\nPOM (Fig. 5A), revealing a potential space of odorous molecules that is much larger than the\\nmuch smaller space covered by current fragrance catalogs (~5,000 purchasable,\\ncharacterized odorants). These molecules would take approximately 70 person-years of\\ncontinuous smelling time to collect using our trained human panel.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='81573862-9886-4823-b747-4336178f7fed', embedding=None, metadata={'page_label': '10', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Fig. 5. POM solves a fundamental set of olfactory prediction tasks. (A) 2D trimap embedding of 500,000\\nunique likely odorants previously uncharacterized. The position of each point (molecule) is determined by POM\\ncoordinates, and the RGB values of each point correspond to their coordinates in the \\x00rst 3 dimensions of a\\nnon-negative matrix factorization of the predicted odor labels. (B) Intuitive geometric measures like vector\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='5ee42e52-6414-43e9-be5b-625c353195f4', embedding=None, metadata={'page_label': '11', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='length, vector distance, and vector projection correspond to the odor prediction tasks of odor detectability,\\nsimilarity, and descriptor applicability. Equation shows that the projected space Y represents the dot product\\nbetween POM and a task-speci\\x00c projection matrix X. (C) A linear model atop POM outpe\\x00orms a\\nchemoinformatic SVM baseline at predicting odor applicability on two extant datasets, Dravnieks (25) and\\nDREAM (6), as well as the current data. (D) A linear model atop POM outpe\\x00orms a chemoinformatic SVM\\nbaseline at predicting odor detection threshold using data from Abraham et al, 2011 (26). (E) A linear model atop\\nPOM outpe\\x00orms a chemoinformatic SVM baseline at predicting perceptual similarity on Snitz et al, 2013 (4).\\nWe show that the POM has a meaningful interpretation by extracting intuitive, geometric\\nmeasures and mapping them to several olfactory prediction tasks (Fig. 5B). The applicability\\nof any set of odor descriptors corresponds to a projection of the POM coordinates onto axes\\ncorresponding to those descriptors; odor strength (detectability) corresponds to the\\nmagnitude of this projection (Fig. S12), and odor similarity corresponds to the distance\\nbetween such projects for di\\x00erent molecules. We \\x00nd that a simple linear model applied to\\nPOM and using these geometric interpretations has comparable or superior pe\\x00ormance to\\na chemoinformatic suppo\\x00 vector machine (SVM) model across multiple published datasets\\n(Fig 5C, D, E), collectively representing some of the most thorough previous public e\\x00o\\x00s to\\ncharacterize these features of odor.\\nDiscussion\\nThere is no universally accepted method for quantifying and categorizing an odor percept. In\\nother words, olfaction has been a sense without a map. Systems of odor classi\\x00cation have\\nbeen proposed: \\x00rst intuitive categorizations (28), then empirically-suppo\\x00ed universal\\nspaces (29, 30), and later a\\x00empts to incorporate receptor mechanisms (31, 32). However,\\nthese systems do not tie stimulus prope\\x00ies to perception, and none have reached broad\\nacceptance. Here we propose and validate a novel, data-driven, high-dimensional map of\\nhuman olfaction. We have shown that this map recapitulates the structure and relationships\\nof odor perceptual categories evoked by single molecules, that it can be used to achieve\\nprospective predictive accuracy in odor description that exceeds that of the typical individual\\nhuman, and that it is broadly transferrable to arbitrary olfactory perceptual tasks using natural\\nand interpretable transformations. This map represents for odor what the CIE color space\\nrepresents for vision.\\nNearly all published chemosensory models were \\x00t to the data used in their construction.\\nEven using cross-validation, the oppo\\x00unity for over-\\x00\\x00ing is high, because the data comes\\nfrom a single distribution, task, or experimental source. Prospective validation on new data\\nfrom a new source with no adjustments, as we pe\\x00ormed, represents a much more stringent\\ntest of real-world utility.  In this prospective context, we found that our model pe\\x00orms\\nroughly on par with the median human panelist, beating a chemoinformatic baseline.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='f1027c04-56eb-4acd-88dd-cddc73d825ba', embedding=None, metadata={'page_label': '12', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='However, in a real-world se\\x00ing, models can and should be updated as new data becomes\\navailable. This process is called ‘online learning’ (27), and is a central capability of many\\nreal-world ML systems. Fig 5C demonstrates that a linear model atop POM reaches an even\\nhigher level of pe\\x00ormance when the POM is tuned to the new dataset.\\nThe success of this model is not merely an advance in predictive modeling.  It o\\x00ers a simple,\\nintuitive, contiguous, hierarchical, parseable map of molecular space in terms of odor, much\\nas color spaces represents wavelengths of light in terms of colors and color components.  It\\nenables human-level pe\\x00ormance not only for odor description but also generalizes to a\\ngamut of other olfactory tasks.  It o\\x00ers the oppo\\x00unity to reason, intuitively and\\ncomputationally, about the relationships within and between molecular and odor spaces.\\nThere are some practical considerations to keep in mind when using this map. First, the\\nconcentration of an odor in\\x00uences odor character, but is not explicitly included in the map.\\nSo while it can predict detection thresholds, a prope\\x00y of the odorant molecule, it cannot\\npredict suprathreshold intensity, a function of the odorant and its concentration.  Many\\nmolecules have no odor, which we addressed by pre-screening with a separate, simpler\\nmodel (33), and we diluted odorants to standardize intensity. Second, predictive pe\\x00ormance\\nis strong for organic molecules, the vast majority of odorants we encounter, but we could not\\nextend the predictions into halides or molecules that include novel elements due to the lack\\nof safety data for those molecules. Given uniformly strong pe\\x00ormance across broad\\nchemical classes tested in our prospective validation set (Fig. 3C), we expect high accuracy\\non novel chemicals within these chemical classes, but we would not expect high pe\\x00ormance\\nfor molecules that have chemical motifs not represented in our training set. For instance, if\\nour training dataset did not contain any molecules with carbon macrocycles, we would not\\nexpect the model to accurately predict the odor of an unseen macrocyclic musk (Fig. 3A).\\nThird, many chemical stimuli have odorous contaminants (24), pa\\x00icularly those that have not\\nbeen developed for use in fragrance applications. Neural networks are known to pe\\x00orm\\nwell, even with substantial noise in the training and test sets, which we see in the present\\nwork. Nonetheless, we recommend isolating the compound of interest from odorous\\ncontaminants, and/or characterizing the perceptual quality of contaminants. Finally, datasets\\nin real-world se\\x00ings are not static, but grow in size, and shi\\x00 in distribution —\\xa0models should\\nbe periodically retrained to incorporate new data. We showed that model pe\\x00ormance tends\\nto improve with increased training data (Fig. 3B) and data quality (Fig. S13), consistent with ML\\napplications in other areas (34, 35). Indeed, the most impo\\x00ant future work -- work which will\\nincrease the accuracy and resolution of the map and any model that uses it -- will be scaling\\nthe volume and quality of training data.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='e73ab585-c5bb-4145-9627-19538c19de5a', embedding=None, metadata={'page_label': '13', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Progress in neuroscience is o\\x00en measured by the creation and discovery of new maps of\\nthe world suppo\\x00ed by neural circuitry—maps of space in hippocampus, faces in the\\nsuperior temporal sulcus, tonotopy in auditory co\\x00ex, and retinotopy and Gabor \\x00lters in V1\\nvisual co\\x00ex, among others. Each is only possible because scientists \\x00rst possessed a map of\\nthe external world, and then measured how responses in the brain varied with stimulus\\nposition on the map. We have had no such map for odor, but this study proposes and\\nvalidates a novel data-driven map of human olfaction. We hope this map will be useful to\\nresearchers in chemistry, olfactory neuroscience, and psychophysics: \\x00rst, as a drop-in\\nreplacement for chemoinformatic descriptors, and more broadly as a new tool for\\ninvestigating the nature of olfactory sensation.\\nAcknowledgements: The authors wish to acknowledge Zelda Mariet for experimental\\ndesign, Yoni Halpern, Bob Da\\x00a, Steven Kearnes, Christina Zelano, Ari Morcos, Dan Bear, and\\nAlex Koulakov for feedback on dra\\x00, contributions to GC-MS/O analysis from Dr. J.S. Elmore,\\nand domain expe\\x00ise from Christophe Laudemiel.\\nFunding: National Institutes of Health grant F32 DC019030 (EJM); National Institutes of\\nHealth grant T32 DC000014 (EJM)\\nAuthor contributions: Conceptualization: ABW; Methodology: BKL, EJM, RCG, JDM, BSL,\\nJKP; So\\x00ware: BKL, BSL, RCG, JNW; Validation: RCG, BKL; Formal analysis: BKL, EJM, RCG;\\nInvestigation: EJM, KAL, MA, BBN, TM, JKP, JDM, BSL, WWQ, JNW; Data curation: BKL, BSL,\\nRCG, EJM, MA, KAL, JKP; Writing – original dra\\x00: EJM, BKL, ABW, JDM, RCG; Writing – review\\n& editing: EJM, BKL, ABW, JDM, RCG, JKP, BSL, WWQ, JNW; Visualization: EJM, RCG, BKL,\\nBSL; Supervision: ABW, JDM, EJM, JKP; Funding acquisition: ABW, JDM, EJM; Project\\nadministration: ABW, JDM\\nCompeting interests: BKL, JNW, BSL, WWQ, RCG, and ABW are employees of Google.  We\\ndeclare no competing \\x00nancial interests.\\nData and materials availability: Human psychophysics data, model predictions, and model\\nembeddings for all 400 tested odorants are included, and will be deposited at the olfactory\\ndata repository py\\x00ume.org upon publication; odorant identities will be released pending\\nlegal review upon publication. Lightweight reproduction notebooks and scripts will be shared\\nvia h\\x00ps://github.com/google-research/google-research.\\nReferences:\\n1. C. R. des Séances, Commission Internationale de l’Eclairage (1931).\\n2. E. F. Evans, Frequency selectivity at high signal levels of single units in cochlear nerve and nucleus.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='733e608d-4c2d-4d50-a69a-5ba73fe08206', embedding=None, metadata={'page_label': '14', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Psychophysics and physiology of hearing (1977).\\n3. C. S. Sell, On the Unpredictability of Odor. Angew. Chem. Int. Ed. 45, 6254–6261 (2006).\\n4. K. Snitz, A. Yablonka, T. Weiss, I. Frumin, R. M. Khan, N. Sobel, Predicting Odor Perceptual Similarity from\\nOdor Structure. PLoS Comput. Biol. 9 (2013), doi:10.1371/journal.pcbi.1003184.\\n5. A. Ravia, K. Snitz, D. Honigstein, M. Finkel, R. Zirler, O. Perl, L. Secundo, C. Laudamiel, D. Harel, N. Sobel, A\\nmeasure of smell enables the creation of olfactory metamers. Nature. 588, 118–123 (2020).\\n6. A. Keller, R. C. Gerkin, Y. Guan, A. Dhurandhar, G. Turu, B. Szalai, J. D. Mainland, Y. Ihara, C. W. Yu, R.\\nWol\\x00nger, C. Vens, L. Schietgat, K. De Grave, R. Norel, G. Stolovitzky, G. A. Cecchi, L. B. Vosshall, P. Meyer,\\nPredicting human olfactory perception from chemical features of odor molecules. Science. 355 (2017),\\ndoi:10.1126/science.aal2014.\\n7. J. Gilmer, S. S. Schoenholz, P. F. Riley, O. Vinyals, G. E. Dahl, in Machine Learning Meets Quantum Physics, K.\\nT. Schü\\x00, S. Chmiela, O. A. von Lilienfeld, A. Tkatchenko, K. Tsuda, K.-R. Müller, Eds. (Springer International\\nPublishing, Cham, 2020), pp. 199–214.\\n8. B. Sanchez-Lengeling, E. Reif, A. Pearce, A. Wiltschko, A gentle introduction to graph neural networks. Distill.\\n6 (2021), doi:10.23915/distill.00033.\\n9. H. L. Morgan, The Generation of a Unique Machine Description for Chemical Structures-A Technique\\nDeveloped at Chemical Abstracts Service. J. Chem. Doc. 5, 107–113 (1965).\\n10. A. Krizhevsky, I. Sutskever, G. E. Hinton, ImageNet classi\\x00cation with deep convolutional neural networks.\\nCommunications of the ACM. 60 (2017), pp. 84–90.\\n11. Schro\\x00, Kalenichenko, Philbin, Facenet: A uni\\x00ed embedding for face recognition and clustering. Proc. IEEE\\n(available at\\nh\\x00ps://www.cv-foundation.org/openaccess/content_cvpr_2015/html/Schro\\x00_FaceNet_A_Uni\\x00ed_2015_CVP\\nR_paper.html).\\n12. N. Jaitly, P. Nguyen, A. Senior, V. Vanhoucke, Application of pretrained deep neural networks to large\\nvocabulary speech recognition (2012) (available at h\\x00ps://research.google/pubs/pub38130/).\\n13. W. Luebke, The Good Scents Company Information System. Online Access: h\\x00p://www.\\nthegoodscentscompany. com (2019).\\n14. J. C. Le\\x00ngwell, Le\\x00ngwell & Associates (2015), (available at\\nh\\x00ps://www.le\\x00ngwell.com/cooler_than_menthol.htm).\\n15. B. Sanchez-Lengeling, J. N. Wei, B. K. Lee, Machine Learning for Scent: Learning Generalizable Perceptual\\nRepresentations of Small Molecules. arXiv preprint arXiv (2019) (available at h\\x00ps://arxiv.org/abs/1910.10685).\\n16. S. Kearnes, Pursuing a Prospective Perspective. TRECHEM. 3, 77–79 (2021).\\n17. H. T. Lawless, H. Heymann, in Sensory Evaluation of Food: Principles and Practices, H. T. Lawless, H.\\nHeymann, Eds. (Springer New York, New York, NY, 2010), pp. 227–257.\\n18. C. Trimmer, A. Keller, N. R. Murphy, L. L. Snyder, J. R. Willer, M. H. Nagai, N. Katsanis, L. B. Vosshall, H.\\nMatsunami, J. D. Mainland, Genetic variation across the human olfactory receptor repe\\x00oire alters odor\\nperception. Proc. Natl. Acad. Sci. U. S. A. (2019), doi:10.1073/pnas.1804106115.\\n19. A. Keller, M. Hempstead, I. A. Gomez, A. N. Gilbe\\x00, L. B. Vosshall, An olfactory demography of a diverse\\nmetropolitan population. BMC Neurosci. 13, 122 (2012).\\n20. A. Dravnieks, Odor quality: semantically generated multidimensional pro\\x00les are stable. Science. 218,\\n799–801 (1982).\\n21. K. J. Rossiter, Structure-odor relationships. Chem. Rev. 96, 3201–3240 (1996).\\n22. O. R. P. David, A Chemical History of Polycyclic Musks. Chemistry. 26, 7537–7555 (2020).\\n23. B. Li, M. L. Kamarck, Q. Peng, F.-L. Lim, A. Keller, M. A. M. Smeets, J. D. Mainland, S. Wang, From musk to\\nbody odor: decoding olfaction through genetic variation. PLOS Genetics (2022),\\ndoi:10.1101/2021.04.27.441177.\\n24. M. Paoli, D. Münch, A. Haase, E. Skoulakis, L. Turin, C. G. Galizia, Minute Impurities Contribute Signi\\x00cantly to\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='b1f3b381-ec48-455e-b0dd-c63d9761c7f8', embedding=None, metadata={'page_label': '15', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Olfactory Receptor Ligand Studies: Tales from Testing the Vibration Theory. eneuro. 4,\\nENEURO.0070–17.2017 (2017).\\n25. A. Dravnieks, Atlas of odor character pro\\x00les (1985).\\n26. M. H. Abraham, R. Sánchez-Moreno, J. E. Come\\x00o-Muñiz, W. S. Cain, An algorithm for 353 odor detection\\nthresholds in humans. Chem. Senses. 37, 207–218 (2012).\\n27. G. I. Parisi, R. Kemker, J. L. Pa\\x00, C. Kanan, S. Wermter, Continual lifelong learning with neural networks: A\\nreview. Neural Netw. 113, 54–71 (2019).\\n28. H. Zwaardemaker, Die Physiologie Des Geruchs (Рипол Классик, 1895).\\n29. H. Henning, Der geruch (JA Ba\\x00h, 1916).\\n30. E. C. Crocker, L. F. Henderson, Analysis and classi\\x00cation of odors: an e\\x00o\\x00 to develop a workable method.\\nAm Pe\\x00um Essent Oil Rev. 22, 325 (1927).\\n31. M. Guillot, Anosmies pa\\x00ielles et odeurs fondamentales. C. R. Acad. Sci. . 226, 1307–1309 (1948).\\n32. J. E. Amoore, C. Pfa\\x00mann, A plan to identify most of the primary odors. Olfaction & Taste III, 158–171 (1969).\\n33. E. J. Mayhew, C. J. Arayata, R. C. Gerkin, B. K. Lee, J. M. Magill, L. L. Snyder, K. A. Li\\x00le, C. W. Yu, J. D.\\nMainland, Transpo\\x00 features predict if a molecule is odorous. Proc. Natl. Acad. Sci. U. S. A. 119, e2116576119\\n(2022).\\n34. Brown, Mann, Ryder, Subbiah, Kaplan, Dhariwal, Neelakantan, Shyam, Sastry, Askell, Agarwal, Herbe\\x00-Voss,\\nKrueger, Henighan, Child, Ramesh, Ziegler, Wu, Winter, Hesse, Chen, Sigler, Litwin, Gray, Chess, Clark,\\nBerner, McCandlish, Radford, Sutskever, Amodei, Language Models are Few-Shot Learners. Adv. Neural Inf.\\nProcess. Syst. (available at\\nh\\x00ps://proceedings.neurips.cc/paper/2020/\\x00le/1457c0d6bfcb4967418b\\x008ac142f64a-Paper.pdf).\\n35. Gwern, The Scaling Hypothesis. gwern.net, (available at h\\x00ps://www.gwern.net/Scaling-hypothesis).\\n36. RDKit: Open-source cheminformatics, (available at h\\x00p://www.rdkit.org).\\n37. M. Meyners, S. R. Jaeger, G. Ares, On the analysis of Rate-All-That-Apply (RATA) data. Food Qual. Prefer. 49,\\n1–10 (2016).\\n38. N. Hauser, P. Kra\\x00, E. M. Carreira, The Serendipitous Discovery of a Rose Odorant. Chimia . 74, 247–251\\n(2020).\\n39. M. Stoll, Drug Cosmet. Ind. 38, 334–337 (1936).\\n40. G. Ohlo\\x00, W. Pickenhagen, P. Kra\\x00, Scent and Chemistry: The Molecular World of Odors (Wiley-VCH, ed. 1,\\n2012).\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='d21bdd56-6278-4f7b-b6e0-11b90dcaccc7', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='1, Propanolol, 1, [Cl].CC(C)NCC(O)COc1cccc2ccccc12\\n2, Terbutylchlorambucil, 1, C(=O)(OC(C)(C)C)CCCc1ccc(cc1)N(CCCl)CCCl\\n3, 40730, 1, c12c3c(N4CCN(C)CC4)c(F)cc1c(c(C(O)=O)cn2C(C)CO3)=O\\n4, 24, 1, C1CCN(CC1)Cc1cccc(c1)OCCCNC(=O)C\\n5, cloxacillin, 1, Cc1onc(c2ccccc2Cl)c1C(=O)N[C@H]3[C@H]4SC(C)(C)[C@@H](N4C3=O)C(O)=O\\n6, cefoperazone, 1, CCN1CCN(C(=O)N[C@@H](C(=O)N[C@H]2[C@H]3SCC(=C(N3C2=O)C(O)=O)CSc4nnnn4C)c5ccc(O)cc5)C(=O)C1=O\\n7, rolitetracycline, 1, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)cccc4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(/O)NCN5CCCC5)C1=O\\n8, ondansetron, 1, Cn1c2CCC(Cn3ccnc3C)C(=O)c2c4ccccc14\\n9, diltiazem, 1, COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@@H]2OC(C)=O\\n10, Amiloride, 1, NC(N)=NC(=O)c1nc(Cl)c(N)nc1N\\n11, M2L-663581, 0, OCC(C)(O)c1onc(c2ncn3c2CN(C)C(c4c3cccc4Cl)=O)n1\\n12, alovudine, 1, CC1=CN([C@H]2C[C@H](F)[C@@H](CO)O2)C(=O)NC1=O\\n13, 18, 1, C(Cl)Cl\\n14, 24959, 1, C1N(C(CC2CCCCC12)C(NC(C)(C)C)=O)CC(C(Cc1ccccc1)NC(C(NC(c1nc2c(cccc2)cc1)=O)CC(N)=O)=O)O\\n15, methadone, 1, CCC(=O)C(CC(C)N(C)C)(c1ccccc1)c2ccccc2\\n16, alfentanil, 1, CCN1N=NN(CCN2CCC(CC2)(COC)N(C(=O)CC)c3ccccc3)C1=O\\n17, BRL53080, 1, CN(C)C(=O)C(CCN1CCC(O)(CC1)c1ccc(Cl)cc1)(c1ccccc1)c1ccccc1\\n18, hyoscyamine, 1, CN1C2CCC1CC(C2)OC(=O)[C@H](CO)c3ccccc3\\n19, glyburide, 1, COc1ccc(Cl)cc1C(=O)NCCc2ccc(cc2)[S](=O)(=O)NC(=O)NC3CCCCC3\\n20, lamotrigine, 1, Nc1nnc(c(N)n1)c2cccc(Cl)c2Cl\\n21, 2-Methylpentane, 1, CCCC(C)C\\n22, Dexamfetamine, 1, C[C@H](N)Cc1ccccc1\\n23, SKF-93619, 0, c1cc2c(cc(CC3=CNC(=NC3=O)NCCSCc3oc(cc3)CN(C)C)cc2)cc1\\n24, nadolol, 1, CC(C)(C)NCC(O)COc1cccc2C[C@@H](O)[C@@H](O)Cc12\\n25, sufentanil, 1, CCC(=O)N(c1ccccc1)C2(CCN(CCc3sccc3)CC2)COC\\n26, 12, 0, CCNC(=NC#N)NCCSCc1c(cccn1)Br\\n27, zolpidem, 1, CN(C)C(=O)Cc1n2cc(C)ccc2nc1c3ccc(C)cc3\\n28, clozapine, 1, CN1CCN(CC1)C2=C3C=CC=CC3=Nc4ccc(Cl)cc4N2\\n29, Org12692, 1, FC(F)(F)c1c(Cl)nc(N2CCNCC2)cc1\\n30, piperacillin, 1, O.CCN1CCN(C(=O)N[C@@H](C(=O)N[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O)c4ccccc4)C(=O)C1=O\\n31, Pentobarbital, 1, CCCC(C)C1(CC)C(=O)NC(=O)NC1=O\\n32, Levofloxacin, 1, C[C@H]1COc2c(N3CCN(C)CC3)c(F)cc4C(=O)C(=CN1c24)C(O)=O\\n33, compound 40, 1, c1ccc(C(NCCCOc2cc(CN3CCCCC3)ccc2)=O)cc1\\n34, RO19-4603, 1, CC(C)(C)OC(=O)C1=C2CN(C(=O)C3=C(N2C=N1)C=CS3)C\\n35, fentanyl, 1, CCC(=O)N(C1CCN(CC1)CCc2ccccc2)c3ccccc3\\n36, etomidate, 1, CCOC(=O)c1cncn1C(C)c2ccccc2\\n37, 11a, 0, CN(C)c1cc(C2=NC(N)=NN2)ccn1\\n38, prochlorperazine, 1, CN1CCN(CCCN2c3ccccc3Sc4ccc(Cl)cc24)CC1\\n39, 2-(3\\'-Iodo-4\\'-aminophenyl)-6-hydroxybenzothiazole, 1, IC1=C(N([H])[H])C=CC(C2=NC3=CC=C(O[H])C=C3S2)=C1\\n40, compound 34, 0, O=C(C)Nc1cc(c2csc(N=C(N)N)n2)ccc1\\n41, granisetron, 1, CN1[C@H]2CCC[C@@H]1CC(C2)NC(=O)c3nn(C)c4ccccc34\\n42, thiamphenicol, 1, C[S](=O)(=O)c1ccc(cc1)[C@@H](O)[C@@H](CO)NC(=O)C(Cl)Cl\\n43, 16, 1, c1ccc(cc1)c1csc(n1)N=C(N)N\\n44, mexiletine, 1, CC(N)COc1c(C)cccc1C\\n45, dextropropoxyphene, 1, CCC(=O)O[C@@](Cc1ccccc1)([C@H](C)CN(C)C)c2ccccc2\\n46, Nor-1-chlorpromazine, 1, ClC1=CC=C(C2=C1)SC3=C(N2CCCNC)C=CC=C3\\n47, compound 37, 1, CN(C)Cc1ccc(c2cccc(NC3C([N+]([O-])=O)=CC=N3)c2)o1\\n48, cefamandole, 1, Cn1nnnc1SCC2=C(N3[C@H](SC2)[C@H](NC(=O)[C@H](O)c4ccccc4)C3=O)C(O)=O\\n49, Y-G 15, 1, n(ccc1)c(c1)CCN(C)C\\n50, Org13011, 1, O=C1N(CCC1)CCCCN2CCN(CC2)c3cc(C(F)(F)F)ccn3\\n51, Quinidine, 1, COc1ccc2nccc([C@H](O)[C@H]3C[C@@H]4CCN3C[C@@H]4C=C)c2c1\\n52, 38480, 1, c1(ccccc1)CC\\n53, 17, 1, c1ccc(cc1N)c1csc(n1)N=C(N)N\\n54, vinblastine, 1, CC[C@]1(O)C[C@H]2CN(CCc3c([nH]c4ccccc34)[C@@](C2)(C(=O)OC)c5cc6c(cc5OC)N(C)[C@H]7[C@](O)([C@H](OC(C)=O)[C@]8(CC)C=CCN9CC[C@]67[C@H]89)C(=O)OC)C1\\n55, Bacitracin, 1, c1(c(cc(NC(CCC)=O)cc1)C(C)=O)OCC(CNC(C)C)O\\n56, Nor-2-chlorpromazine, 1, ClC1=CC=C(SC(C=CC=C2)=C2N3CCCN)C3=C1\\n57, compound 36, 1, CN(C)Cc1ccc(CSCCNC2=C([N+]([O-])=O)C(Cc3ccccc3)=CN2)o1\\n58, 19, 0, CNC(=NC#N)Nc1cccc(c1)c1csc(n1)N=C(N)N\\n59, Y-G 14, 1, n(ccc1)c(c1)CCNC\\n60, 15, 1, O=N([O-])C1=C(CN=C1NCCSCc2ncccc2)Cc3ccccc3\\n61, sertraline, 1, [Cl-].CN[C@H]1CC[C@@H](c2ccc(Cl)c(Cl)c2)c3ccccc13.[H+]\\n62, 22767, 1, c1(nc(NC(N)=[NH2])sc1)CSCCNC(=[NH]C#N)NC\\n64, vincristine, 1, CC[C@]1(O)C[C@H]2CN(CCc3c([nH]c4ccccc34)[C@@](C2)(C(=O)OC)c5cc6c(cc5OC)N(C=O)[C@H]7[C@](O)([C@H](OC(C)=O)[C@]8(CC)C=CCN9CC[C@]67[C@H]89)C(=O)OC)C1\\n65, SB-222200, 1, CCC(NC(=O)c1c(c(nc2c1cccc2)c1ccccc1)C)c1ccccc1\\n66, SB-656104-A, 1, ClC1=CC=CC(OC2CCN(CCC3CCCN3S(C4=CC(N([H])C=C5)=C5C=C4)(=O)=O)CC2)=C1\\n67, compound 38, 1, CN(C)Cc1ccnc(c2cccc(NC3C([N+]([O-])=O)=CC=N3)c2)c1\\n68, maprotiline, 1, CNCCCC12CCC(c3ccccc13)c4ccccc24\\n69, compound 43, 1, N1(Cc2cc(OCCCNc3nccs3)ccc2)CCCCC1\\n70, MIL-663581, 0, OC(C)(C)c1onc(c2ncn3c2CN(C)C(c4c3cccc4Cl)=O)n1\\n71, cefotaxime, 1, [Na+].CO\\\\N=C(C(=O)N[C@@H]1[C@@H]2SCC(=C(N2C1=O)C([O-])=O)COC(C)=O)\\\\c3csc(N)n3\\n72, carbamazepine-10, 1, C1=CC=C2C(=C1)C3C(O3)C4=CC=CC=C4N2C(=O)N\\n73, Cyclohexane, 1, C1CCCCC1\\n74, 2, 1, Cc1csc(n1)N=C(N)N\\n75, SKF89124, 1, CCCN(CCc1c2c(c(cc1)O)NC(=O)C2)CCC\\n76, Org5222, 1, Clc1cc2c(Oc3ccccc3C3CN(CC32)C)cc1\\n77, compound 35, 0, CN(C)Cc1oc(CSCCNC2C([N+]([O-])=O)=CC=N2)cc1\\n78, cefuroxime, 1, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)COC(N)=O)/c3occc3\\n79, compound 45, 1, N1(Cc2cc(OCCCNc3oc4ccccc4n3)ccc2)CCCCC1\\n80, Org4428, 1, OC12c3c(Oc4c(C)cccc4C2CN(CC1)C)cccc3\\n81, compound 32, 1, [O-][N+](C1=CC=NC1NCCSCc2ccccn2)=O\\n82, Bretyliumtosilate, 1, c12c(nc([nH]1)NC(OC)=O)cc(SCCC)cc2\\n83, Y-G 20, 0, c1cccn2c1nc(c2)CCN\\n84, ornidazole, 1, Cc1ncc(n1CC(O)CCl)[N+]([O-])=O\\n85, SKF101468, 1, CCCN(CCc1c2c(ccc1)NC(=O)C2)CCC\\n86, Org30526, 1, Clc1cc2c(Oc3ccccc3C3CNCC32)cc1\\n87, 36, 1, CN(CCC=C1c2ccccc2CCc2ccccc12)C\\n88, ceftriaxone, 1, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)CSC3=NC(=O)C(=O)NN3C)/c4csc(N)n4\\n89, compound 42, 1, N1(Cc2cccc(OCCCNc3ccccn3)c2)CCCCC1\\n90, Org32104, 1, OC12c3c(Oc4c(C)cccc4C2CNCC1)cccc3\\n91, compound 31, 1, [O-][N+](C1=CC=NC1NCCSCc2ncccc2Br)=O\\n92, SB204457, 1, c12c3c(C(NC(C)=O)CCc2cc(c(c1OC)OC)OC)cc(=O)c(cc3)OC\\n93, compound 41, 1, C1CCN(CC1)Cc1cc(OCCCO)ccc1\\n94, clomethiazole, 1, Cc1ncsc1CCCl\\n95, celiprolol, 1, CCN(CC)C(=O)Nc1ccc(OCC(O)CNC(C)(C)C)c(c1)C(C)=O\\n96, \"1,1,1-Trifluro-2-chloroethane\", 1, ClCC(F)(F)F\\n97, 33, 1, CN1C(=O)c2c(cccc2Cl)n2cnc(c2C1)c1noc(n1)C(C)C\\n98, cefoxitin, 1, CO[C@]1(NC(=O)Cc2sccc2)[C@H]3SCC(=C(N3C1=O)C(O)=O)COC(N)=O\\n99, Org34167, 1, NC(CC=C)c1ccccc1c2noc3c2cccc3\\n100, 9-OH Risperidone, 1, OC1C(N2CCC1)=NC(C)=C(CCN3CCC(CC3)c4c5ccc(F)cc5on4)C2=O\\n101, acetaminophen, 1, CC(=O)Nc1ccc(O)cc1\\n102, acetylsalicylate, 0, CC(=O)Oc1ccccc1C(O)=O\\n103, allopurinol, 0, O=C1N=CN=C2NNC=C12\\n104, Alprostadil, 0, CCCCC[C@H](O)/C=C/[C@H]1[C@H](O)CC(=O)[C@@H]1CCCCCCC(O)=O\\n105, aminophylline, 0, CN1C(=O)N(C)c2nc[nH]c2C1=O.CN3C(=O)N(C)c4nc[nH]c4C3=O.NCCN\\n106, Amiodarone, 0, CCCCc1oc2ccccc2c1C(=O)c3cc(I)c(OCCN(CC)CC)c(I)c3\\n107, amoxicillin, 0, O.O.O.CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](N)c3ccc(O)cc3)C(=O)N2[C@H]1C(O)=O\\n108, ampicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](N)c3ccccc3)C(=O)N2[C@H]1C(O)=O\\n109, androsan, 0, C[C@]1(O)CC[C@H]2[C@@H]3CCC4=CC(=O)CC[C@]4(C)[C@H]3CC[C@]12C\\n110, oxaflumazine, 1, OC(=O)CCC(O)=O.FC(F)(F)c1ccc2Sc3ccccc3N(CCCN4CCN(CCC5OCCCO5)CC4)c2c1\\n111, oxazepam, 1, OC1N=C(c2ccccc2)c3cc(Cl)ccc3NC1=O\\n112, oxiperomide, 1, O=C1Nc2ccccc2N1C3CCN(CCOc4ccccc4)CC3\\n113, oxycodone, 1, COc1ccc2C[C@H]3N(C)CC[C@@]45[C@@H](Oc1c24)C(=O)CC[C@@]35O\\n114, oxymorphone, 1, CN1CC[C@]23[C@H]4Oc5c(O)ccc(C[C@@H]1[C@]2(O)CCC4=O)c35\\n115, pecazine, 1, CN1CCCC(C1)CN2c3ccccc3Sc4ccccc24\\n116, pemoline, 1, NC1=NC(=O)C(O1)c2ccccc2\\n117, pentazocine, 1, C[C@@H]1[C@H]2Cc3ccc(O)cc3[C@]1(C)CCN2CC=C(C)C\\n118, Ethionamide, 1, CCc1cc(ccn1)C(N)=S\\n119, Etoposide, 0, COc1cc(cc(OC)c1O)[C@H]2[C@@H]3C(COC3=O)[C@H](O[C@@H]4O[C@@H]5CO[C@@H](C)O[C@H]5[C@H](O)[C@H]4O)c6cc7OCOc7cc26\\n120, Famotidine, 1, NC(N)=Nc1scc(CSCCC(N)=N[S](N)(=O)=O)n1\\n121, Fluconazole, 1, OC(Cn1cncn1)(Cn2cncn2)c3ccc(F)cc3F\\n122, Foscarnetsodium, 1, [Na+].[Na+].[Na+].[O-]C(=O)[P]([O-])([O-])=O\\n123, Gabapentin, 1, NCC1(CCCCC1)CC(O)=O\\n124, Nafcillin, 0, [Na+].CCOc1ccc2ccccc2c1C(=O)N[C@H]3[C@H]4SC(C)(C)[C@@H](N4C3=O)C([O-])=O\\n125, Nalidixicacid, 0, CCN1C=C(C(O)=O)C(=O)c2ccc(C)nc12\\n126, Nelfinavir, 0, Cc1c(O)cccc1C(=O)N[C@@H](CSc2ccccc2)[C@H](O)CN3C[C@H]4CCCC[C@H]4C[C@H]3C(=O)NC(C)(C)C\\n127, Astemizole, 0, COc1ccc(CCN2CCC(CC2)Nc3nc4ccccc4n3Cc5ccc(F)cc5)cc1\\n128, atropine, 0, CN1C2CCC1CC(C2)OC(=O)C(CO)c3ccccc3\\n129, Auxeomycin, 0, [Cl-].CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)ccc(Cl)c4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O.[H+]\\n130, Betamethasone, 0, C[C@H]1C[C@H]2[C@@H]3CCC4=CC(=O)C=C[C@]4(C)[C@@]3(F)[C@@H](O)C[C@]2(C)[C@@]1(O)C(=O)CO\\n131, carbenicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C(C(O)=O)c3ccccc3)C(=O)N2[C@H]1C(O)=O\\n132, carbidopa, 0, C[C@@](Cc1ccc(O)c(O)c1)(NN)C(O)=O\\n133, carteolol, 0, CC(C)(C)NCC(O)COc1cccc2NC(=O)CCc12\\n134, Cefazolin, 0, [Na+].Cc1sc(SCC2=C(N3[C@H](SC2)[C@H](NC(=O)Cn4cnnn4)C3=O)C([O-])=O)nn1\\n135, cephapirin, 0, CC(=O)OCC1=C(N2[C@H](SC1)[C@H](NC(=O)CSc3ccncc3)C2=O)C(O)=O\\n136, chlorambucil, 0, OC(=O)CCCc1ccc(cc1)N(CCCl)CCCl\\n137, pergolide, 1, CCCN1C[C@H](CSC)C[C@H]2[C@H]1Cc3c[nH]c4cccc2c34\\n138, perlapine, 1, CN1CCN(CC1)C2=Nc3ccccc3Cc4ccccc24\\n139, perphenazine-HCl, 1, OCCN1CCN(CCCN2c3ccccc3Sc4ccc(Cl)cc24)CC1\\n140, pethidine, 1, CCOC(=O)C1(CCN(C)CC1)c2ccccc2\\n141, phencyclidine, 1, C1CCN(CC1)C2(CCCCC2)c3ccccc3\\n142, phenelzine sulfate, 1, NNCCc1ccccc1.O[S](O)(=O)=O\\n143, phenobarbital, 1, CCC1(C(=O)NC(=O)NC1=O)c2ccccc2\\n144, phenprobamate, 1, NC(=O)OCCCc1ccccc1\\n145, physostigmine, 1, CNC(=O)Oc1ccc2N(C)[C@H]3N(C)CC[C@@]3(C)c2c1\\n146, Norepinephrine, 0, NCC(O)c1ccc(O)c(O)c1\\n147, Omeprazole, 0, COc1ccc2nc([nH]c2c1)[S](=O)Cc3ncc(C)c(OC)c3C\\n148, Oxacillin, 0, Cc1onc(c2ccccc2)c1C(=O)N[C@H]3[C@H]4SC(C)(C)[C@@H](N4C3=O)C(O)=O\\n149, Pentostatin, 1, OC[C@H]1O[C@H](C[C@@H]1O)n2cnc3[C@H](O)CN=CNc23\\n150, Phenoxymethylpenicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)COc3ccccc3)C(=O)N2[C@H]1C([O-])=O\\n151, chloramphenicol, 0, OCC(NC(=O)C(Cl)Cl)C(O)c1ccc(cc1)[N+]([O-])=O\\n152, chlorothiazide, 0, N[S](=O)(=O)c1cc2c(NC=N[S]2(=O)=O)cc1Cl\\n153, chlorpropamide, 0, CCCNC(=O)N[S](=O)(=O)c1ccc(Cl)cc1\\n154, Chlortetracycline, 0, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)ccc(Cl)c4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O\\n155, chlorthalidone, 0, N[S](=O)(=O)c1cc(ccc1Cl)C2(O)NC(=O)c3ccccc23\\n156, clofibrate, 0, CCOC(=O)C(C)(C)Oc1ccc(Cl)cc1\\n157, clofibric acid, 0, CC(C)(Oc1ccc(Cl)cc1)C(O)=O\\n158, clotrimazole, 0, Clc1ccccc1C(n2ccnc2)(c3ccccc3)c4ccccc4\\n159, corticosterone, 0, C[C@]12C[C@H](O)[C@H]3[C@@H](CCC4=CC(=O)CC[C@]34C)[C@@H]1CC[C@@H]2C(=O)CO\\n160, Cortisone, 0, C[C@]12CCC(=O)C=C1CC[C@H]3[C@@H]4CC[C@](O)(C(=O)CO)[C@@]4(C)CC(=O)[C@H]23\\n161, piflutixol, 1, OCC[C@@H]1CCN(CC\\\\C=C2/c3ccc(F)cc3Sc4ccc(cc24)C(F)(F)F)CC1\\n162, pimozide, 1, Fc1ccc(cc1)C(CCCN2CCC(CC2)N3C(=O)Nc4ccccc34)c5ccc(F)cc5\\n163, pinoxepin, 1, OCCN1CCN(CC\\\\C=C/2c3ccccc3COc4ccc(Cl)cc24)CC1\\n164, pipamperone, 1, NC(=O)C1(CCN(CCCC(=O)c2ccc(F)cc2)CC1)N3CCCCC3\\n165, prazepam, 1, Clc1ccc2N(CC3CC3)C(=O)CN=C(c4ccccc4)c2c1\\n166, pregnanolone, 1, CC(=O)[C@H]1CC[C@H]2[C@@H]3CC[C@@H]4C[C@H](O)CC[C@]4(C)[C@H]3CC[C@]12C\\n167, primidone, 1, CCC1(C(=O)NCNC1=O)c2ccccc2\\n168, procaine, 1, CCN(CC)CCOC(=O)c1ccc(N)cc1\\n169, procyclidine, 1, OC(CCN1CCCC1)(C2CCCCC2)c3ccccc3\\n170, promazine, 1, CN(C)CCCN1c2ccccc2Sc3ccccc13\\n171, Pindolol, 1, CC(C)NCC(O)COc1cccc2[nH]ccc12\\n172, Plicamycin, 1, CO[C@@H]([C@@H]1Cc2cc3cc(O[C@H]4C[C@@H](O[C@H]5C[C@@H](O)[C@H](O)[C@@H](C)O5)[C@H](O)[C@@H](C)O4)c(C)c(O)c3c(O)c2C(=O)[C@H]1O[C@H]6C[C@@H](O[C@H]7C[C@@H](O[C@H]8C[C@](C)(O)[C@H](O)[C@@H](C)O8)[C@H](O)[C@@H](C)O7)[C@H](O)[C@@H](C)O6)C(=O)[C@@H](O)[C@@H](C)O\\n173, Pravastatin, 0, CC[C@H](C)C(=O)O[C@H]1C[C@H](O)C=C2C=C[C@H](C)[C@H](CC[C@@H](O)C[C@@H](O)CC(O)=O)[C@@H]12\\n174, Praziquantel, 1, O=C1CN(CC2N1CCc3ccccc23)C(=O)C4CCCCC4\\n175, Trimethoprim, 1, COc1cc(Cc2cnc(N)nc2N)cc(OC)c1OC\\n176, Trimetrexate, 0, COc1cc(NCc2ccc3nc(N)nc(N)c3c2C)cc(OC)c1OC\\n177, Trometamol, 0, NC(CO)(CO)CO\\n178, Trovafloxacin, 1, NC1[C@H]2CN(C[C@@H]12)c3nc4N(C=C(C(O)=O)C(=O)c4cc3F)c5ccc(F)cc5F\\n179, Urea, 1, NC(N)=O\\n180, coumarin, 0, O=C1Oc2ccccc2C=C1\\n181, dapsone, 0, Nc1ccc(cc1)[S](=O)(=O)c2ccc(N)cc2\\n182, Dehydrocholic acid, 0, C[C@H](CCC(O)=O)[C@H]1CC[C@H]2[C@H]3[C@H](CC(=O)[C@]12C)[C@@]4(C)CCC(=O)C[C@H]4CC3=O\\n183, dicumarol, 0, OC1=C(CC2=C(O)Oc3ccccc3C2=O)C(=O)c4ccccc4O1\\n184, dicyclomine, 0, CCN(CC)CCOC(=O)C1(CCCCC1)C2CCCCC2\\n185, Diethylstilbestrol, 0, CC\\\\C(c1ccc(O)cc1)=C(\\\\CC)c2ccc(O)cc2\\n186, Digitoxin, 0, C[C@H]1O[C@H](C[C@H](O)[C@@H]1O)O[C@H]2[C@@H](O)C[C@@H](O[C@@H]2C)O[C@H]3[C@@H](O)C[C@@H](O[C@@H]3C)O[C@H]4CC[C@@]5(C)[C@H](CC[C@@H]6[C@@H]5CC[C@]7(C)[C@@H](CC[C@]67O)C8=CC(=O)OC8)C4\\n187, Digoxin, 0, C[C@H]1O[C@H](C[C@H](O)[C@@H]1O)O[C@H]2[C@@H](O)C[C@@H](O[C@@H]2C)O[C@H]3[C@@H](O)C[C@@H](O[C@@H]3C)O[C@H]4CC[C@@]5(C)[C@H](CC[C@@H]6[C@@H]5C[C@@H](O)[C@]7(C)[C@H](CC[C@]67O)C8=CC(=O)OC8)C4\\n188, Domperidone, 0, Clc1ccc2N(C3CCN(CCCN4C(=O)Nc5ccccc45)CC3)C(=O)Nc2c1\\n189, promethazine-HCl, 1, CC(CN1c2ccccc2Sc3ccccc13)N(C)C\\n190, propiomazine, 1, CCC(=O)c1ccc2Sc3ccccc3N(CC(C)N(C)C)c2c1\\n191, propofol, 1, CC(C)c1cccc(C(C)C)c1O\\n192, protriptyline, 1, CNCCCC1c2ccccc2C=Cc3ccccc13\\n193, pyridazinone, 1, O=C1C=CNN=C1\\n194, quazepam, 1, Fc1ccccc1C2=NCC(=S)N(CC(F)(F)F)c3ccc(Cl)cc23\\n195, rescinnamine, 1, CO[C@H]1[C@@H](C[C@@H]2CN3CCc4c([nH]c5cc(OC)ccc45)[C@H]3C[C@@H]2[C@@H]1C(=O)OC)OC(=O)\\\\C=C\\\\c6cc(OC)c(OC)c(OC)c6\\n196, reserpine, 1, CO[C@H]1[C@@H](C[C@@H]2CN3CCc4c([nH]c5cc(OC)ccc45)[C@H]3C[C@@H]2[C@@H]1C(=O)OC)OC(=O)c6cc(OC)c(OC)c(OC)c6\\n197, secobarbital, 1, CCCC(C)C1(CC=C)C(=O)NC(=O)NC1=O\\n198, spiclomazine, 1, [H+].[Cl-].Clc1ccc2Sc3ccccc3N(CCCN4CCC5(CC4)NC(=O)CS5)c2c1\\n199, Atovaquone, 0, OC1=C(C2CCC(CC2)c3ccc(Cl)cc3)C(=O)C(=O)c4ccccc14\\n200, Azelaicacid, 0, OC(=O)CCCCCCCC(O)=O\\n201, Azithromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(OC)[C@@H](O)[C@H](C)O2)C(C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)[C@](C)(O)C[C@@H](C)CN(C)[C@H](C)[C@@H](O)[C@]1(C)O\\n202, Bacitracin, 0, CCC(C)C(N)C1=NC(CS1)C(=O)N[C@@H](CC(C)C)C(=O)N[C@H](CCC(O)=O)C(=O)N[C@@H]([C@@H](C)CC)C(=O)NCCCC[C@@H]2NC(=O)[C@H](CC(N)=O)NC(=O)[C@@H](CC(O)=O)NC(=O)[C@H](Cc3[nH]cnc3)NC(=O)[C@@H](Cc4ccccc4)NC(=O)[C@@H](NC(=O)[C@@H](CCCN)NC2=O)[C@@H](C)CC\\n203, Bethanechol, 0, [Cl-].CC(C[N+](C)(C)C)OC(N)=O\\n204, Bretyliumtosilate, 0, CC[N+](C)(C)Cc1ccccc1Br.Cc2ccc(cc2)[S]([O-])(=O)=O\\n205, Bromocriptine, 0, CC(C)C[C@@H]1N2C(=O)[C@](NC(=O)[C@H]3CN(C)[C@@H]4Cc5c(Br)[nH]c6cccc(C4=C3)c56)(O[C@@]2(O)[C@@H]7CCCN7C1=O)C(C)C\\n206, Buprenorphine, 1, c12[C@]34[C@@]56[C@H]([N@@](CC7CC7)CC4)Cc2ccc(c1O[C@H]3[C@](OC)([C@H](C5)[C@](C(C)(C)C)(C)O)CC6)O\\n207, dopamine, 0, NCCc1ccc(O)c(O)c1\\n208, doxorubicin, 0, COc1cccc2C(=O)c3c(O)c4C[C@](O)(C[C@H](O[C@H]5C[C@H](N)[C@H](O)[C@H](C)O5)c4c(O)c3C(=O)c12)C(=O)CO\\n209, doxycycline, 0, O.C[C@@H]1[C@H]2[C@H](O)[C@H]3[C@H](N(C)C)C(=O)C(=C(N)/O)/C(=O)[C@@]3(O)C(=O)C2=C(O)c4c(O)cccc14\\n210, Dyclonine, 0, [Cl].CCCCOc1ccc(cc1)C(=O)CCN2CCCCC2\\n211, Econazole, 0, Clc1ccc(COC(Cn2ccnc2)c3ccc(Cl)cc3Cl)cc1\\n212, enkephalin, 0, CC(C)[C@H](NC(=O)[C@H](Cc1ccccc1)NC(=O)CNC(=O)[C@@H](C)NC(=O)[C@@H](N)Cc2ccc(O)cc2)C(N)=O\\n213, Ephedrine, 0, CN[C@@H](C)[C@H](O)c1ccccc1\\n214, epinephrine, 0, CNC[C@H](O)c1ccc(O)c(O)c1\\n215, erythromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(OC)[C@@H](O)[C@H](C)O2)[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)[C@](C)(O)C[C@@H](C)C(=O)[C@H](C)[C@@H](O)[C@]1(C)O\\n216, estradiol, 0, C[C@]12CC[C@H]3[C@@H](CCc4cc(O)ccc34)[C@@H]1CC[C@@H]2O\\n217, tacrine, 1, Nc1c2CCCCc2nc3ccccc13\\n218, temazepam, 1, CN1C(=O)C(O)N=C(c2ccccc2)c3cc(Cl)ccc13\\n219, tetrabenzine, 1, COc1cc2CCN3CC(CC(C)C)C(=O)CC3c2cc1OC\\n220, thc, 1, CCCCCc1cc(O)c2[C@@H]3C=C(C)CC[C@H]3C(C)(C)Oc2c1\\n221, thebaine, 1, COC1=CC=C2[C@H]3Cc4ccc(OC)c5O[C@@H]1[C@]2(CCN3C)c45\\n222, thiethylperazine, 1, CCSc1ccc2Sc3ccccc3N(CCCN4CCN(C)CC4)c2c1\\n223, thioridazine, 1, CSc1ccc2Sc3ccccc3N(CCC4CCCCN4C)c2c1\\n224, thiothixene, 1, CN1CCN(CC/C=C/2c3ccccc3Sc4ccc(cc24)[S](=O)(=O)N(C)C)CC1\\n225, tranylcypromine, 1, [Cl-].N[C@@H]1C[C@H]1c2ccccc2.[H+]\\n226, trazodone, 1, Clc1cccc(c1)N2CCN(CCCN3N=C4C=CC=CN4C3=O)CC2\\n227, Busulfan, 1, C[S](=O)(=O)OCCCCO[S](C)(=O)=O\\n228, Cefradine, 0, CC1=C(N2[C@H](SC1)[C@H](NC(=O)[C@H](N)C3=CCC=CC3)C2=O)C(O)=O\\n229, Chloroquine, 1, CCN(CC)CCCC(C)Nc1ccnc2cc(Cl)ccc12\\n230, Cisapride, 1, O.COC1CN(CCCOc2ccc(F)cc2)CCC1NC(=O)c3cc(Cl)c(N)cc3OC\\n231, Cladribine, 1, Nc1nc(Cl)nc2n(cnc12)[C@H]3C[C@H](O)[C@@H](CO)O3\\n232, Clidiniumbromide, 0, [Br-].C[N+]12CCC(CC1)C(C2)OC(=O)C(O)(c3ccccc3)c4ccccc4\\n233, estrone, 0, C[C@]12CC[C@H]3[C@@H](CCc4cc(O)ccc34)[C@@H]1CCC2=O\\n234, Ethacrynic acid, 0, CCC(=C)C(=O)c1ccc(OCC(O)=O)c(Cl)c1Cl\\n235, ethinylestradiol, 0, C[C@]12CC[C@H]3[C@@H](CCc4cc(O)ccc34)[C@@H]1CC[C@@]2(O)C#C\\n236, fenoterol, 0, CC(Cc1ccc(O)cc1)NCC(O)c2cc(O)cc(O)c2\\n237, floxuridine, 0, OC[C@H]1O[C@H](C[C@@H]1O)N2C=C(F)C(=O)NC2=O\\n238, Flucloxacillin, 0, Cc1onc(c1C(=O)N[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O)c4c(F)cccc4Cl\\n239, fluorouracil, 0, FC1=CNC(=O)NC1=O\\n240, furosemide, 0, N[S](=O)(=O)c1cc(C(O)=O)c(NCc2occc2)cc1Cl\\n241, ganciclovir, 0, NC1=NC(=O)c2ncn(COC(CO)CO)c2N1\\n242, glycopyrrolate, 0, [Br-].C[N+]1(C)CCC(C1)OC(=O)C(O)(C2CCCC2)c3ccccc3\\n243, trifluperidol, 1, OC1(CCN(CCCC(=O)c2ccc(F)cc2)CC1)c3cccc(c3)C(F)(F)F\\n244, triflupromazine, 1, CN(C)CCCN1c2ccccc2Sc3ccc(cc13)C(F)(F)F\\n245, trifluroperazine, 1, CN1CCN(CCCN2c3ccccc3Sc4ccc(cc24)C(F)(F)F)CC1\\n246, trihexyphenidyl, 1, OC(CCN1CCCCC1)(C2CCCCC2)c3ccccc3\\n247, trimetozine, 1, COc1cc(cc(OC)c1OC)C(=O)N2CCOCC2\\n248, trimipramine, 1, CC(CN(C)C)CN1c2ccccc2CCc3ccccc13\\n249, tybamate, 1, CCCCNC(=O)OCC(C)(CCC)COC(N)=O\\n250, uridine, 1, OCC1OC(C(O)C1O)N2C=CC(=O)NC2=O\\n251, valproicacid, 1, CCCC(CCC)C(O)=O\\n252, zimelidine, 1, CN(C)C/C=C(/c1ccc(Br)cc1)c2cccnc2\\n253, Clofazimine, 0, CC(C)N=C1C=C2N(c3ccc(Cl)cc3)c4ccccc4N=C2C=C1Nc5ccc(Cl)cc5\\n254, Colchicine, 0, COC1=CC=C2C(=CC1=O)[C@H](CCc3cc(OC)c(OC)c(OC)c23)NC(C)=O\\n255, Cromoglicicacid, 0, OC(COc1ccc2OC(=CC(=O)c2c1)C(O)=O)COc3cccc4OC(=CC(=O)c34)C(O)=O\\n256, Cycloserine, 1, NC1CONC1=O\\n257, Hydroxycarbamide, 1, NC(=O)NO\\n258, Ipratropiumbromide, 0, [Br-].CC(C)[N+]1(C)C2CCC1CC(C2)OC(=O)C(CO)c3ccccc3\\n259, guanabenz, 0, NC(N)=N\\\\N=C\\\\c1c(Cl)cccc1Cl\\n260, guanethidinesulfate, 0, NC(N)=NCCN1CCCCCCC1\\n261, homatropine, 0, CN1C2CCC1CC(C2)OC(=O)C(O)c3ccccc3\\n262, hydralazine, 0, NNc1nncc2ccccc12\\n263, Hydrochlorothiazide, 0, N[S](=O)(=O)c1cc2c(NCN[S]2(=O)=O)cc1Cl\\n264, Hydrocortisone, 0, C[C@]12CCC(=O)C=C1CC[C@H]3[C@@H]4CC[C@](O)(C(=O)CO)[C@@]4(C)C[C@H](O)[C@H]23\\n265, hydroflumethiazide, 0, N[S](=O)(=O)c1cc2c(NCN[S]2(=O)=O)cc1C(F)(F)F\\n266, ibuprofen, 0, CC(C)Cc1ccc(cc1)C(C)C(O)=O\\n267, indomethacin, 0, COc1ccc2n(c(C)c(CC(O)=O)c2c1)C(=O)c3ccc(Cl)cc3\\n268, 35, 0, c12c(c(ccc1)Cl)C(N(Cc1n2cnc1c1nc(on1)[C@](CO)(C)O)C)=O\\n269, 1-47, 0, NCCc1cn2c(n1)cccc2\\n270, 1-48, 1, CCCN(CCC)CCc1ccc(c2c1CC(N2)=C)O\\n271, 9-OH-risperidone, 1, c1(c2c(cc(F)cc2)on1)C1CCN(CCc2c(n3c([C@@H](CCC3)O)nc2C)=O)CC1\\n272, Alprazolam, 1, Cc1nnc2CN=C(c3ccccc3)c4cc(Cl)ccc4n12\\n273, BCNU, 1, ClCCNC(=O)N(CCCl)N=O\\n274, Itraconazole, 0, CCC(C)N1N=CN(C1=O)c2ccc(cc2)N3CCN(CC3)c4ccc(OC[C@H]5CO[C@@](Cn6cncn6)(O5)c7ccc(Cl)cc7Cl)cc4\\n275, Kanamycin, 0, NCC1OC(OC2C(N)CC(N)C(OC3OC(CO)C(O)C(N)C3O)C2O)C(O)C(O)C1O\\n276, Ketoprofen, 1, CC(C(O)=O)c1cccc(c1)C(=O)c2ccccc2\\n277, Ketorolac, 0, OC(=O)C1CCn2c1ccc2C(=O)c3ccccc3\\n278, Metaraminol, 0, C[C@H](N)[C@H](O)c1cccc(O)c1\\n279, isoniazid, 0, NNC(=O)c1ccncc1\\n280, Isoproterenol, 0, CC(C)NCC(O)c1ccc(O)c(O)c1\\n281, Isoquercitrin, 0, OC[C@H]1O[C@@H](OC2=C(Oc3cc(O)cc(O)c3C2=O)c4ccc(O)c(O)c4)[C@H](O)[C@@H](O)[C@@H]1O\\n282, Ketoconazole, 0, CC(=O)N1CCN(CC1)c2ccc(OC[C@H]3CO[C@@](Cn4ccnc4)(O3)c5ccc(Cl)cc5Cl)cc2\\n283, labetalol, 0, CC(CCc1ccccc1)NCC(O)c2ccc(O)c(c2)C(N)=O\\n284, levodopa, 0, N[C@@H](Cc1ccc(O)c(O)c1)C(O)=O\\n285, mannitol, 0, OCC(O)C(O)C(O)C(O)CO\\n286, Meclocycline, 0, CN(C)C1C2C(O)C3C(=C)c4c(Cl)ccc(O)c4C(=C3C(=O)C2(O)C(=O)\\\\C(=C(N)/O)C1=O)O\\n287, Meclofenamate, 0, Cc1ccc(Cl)c(Nc2ccccc2C(O)=O)c1Cl\\n288, Mefenamic_acid, 0, Cc1cccc(Nc2ccccc2C(O)=O)c1C\\n289, Cimetidine, 0, CN=C(NCCSCc1nc[nH]c1C)NC#N\\n290, Clonidine, 1, Clc1cccc(Cl)c1NC2=NCCN2\\n291, Didanosine, 0, OC[C@@H]1CC[C@@H](O1)n2cnc3C(=O)N=CNc23\\n292, Metoprolol, 1, COCCc1ccc(OCC(O)CNC(C)C)cc1\\n293, Metronidazole, 1, Cc1ncc(n1CCO)[N+]([O-])=O\\n294, Mifepristone, 0, CC#C[C@]1(O)CC[C@H]2[C@@H]3CCC4=CC(=O)CCC4=C3[C@H](C[C@]12C)c5ccc(cc5)N(C)C\\n295, Minocycline, 1, C1[C@@H]2[C@](C(=C3C(c4c(ccc(c4C[C@@H]13)N(C)C)O)=O)O)(C(C(C(N)=O)=C([C@H]2N(C)C)O)=O)O\\n296, Moclobemide, 1, Clc1ccc(cc1)C(=O)NCCN2CCOCC2\\n297, Propanthelinebromide, 0, [Br-].CC(C)[N+](C)(CCOC(=O)C1c2ccccc2Oc3ccccc13)C(C)C\\n298, Pyrazinamide, 1, NC(=O)c1cnccn1\\n299, Pyridostigminebromide, 1, [Br-].CN(C)C(=O)Oc1ccc[n+](C)c1\\n300, Pyrimethamine, 1, CCc1nc(N)nc(N)c1c2ccc(Cl)cc2\\n301, Quinine, 0, COc1ccc2nccc([C@@H](O)[C@@H]3C[C@H]4CCN3C[C@@H]4C=C)c2c1\\n302, melphalan, 0, NC(Cc1ccc(cc1)N(CCCl)CCCl)C(O)=O\\n303, mepenzolatebromide, 0, [Br-].C[N+]1(C)CCCC(C1)OC(=O)C(O)(c2ccccc2)c3ccccc3\\n304, mepiramine, 0, COc1ccc(CN(CCN(C)C)c2ccccn2)cc1\\n305, mequitazine, 0, C1CN2CCC1C(C2)CN3c4ccccc4Sc5ccccc35\\n306, mestranol, 0, COc1ccc2[C@H]3CC[C@@]4(C)[C@@H](CC[C@@]4(O)C#C)[C@@H]3CCc2c1\\n307, methantheline, 0, CC[N+](C)(CC)CCOC(=O)C1c2ccccc2Oc3ccccc13\\n308, methotrexate, 0, CN(Cc1cnc2nc(N)nc(N)c2n1)c3ccc(cc3)C(=O)N[C@@H](CCC(O)=O)C(O)=O\\n309, Methyclothiazide, 0, CN1C(CCl)Nc2cc(Cl)c(cc2[S]1(=O)=O)[S](N)(=O)=O\\n310, methyldopa, 0, C[C@](N)(Cc1ccc(O)c(O)c1)C(O)=O\\n311, Icotidine, 0, [Cl-].[Cl-].[Cl-].COc1cccnc1CCCCNC2=NC=C(Cc3ccc(C)nc3)C(=O)N2.[H+].[H+].[H+]\\n312, L-663581, 1, CC(C)c1onc(n1)c2ncn3c2CN(C)C(=O)c4c(Cl)cccc34\\n313, Lupitidine, 0, Cc1ccc(CC2=CN=C(NCCSCc3oc(cc3)C(C)(C)N)NC2=O)cn1\\n314, Mianserin, 1, CN1CCN2C(C1)c3ccccc3Cc4ccccc24\\n315, Midazolam, 1, Cc1ncc2CN=C(c3ccccc3F)c4cc(Cl)ccc4n12\\n316, Ritonavir, 0, CC(C)[C@H](NC(=O)N(C)Cc1csc(n1)C(C)C)C(=O)N[C@H](C[C@H](O)[C@H](Cc2ccccc2)NC(=O)OCc3scnc3)Cc4ccccc4\\n317, Saquinavir, 0, CC(C)(C)NC(=O)[C@@H]1C[C@@H]2CCCC[C@@H]2CN1C[C@@H](O)[C@H](Cc3ccccc3)NC(=O)[C@H](CC(N)=O)NC(=O)c4ccc5ccccc5n4\\n318, Sevoflurane, 1, FCOC(C(F)(F)F)C(F)(F)F\\n319, Sodiumamidotrizoate, 0, [Na+].CC(=O)Nc1c(I)c(NC(C)=O)c(I)c(C([O-])=O)c1I\\n320, Sotalol, 1, CC(C)NCC(O)c1ccc(N[S](C)(=O)=O)cc1\\n321, Stavudine, 1, CC1=CN([C@@H]2O[C@H](CO)C=C2)C(=O)NC1=O\\n322, Streptomycin, 0, CN[C@H]1[C@H](O)[C@@H](O)[C@H](CO)O[C@H]1O[C@H]2[C@@H](O[C@@H](C)[C@]2(O)C=O)O[C@H]3[C@H](O)[C@@H](O)[C@H](N=C(N)N)[C@@H](O)[C@@H]3N=C(N)N\\n323, Sulfadiazine, 1, Nc1ccc(cc1)[S](=O)(=O)Nc2ncccn2\\n324, Sulfafurazole, 1, Cc1noc(N[S](=O)(=O)c2ccc(N)cc2)c1C\\n325, Methylprednisolone, 0, C[C@H]1C[C@H]2[C@@H]3CC[C@](O)(C(=O)CO)[C@@]3(C)C[C@H](O)[C@@H]2[C@@]4(C)C=CC(=O)C=C14\\n326, Miconazole, 0, Clc1ccc(COC(Cn2ccnc2)c3ccc(Cl)cc3Cl)c(Cl)c1\\n327, nandrolone, 0, C[C@]12CC[C@H]3[C@@H](CCC4=CC(=O)CC[C@H]34)[C@@H]1CC[C@@H]2O\\n328, neodicoumarol, 0, CCOC(=O)C(C1=C(O)Oc2ccccc2C1=O)C3=C(O)Oc4ccccc4C3=O\\n329, Nicotinyl alcohol, 0, OCc1cccnc1\\n330, norethindrone, 0, C[C@]12CC[C@H]3[C@@H](CCC4=CC(=O)CC[C@H]34)[C@@H]1CC[C@@]2(O)C#C\\n331, norfenefrine, 0, NCC(O)c1cccc(O)c1\\n332, nylidrin, 0, CC(CCc1ccccc1)NC(C)C(O)c2ccc(O)cc2\\n333, Oxandrolone, 0, C[C@]1(O)CC[C@H]2[C@@H]3CC[C@H]4CC(=O)OC[C@]4(C)[C@H]3CC[C@]12C\\n334, Oxaprozin, 0, OC(=O)CCc1oc(c2ccccc2)c(n1)c3ccccc3\\n335, Mirtazapine, 1, CN1CCN2C(C1)c3ccccc3Cc4cccnc24\\n336, Nevirapine, 1, Cc1ccnc2N(C3CC3)c4ncccc4C(=O)Nc12\\n337, Org12962, 1, C1NCCN(C1)c1ccc(c(n1)Cl)C(F)(F)F\\n338, Org30526, 1, c1c2c(cc(c1)Cl)[C@@H]1[C@H](c3ccccc3O2)CNC1\\n339, Org32104, 1, c1c2c(ccc1)[C@@]1([C@@H](c3cccc(c3O2)C)CNCC1)O\\n340, Org34167, 1, c1ccc(c(c1)[C@H](CC=C)N)c1c2c(on1)cccc2\\n341, Org4428, 1, c1c2c(ccc1)[C@@]1([C@@H](c3cccc(c3O2)C)C[N@](CC1)C)O\\n342, Org5222, 1, c1c2c(cc(c1)Cl)[C@@H]1[C@H](c3ccccc3O2)CN(C1)C\\n343, Ranitidine, 0, CNC(/NCCSCc1oc(CN(C)C)cc1)=C\\\\[N+]([O-])=O\\n344, Sulfamethoxazole, 1, Cc1onc(N[S](=O)(=O)c2ccc(N)cc2)c1\\n345, Tamoxifen, 1, CC/C(c1ccccc1)=C(c2ccccc2)/c3ccc(OCCN(C)C)cc3\\n346, Ticarcillin, 1, [Na+].[Na+].CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](C([O-])=O)c3cscc3)C(=O)N2[C@H]1C([O-])=O\\n347, Topotecan, 1, CC[C@@]1(O)C(=O)OCC2=C1C=C3N(Cc4cc5c(CN(C)C)c(O)ccc5nc34)C2=O\\n348, Zalcitabine, 1, NC1=NC(=O)N(C=C1)[C@H]2CC[C@@H](CO)O2\\n349, Zonisamide, 1, N[S](=O)(=O)Cc1noc2ccccc12\\n350, AmphotericinB, 0, C[C@H]1O[C@@H](O[C@@H]\\\\2C[C@@H]3O[C@](O)(C[C@@H](O)C[C@@H](O)[C@H](O)CC[C@@H](O)C[C@@H](O)CC(=O)O[C@@H](C)[C@H](C)[C@H](O)[C@@H](C)\\\\C=C/C=C\\\\C=C/C=C\\\\C=C/C=C\\\\C=C2)C[C@H](O)[C@H]3C(O)=O)[C@@H](O)[C@@H](N)[C@@H]1O\\n351, Calcium-folinate, 1, [Ca++].NC1=NC(=O)C2=C(NCC(CNc3ccc(cc3)C(=O)N[C@@H](CCC([O-])=O)C([O-])=O)N2C=O)N1\\n352, oxytetracycline, 0, O.O.CN(C)[C@H]1[C@@H]2[C@@H](O)[C@H]3C(=C(O)c4c(O)cccc4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O\\n353, papaverine, 0, COc1ccc(Cc2nccc3cc(OC)c(OC)cc23)cc1OC\\n354, penicillin G, 0, CC1(C)S[C@@H]2[C@H](NC(=O)Cc3ccccc3)C(=O)N2[C@H]1C(O)=O\\n355, phenacetin, 0, CCOc1ccc(NC(C)=O)cc1\\n356, phenazopyridine, 0, Nc1ccc(N=Nc2ccccc2)c(N)n1\\n357, pheniramine, 0, CN(C)CCC(c1ccccc1)c2ccccn2\\n358, Phenolphthalein, 0, Oc1ccc(cc1)C2(OC(=O)c3ccccc23)c4ccc(O)cc4\\n359, phenoxybenzamine, 0, CC(COc1ccccc1)N(CCCl)Cc2ccccc2\\n360, Phenylbutazone, 0, CCCCC1C(=O)N(N(C1=O)c2ccccc2)c3ccccc3\\n361, pirbuterol, 0, CC(C)(C)NCC(O)c1ccc(O)c(CO)n1\\n362, Risperidone, 1, CC1=C(CCN2CCC(CC2)c3noc4cc(F)ccc34)C(=O)N5CCCCC5=N1\\n363, Phenserine, 1, CN1CC[C@]2(C)[C@H]1N(C)c3ccc(OC(=O)Nc4ccccc4)cc23\\n364, Temelastine, 0, Cc1ccc(CC2=CN=C(NCCCCc3ncc(Br)cc3C)NC2=O)cn1\\n365, Tibolone, 1, C[C@@H]1CC2=C(CCC(=O)C2)[C@H]3CC[C@@]4(C)[C@@H](CC[C@@]4(O)C#C)[C@H]13\\n366, Ceftazidime, 1, CC(C)(O/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C([O-])=O)C[n+]3ccccc3)/c4csc(N)n4)C(O)=O\\n367, Flucytosine, 1, NC1=C(F)C=NC(=O)N1\\n368, Guanadrel, 0, NC(N)=NCC1COC2(CCCCC2)O1\\n369, Lomustine, 1, ClCCN(N=O)C(=O)NC1CCCCC1\\n370, Mesna, 0, [Na+].[O-][S](=O)(=O)CCS\\n371, Streptozotocin, 0, CN(N=O)C(=O)N[C@H]1[C@@H](O)O[C@H](CO)[C@@H](O)[C@@H]1O\\n372, Teniposide, 1, COc1cc(cc(OC)c1O)[C@H]2[C@@H]3[C@H](COC3=O)[C@H](OC4OC5COC(OC5C(O)C4O)c6sccc6)c7cc8OCOc8cc27\\n373, Thiotepa, 1, S=[P](N1CC1)(N2CC2)N3CC3\\n374, Vancomycin, 0, CN[C@H](CC(C)C)C(=O)NC1[C@H](O)c2ccc(Oc3cc4cc(Oc5ccc(cc5Cl)[C@@H](O)[C@@H]6NC(=O)[C@H](NC(=O)[C@@H]4NC(=O)[C@H](CC(N)=O)NC1=O)c7ccc(O)c(c7)c8c(O)cc(O)cc8[C@@H](NC6=O)C(O)=O)c3O[C@@H]9O[C@H](CO)[C@@H](O)[C@H](O)[C@H]9O[C@H]%10C[C@](C)(N)[C@H](O)[C@H](C)O%10)c(Cl)c2\\n375, Polythiazide, 0, CN1C(CSCC(F)(F)F)Nc2cc(Cl)c(cc2[S]1(=O)=O)[S](N)(=O)=O\\n376, prednisolone, 0, C[C@]12C[C@H](O)[C@H]3[C@@H](CCC4=CC(=O)C=C[C@]34C)[C@@H]1CC[C@]2(O)C(=O)CO\\n377, prednisone, 0, C[C@]12CC(=O)[C@H]3[C@@H](CCC4=CC(=O)C=C[C@]34C)[C@@H]1CC[C@]2(O)C(=O)CO\\n378, probenecid, 0, CCCN(CCC)[S](=O)(=O)c1ccc(cc1)C(O)=O\\n379, progesterone, 0, CC(=O)[C@H]1CC[C@H]2[C@@H]3CCC4=CC(=O)CC[C@]4(C)[C@H]3CC[C@]12C\\n380, propranolol, 0, CC(C)NCC(O)COc1cccc2ccccc12\\n381, proscillaridin, 0, CC1OC(O[C@H]2CC[C@]3(C)[C@H]4CC[C@]5(C)[C@H](CC[C@]5(O)[C@@H]4CCC3=C2)C6=COC(=O)C=C6)C(O)C(O)C1O\\n382, puromycin, 0, COc1ccc(C[C@H](N)C(=O)NC2[C@@H](O)[C@@H](O[C@@H]2CO)n3cnc4c(ncnc34)N(C)C)cc1\\n383, ribavirin, 0, NC(=O)c1ncn(n1)[C@@H]2O[C@H](CO)[C@@H](O)[C@H]2O\\n384, Tiotidine, 1, CN=C(NCCSCc1csc(N=C(N)N)n1)NC#N\\n385, Zolantidine, 1, OC(=O)\\\\C=C/C(O)=O.OC(=O)\\\\C=C/C(O)=O.C1CCN(CC1)Cc2cccc(OCCCNc3sc4ccccc4n3)c2\\n386, 1-hydroxymidazolam, 1, OCc1ncc2CN=C(c3ccccc3F)c4cc(Cl)ccc4n12\\n387, 111-trichloroethane, 1, CC(Cl)(Cl)Cl\\n388, 111-trifluoro-2-chloroethane, 1, C(CCl)(F)(F)F\\n389, 2-methylpropanol, 1, CC(C)CO\\n390, \"2,2-dimethylbutane\", 1, CCC(C)(C)C\\n391, Vidarabine, 1, Nc1ncnc2n(cnc12)C3OC(CO)C(O)C3O\\n392, Bis-hydroxylated-L-663581, 0, c12c(C(N(Cc3n1cnc3c1noc(n1)C(CO)(CO)C)C)=O)c(ccc2)Cl\\n393, ICI17148, 1, Cc1nc(sc1)\\\\[NH]=C(\\\\N)N\\n394, Mono-hydroxylated-L-663581, 0, c12c(C(N(Cc3n1cnc3c1noc(n1)C(O)(C)C)C)=O)c(ccc2)Cl\\n395, Thioperamide, 1, S=C(NC1CCCCC1)N2CCC(CC2)c3[nH]cnc3\\n396, rifampin, 0, CC1C=CC=C(C(=O)NC2=C(C3=C(C(=C4C(=C3C(=O)C2=CNN5CCN(CC5)C)C(=O)C(O4)(OC=CC(C(C(C(C(C(C1O)C)O)C)OC(=O)C)C)OC)C)C)O)O)C\\n397, salicylic acid, 0, OC(=O)c1ccccc1O\\n398, scopolamine, 0, CN1[C@@H]2CC(C[C@H]1[C@@H]3O[C@H]23)OC(=O)[C@H](CO)c4ccccc4\\n399, Spironolactone, 0, CC(=O)S[C@@H]1CC2=CC(=O)CC[C@]2(C)[C@H]3CC[C@@]4(C)[C@@H](CC[C@@]45CCC(=O)O5)[C@H]13\\n400, Stanozolol, 0, C[C@]1(O)CC[C@H]2[C@@H]3CC[C@H]4Cc5[nH]ncc5C[C@]4(C)[C@H]3CC[C@]12C\\n401, sulfasalazine, 0, OC(=O)C1=C\\\\C(C=CC1=O)=N/Nc2ccc(cc2)[S](=O)(=O)Nc3ccccn3\\n402, Terconazole, 0, CC(C)N1CCN(CC1)c2ccc(OC[C@H]3CO[C@@](Cn4cncn4)(O3)c5ccc(Cl)cc5Cl)cc2\\n403, testolactone, 0, C[C@]12CC[C@H]3[C@@H](CCC4=CC(=O)C=C[C@]34C)[C@@H]1CCC(=O)O2\\n404, testosterone, 0, C[C@]12CC[C@H]3[C@@H](CCC4=CC(=O)CC[C@]34C)[C@@H]1CC[C@@H]2O\\n405, TetraCycline, 0, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)cccc4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O\\n406, 3-methylhexane, 1, CCCC(C)CC\\n407, 3-methylpentane, 1, CCC(C)CC\\n408, 4-hydroxymidazolam, 1, Cc1ncc2C(O)N=C(c3ccccc3F)c4cc(Cl)ccc4n12\\n409, Aminopyrine, 1, CN(C)C1=C(C)N(C)N(c2ccccc2)C1=O\\n410, Atenolol, 0, CC(C)NCC(O)COc1ccc(CC(N)=O)cc1\\n411, Benzene, 1, c1ccccc1\\n412, Bretazenil, 1, CC(C)(C)OC(=O)c1ncn2c3cccc(Br)c3C(=O)N4CCC[C@H]4c12\\n413, Bromperidol, 1, OC1(CCN(CCCC(=O)c2ccc(F)cc2)CC1)c3ccc(Br)cc3\\n414, Mefloquine, 1, OC(C1CCCCN1)c2cc(nc3c2cccc3C(F)(F)F)C(F)(F)F\\n415, Ofloxacin, 1, CC1COc2c(N3CCN(C)CC3)c(F)cc4C(=O)C(=CN1c24)C(O)=O\\n416, Tifluadom, 1, CN1C(CNC(=O)c2cscc2)CN=C(c3ccccc3F)c4ccccc14\\n417, aldosterone, 0, C[C@]12CCC(=O)C=C1CC[C@H]3[C@@H]4CC[C@H](C(=O)CO)[C@]4(C[C@H](O)[C@H]23)C=O\\n418, BRL52537, 1, c1(cc(c(cc1)Cl)Cl)CC(N1CCCC[C@H]1CN1CCCC1)=O\\n419, BRL52580, 1, c1(cc(c(cc1)Cl)Cl)CC(N1[C@@H](c2c(CC1)cccc2)CN1CCCC1)=O\\n420, BRL52656, 1, c1(ccc(cc1)C(F)(F)F)CC(N1[C@@H](CCCC1)CN1CCCC1)=O\\n421, BRL52871, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@@H](c2c(CC1)scc2)CN1CCCC1)=O\\n422, theophylline, 0, CN1C(=O)N(C)c2nc[nH]c2C1=O\\n423, thioguanine, 0, NC1=NC(=S)c2[nH]cnc2N1\\n424, tolazamide, 0, Cc1ccc(cc1)[S](=O)(=O)NC(=O)NN2CCCCCC2\\n425, tolbutamide, 0, CCCCNC(=O)N[S](=O)(=O)c1ccc(C)cc1\\n426, Triamcinolone, 0, C[C@]12C[C@H](O)[C@@]3(F)[C@@H](CCC4=CC(=O)C=C[C@]34C)[C@@H]1C[C@@H](O)[C@]2(O)C(=O)CO\\n427, Triamterene, 0, Nc1nc(N)c2nc(c3ccccc3)c(N)nc2n1\\n428, Trichlormethiazide, 0, N[S](=O)(=O)c1cc2c(NC(N[S]2(=O)=O)C(Cl)Cl)cc1Cl\\n429, valinomycin, 0, CC(C)[C@@H]1NC(=O)[C@H](C)OC(=O)C(NC(=O)[C@H](OC(=O)[C@@H](NC(=O)[C@H](C)OC(=O)[C@H](NC(=O)[C@H](OC(=O)[C@@H](NC(=O)[C@H](C)OC(=O)[C@H](NC(=O)[C@H](OC1=O)C(C)C)C(C)C)C(C)C)C(C)C)C(C)C)C(C)C)C(C)C)C(C)C\\n430, verapamil, 0, COc1ccc(CCN(C)CCCC(C#N)(C(C)C)c2ccc(OC)c(OC)c2)cc1OC\\n431, warfarin, 0, CC(=O)CC(c1ccccc1)C2=C(O)Oc3ccccc3C2=O\\n432, Butanone, 1, CCC(C)=O\\n433, Cyclopropane, 1, C1CC1\\n434, Desmethyldesipramine, 1, N1(c2c(CCc3c1cccc3)cccc2)CCCN\\n435, Dichloromethane, 1, ClCCl\\n436, Diethylether, 1, CCOCC\\n437, Divinylether, 1, C=COC=C\\n438, Enflurane, 1, FC(F)OC(F)(F)C(F)Cl\\n439, BRL52974, 0, c1(ccc(c(c1)Cl)Cl)CC(N1[C@@H](c2c(CC1)[nH]cn2)CN1CCCC1)=O\\n440, BRL53087, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@@H](CN2CCCC2)CC(CC1)(C)C)=O\\n441, carebastine, 0, CC(C)(C(O)=O)c1ccc(cc1)C(=O)CCCN2CCC(CC2)OC(c3ccccc3)c4ccccc4\\n442, cetirizine, 0, OC(=O)COCCN1CCN(CC1)C(c2ccccc2)c3ccc(Cl)cc3\\n443, ciprofloxacin, 0, OC(=O)C1=CN(C2CC2)c3cc(N4CCNCC4)c(F)cc3C1=O\\n444, cp102, 0, c1cn(CCO)c(CC)c(O)c1=O\\n445, cp107, 0, c1cn(CCCCO)c(CC)c(O)c1=O\\n446, yohimbine, 0, COC(=O)[C@H]1[C@@H](O)CC[C@H]2CN3CCc4c([nH]c5ccccc45)[C@@H]3C[C@H]12\\n447, zidovudine, 0, CC1=CN([C@H]2C[C@H](N=[N+]=[N-])[C@@H](CO)O2)C(=O)NC1=O\\n448, acepromazine, 1, CN(C)CCCN1c2ccccc2Sc3ccc(cc13)C(C)=O\\n449, acetophenazine, 1, CC(=O)c1ccc2Sc3ccccc3N(CCCN4CCN(CCO)CC4)c2c1\\n450, acetozalamide, 1, CC(=O)Nc1sc(nn1)[S](N)(=O)=O\\n451, allobarbital, 1, C=CCC1(CC=C)C(=O)NC(=O)NC1=O\\n452, amantadine, 1, [Cl-].NC12CC3CC(CC(C3)C1)C2.[H+]\\n453, amitriptyline, 1, CN(C)CCC=C1c2ccccc2CCc3ccccc13\\n454, amobarbital, 1, CCC1(CCC(C)C)C(=O)NC(=O)NC1=O\\n455, amphetamine, 1, CC(N)Cc1ccccc1\\n456, Ethanol, 1, CCO\\n457, Ethylbenzene, 1, CCc1ccccc1\\n458, Flumazenil, 1, CCOC(=O)c1ncn2c1CN(C)C(=O)c3cc(F)ccc23\\n459, Flunitrazepam, 1, CN1C(=O)CN=C(c2ccccc2F)c3cc(ccc13)[N+]([O-])=O\\n460, Fluroxene, 1, FC(F)(F)COC=C\\n461, Halothane, 1, FC(F)(F)C(Cl)Br\\n462, Heptane, 1, CCCCCCC\\n463, Hexane, 1, CCCCCC\\n464, cp20, 1, CN1C=CC(=O)C(=C1C)O\\n465, cp21, 1, CCN1C=CC(=O)C(=C1C)O\\n466, cp24, 1, CCCCN1C=CC(=O)C(=C1C)O\\n467, cp25, 1, CCCCCN1C=CC(=O)C(=C1C)O\\n468, cp29, 1, c1cn(CCCCC)c(C)c(O)c1=O\\n469, cp41, 0, c1cn(CCCO)c(C)c(O)c1=O\\n470, cp94, 1, CCN1C=CC(=O)C(=C1CC)O\\n471, difloxacin, 0, CN1CCN(CC1)c2cc3N(C=C(C(O)=O)C(=O)c3cc2F)c4ccc(F)cc4\\n472, antipyrine, 1, CN1N(C(=O)C=C1C)c2ccccc2\\n473, apomorphine, 1, CN1CCc2cccc3c2[C@H]1Cc4ccc(O)c(O)c34\\n474, arecoline, 1, COC(=O)C1=CCCN(C)C1\\n475, azaperone, 1, Fc1ccc(cc1)C(=O)CCCN2CCN(CC2)c3ccccn3\\n476, benactyzine, 1, [H+].[Cl-].CCN(CC)CCOC(=O)C(O)(c1ccccc1)c2ccccc2\\n477, benperidol, 1, Fc1ccc(cc1)C(=O)CCCN2CCC(CC2)N3C(=O)Nc4ccccc34\\n478, benzocaine, 1, CCOC(=O)c1ccc(N)cc1\\n479, benzquinamide, 1, CCN(CC)C(=O)C1CN2CCc3cc(OC)c(OC)cc3C2CC1OC(C)=O\\n480, benztropine, 1, CN1C2CCC1CC(C2)OC(c3ccccc3)c4ccccc4.O[S](O)(=O)=O\\n481, biperiden, 1, OC(CCN1CCCCC1)(C2CC3CC2C=C3)c4ccccc4\\n482, Hexobarbital, 1, CN1C(=O)NC(=O)C(C)(C1=O)C2=CCCCC2\\n483, Indinavir, 1, CC(C)(C)NC(=O)[C@@H]1CN(CCN1C[C@@H](O)C[C@@H](Cc2ccccc2)C(=O)N[C@@H]3[C@H](O)Cc4ccccc34)Cc5cccnc5\\n484, Isoflurane, 1, FC(F)OC(Cl)C(F)(F)F\\n485, Methohexital, 1, CCC#CC(C)C1(CC=C)C(=O)NC(=O)N(C)C1=O\\n486, Methoxyflurane, 1, COC(F)(F)C(Cl)Cl\\n487, Methylcyclopentane, 1, CC1CCCC1\\n488, Nor-1-chlorpromazine, 1, N1(c2c(Sc3c1cccc3)ccc(c2)Cl)CCCNC\\n489, Nor-2-chlorpromazine, 1, N1(c2c(Sc3c1cccc3)ccc(c2)Cl)CCCN\\n490, doxylamine, 1, CN(C)CCOC(C)(c1ccccc1)c2ccccn2\\n491, EMD60400, 0, c1ccc([C@H](C[N@]2CC[C@H](C2)O)N(C(Cc2ccccc2N)=O)C)cc1\\n492, enoxacin, 0, CCN1C=C(C(O)=O)C(=O)c2cc(F)c(nc12)N3CCNCC3\\n493, fleroxacin, 0, CN1CCN(CC1)c2c(F)cc3C(=O)C(=CN(CCF)c3c2F)C(O)=O\\n494, GR85571, 1, c1(cc(c(cc1)Cl)Cl)CC(N1[C@@H](CN2CCCC2)CN(CC1)C(=O)C)=O\\n495, GR88377, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@@H](c2c(CC1)occ2)CN1CCCC1)=O\\n496, GR89696, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@H](CN(CC1)C(=O)OC)CN1CCCC1)=O\\n497, bromazepam, 1, Brc1ccc2NC(=O)CN=C(c3ccccn3)c2c1\\n498, buclizine, 1, CC(C)(C)c1ccc(CN2CCN(CC2)C(c3ccccc3)c4ccc(Cl)cc4)cc1\\n499, bufotenine, 1, CN(C)CCc1c[nH]c2ccc(O)cc12\\n500, bupropion, 1, CC(NC(C)(C)C)C(=O)c1cccc(Cl)c1\\n501, buramate, 1, OCCOC(=O)NCc1ccccc1\\n502, butabarbital, 1, CCC(C)C1(CC)C(=O)NC(=O)NC1=O\\n503, caffeine, 1, Cn1cnc2N(C)C(=O)N(C)C(=O)c12\\n504, cannabidiol, 1, CCCCCc1cc(O)c(C2C=C(C)CC[C@H]2C(C)=C)c(O)c1\\n505, capuride, 1, CCC(C)C(CC)C(=O)NC(N)=O\\n506, carbamazepine, 1, NC(=O)N1c2ccccc2C=Cc3ccccc13\\n507, Northioridazine, 1, CSc1ccc2Sc3ccccc3N(CCC4CCCNC4)c2c1\\n508, Org13011, 1, C1N(CCN(C1)c1cc(ccn1)C(F)(F)F)CCCCN1C(CCC1)=O\\n509, Paraxanthine, 1, Cn1cnc2NC(=O)N(C)C(=O)c12\\n510, Pentane, 1, CCCCC\\n511, Propanol, 1, CCCO\\n512, Propanone, 1, CC(C)=O\\n513, GR89696_et, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@H](CN(CC1)C(=O)OCC)CN1CCCC1)=O\\n514, GR89696_pr, 1, c1(CC(N2[C@H](CN(CC2)C(=O)OCCC)CN2CCCC2)=O)cc(c(cc1)Cl)Cl\\n515, GR91272, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@@H](c2c(CC1)occ2)C[N@@]1C[C@@H](CC1)O)=O\\n516, GR94839, 0, c1(ccc(c(c1)Cl)Cl)CC(=O)N1[C@H](CN(C(=O)C)CC1)C[N@@]1C[C@H](O)CC1\\n517, GR94839_A, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](C2)O)=O)ccc(C(F)(F)F)cc1\\n518, GR94839_B, 0, c1(ccc(cc1)SC)CC(N1[C@H](CN(CC1)C(=O)C)C[N@]1CC[C@H](C1)O)=O\\n519, GR94839_C, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](C2)O)=O)cc(cc(c1)F)F\\n520, GR94839_D, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](C2)O)=O)cc(ccc1)OC\\n521, GR94839_E, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(N(=O)=O)cc1\\n522, GR94839_F, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](C2)O)=O)ccc(OC)cc1\\n523, carphenazine, 1, CCC(=O)c1ccc2Sc3ccccc3N(CCCN4CCN(CCO)CC4)c2c1\\n524, cartazolate, 1, CCCCNc1c(cnc2n(CC)ncc12)C(=O)OCC\\n525, centazolone, 1, NN1C=Nc2cc3ccccc3cc2C1=O\\n526, chloralhydrate, 1, OC(O)C(Cl)(Cl)Cl\\n527, chlordiazepoxide, 1, CN=C1CN(O)C(=C2C=C(Cl)C=CC2=N1)c3ccccc3\\n528, chlorpromazine, 1, CN(C)CCCN1c2ccccc2Sc3ccc(Cl)cc13\\n529, chlorprothixene, 1, CN(C)CC\\\\C=C/1c2ccccc2Sc3ccc(Cl)cc13\\n530, chlorzoxazone, 1, Clc1ccc2OC(=O)Nc2c1\\n531, choline, 1, C[N+](C)(C)CCO\\n532, clobazam, 1, CN1C(=O)CC(=O)N(c2ccccc2)c3cc(Cl)ccc13\\n533, RO19-4603, 1, CN1Cc2n(cnc2C(=O)OC(C)(C)C)c3ccsc3C1=O\\n534, SKF101468, 1, C1C(Nc2c1c(ccc2)CCN(CCC)CCC)=O\\n535, SKF89124, 1, C1C(Nc2c1c(ccc2O)CCN(CCC)CCC)=O\\n536, Salicyluricacid, 1, OC(=O)CNC(=O)c1ccccc1O\\n537, Sulforidazine, 1, CN1CCCCC1CCN2c3ccccc3Sc4ccc(cc24)[S](C)(=O)=O\\n538, Teflurane, 1, FC(Br)C(F)(F)F\\n539, Theobromine, 1, Cn1cnc2N(C)C(=O)NC(=O)c12\\n540, Thiopental, 0, CCCC(C)C1(CC)C(=O)NC(=S)NC1=O\\n541, Toluene, 1, Cc1ccccc1\\n542, Triazolam, 1, Cc1nnc2CN=C(c3ccccc3Cl)c4cc(Cl)ccc4n12\\n543, GR94839_G, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)cc(ccc1)N(=O)=O\\n544, GR94839_H, 0, c1(ccccc1)CC(N1[C@H](CN(CC1)C(=O)C)C[N@]1CC[C@H](C1)O)=O\\n545, GR94839_I, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc([S@](=O)C)cc1\\n546, GR94839_L, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(S(=O)(=O)C)cc1\\n547, ICI197067, 1, CC(C)[C@@H](CN1CCCC1)N(C)C(=O)Cc2ccc(Cl)c(Cl)c2\\n548, ICI199441, 1, c1(ccccc1)[C@H](CN1CCCC1)N(C(=O)Cc1ccc(c(c1)Cl)Cl)C\\n549, isoxicam, 0, CN1C(=C(\\\\O)Nc2cc(C)on2)/C(=O)c3ccccc3[S]1(=O)=O\\n550, Levallorphan, 1, Oc1ccc2C[C@@H]3[C@@H]4CCCC[C@]4(CCN3CC=C)c2c1\\n551, lomefloxacin, 0, CCN1C=C(C(O)=O)C(=O)c2cc(F)c(N3CCNC(C)C3)c(F)c12\\n552, loperamide, 0, CN(C)C(=O)C(CCN1CCC(O)(CC1)c2ccc(Cl)cc2)(c3ccccc3)c4ccccc4\\n553, clomipramine, 1, CN(C)CCCN1c2ccccc2CCc3ccc(Cl)cc13\\n554, clonazepam, 1, [O-][N+](=O)c1ccc2NC(=O)CN=C(c3ccccc3Cl)c2c1\\n555, clopenthixol, 1, OCCN1CCN(CC\\\\C=C/2c3ccccc3Sc4ccc(Cl)cc24)CC1\\n556, clopimozide, 1, Fc1ccc(cc1)C(CCCN2CCC(CC2)N3C(=O)Nc4cc(Cl)ccc34)c5ccc(F)cc5\\n557, cocaine, 1, COC(=O)[C@H]1[C@H](CC2CCC1N2C)OC(=O)c3ccccc3\\n558, codeine, 1, COc1ccc2C[C@@H]3[C@@H]4C=C[C@H](O)[C@@H]5Oc1c2[C@]45CCN3C\\n559, cyclazocine, 1, CC1C2Cc3ccc(O)cc3C1(C)CCN2CC4CC4\\n560, cyclophenazine, 1, FC(F)(F)c1ccc2Sc3ccccc3N(CCCN4CCN(CC4)C5CC5)c2c1\\n561, deanol, 1, CN(C)CCO\\n562, demerol, 1, [H+].[Cl-].CCOC(=O)C1(CCN(C)CC1)c2ccccc2\\n563, Trichloroethylene, 1, ClC=C(Cl)Cl\\n564, Trichloromethane, 1, ClC(Cl)Cl\\n565, Y-G14, 1, c1c(nccc1)CCNC\\n566, Y-G15, 1, c1c(nccc1)CCN(C)C\\n567, Y-G16, 1, n1c(scc1)CCN\\n568, Y-G19, 0, n1c(scc1c1ccccc1)CCN\\n569, Y-G20, 0, c1c(nc2n1cccc2)CCN\\n570, m-Xylene, 1, Cc1cccc(C)c1\\n571, loratadine, 0, CCOC(=O)N1CCC(CC1)=C2c3ccc(Cl)cc3CCc4cccnc24\\n572, meloxicam, 0, CN1C(=C(/O)Nc2sc(C)cn2)/C(=O)c3ccccc3[S]1(=O)=O\\n573, Nalorphine, 1, O[C@H]1C=C[C@H]2[C@H]3Cc4ccc(O)c5O[C@@H]1[C@]2(CCN3CC=C)c45\\n574, naltrexone, 1, Oc1ccc2C[C@H]3N(CC[C@@]45[C@@H](Oc1c24)C(=O)CC[C@@]35O)CC6CC6\\n575, norfloxacin, 0, CCN1C=C(C(O)=O)C(=O)c2cc(F)c(cc12)N3CCNCC3\\n576, pefloxacin, 0, CCN1C=C(C(O)=O)C(=O)c2cc(F)c(cc12)N3CCN(C)CC3\\n577, piroxicam, 0, CN1C(=C(/O)Nc2ccccn2)/C(=O)c3ccccc3[S]1(=O)=O\\n578, roxindole, 1, C[S](O)(=O)=O.Oc1ccc2[nH]cc(CCCCN3CCC(=CC3)c4ccccc4)c2c1\\n579, deserpidine, 1, CO[C@H]1[C@@H](C[C@@H]2CN3CCc4c([nH]c5ccccc45)[C@H]3C[C@@H]2[C@@H]1C(=O)OC)OC(=O)c6cc(OC)c(OC)c(OC)c6\\n580, desipramine, 1, CNCCCN1c2ccccc2CCc3ccccc13\\n581, diazepam, 1, CN1C(=O)CN=C(c2ccccc2)c3cc(Cl)ccc13\\n582, dihydrocodeine, 1, COc1ccc2C[C@@H]3[C@@H]4CC[C@H](O)[C@@H]5Oc1c2[C@]45CCN3C\\n583, dihydromorphine, 1, CN1CCC23C4CCC(O)C2Oc5c(O)ccc(CC14)c35\\n584, diphenhydramine, 1, CN(C)CCOC(c1ccccc1)c2ccccc2\\n585, diphenhylhydantoin, 1, O=C1NC(=O)C(N1)(c2ccccc2)c3ccccc3\\n586, dixyrazine, 1, CC(CN1CCN(CCOCCO)CC1)CN2c3ccccc3Sc4ccccc24\\n587, DMT, 1, COC(=O)c1ccc(cc1)C(=O)OC\\n588, o-Xylene, 1, Cc1ccccc1C\\n589, p-Xylene, 1, Cc1ccc(C)cc1\\n590, 2-propanol, 1, CC(C)O\\n591, Carbamazepineepoxide, 1, NC(=O)N1c2ccccc2C3OC3c4ccccc14\\n592, SB-222200, 1, c12c(nc(c(c1C(N[C@@H](CC)c1ccccc1)=O)C)c1ccccc1)cccc2\\n593, Tertbutylchlorambucil, 1, c1cc(ccc1CCCC(OC(C)(C)C)=O)N(CCCl)CCCl\\n594, Delavirdine, 0, CC(C)Nc1cccnc1N2CCN(CC2)C(=O)c3[nH]c4ccc(N[S](C)(=O)=O)cc4c3\\n595, Ebastine, 0, CC(C)(C)c1ccc(cc1)C(=O)CCCN2CCC(CC2)OC(c3ccccc3)c4ccccc4\\n596, Pirenzepine, 1, CN1CCN(CC1)CC(=O)N2c3ccccc3C(=O)Nc4cccnc24\\n597, RP60180, 1, c1ccc2N([C@H](CN3CCCC3)C)c3c(ccc(C(=O)NCCC)c3)Sc2c1\\n598, rufloxacin, 0, CN1CCN(CC1)c2c(F)cc3C(=O)C(=CN4CCSc2c34)C(O)=O\\n599, salbutamol, 0, CC(C)(C)NCC(O)c1ccc(O)c(CO)c1\\n600, Sankyo, 1, c1(ccc(c(c1)Cl)Cl)CC(=O)N1[C@H](C[N@@](C)CC1)CN1CCCC1\\n601, SB201708, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@H](CN2CCCC2)c2n(CC1)ccn2)=O\\n602, SB204454, 0, c1(ccc(c(c1)Cl)Cl)CC(N1[C@H](C[C@]2(CC1)NC(NC2=O)=O)CN1CCCC1)=O\\n603, SB204459, 0, c1(ccc(c(c1)Cl)Cl)CC(N1[C@H](C[N@@]2C[C@@H](CC2)O)c2n(CC1)ccn2)=O\\n604, SB204484, 1, c1(ccc(c(c1)Cl)Cl)CC(N1[C@H](CN2CCCC2)c2n(CC1)ncn2)=O\\n605, SB205563, 0, c1(ccc(c(c1)Cl)Cl)CC(N1[C@@H](C[N@@]2C[C@@H](CC2)O)C[N@](CC1)CCO)=O\\n606, doxepin, 1, [H+].[Cl-].CN(C)CC\\\\C=C/1c2ccccc2COc3ccccc13\\n607, doxylamine succinate, 1, CN(C)CCOC(C)(c1ccccc1)c2ccccn2.OC(=O)CCC(O)=O\\n608, droperidol, 1, Fc1ccc(cc1)C(=O)CCCN2CCC(=CC2)N3C(=O)Nc4ccccc34\\n609, ectylurea, 1, CC\\\\C(=C/C)C(=O)NC(N)=O\\n610, emylcamate, 1, CCC(C)(CC)OC(N)=O\\n611, estazolam, 1, Clc1ccc2n3cnnc3CN=C(c4ccccc4)c2c1\\n612, etaqualone, 1, CCc1ccccc1N2C(=Nc3ccccc3C2=O)C\\n613, etazolate, 1, CCOC(=O)c1cnc2n(CC)ncc2c1NN=C(C)C\\n614, ethchlorvynol, 1, CCC(O)(\\\\C=C\\\\Cl)C#C\\n615, ethinamate, 1, NC(=O)OC1(CCCCC1)C#C\\n616, 5-6, 1, s1cc(CSCCN\\\\C(NC)=[NH]\\\\C#N)nc1\\\\[NH]=C(\\\\N)N\\n617, 5-66, 1, c1cc(CCNC)ncc1\\n618, 5-67, 1, c1cc(CCN(C)C)ncc1\\n619, 5-68, 1, s1c(ncc1)CCN\\n620, 5-69, 1, c1nc(C2CCN(CC2)C(NC2CCCCC2)=S)c[nH]1\\n621, 7-K06, 1, c1ccc2Oc3c(cc(cc3)Cl)[C@@H]3[C@@H](c2c1)C[N@](CC3)C\\n622, 7-K08, 1, c1c(Cl)nc(N2CCN(CCCCN3C(CCC3)=O)CC2)cc1C(F)(F)F\\n623, 7-K10, 1, c1ccc2Oc3c(cc(cc3)Cl)[C@@H]3[C@@H](c2c1)CNCC3\\n624, SB205605, 0, c1(ccc(c(c1)Cl)Cl)CC(N1[C@H](C[N@@]2C[C@@H](CC2)O)c2c(CC1)[nH]cn2)=O\\n625, skb_a, 1, c1ccc2nc(c(c(C(N[C@@H](CC)c3ccccc3)=O)c2c1)O)c1ccccc1\\n626, skb_c, 1, c1cc2c(C(N[C@@H](CC)c3ccccc3)=O)c(c(nc2cc1)c1ccccc1)OCCCN(C)C\\n627, skb-d, 0, c1cc2c(C(N[C@@H](CC)c3ccccc3)=O)c(c(nc2cc1)c1ccccc1)OCC(O)=O\\n628, skb-e, 0, c1cc2c(C(N[C@@H](CC)c3ccccc3)=O)c(c(c3ccccc3)nc2cc1)OCCCC(O)=O\\n629, skb-f, 0, c1ccc2c(c(C(N[C@@H](CC)c3ccccc3)=O)c(OCCNC(Cc3c(cccc3)C(=O)O)=O)c(c3ccccc3)n2)c1\\n630, skb-g, 0, c1cc2c(c(OCCNC(Cc3ncccc3)=O)c(c3ccccc3)nc2cc1)C(N[C@@H](CC)c1ccccc1)=O\\n631, skb-h, 0, c1cc2c(c(c(c3ccccc3)nc2cc1)OCCNC([C@@H]1CCCN1)=O)C(=O)N[C@@H](CC)c1ccccc1\\n632, skb-i, 0, c1cc2c(C(N[C@@H](CC)c3ccccc3)=O)c(c(nc2cc1)c1ccccc1)Cn1cncc1\\n633, sparfloxacin, 0, C[C@H]1CN(C[C@@H](C)N1)c2c(F)c(N)c3C(=O)C(=CN(C4CC4)c3c2F)C(O)=O\\n634, ethosuximide, 1, CCC1(C)CC(=O)NC1=O\\n635, etymemazine, 1, CCc1ccc2Sc3ccccc3N(CC(C)CN(C)C)c2c1\\n636, flupentixol, 1, OCCN1CCN(CC\\\\C=C/2c3ccccc3Sc4ccc(cc24)C(F)(F)F)CC1\\n637, fluphenazine, 1, OCCN1CCN(CCCN2c3ccccc3Sc4ccc(cc24)C(F)(F)F)CC1\\n638, flurazepam, 1, CCN(CC)CCN1C(=O)CN=C(c2ccccc2F)c3cc(Cl)ccc13\\n639, glaziovine, 1, COc1cc2CCN(C)C3CC4(C=CC(=O)C=C4)c(c1O)c23\\n640, glutethimide, 1, CCC1(CCC(=O)NC1=O)c2ccccc2\\n641, haloperidol, 1, OC1(CCN(CCCC(=O)c2ccc(F)cc2)CC1)c3ccc(Cl)cc3\\n642, harmalol, 1, CC1=C2NC3=CC(=O)C=CC3=C2C=CN1\\n643, harman, 1, Cc1nccc2c1[nH]c3ccccc23\\n644, 12, 0, c1c(c(ncc1)CSCCN\\\\C(=[NH]\\\\C#N)NCC)Br\\n645, 14, 1, c1cc(ncc1)CSCCNc1c(cc[nH]1)[N+](=O)[O-]\\n646, 15, 1, c1(cc(nc(c1C)C)C(SC(CNc1c(cc[nH]1)[N+](=O)[O-])(C)C)(C)C)C\\n647, 16, 1, n1c(csc1\\\\[NH]=C(\\\\N)N)c1ccccc1\\n648, 17, 0, n1c(csc1\\\\[NH]=C(\\\\N)N)c1cccc(c1)N\\n649, 18, 0, n1c(csc1\\\\[NH]=C(\\\\N)N)c1cccc(c1)NC(C)=O\\n650, 19, 0, n1c(csc1\\\\[NH]=C(\\\\N)N)c1cccc(c1)N\\\\C(NC)=[NH]\\\\C#N\\n651, 2, 1, s1cc(nc1\\\\[NH]=C(\\\\N)N)C\\n652, 22, 1, c1(c(oc(c1C)C(N(C(C)(N)C)C(N)(C)C)(C)C)C)C.C[NH+]O.[OH-]\\n653, 23, 1, [NH]C(CC(C)C([N@@](C(C)(C)C)C(N)(C)N)(C)C)c1c(c(c[nH+][o+]1)C)[O-]\\n654, tamitinol, 1, CCNCc1c(O)c(C)ncc1CSC\\n655, tenoxicam, 0, CN1C(=C(\\\\O)Nc2ccccn2)/C(=O)c3sccc3[S]1(=O)=O\\n656, terfenadine, 0, CC(C)(C)c1ccc(cc1)C(O)CCCN2CCC(CC2)C(O)(c3ccccc3)c4ccccc4\\n657, U50488, 1, CN(C1CCCC[C@H]1N2CCCC2)C(=O)Cc3ccc(Cl)c(Cl)c3\\n658, heroin, 1, CN1CC[C@]23[C@H]4Oc5c(OC(C)=O)ccc(C[C@@H]1[C@@H]2C=C[C@@H]4OC(C)=O)c35\\n659, hexapropymate, 1, NC(=O)OC1(CCCCC1)CC#C\\n660, homophenazine, 1, [H+].[H+].[Cl-].[Cl-].OCCN1CCCN(CCCN2c3ccccc3Sc4cc(ccc24)C(F)(F)F)CC1\\n661, hydrocodone, 1, COc1ccc2C[C@@H]3[C@@H]4CCC(=O)[C@@H]5Oc1c2[C@]45CCN3C\\n662, hydroxyphenamate, 1, CCC(O)(COC(N)=O)c1ccccc1\\n663, hydroxyzine, 1, OCCOCCN1CCN(CC1)C(c2ccccc2)c3ccc(Cl)cc3\\n664, imipramine, 1, CN(C)CCCN1c2ccccc2CCc3ccccc13\\n665, ketamine, 1, CNC1(CCCCC1=O)c2ccccc2Cl\\n666, kynurenic acid, 1, OC(=O)C1=CC(=O)c2ccccc2N1\\n667, 24, 1, C1CN(CCC1)Cc1cccc(c1)OCCCNC(=O)C\\n668, 25, 1, C1CN(CCC1)Cc1cccc(c1)OCCCNC(=O)c1ccccc1\\n669, 26, 1, C1CN(CCC1)Cc1cccc(c1)OCCCO\\n670, 27, 1, C1CN(CCC1)Cc1cccc(c1)OCCCNc1ncccc1\\n671, 28, 1, C1CN(CCC1)Cc1cccc(c1)OCCCNc1nccs1\\n672, 30, 1, C1CN(CCC1)Cc1cccc(c1)OCCCNc1nc2c(o1)cccc2\\n673, 4, 0, c1c(oc(c1)CN(C)C)CSCCNc1nc(c(c[nH]1)Cc1cc2c(cc1)cccc2)=O\\n674, 8-5, 0, CN(C)Cc1oc(cc1)CSCCNC=1NC=C(CN1)Cc1cnc(cc1)C\\n675, BBcpd10, 0, c1(nccc(c1)c1nc([nH]n1)N)N(C)C\\n676, 9-C1, 1, n1(c(c2nc[nH]c2n(c1=O)C)=O)C\\n677, lenperone, 1, Fc1ccc(cc1)C(=O)CCCN2CCC(CC2)C(=O)c3ccc(F)cc3\\n678, levorphanol, 1, CN1CC[C@]23CCCC[C@H]2[C@H]1Cc4ccc(O)cc34\\n679, lidocaine, 1, CCN(CC)CC(=O)Nc1c(C)cccc1C\\n680, lorazepam, 1, OC1N=C(c2ccccc2Cl)c3cc(Cl)ccc3NC1=O\\n681, lorcainide, 1, CC(C)N1CCC(CC1)N(C(=O)Cc2ccccc2)c3ccc(Cl)cc3\\n682, meclofenoxate, 1, CN(C)CCOC(=O)COc1ccc(Cl)cc1\\n683, mecloqualone, 1, CC1=Nc2ccccc2C(=O)N1c3ccccc3Cl\\n684, medazepam, 1, CN1CCN=C(c2ccccc2)c3cc(Cl)ccc13\\n685, meprobamate, 1, CCCC(C)(COC(N)=O)COC(N)=O\\n686, 9-C15, 1, c1c2CCOc2c(cc1OC)CN[C@@H]1CCCN[C@H]1c1ccccc1\\n687, 9-C17, 1, c1(cc(N\\\\C(=[NH]\\\\c2cccc(c2)CC)C)ccc1)CC\\n688, 9-C7, 1, c1(c(cc2n(ncc2c1)CCN)Cl)Cl\\n689, Abacavir, 1, Nc1nc(NC2CC2)c3ncn([C@@H]4C[C@H](CO)C=C4)c3n1\\n690, Acebutolol, 1, CCCC(=O)Nc1ccc(OCC(O)CNC(C)C)c(c1)C(C)=O\\n691, Aciclovir, 1, NC1=NC(=O)c2ncn(COCCO)c2N1\\n692, Albendazole, 1, CCCSc1ccc2nc(NC(=O)OC)[nH]c2c1\\n693, Amifostine, 0, NCCCNCCS[P](O)(O)=O\\n694, mescaline, 1, COc1cc(CCN)cc(OC)c1OC\\n695, mesoridazine, 1, CN1CCCCC1CCN2c3ccccc3Sc4ccc(cc24)[S](C)=O\\n696, methamphetamine, 1, CN[C@@H](C)Cc1ccccc1\\n697, methaqualone, 1, Cc1ccccc1N2C(=Nc3ccccc3C2=O)C\\n698, methocarbamol, 1, COc1ccccc1OCC(O)COC(N)=O\\n699, methopromazine, 1, COc1ccc2Sc3ccccc3N(CCCN(C)C)c2c1\\n700, methotrimeprazine, 1, COc1ccc2Sc3ccccc3N(C[C@H](C)CN(C)C)c2c1\\n701, methylpentynol, 1, CCC(C)(O)C#C\\n702, methylphenidate, 1, COC(=O)C(C1CCCCN1)c2ccccc2\\n703, methyprylon, 1, CCC1(CC)C(=O)NCC(C)C1=O\\n704, Amikacin, 1, NCC[C@H](O)C(=O)N[C@@H]1C[C@H](N)[C@@H](O[C@H]2O[C@H](CN)[C@@H](O)[C@H](O)[C@H]2O)[C@H](O)[C@H]1O[C@H]3O[C@H](CO)[C@@H](O)[C@H](N)[C@H]3O\\n705, Aminosalicylicacid, 0, Nc1ccc(C(O)=O)c(O)c1\\n706, Cytarabine, 1, NC1=NC(=O)N(C=C1)[C@@H]2O[C@H](CO)[C@@H](O)[C@@H]2O\\n707, Dacarbazine, 1, CN(C)N\\\\N=C/1N=CN=C1C(N)=O\\n708, Daunorubicin, 0, COc1cccc2C(=O)c3c(O)c4C[C@](O)(C[C@H](O[C@H]5C[C@H](N)[C@H](O)[C@H](C)O5)c4c(O)c3C(=O)c12)C(C)=O\\n709, Demeclocycline, 0, CN(C)[C@H]1[C@@H]2C[C@@H]3[C@H](O)c4c(Cl)ccc(O)c4C(=C3C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O)O\\n710, Desmethylclobazam, 1, Clc1ccc2NC(=O)CC(=O)N(c3ccccc3)c2c1\\n711, Desmonomethylpromazine, 1, CNCCCN1c2ccccc2Sc3ccccc13\\n712, Dexrazoxane, 0, C[C@@H](CN1CC(=O)NC(=O)C1)N2CC(=O)NC(=O)C2\\n713, midazolam maleate, 1, Cc1ncc2CN=C(c3ccccc3F)c4cc(Cl)ccc4n12.OC(=O)\\\\C=C/C(O)=O\\n714, moperone, 1, Cc1ccc(cc1)C2(O)CCN(CCCC(=O)c3ccc(F)cc3)CC2\\n715, morphine, 1, CN1CC[C@]23[C@H]4Oc5c(O)ccc(C[C@@H]1[C@@H]2C=CC4O)c35\\n716, MPTP, 1, CN1CCC(=CC1)c2ccccc2\\n717, naloxone, 1, Oc1ccc2C[C@H]3N(CC[C@@]45[C@@H](Oc1c24)C(=O)CC[C@@]35O)CC=C\\n718, nicotine, 1, CN1CCC[C@H]1c2cccnc2\\n719, nitrazepam, 1, [O-][N+](=O)c1ccc2NC(=O)CN=C(c3ccccc3)c2c1\\n720, nordazepam, 1, Clc1ccc2NC(=O)CN=C(c3ccccc3)c2c1\\n721, nortriptyline, 1, CNCCC=C1c2ccccc2CCc3ccccc13\\n722, orphenadrine, 1, CN(C)CCOC(c1ccccc1)c2ccccc2C\\n723, Dextromethorphan, 1, COc1ccc2C[C@H]3[C@H]4CCCC[C@@]4(CCN3C)c2c1\\n724, Dihydroergotamine, 0, CN1C[C@@H](C[C@H]2[C@H]1Cc3c[nH]c4cccc2c34)C(=O)N[C@]5(C)O[C@@]6(O)[C@@H]7CCCN7C(=O)[C@H](Cc8ccccc8)N6C5=O\\n725, Disulfiram, 1, CCN(CC)C(=S)SSC(=S)N(CC)CC\\n726, Efavirenz, 0, FC(F)(F)[C@]1(OC(=O)Nc2ccc(Cl)cc12)C#CC3CC3\\n727, Enalapril, 0, CCOC(=O)[C@H](CCc1ccccc1)N[C@@H](C)C(=O)N2CCC[C@H]2C(O)=O\\n728, Ethambutol, 0, CC[C@@H](CO)NCCN[C@@H](CC)CO\\n729, aclarubicin, 0, CC[C@@]1(O)C[C@H](OC2CC(C(OC3CC(O)C(OC4CCC(=O)C(C)O4)C(C)O3)C(C)O2)N(C)C)c5c(O)c6C(=O)c7c(O)cccc7C(=O)c6cc5[C@H]1C(=O)OC\\n730, adicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)CCC[C@@H](N)C(O)=O)C(=O)N2[C@H]1C(O)=O\\n731, adrenaline, 0, CNCC(O)c1ccc(O)c(O)c1\\n732, adriamycinol-aglycone, 0, COc1cccc2C(=O)c3c(O)c4CC(O)(CC(O)c4c(O)c3C(=O)c12)C(=O)CO\\n733, almecillin, 0, [Na+].CC1(C)SC2C(NC(=O)CSCC=C)C(=O)N2C1C([O-])=O\\n734, amantocillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C34C[C@H]5C[C@H](CC(N)(C5)C3)C4)C(=O)N2[C@H]1C(O)=O\\n735, amdinocillinpivoxil, 0, CC(C)(C)C(=O)OCOC(=O)C1N2[C@H](SC1(C)C)[C@H](N=CN3CCCCCC3)C2=O\\n736, amicycline, 0, CN(C)[C@H]1[C@@H]2C[C@@H]3Cc4ccc(N)c(O)c4C(=C3C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O)O\\n737, amrubicin, 0, [Cl-].CC(=O)[C@@]1(N)C[C@H](O[C@H]2C[C@H](O)[C@H](O)CO2)c3c(O)c4C(=O)c5ccccc5C(=O)c4c(O)c3C1.[H+]\\n738, apalcillin, 0, CC1(C)S[C@@H]2C(NC(=O)[C@H](NC(=O)C3=CNc4cccnc4C3=O)c5ccccc5)C(=O)N2[C@H]1C(O)=O\\n739, apicycline, 0, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)cccc4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(/O)NC(N5CCN(CCO)CC5)C(O)=O)C1=O\\n740, apramycin, 0, CN[C@H]1[C@@H](O)[C@H]2O[C@@H](O[C@@H]3[C@@H](N)C[C@@H](N)[C@H](O)[C@H]3O)[C@H](N)C[C@@H]2O[C@@H]1O[C@H]4O[C@H](CO)[C@@H](N)[C@H](O)[C@H]4O\\n741, arbekacin, 0, NCC[C@H](O)C(=O)N[C@@H]1C[C@H](N)[C@@H](O[C@H]2O[C@H](CN)CC[C@H]2N)[C@H](O)[C@H]1O[C@H]3O[C@H](CO)[C@@H](O)[C@H](N)[C@H]3O\\n742, aspoxicillin, 0, CNC(=O)C[C@@H](N)C(=O)N[C@@H](C(=O)N[C@H]1[C@H]2SC(C)(C)[C@@H](N2C1=O)C(O)=O)c3ccc(O)cc3\\n743, astromicin, 0, CO[C@H]1[C@@H](O)[C@H](N)[C@@H](O[C@H]2O[C@@H](CC[C@H]2N)C(C)N)[C@H](O)[C@@H]1N(C)C(=O)CN\\n744, aztreonam, 0, C[C@H]1[C@H](NC(=O)C(=N/OC(C)(C)C(O)=O)\\\\c2csc([NH3+])n2)C(=O)N1[S]([O-])(=O)=O\\n745, bacmecillinam, 0, CCOC(=O)OC(C)OC(=O)[C@@H]1N2[C@H](SC1(C)C)[C@H](N=CN3CCCCCC3)C2=O\\n746, bekanamycin, 0, NCC1OC(OC2C(N)CC(N)C(OC3OC(CO)C(O)C(N)C3O)C2O)C(N)C(O)C1O\\n747, benserazide, 0, NC(CO)C(=O)NNCc1ccc(O)c(O)c1O\\n748, bepridil, 0, CC(C)COCC(CN(Cc1ccccc1)c2ccccc2)N3CCCC3\\n749, berythromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](OC2CC(C)(OC)C(O)C(C)O2)C(C)[C@@H](OC3OC(C)CC(C3O)N(C)C)[C@](C)(O)C[C@@H](C)C(=O)[C@H](C)[C@@H](O)[C@H]1C\\n750, butikacin, 0, NCCC(O)CNC1CC(N)C(OC2OC(CN)C(O)C(O)C2O)C(O)C1OC3OC(CO)C(O)C(N)C3O\\n751, carbuterol, 0, CC(C)(C)NCC(O)c1ccc(O)c(NC(N)=O)c1\\n752, carfecillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C(C(=O)Oc3ccccc3)c4ccccc4)C(=O)N2[C@H]1C(O)=O\\n753, carindacillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C(C(=O)Oc3ccc4CCCc4c3)c5ccccc5)C(=O)N2[C@H]1C(O)=O\\n754, carubicin, 0, CC1OC(CC(N)C1O)OC2C[C@@](O)(Cc3c(O)c4C(=O)c5cccc(O)c5C(=O)c4c(O)c23)C(C)=O\\n755, carumonam, 0, NC(=O)OCC1[C@H](NC(=O)C(=N/OCC(O)=O)\\\\c2csc(N)n2)C(=O)N1[S](O)(=O)=O\\n756, cefaclor, 0, N[C@@H](C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)Cl)c3ccccc3\\n757, cefadroxil, 0, CC1=C(N2[C@H](SC1)[C@H](NC(=O)[C@H](N)c3ccc(O)cc3)C2=O)C(O)=O\\n758, cefalexin, 0, CC1=C(N2[C@H](SC1)[C@H](NC(=O)[C@H](N)c3ccccc3)C2=O)C(O)=O\\n759, cefaloram, 0, CC(=O)OCC1=C(N2[C@H](SC1)[C@H](NC(=O)Cc3ccccc3)C2=O)C(O)=O\\n760, cefaparole, 0, Cc1sc(SCC2=C(N3[C@H](SC2)[C@H](NC(=O)C(N)c4ccc(O)cc4)C3=O)C(O)=O)nn1\\n761, cefatrizine, 0, NC(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)CSc3cn[nH]n3)c4ccc(O)cc4\\n762, cefazaflur, 0, [Na+].Cn1nnnc1SCC2=C(N3[C@H](SC2)[C@H](NC(=O)CSC(F)(F)F)C3=O)C([O-])=O\\n763, cefazedone, 0, Cc1sc(SCC2=C(N3[C@H](SC2)[C@H](NC(=O)CN4C=C(Cl)C(=O)C(=C4)Cl)C3=O)C(O)=O)nn1\\n764, cefcanel, 0, Cc1sc(SCC2=C(N3[C@H](SC2)[C@H](NC(=O)[C@H](O)c4ccccc4)C3=O)C(O)=O)nn1\\n765, cefcanel-daloxate, 0, CC(N)C(=O)OC(C(=O)NC1C2SCC(=C(N2C1=O)C(=O)OCC3=C(C)OC(=O)O3)CSc4sc(C)nn4)c5ccccc5\\n766, cefcapene, 0, CC/C=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)COC(N)=O)/c3csc(N)n3\\n767, cefdaloxime, 0, COCC1=C(N2[C@H](SC1)[C@H](NC(=O)C(=N\\\\O)/c3csc(N)n3)C2=O)C(O)=O\\n768, cefdinir, 0, Nc1scc(n1)\\\\C(=N\\\\O)C(=O)N[C@H]2[C@H]3SCC(=C(N3C2=O)C(O)=O)C=C\\n769, cefditoren, 0, CO\\\\N=C(C(=O)NC1[C@H]2SCC(=C(N2C1=O)C(O)=O)\\\\C=C/c3scnc3C)\\\\c4csc(N)n4\\n770, cefedrolor, 0, CC1=C(N2[C@H](SC1)[C@H](NC(=O)[C@H](N)c3ccc(O)c(Cl)c3)C2=O)C(O)=O\\n771, cefetamet, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)C)/c3csc(N)n3\\n772, cefetrizole, 0, OC(=O)C1=C(CS[C@@H]2[C@H](NC(=O)Cc3sccc3)C(=O)N12)CSc4[nH]ncn4\\n773, cefivitril, 0, Cn1nnnc1SCC2=C(N3[C@H](SC2)[C@H](NC(=O)CS\\\\C=C/C#N)C3=O)C(O)=O\\n774, cefixime, 0, Nc1scc(n1)\\\\C(=N\\\\OCC(O)=O)C(=O)N[C@H]2[C@H]3SCC(=C(N3C2=O)C(O)=O)C=C\\n775, cefmenoxime, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)CSc3nnnn3C)/c4csc(N)n4\\n776, cefmetazole, 0, CO[C@]1(NC(=O)CSCC#N)[C@H]2SCC(=C(N2C1=O)C(O)=O)CSc3nnnn3C\\n777, cefminox, 0, CO[C@]1(NC(=O)CSC[C@@H](N)C(O)=O)[C@H]2SCC(=C(N2C1=O)C(O)=O)CSc3nnnn3C\\n778, cefodizime, 0, CO\\\\N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)CSc3sc(CC(O)=O)c(C)n3)/c4csc(N)n4\\n779, cefonicid, 0, OC(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)CSc3nnnn3C[S](O)(=O)=O)c4ccccc4\\n780, ceforanide, 0, NCc1ccccc1CC(=O)N[C@H]2[C@H]3SCC(=C(N3C2=O)C(O)=O)CSc4nnnn4CC(O)=O\\n781, cefotetan, 0, CO[C@]1(NC(=O)C2S\\\\C(S2)=C(\\\\C(N)=O)C(O)=O)[C@H]3SCC(=C(N3C1=O)C(O)=O)CSc4nnnn4C\\n782, cefotiam, 0, CN(C)CCn1nnnc1SCC2=C(N3[C@H](SC2)[C@H](NC(=O)Cc4csc(N)n4)C3=O)C(O)=O\\n783, cefoxazole, 0, CC(=O)OCC1=C(N2[C@H](SC1)[C@H](NC(=O)c3c(C)onc3c4ccccc4Cl)C2=O)C(O)=O\\n784, cefpiramide, 0, Cn1nnnc1SCC2=C(N3[C@H](SC2)C(NC(=O)[C@H](NC(=O)C4=CNC(=CC4=O)C)c5ccc(O)cc5)C3=O)C(O)=O\\n785, cefpodoxime, 0, [Na+].COCC1=C(N2[C@H](SC1)[C@H](NC(=O)C(=N/OC)\\\\c3csc(N)n3)C2=O)C([O-])=O\\n786, cefprozil, 0, O.C/C=C/C1=C(N2[C@H](SC1)[C@H](NC(=O)[C@H](N)c3ccc(O)cc3)C2=O)C(O)=O\\n787, cefrotil, 0, CC1=C(N2[C@H](SC1)[C@H](NC(=O)Cc3ccc(cc3)C4=NCCCN4)C2=O)C(O)=O\\n788, cefroxadine, 0, COC1=C(N3C(SC1)C(NC(=O)C(N)C2C=CCC=C2)C3=O)C(O)=O\\n789, cefsumide, 0, CC1=C(N2[C@H](SC1)[C@H](NC(=O)[C@H](N)c3cccc(N[S](C)(=O)=O)c3)C2=O)C(O)=O\\n790, cefteram, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)Cn3nnc(C)n3)/c4csc(N)n4\\n791, ceftezole, 0, OC(=O)C1=C(CS[C@@H]2[C@H](NC(=O)Cn3cnnn3)C(=O)N12)CSc4scnn4\\n792, ceftibuten, 0, O.O.Nc1scc(n1)\\\\C(=C\\\\CC(O)=O)C(=O)N[C@H]2[C@H]3SCC=C(N3C2=O)C(O)=O\\n793, ceftiofur, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)CSC(=O)c3occc3)/c4csc(N)n4\\n794, ceftiolene, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)\\\\C=C\\\\SC3=NNC(=O)C(=O)N3CC=O)/c4csc(N)n4\\n795, ceftioxide, 0, CO/N=C(C(=O)N[C@H]1[C@@H]2N(C1=O)C(=C(COC(C)=O)C[S]2=O)C(O)=O)/c3csc(N)n3\\n796, ceftizoxime, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC=C(N2C1=O)C(O)=O)/c3csc(N)n3\\n797, cefuracetime, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)COC(C)=O)/c3occc3\\n798, cefuroxime-axetil, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(=O)OC(C)OC(C)=O)COC(N)=O)/c3occc3\\n799, cefuroxime-pivoxetil, 0, COCCCC(=O)OC(C)OC(=O)C1=C(COC(N)=O)CS[C@@H]2[C@H](NC(=O)C(=N/OC)\\\\c3occc3)C(=O)N12\\n800, cefuzonam, 0, CO/N=C(C(=O)N[C@H]1[C@H]2SCC(=C(N2C1=O)C(O)=O)CSc3snnc3)/c4csc(N)n4\\n801, cephacetrile, 0, [Na+].CC(=O)OCC1=C(N2[C@@H](SC1)[C@@H](NC(=O)CC#N)C2=O)C([O-])=O\\n802, cephaloglycin, 0, O.O.CC(=O)OCC1=C(N2[C@H](SC1)[C@H](NC(=O)[C@H](N)c3ccccc3)C2=O)C(O)=O\\n803, cephalosporin-c, 0, CC(=O)OCC1=C(N2[C@H](SC1)[C@H](NC(=O)CCC[C@@H](N)C(O)=O)C2=O)C(O)=O\\n804, cephalothin, 0, CC(=O)OCC1=C(N2[C@H](SC1)[C@H](NC(=O)Cc3sccc3)C2=O)C(O)=O\\n805, cetocycline, 0, CC(=O)C1=C(O)[C@]2(O)[C@@H](Cc3c(C)c4ccc(C)c(O)c4c(O)c3C2=O)[C@@H](N)C1=O\\n806, clarithromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(OC)[C@@H](O)[C@H](C)O2)[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)[C@@](C)(C[C@@H](C)C(=O)[C@H](C)[C@@H](O)[C@]1(C)O)OC\\n807, clindamycin, 0, CCC[C@@H]1C[C@H](N(C)C1)C(=O)NC(C(C)Cl)[C@H]2O[C@H](SC)[C@H](O)[C@@H](O)[C@H]2O\\n808, clindamycin-phosphate, 0, CCC[C@@H]1C[C@H](N(C)C1)C(=O)NC(C(C)Cl)[C@H]2O[C@H](SC)[C@H](O[P](O)(O)=O)[C@@H](O)[C@H]2O\\n809, clometocillin, 0, COC(C(=O)N[C@H]1[C@H]2SC(C)(C)[C@@H](N2C1=O)C(O)=O)c3ccc(Cl)c(Cl)c3\\n810, clomocycline, 0, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)ccc(Cl)c4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(/O)NCO)C1=O\\n811, cyclacillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C3(N)CCCCC3)C(=O)N2[C@H]1C(O)=O\\n812, cyclosporina, 0, CCC1C(=O)N(CC(=O)N(C(C(=O)NC(C(=O)N(C(C(=O)NC(C(=O)NC(C(=O)N(C(C(=O)N(C(C(=O)N(C(C(=O)N(C(C(=O)N1)C(C(C)CC=CC)O)C)C(C)C)C)CC(C)C)C)CC(C)C)C)C)C)CC(C)C)C)C(C)C)CC(C)C)C)C\\n813, dactinomycin, 0, CC(C)C1NC(=O)C(NC(=O)c2ccc(C)c3OC4=C(C)C(=O)C(=C(C(=O)NC5C(C)OC(=O)C(C(C)C)N(C)C(=O)CN(C)C(=O)C6CCCN6C(=O)C(NC5=O)C(C)C)C4=Nc23)N)C(C)OC(=O)C(C(C)C)N(C)C(=O)CN(C)C(=O)C7CCCN7C1=O\\n814, demecycline, 0, CN(C)[C@H]1[C@@H]2C[C@@H]3[C@H](O)c4cccc(O)c4C(=C3C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O)O\\n815, detorubicin, 0, CCOC(OCC)C(=O)OCC(=O)C1(O)CC(OC2CC(N)C(O)C(C)O2)c3c(O)c4C(=O)c5c(OC)cccc5C(=O)c4c(O)c3C1\\n816, dibekacin, 0, NCC1CCC(N)C(O1)OC2C(N)CC(N)C(OC3OC(CO)C(O)C(N)C3O)C2O\\n817, dicloxacillin, 0, Cc1onc(c1C(=O)N[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O)c4c(Cl)cccc4Cl\\n818, dihydrostreptomycin, 0, CN[C@H]1[C@H](O)[C@@H](O)[C@H](CO)O[C@H]1O[C@H]2[C@@H](O[C@@H](C)[C@]2(O)CO)O[C@@H]3[C@@H](O)[C@H](O)[C@@H](N=C(N)N)[C@H](O)[C@H]3N=C(N)N\\n819, diosmin, 0, COc1ccc(cc1O)C2=CC(=O)c3c(O)cc(O[C@@H]4O[C@H](CO[C@@H]5O[C@@H](C)[C@H](O)[C@@H](O)[C@H]5O)[C@@H](O)[C@H](O)[C@H]4O)cc3O2\\n820, dipivefrin, 0, CNCC(O)c1ccc(OC(=O)C(C)(C)C)c(OC(=O)C(C)(C)C)c1\\n821, diproleandomycin, 0, CCC(=O)O[C@H]1[C@@H](C)[C@@H](C)OC(=O)C(C)[C@@H](O[C@H]2C[C@H](OC)[C@@H](OC(=O)CC)[C@H](C)O2)[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)C(C)C[C@@]4(CO4)C(=O)[C@@H]1C\\n822, dipyridamole, 0, OCCN(CCO)c1nc(N2CCCCC2)c3nc(nc(N4CCCCC4)c3n1)N(CCO)CCO\\n823, dirithromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(OC)[C@@H](O)[C@H](C)O2)[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)[C@](C)(O)C[C@@H](C)[C@@H]4N[C@@H](COCCOC)O[C@H]([C@H]4C)[C@]1(C)O\\n824, docetaxel, 0, CC(=O)O[C@@]12CO[C@@H]1C[C@H](O)[C@]3(C)[C@@H]2[C@H](OC(=O)c4ccccc4)[C@]5(O)C[C@H](OC(=O)[C@H](O)[C@@H](NC(=O)OC(C)(C)C)c6ccccc6)C(=C([C@@H](O)C3=O)C5(C)C)C\\n825, efrotomycin, 0, CCC(C(=O)NC\\\\C=C\\\\C=C(/C)C(OC)C(C)C1OC(\\\\C=C\\\\C=C\\\\C=C(/C)C(=O)C2=C(O)N(C)C=CC2=O)C(O)C1O)C3(O)OC(\\\\C=C\\\\C=C\\\\C)C(C)(C)C(OC4OC(C)C(OC5OC(C)C(OC)C(O)C5OC)C(OC)C4O)C3O\\n826, emetine, 0, [Cl].CC[C@H]1CN2CCc3cc(OC)c(OC)cc3[C@@H]2C[C@@H]1C[C@H]4NCCc5cc(OC)c(OC)cc45\\n827, epicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](N)C3=CCC=CC3)C(=O)N2[C@H]1C(O)=O\\n828, eqvalan, 0, CCC(C)[C@H]1O[C@]2(CC[C@@H]1C)CC3C[C@@H](C\\\\C=C(C)\\\\[C@@H](O[C@H]4C[C@H](OC)[C@@H](OC5C[C@H](OC)[C@@H](O)[C@H](C)O5)[C@H](C)O4)[C@@H](C)/C=C/C=C/6CO[C@@H]7[C@H](O)C(=C[C@@H](C(=O)O3)[C@]67O)C)O2.CO[C@H]8CC(O[C@@H](C)[C@@H]8O)O[C@H]9[C@H](C)O[C@H](C[C@@H]9OC)O[C@H]/%10[C@@H](C)/C=C/C=C/%11CO[C@@H]%12[C@H](O)C(=C[C@@H](C(=O)OC%13C[C@@H](C\\\\C=C%10C)O[C@@]%14(CC[C@H](C)[C@H](O%14)C(C)C)C%13)[C@]%11%12O)C\\n829, esorubicin, 0, COc1cccc2C(=O)c3c(O)c4C[C@@](O)(C[C@@H](OC5CC(N)CC(C)O5)c4c(O)c3C(=O)c12)C(=O)CO\\n830, etamocycline, 0, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)cccc4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(/O)NCN(C)CCN(C)CNC(/O)=C/5C(=O)[C@H]([C@@H]6C[C@H]7C(=C(O)c8c(O)cccc8[C@@]7(C)O)C(=O)[C@]6(O)C5=O)N(C)C)C1=O\\n831, etilefrine, 0, CCNCC(O)c1cccc(O)c1\\n832, felodipine, 0, CCOC(=O)C1=C(C)NC(=C(C1c2cccc(Cl)c2Cl)C(=O)OC)C\\n833, fenbenicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C(Oc3ccccc3)c4ccccc4)C(=O)N2[C@H]1C(O)=O\\n834, fenoxazoline, 0, CC(C)c1ccccc1OCC2=NCCN2\\n835, fexofenadine, 0, CC(C)(C(O)=O)c1ccc(cc1)C(O)CCCN2CCC(CC2)C(O)(c3ccccc3)c4ccccc4\\n836, fibracillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](NC(=O)C(C)(C)Oc3ccc(Cl)cc3)c4ccccc4)C(=O)N2[C@H]1C(O)=O\\n837, flomoxef, 0, [Na].CO[C@]1(NC(=O)CSC(F)(F)F)[C@H]2OCC(=C(N2C1=O)C(O)=O)CSc3nnnn3CCO\\n838, flurithromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(OC)[C@@H](O)[C@H](C)O2)[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)[C@](C)(O)C[C@](C)(F)C(=O)[C@H](C)[C@@H](O)[C@]1(C)O\\n839, fropenem, 0, C[C@@H](O)[C@@H]1[C@H]2SC(=C(N2C1=O)C(O)=O)[C@H]3CCCO3\\n840, fumoxicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@@H](N=Cc3occc3)c4ccc(O)cc4)C(=O)N2[C@H]1C(O)=O\\n841, furbucillin, 0, CC(C)CC(OC(=O)c1occc1)C(=O)N[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O\\n842, fusidic-acid, 0, C[C@@H]1[C@H](O)CC[C@@]2(C)[C@H]1CC[C@@]3(C)[C@H]2[C@H](O)C[C@H]4\\\\C([C@H](C[C@]34C)OC(C)=O)=C(/CCC=C(C)C)C(O)=O\\n843, gentamicin, 0, CNC(C)C1CCC(N)C(O1)OC2C(N)CC(N)C(OC3OCC(C)(O)C(NC)C3O)C2O\\n844, gloximonam, 0, CO/N=C(C(=O)N[C@H]1[C@H](C)N(OCC(=O)OCC(=O)OC(C)(C)C)C1=O)/c2csc(N)n2\\n845, gramicidin-s-2, 0, CC(C)CC1C(=O)NC(C(=O)N2CCCC2C(=O)NC(C(=O)NC(C(=O)NC(C(=O)NC(C(=O)N3CCCC3C(=O)NC(C(=O)NC(C(=O)N1)CCCN)C(C)C)CC4=CC=CC=C4)CC(C)C)CCCN)C(C)C)CC5=CC=CC=C5\\n846, grepafloxacin, 0, CC1CN(CCN1)c2cc3N(C=C(C(O)=O)C(=O)c3c(C)c2F)C4CC4\\n847, guamecycline, 0, CN(C)C1C2CC3C(=C(O)c4c(O)cccc4C3(C)O)C(=O)C2(O)C(=O)\\\\C(=C(/O)NCN5CCN(CC5)C(=N)N=C(N)N)C1=O\\n848, hetacillin, 0, CC1(C)N[C@@H](C(=O)N1[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O)c4ccccc4\\n849, hydroxyamphetamine, 0, CC(Cc1ccccc1)NO\\n850, hydroxyephedrine, 0, CNC(C)C(O)c1ccc(O)cc1\\n851, hymecromone, 0, CC1=CC(=O)Oc2cc(O)ccc12\\n852, idarubicin, 0, C[C@@H]1O[C@H](C[C@H](N)[C@@H]1O)O[C@H]2C[C@@](O)(Cc3c(O)c4C(=O)c5ccccc5C(=O)c4c(O)c23)C(C)=O\\n853, imipenem, 0, O.C[C@H](O)[C@@H]1[C@H]2CC(=C(N2C1=O)C(O)=O)SCCN=CN\\n854, isepamicin, 0, CNC1C(O)C(OCC1(C)O)OC2C(O)C(OC3OC(CN)C(O)C(O)C3O)C(N)CC2NC(=O)C(O)CN\\n855, isopropicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C(C)(C)Oc3ccccc3)C(=O)N2[C@H]1C(O)=O\\n856, josamycin, 0, CO[C@H]1[C@@H](CC(=O)O[C@H](C)C/C=C/C=C/[C@H](O)[C@H](C)C[C@H](CC=O)[C@@H]1O[C@@H]2O[C@H](C)[C@@H](O[C@H]3C[C@@](C)(O)[C@@H](OC(=O)CC(C)C)[C@H](C)O3)[C@@H]([C@H]2O)N(C)C)OC(C)=O\\n857, lenampicillin, 0, CC1=C(COC(=O)[C@@H]2N3[C@H](SC2(C)C)[C@H](NC(=O)[C@H](N)c4ccccc4)C3=O)OC(=O)O1\\n858, lenapenem, 0, CNCC[C@@H](O)[C@@H]1C[C@@H](CN1)SC2=C(N3[C@H]([C@H]2C)[C@@H]([C@@H](C)O)C3=O)C(O)=O\\n859, leurubicin, 0, COc1cccc2C(=O)c3c(O)c4C[C@](O)(C[C@H](O[C@H]5C[C@H](NC(=O)[C@@H](N)CC(C)C)[C@H](O)[C@H](C)O5)c4c(O)c3C(=O)c12)C(=O)CO\\n860, lexithromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](OC2CC(C)(OC)C(O)C(C)O2)C(C)[C@@H](OC3OC(C)CC(C3O)N(C)C)[C@](C)(O)C[C@@H](C)\\\\C(=N/OC)[C@H](C)[C@@H](O)[C@]1(C)O\\n861, lincomycin, 0, CCCC1CC(N(C)C1)C(=O)NC(C(C)O)C2OC(SC)C(O)C(O)C2O\\n862, loracarbef, 0, O.N[C@@H](C(=O)NC1C2CCC(=C(N2C1=O)C(O)=O)Cl)c3ccccc3\\n863, losartan, 0, CCCCc1nc(Cl)c(CO)n1Cc2ccc(cc2)c3ccccc3c4n[nH]nn4\\n864, lymecycline, 0, [Cl-].CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)ccc(Cl)c4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(N)\\\\O)C1=O.[H+]\\n865, mecillinam, 0, CC1(C)S[C@@H]2[C@H](N=CN3CCCCCC3)C(=O)N2[C@H]1C(O)=O\\n866, mecloxamine, 0, CC(CN(C)C)OC(C)(c1ccccc1)c2ccc(Cl)cc2\\n867, medorubicin, 0, CC1OC(CC(N)C1O)OC2CC(O)(Cc3c(O)c4C(=O)c5ccccc5C(=O)c4c(O)c23)C(=O)CO\\n868, meropenem, 0, O.O.O.C[C@H](O)C1C2[C@@H](C)C(=C(N2C1=O)C(O)=O)S[C@@H]3CN[C@@H](C3)C(=O)N(C)C\\n869, metampicillin, 0, CC1(C)SC2C(NC(=O)C(N=C)c3ccccc3)C(=O)N2C1C(O)=O\\n870, methacycline, 0, CN(C)[C@H]1[C@@H]2[C@@H](O)[C@@H]3C(=C)c4cccc(O)c4C(=C3C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O)O\\n871, methoxyphedrine, 0, CNC(C)C(=O)c1ccc(OC)cc1\\n872, meticillin, 0, COc1cccc(OC)c1C(=O)N[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O\\n873, mezlocillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](NC(=O)N3CCN(C3=O)[S](C)(=O)=O)c4ccccc4)C(=O)N2[C@H]1C(O)=O\\n874, mibefradil, 0, COCC(=O)O[C@]1(CCN(C)CCCc2[nH]c3ccccc3n2)CCc4cc(F)ccc4[C@@H]1C(C)C\\n875, midecamycin, 0, CCC(=O)OC1CC(=O)OC(C)C\\\\C=C\\\\C=C/C(O)C(C)CC(CC=O)C(O[C@@H]2O[C@H](C)[C@@H](O[C@H]3C[C@@](C)(O)[C@H](OC(=O)CC)[C@@H](C)O3)[C@@H]([C@H]2O)N(C)C)C1OC\\n876, mikamycin, 0, CCC1NC(=O)C(NC(=O)c2ncccc2O)C(C)OC(=O)C(NC(=O)C3CC(=O)CCN3C(=O)C(Cc4ccc(cc4)N(C)C)N(C)C(=O)C5CCCN5C1=O)c6ccccc6\\n877, mirincamycin, 0, CCCCCC1CNC(C1)C(=O)NC(C(C)Cl)C2OC(SC)C(O)C(O)C2O\\n878, mitomycin, 0, CO[C@@]12[C@H](COC(N)=O)C3=C(N1C[C@@H]4N[C@H]24)C(=O)C(=C(N)C3=O)C\\n879, moxalactam, 0, CO[C@]1(NC(=O)C(C(O)=O)c2ccc(O)cc2)[C@H]3OCC(=C(N3C1=O)C(O)=O)CSc4nnnn4C\\n880, neamine, 0, NC[C@H]1O[C@H](O[C@@H]2[C@@H](N)C[C@@H](N)[C@H](O)[C@H]2O)[C@H](N)[C@@H](O)[C@@H]1O\\n881, netilmicin, 0, CCN[C@@H]1C[C@H](N)[C@@H](O[C@H]2OC(=CC[C@H]2N)CN)[C@H](O)[C@H]1O[C@H]3OC[C@](C)(O)[C@H](NC)[C@H]3O\\n882, nicardipine, 0, COC(=O)C1=C(C)NC(=C(C1c2cccc(c2)[N+]([O-])=O)C(=O)OCCN(C)Cc3ccccc3)C\\n883, nifedipine, 0, COC(=O)C1=C(C)NC(=C(C1c2ccccc2[N+]([O-])=O)C(=O)OC)C\\n884, nitrendipine, 0, CCOC(=O)C1=C(C)NC(=C(C1c2cccc(c2)[N+]([O-])=O)C(=O)OC)C\\n885, nitrocycline, 0, CN(C)[C@H]1[C@@H]2C[C@@H]3Cc4c(ccc(O)c4C(=C3C(=O)[C@]2(O)C(=O)\\\\C(=C(N)/O)C1=O)O)[N+]([O-])=O\\n886, novobiocin, 0, CO[C@@H]1[C@@H](OC(N)=O)[C@@H](O)[C@H](Oc2ccc3C(=O)C(=C(O)Oc3c2C)NC(=O)c4ccc(O)c(CC=C(C)C)c4)OC1(C)C\\n887, oleandomycin, 0, COC1CC(OC(C)C1O)O[C@H]2[C@H](C)[C@@H](OC3OC(C)CC(C3O)N(C)C)[C@@H](C)C[C@]4(CO4)C(=O)[C@H](C)[C@@H](O)[C@@H](C)[C@@H](C)OC(=O)[C@@H]2C\\n888, oxetacillin, 0, CC1(C)NC(C(=O)[NH+]1[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O)c4ccc(O)cc4\\n889, oximonam, 0, CO\\\\N=C(C(=O)NC1[C@H](C)N(OCC(O)=O)C1=O)\\\\c2csc(N)n2\\n890, paclitaxel, 0, CC(=O)O[C@H]1C(=O)[C@]2(C)[C@@H](O)C[C@H]3OC[C@@]3(OC(C)=O)C2[C@H](OC(=O)c4ccccc4)[C@]5(O)C[C@H](OC(=O)[C@H](O)[C@@H](NC(=O)c6ccccc6)c7ccccc7)C(=C1C5(C)C)C\\n891, pecocycline, 0, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)cccc4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(/O)NCN5CCCC(C5)C(O)=O)C1=O\\n892, penamecillin, 0, CC(=O)OCOC(=O)C1N2C(SC1(C)C)C(NC(=O)Cc3ccccc3)C2=O\\n893, penicillin, 0, [Na].CC1(C)SC2C(NC(=O)Cc3ccccc3)C(=O)N2C1C(O)=O\\n894, penimocycline, 0, CN(C)[C@H]1[C@@H]2C[C@H]3C(=C(O)c4c(O)cccc4[C@@]3(C)O)C(=O)[C@]2(O)C(=O)\\\\C(=C(/O)NCNC(C(=O)NC5C6SC(C)(C)C(N6C5=O)C(O)=O)c7ccccc7)C1=O\\n895, phenethicillin, 0, CC(Oc1ccccc1)C(=O)NC2C3SC(C)(C)C(N3C2=O)C(O)=O\\n896, pipacycline, 0, CN(C)C1C2CC3C(=C(O)c4c(O)cccc4C3(C)O)C(=O)C2(O)C(=O)\\\\C(=C(/O)NCN5CCN(CCO)CC5)C1=O\\n897, pirarubicin, 0, COc1cccc2C(=O)c3c(O)c4C[C@](O)(C[C@H](O[C@H]5C[C@H](N)[C@H](O[C@H]6CCCCO6)[C@H](C)O5)c4c(O)c3C(=O)c12)C(=O)CO\\n898, pirbenicillin, 0, CC1(C)SC2C(N([C@H](C(=O)NC(=O)CN)c3ccccc3)C(=N)c4ccncc4)C(=O)N2C1C(O)=O\\n899, piridicillin, 0, [Na+].CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](NC(=O)C3=CC=C(NC3=O)c4ccc(cc4)[S](=O)(=O)N(CCO)CCO)c5ccc(O)cc5)C(=O)N2[C@H]1C([O-])=O\\n900, piroxicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](NC(=O)NC3=CN=C(NC3=O)Nc4ccc(cc4)[S](N)(=O)=O)c5ccc(O)cc5)C(=O)N2[C@H]1C(O)=O\\n901, pivampicillin, 0, CC(C)(C)C(=O)OCOC(=O)[C@@H]1N2[C@H](SC1(C)C)[C@H](NC(=O)[C@H](N)c3ccccc3)C2=O\\n902, pivenfrine, 0, CNCC(O)c1cccc(OC(=O)C(C)(C)C)c1\\n903, prazocillin, 0, Cc1cnn(c1C(=O)N[C@H]2[C@H]3SC(C)(C)[C@@H](N3C2=O)C(O)=O)c4c(Cl)cccc4Cl\\n904, pristinamycin-ic, 0, CC1OC(=O)C(NC(=O)C2CC(=O)CCN2C(=O)C(Cc3ccc(cc3)N(C)C)N(C)C(=O)C4CCCN4C(=O)C(C)NC(=O)C1NC(=O)c5ncccc5O)c6ccccc6\\n905, propafenone, 0, CCCNCC(O)COc1ccccc1C(=O)CCc2ccccc2\\n906, propicillin, 0, CCC(Oc1ccccc1)C(=O)NC2C3SC(C)(C)C(N3C2=O)C(O)=O\\n907, propikacin, 0, NCC1OC(OC2C(N)CC(NC(CO)CO)C(OC3OC(CO)C(O)C(N)C3O)C2O)C(N)C(O)C1O\\n908, quinacillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)c3nc4ccccc4nc3C(O)=O)C(=O)N2[C@H]1C(O)=O\\n909, rifabutin, 0, CO[C@H]1/C=C/O[C@@]2(C)Oc3c(C)c(O)c4C(=O)C(=C5NC6(CCN(CC6)CC(C)C)N=C5c4c3C2=O)NC(=O)C(=C\\\\C=C\\\\[C@H](C)[C@H](O)[C@@H](C)[C@@H](O)[C@@H](C)[C@H](OC(C)=O)[C@@H]1C)/C\\n910, rifametane, 0, CCN(CC)C(C)=NN=Cc1c(O)c2c3C(=O)C4(C)OC=CC(OC)C(C)C(OC(C)=O)C(C)C(O)C(C)C(O)C(C)C=CC=C(C)C(=O)Nc1c(O)c2c(O)c(C)c3O4\\n911, rifamexil, 0, CCN(CC)c1sc2c3NC(=O)C(=C\\\\C=C\\\\[C@H](C)[C@H](O)[C@@H](C)[C@@H](O)[C@@H](C)[C@H](OC(C)=O)[C@H](C)[C@@H](OC)/C=C/O[C@@]4(C)Oc5c(C)c(O)c(c3O)c(c2n1)c5C4=O)/C\\n912, rifamide, 0, CCN(CC)C(=O)COc1cc2NC(=O)C(=C\\\\C=C\\\\[C@H](C)[C@H](O)[C@@H](C)[C@@H](O)[C@@H](C)[C@H](OC(C)=O)[C@H](C)[C@@H](OC)/C=C/O[C@@]3(C)Oc4c(C)c(O)c(c2O)c1c4C3=O)/C\\n913, rifampin, 0, COC1C=COC5(C)Oc4c(C)c(O)c3c(O)c(NC(=O)C(=CC=CC(C)C(O)C(C)C(O)C(C)C(OC(C)=O)C1C)C)c(C=NN2CCN(C)CC2)c(O)c3c4C5=O\\n914, rifamycin, 0, CO[C@H]1/C=C/O[C@@]2(C)Oc3c(C)c(O)c4c(O)c(NC(=O)C(=C\\\\C=C\\\\[C@H](C)[C@H](O)[C@@H](C)[C@@H](O)[C@@H](C)[C@H](OC(C)=O)[C@@H]1C)/C)cc(O)c4c3C2=O\\n915, rifapentine, 0, CO[C@H]1\\\\C=C\\\\O[C@@]2(C)Oc3c(C)c(O)c4C(=C(NC(=O)C(=C/C=C/[C@H](C)[C@H](O)[C@@H](C)[C@@H](O)[C@@H](C)[C@H](OC(C)=O)[C@@H]1C)\\\\C)C(=C/NN5CCN(CC5)C6CCCC6)\\\\C(=O)c4c3C2=O)O\\n916, rifaximin, 0, CO[C@H]1/C=C/O[C@@]2(C)Oc3c(C)c(O)c4c(O)c(NC(=O)C(=C\\\\C=C\\\\[C@H](C)[C@H](O)[C@@H](C)[C@@H](O)[C@@H](C)[C@H](OC(C)=O)[C@@H]1C)/C)c5n6ccc(C)cc6nc5c4c3C2=O\\n917, ritipenem, 0, [Na+].C[C@@H](O)[C@@H]1[C@H]2SC(=C(N2C1=O)C([O-])=O)COC(N)=O\\n918, rodorubicin, 0, CC[C@@]1(O)C[C@H](O[C@H]2C[C@@H]([C@H](O[C@H]3C[C@@H]4O[C@H]5CC(=O)[C@H](C)O[C@H]5O[C@@H]4[C@H](C)O3)[C@H](C)O2)N(C)C)c6c(O)c7C(=O)c8c(O)cccc8C(=O)c7c(O)c6[C@H]1O[C@H]9C[C@@H]([C@H](O)[C@H](C)O9)N(C)C\\n919, rotamicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](NC(=O)Cc3ccc(cc3)C4=NCCCN4)c5ccccc5)C(=O)N2[C@H]1C(O)=O\\n920, roxithromycin, 0, CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(OC)[C@@H](O)[C@H](C)O2)[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)[C@](C)(O)C[C@@H](C)\\\\C(=N\\\\OCOCCOC)[C@H](C)[C@@H](O)[C@]1(C)O\\n921, sanfetrinem, 0, COC1CCCC2C3C(C(C)O)C(=O)N3C(=C12)C([O-])=O\\n922, sarmoxicillin, 0, COCOC(=O)C1N2C(SC1(C)C)C(N3C(=O)C(NC3(C)C)c4ccc(O)cc4)C2=O\\n923, sarpicillin, 0, COCOC(=O)[C@@H]1N2[C@H](SC1(C)C)[C@H](N3C(=O)C(NC3(C)C)c4ccccc4)C2=O\\n924, sedecamycin, 0, CC1C2CC(OC(C)=O)\\\\C=C\\\\C(=C\\\\CC(O)/C=C/C(=C/C(NC(=O)C(C)=O)C(C)(C(=O)O2)C1=O)C)C\\n925, sisomicin, 0, CN[C@@H]1[C@@H](O)[C@H](OC[C@]1(C)O)O[C@H]2[C@H](N)C[C@H](N)[C@@H](O[C@H]3OC(=CC[C@H]3N)CN)[C@@H]2O\\n926, spiramycin, 0, COC1C(O)CC(=O)OC(C)C\\\\C=C\\\\C=C\\\\C(OC2CCC(C(C)O2)N(C)C)C(C)CC(CC=O)C1OC3OC(C)C(OC4CC(C)(O)C(O)C(C)O4)C(C3O)N(C)C\\n927, sulbenicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)C(c3ccccc3)[S](O)(=O)=O)C(=O)N2[C@H]1C(O)=O\\n928, sulopenem, 0, C[C@@H](O)[C@@H]1[C@H]2SC(=C(N2C1=O)C(O)=O)SC3CC[S](=O)C3\\n929, sultamicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](N)c3ccccc3)C(=O)N2[C@H]1C(O)=O.CC4(C)[C@@H](N5[C@@H](CC5=O)[S]4(=O)=O)C(O)=O\\n930, suncillin, 0, [Na+].[Na+].CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](N[S]([O-])(=O)=O)c3ccccc3)C(=O)N2[C@H]1C([O-])=O\\n931, tacrolimus, 0, CO[C@@H]1C[C@@H](CC[C@H]1O)/C=C(C)/[C@H]2OC(=O)[C@@H]3CCCCN3C(=O)C(=O)[C@]4(O)O[C@H]([C@H](C[C@@H](C)CC(=C/[C@@H](CC=C)C(=O)C[C@H](O)[C@H]2C)/C)OC)[C@H](C[C@H]4C)OC\\n932, talampicillin, 0, CC1(C)S[C@@H]2[C@H](NC(=O)[C@H](N)c3ccccc3)C(=O)N2[C@H]1C(=O)OC4OC(=O)c5ccccc45\\n933, tazobactam, 0, C[C@]1(Cn2ccnn2)[C@@H](N3[C@@H](CC3=O)[S]1(=O)=O)C(O)=O\\n934, teicoplanin, 0, CCCCCCCCCC(=O)N[C@@H]1[C@@H](O)[C@H](O)[C@@H](CO)O[C@H]1Oc2c3Oc4ccc(C[C@H]5NC(=O)[C@H](N)c6ccc(O)c(Oc7cc(O)cc(c7)[C@H](NC5=O)C(=O)N[C@H]8C(=O)N[C@H]9C(=O)N[C@@H]([C@H](O[C@@H]%10O[C@H](CO)[C@@H](O)[C@H](O)[C@H]%10NC(C)=O)c%11ccc(Oc2cc8c3)c(Cl)c%11)C(=O)N[C@@H](C(O)=O)c%12cc(O)cc(O[C@H]%13O[C@H](CO)[C@@H](O)[C@H](O)[C@@H]%13O)c%12c%14cc9ccc%14O)c6)cc4Cl\\n935, temocillin, 0, CO[C@]1(NC(=O)C(C(O)=O)c2cscc2)[C@H]3SC(C)(C)[C@@H](N3C1=O)C(O)=O\\n936, theodrenaline, 0, CN1C(=O)N(C)c2ncn(CCNCC(O)c3ccc(O)c(O)c3)c2C1=O\\n937, thiphencillin, 0, CC1(C)SC2C(NC(=O)CSc3ccccc3)C(=O)N2C1C(O)=O\\n938, tigemonam, 0, CC1(C)[C@H](NC(=O)C(=N\\\\OCC(O)=O)/c2csc(N)n2)C(=O)N1O[S](O)(=O)=O\\n939, tobicillin, 0, CC(C)C(=O)OCc1cccc(OC(=O)[C@@H]2N3[C@H](SC2(C)C)[C@H](NC(=O)Cc4ccccc4)C3=O)c1\\n940, tobramycin, 0, NC[C@H]1O[C@H](O[C@@H]2[C@@H](N)C[C@@H](N)[C@H](O[C@H]3O[C@H](CO)[C@@H](O)[C@H](N)[C@H]3O)[C@H]2O)[C@H](N)C[C@@H]1O\\n941, tolcapone, 0, Cc1ccc(cc1)C(=O)c2cc(O)c(O)c(c2)[N+]([O-])=O\\n942, troleandomycin, 0, CO[C@H]1C[C@@H](O[C@@H](C)[C@@H]1OC(C)=O)O[C@H]2[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3OC(C)=O)N(C)C)[C@@H](C)C[C@@]4(CO4)C(=O)[C@H](C)[C@@H](OC(C)=O)[C@@H](C)[C@@H](C)OC(=O)[C@@H]2C\\n943, trospectomycin, 0, CCCC[C@@H]1CC(=O)[C@]2(O)O[C@@H]3[C@@H](NC)[C@@H](O)[C@@H](NC)[C@H](O)[C@H]3O[C@@H]2O1\\n944, tymazoline, 0, CC(C)c1ccc(C)cc1OCC2=NCCN2\\n945, vindesine, 0, CC[C@]1(O)C[C@H]2CN(CCc3c([nH]c4ccccc34)[C@@](C2)(C(=O)OC)c5cc6c(cc5OC)N(C)[C@H]7[C@](O)([C@H](O)[C@]8(CC)C=CCN9CC[C@]67[C@H]89)C(N)=O)C1\\n946, virginiamycin-factor-m1, 0, CC(C)C1OC(=O)C2=CCCN2C(=O)c3coc(CC(=O)CC(O)\\\\C=C(C)/C=C\\\\CNC(=O)\\\\C=C/C1C)n3\\n947, xantocillin, 0, Oc1ccc(cc1)/C=C([N+]#[C-])/C(=C/c2ccc(O)cc2)[N+]#[C-]\\n948, zorubicin, 0, [Cl-].COc1cccc2C(=O)c3c(O)c4C[C@](O)(C[C@H](O[C@H]5C[C@H](N)[C@H](O)[C@H](C)O5)c4c(O)c3C(=O)c12)/C(C)=N/NC(=O)c6ccccc6.[H+]\\n949, cryofluorane, 1, ClC(C(F)(F)Cl)(F)F\\n950, gacyclidine, 1, [C@@]3(C1=CC=CS1)(N2CCCCC2)[C@H](CCCC3)C\\n951, fenetylline, 1, [H+].C1=NC3=C([N]1CCNC(CC2=CC=CC=C2)C)C(=O)N(C(=O)N3C)C.[Cl-]\\n952, acaprazine, 1, C1=C(Cl)C=CC(=C1N2CCN(CCCNC(C)=O)CC2)Cl\\n953, acebrochol, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C([C@@H](C(CC1)OC(C)=O)Br)[C@@H](C2)Br)C)CC3)CC[C@@H]4[C@@H](CCCC(C)C)C)C\\n954, acecarbromal, 1, C(C(C(NC(NC(C)=O)=O)=O)(CC)Br)C\\n955, acefurtiamine, 1, C2=C(CN(C(=C(SC(=O)C1=CC=CO1)\\\\CCOC(=O)COC(=O)C)/C)C=O)C(=NC(=N2)C)N\\n956, aceglutamide, 1, C(C(NC(C)=O)C(O)=O)CC(N)=O\\n957, aceperone, 1, C1=CC=CC=C1C3(CCN(CCCC(C2=CC=C(F)C=C2)=O)CC3)CNC(C)=O\\n958, acetaminosalol, 1, C2=C(C(OC1=CC=C(C=C1)NC(C)=O)=O)C(=CC=C2)O\\n959, acetazolamide, 1, CC(NC1=NN=C(S1)[S](N)(=O)=O)=O\\n960, acetiamine, 1, C1=NC(=NC(=C1CN(C(=C(\\\\CCOC(C)=O)SC(C)=O)/C)C=O)N)C\\n961, acetorphan/racecadotril/ecadotril, 1, C2=C(CC(C(NCC(OCC1=CC=CC=C1)=O)=O)CSC(C)=O)C=CC=C2\\n962, acetorphine, 1, [C@H]23OC1=C(OC(=O)C)C=CC6=C1C25C4(C[C@@H](C3(OC)C=C4)[C@](O)(CCC)C)C(N(CC5)C)C6\\n963, acetylmethadol/levacetylmethadol/alphacetylmethadol/betacetylmethadol, 1, C2=C(C(C1=CC=CC=C1)(C(OC(=O)C)CC)CC(N(C)C)C)C=CC=C2\\n964, acevaltrate, 1, [C@]12(OC1)[C@@H]3C(=C[C@@H]2OC(=O)CC(OC(=O)C)(C)C)C(=CO[C@H]3OC(=O)CC(C)C)COC(=O)C\\n965, acrocinonide, 1, [C@]12(OC(O[C@@H]1CC3C2(CC(O)C4(F)C3CCC5=CC(=O)C=CC45C)C)C=C)C(=O)CO\\n966, acyclovir/aciclovir, 1, C1=NC2=C([N]1COCCO)NC(=NC2=O)N\\n967, adafenoxate, 1, C1=CC(=CC=C1OCC(OCCNC23CC4CC(C2)CC(C3)C4)=O)Cl\\n968, adatanserin, 1, C1=NC(=NC=C1)N5CCN(CCNC(C23CC4CC(C2)CC(C3)C4)=O)CC5\\n969, alaproclate, 1, C1=CC(=CC=C1CC(OC(C(C)N)=O)(C)C)Cl\\n970, alclometasone-dipropionate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)C[C@H]2Cl)C)[C@H](C3)O)C[C@H]4C)C)(C(COC(CC)=O)=O)OC(CC)=O\\n971, alentemol, 1, C1=C3C2=C(C=C1O)C=CC=C2CC(C3)N(CCC)CCC\\n972, alfadolone, 1, [C@@H]12C4([C@@H](CCC1[C@H]3C(CC2=O)(C(CC3)C(=O)CO)C)C[C@H](O)CC4)C\\n973, alfaxalone/renanolone, 1, [C@@H]23[C@H]([C@H]1[C@]([C@@H](C(C)=O)CC1)(C)CC2=O)CC[C@@H]4[C@@]3(CC[C@H](C4)O)C\\n974, algestone-acetonide, 1, CC14C5(C(CC1C3C(C2(C(=CC(=O)CC2)CC3)C)CC4)OC(O5)(C)C)C(C)=O\\n975, aliflurane, 1, COC1(C(F)(F)C1(F)Cl)F\\n976, alletorphine, 1, [C@@](O)(C5C6(OC)C2OC1=C(O)C=CC4=C1C23C(C(N(CC3)CC=C)C4)(C5)C=C6)(CCC)C\\n977, allomethadione, 1, C(N1C(C(C)OC1=O)=O)C=C\\n978, almoxatone, 1, [C@H]3(CN(C2=CC=C(OCC1=CC=CC(=C1)Cl)C=C2)C(O3)=O)CNC\\n979, alnespirone, 1, [H+].C4=C3OCC(N(CCCCN1C(=O)CC2(CC1=O)CCCC2)CCC)CC3=C(OC)C=C4.[Cl-]\\n980, alonimid, 1, C1=CC=CC3=C1C2(C(NC(=O)CC2)=O)CCC3=O\\n981, aloracetam, 1, C1=C([N](C(=C1C=O)C)CCNC(C)=O)C\\n982, alozafone, 1, C1=C(C(=CC=C1Cl)N(C(CN(C(CC#N)C)C)=O)C)C(C2=C(C=CC=C2)F)=O\\n983, alphameprodine/betameprodine, 1, [C@]2(C1=CC=CC=C1)([C@H](CN(C)CC2)CC)OC(CC)=O\\n984, alphamethadol, 1, [C@H](C(C1=CC=CC=C1)(C2=CC=CC=C2)C[C@H](N(C)C)C)(CC)O\\n985, alpiropride, 1, C1=C(C(=CC(=C1[S](NC)(=O)=O)N)OC)C(NCC2N(CCC2)CC=C)=O\\n986, alrestatin, 1, C1=CC=C3C2=C1C(N(CC(O)=O)C(C2=CC=C3)=O)=O\\n987, amcinafal, 1, [C@H]23C([C@@]1(OC(O[C@@H]1C2)(CC)CC)C(=O)CO)(C[C@H](O)[C@@]4(F)C3CCC5=CC(=O)C=CC45C)C\\n988, amcinafide, 1, [C@H]24C([C@@]1(O[C@@](O[C@@H]1C2)(C3=CC=CC=C3)C)C(=O)CO)(C[C@H](O)[C@@]5(F)C4CCC6=CC(=O)C=CC56C)C\\n989, amcinonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@H]4OC6(O5)CCCC6)C)C(COC(C)=O)=O\\n990, amelometasone, 1, [C@H]12[C@@]([C@](OC(=O)CC)([C@H](C1)C)C(=O)COC)(C[C@H](O)[C@@]3(F)[C@H]2CCC4=CC(=O)C=C[C@]34C)C\\n991, ameltolide, 1, C1=C(C(=C(C=C1)C)NC(C2=CC=C(N)C=C2)=O)C\\n992, amesergide, 1, [C@H]4(CC3C2=C1C(=C[N](C1=CC=C2)C(C)C)CC3N(C4)C)C(NC5CCCCC5)=O\\n993, amfetaminil, 1, C2=C(C(NC(CC1=CC=CC=C1)C)C#N)C=CC=C2\\n994, amineptine, 1, C3=C2C([NH2+]CCCCCCC(O)=O)C1=CC=CC=C1CCC2=CC=C3.[Cl-]\\n995, amisulpride, 1, C1=C([S](CC)(=O)=O)C(=CC(=C1C(NCC2N(CCC2)CC)=O)OC)N\\n996, amitriptylinoxide, 1, C1=CC=CC2=C1C(C3=C(CC2)C=CC=C3)=CCC[N+](C)(C)[O-]\\n997, amixetrine, 1, C1=CC=CC=C1C(CN2CCCC2)OCCC(C)C\\n998, amperozide, 1, C3=C(C(C1=CC=C(C=C1)F)CCCN2CCN(C(NCC)=O)CC2)C=CC(=C3)F\\n999, amphenidone, 1, C1=C(C=CC=C1N2C(C=CC=C2)=O)N\\n1000, ampyzine, 1, CN(C)c1cnccn1\\n1001, amylene-hydrate, 1, C(C(C)(C)O)C\\n1002, anileridine, 1, C3=C(CCN1CCC(CC1)(C2=CC=CC=C2)C(OCC)=O)C=CC(=C3)N\\n1003, aniracetam, 1, C1=CC(=CC=C1C(N2C(CCC2)=O)=O)OC\\n1004, anseculin, 1, COc2cc1oc(=O)c(C)c(C)c1cc2OCCCN3CCN(CC3)c4ccccc4OC\\n1005, apadoline, 1, [C@H](N1C3=C(SC2=C1C=CC=C2)C=CC(=C3)C(=O)NCCC)(CN4CCCC4)C\\n1006, aprobarbital, 1, C(C1(C(NC(=O)NC1=O)=O)C(C)C)C=C\\n1007, aripiprazole, 1, C1=C4C(=CC=C1OCCCCN3CCN(C2=CC=CC(=C2Cl)Cl)CC3)CCC(N4)=O\\n1008, ascorbic-acid, 1, C(O)C(O)C1OC(=C(O)C1=O)O\\n1009, aspirin, 1, C1=CC=CC(=C1C(O)=O)OC(C)=O\\n1010, atolide, 1, C1=CC=CC(=C1C(NC2=C(C=C(N(CC)CC)C=C2)C)=O)N\\n1011, atromepine, 1, C1=CC=CC=C1C(C(OC2CC3N(C)C(C2)CC3)=O)(CO)C\\n1012, atropine-oxide, 1, C1=CC=CC=C1C(C(OC2CC3[N+](C(C2)CC3)(C)[O-])=O)CO\\n1013, avizafone, 1, [C@@H](C(NCC(N(C1=CC=C(C=C1C(C2=CC=CC=C2)=O)Cl)C)=O)=O)(CCCCN)N\\n1014, axamozide, 1, C1=CC(=CC5=C1N(C4CCN(CC3OC2=C(C=CC=C2)OC3)CC4)C(N5)=O)Cl\\n1015, azabon, 1, C1=CC(=CC=C1[S](N2CC3CCC(C2)CC3)(=O)=O)N\\n1016, azabuperone, 1, C1=CC(=CC=C1C(CCCN2CC3N(CC2)CCC3)=O)F\\n1017, azacyclonol, 1, C3=C(C(O)(C1=CC=CC=C1)C2CCNCC2)C=CC=C3\\n1018, azaloxan, 1, O=C1NCCN1C4CCN(CCC3COc2ccccc2O3)CC4\\n1019, azaquinzole, 1, C1=CC=CC2=C1C3N(CC2)CCNC3\\n1020, azepindole, 1, C1=CC=CC2=C1[N]3C(=C2)CNCCC3\\n1021, azipramine, 1, CN(CCc2cc3cccc4CCc1ccccc1n2c34)Cc5ccccc5\\n1022, balaperidone, 1, [C@H]13[C@H](C[C@@H]1C2=CC=C(F)C=C2)CN(C3)CCN4C(=O)C5=C(NC4=O)C=CC=C5\\n1023, bamifylline, 1, C1=CC=CC=C1CC2=NC3=C([N]2CCN(CCO)CC)C(N(C)C(N3C)=O)=O\\n1024, barbital, 1, C(C1(C(NC(=O)NC1=O)=O)CC)C\\n1025, batelapine, 1, C1=CC=CC3=C1N=C([N]2C(=NC(=N2)C)C3)N4CCN(C)CC4\\n1026, beclamide, 1, C1=CC=CC=C1CNC(CCCl)=O\\n1027, beclomethasone/beclometasone, 1, [C@]23([C@H]([C@H]1[C@]([C@](C(CO)=O)(O)[C@H](C1)C)(C)C[C@@H]2O)CCC4=CC(=O)C=C[C@]34C)Cl\\n1028, befloxatone, 1, [C@@H]2(CN(C1=CC=C(C=C1)OCC[C@H](C(F)(F)F)O)C(O2)=O)COC\\n1029, bemegride, 1, C(C1(CC(NC(C1)=O)=O)C)C\\n1030, benfotiamine, 1, C2=C(CN(C(=C(SC(=O)C1=CC=CC=C1)/CCO[P](=O)(O)O)/C)C=O)C(=NC(=N2)C)N\\n1031, benorilate, 1, C1=CC=CC(=C1C(OC2=CC=C(NC(C)=O)C=C2)=O)OC(C)=O\\n1032, bentazepam, 1, C4=C(C2=C1C3=C(SC1=NC(=O)CN2)CCCC3)C=CC=C4\\n1033, bentiamine, 1, C3=C(C(SC(/CCOC(C1=CC=CC=C1)=O)=C(N(CC2=C(N=C(N=C2)C)N)C=O)/C)=O)C=CC=C3\\n1034, benzaprinoxide, 1, C1=CC=C(Cl)C3=C1\\\\C(C2=CC=CC=C2C=C3)=C/CC[N+]([O-])(C)C\\n1035, benzindopyrine, 1, [Cl].C1=CC=CC2=C1[N](C=C2CCC3=CC=NC=C3)CC4=CC=CC=C4\\n1036, benzyl-alcohol, 1, C(C1=CC=CC=C1)O\\n1037, berupipam, 1, [C@@H]4(C1=C2C(=CC(=C1)Br)CCO2)C3=C(C=C(Cl)C(=C3)O)CCN(C4)C\\n1038, bervastatin, 1, [C@H](O)(/C=C/C1=C(C3=C(OC12CCCC2)C=CC=C3)C4=CC=C(F)C=C4)C[C@H](O)CC(OCC)=O\\n1039, betamethasone-acibutate, 1, [C@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@@H]4C)C)(OC(C5=CC=CC=C5)=O)C(CO)=O\\n1040, betamethasone-benzoate, 1, [C@]14([C@@]([C@@H](C)C[C@H]1[C@H]3[C@]([C@@]2(C(=CC(=O)C=C2)CC3)C)(F)[C@H](C4)O)(OC(CCCC)=O)C(CO)=O)C\\n1041, binospirone, 1, O=C2CC1(CCCC1)CC(=O)N2CCNCC4COc3ccccc3O4\\n1042, biotin, 1, C(C2C1C(NC(N1)=O)CS2)CCCC(O)=O\\n1043, bipenamol, 1, S1[C@H]4N(CC1(C2CN3[C@H](S2)CC3=O)O)C(C4)=O\\n1044, biriperone, 1, C1=CC=CC2=C1C4=C([NH]2)CN3CCN(CC3C4)CCCC(C5=CC=C(C=C5)F)=O\\n1045, bisbentiamine, 1, C1=NC(=NC(=C1CN(C(=C(SSC(=C(\\\\C)N(C=O)CC2=C(N)N=C(C)N=C2)/CCOC(=O)C3=CC=CC=C3)/CCOC(=O)C4=CC=CC=C4)\\\\C)C=O)N)C\\n1046, bisorcic, 1, [C@H](CCCNC(C)=O)(C(O)=O)NC(C)=O\\n1047, blonanserin, 1, C1=C(C3=C(N=C1N2CCN(CC)CC2)CCCCCC3)C4=CC=C(F)C=C4\\n1048, bremazocine, 1, [C@@]23(C([C@H](N(CC1(CC1)O)CC2)CC4=C3C=C(C=C4)O)(C)C)CC\\n1049, brofoxine, 1, C1=CC(=CC2=C1NC(=O)OC2(C)C)Br\\n1050, bromisoval, 1, CC(C(C(NC(N)=O)=O)Br)C\\n1051, bromoform, 1, C(Br)(Br)Br\\n1052, brosotamide, 1, C1=C(C(=C(C=C1Br)C)O)C(N)=O\\n1053, brosuximide, 1, C2=C(C1C(NC(C1)=O)=O)C=CC=C2Br\\n1054, brotizolam, 1, C1=C(Br)SC2=C1C(=NCC3=NN=C([N]23)C)C4=C(C=CC=C4)Cl\\n1055, buciclovir, 1, [C@@H](CC[N]2C1=C(C(N=C(N1)N)=O)N=C2)(CO)O\\n1056, butacetin, 1, C1=C(C=CC(=C1)NC(C)=O)OC(C)(C)C\\n1057, butaclamol, 1, [C@]1(O)(C[C@@H]4N(CC1)C[C@@H]2C5=C(CCC3=CC=CC=C23)C=CC=C45)C(C)(C)C.[H+].[Cl-]\\n1058, butaperazine, 1, C1=C(C(CCC)=O)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCN(C)CC4\\n1059, butoctamide, 1, C(C(CCCC)CC)NC(CC(C)O)=O\\n1060, butorphanol, 1, [C@]235[C@]([C@H](N(CC1CCC1)CC2)CC4=C3C=C(O)C=C4)(CCCC5)O\\n1061, cabergoline, 1, [C@@H]34C1=CC=CC2=C1C(=C[NH]2)C[C@H]3N(C[C@@H](C4)C(N(C(NCC)=O)CCCN(C)C)=O)CC=C\\n1062, cafedrine, 1, C1=NC3=C([N]1CCNC(C(C2=CC=CC=C2)O)C)C(N(C)C(N3C)=O)=O\\n1063, camazepam, 1, C1=C(Cl)C=CC2=C1C(=NC(OC(N(C)C)=O)C(N2C)=O)C3=CC=CC=C3\\n1064, camphor, 1, CC12C(C(CC1=O)CC2)(C)C\\n1065, caproxamine, 1, C1=C(C(=CC=C1C(/CCCCC)=N\\\\OCCN)C)N\\n1066, captodiame, 1, C1=CC(=CC=C1C(C2=CC=CC=C2)SCCN(C)C)SCCCC\\n1067, carbubarb, 1, C(C1(C(NC(=O)NC1=O)=O)CCCC)COC(N)=O\\n1068, carburazepam, 1, C1=C(Cl)C=CC2=C1C(N(CC(=O)N2C)C(N)=O)C3=CC=CC=C3\\n1069, carfimate, 1, C1=C(C(OC(N)=O)C#C)C=CC=C1\\n1070, carmantadine, 1, O=C(C4N(C12CC3CC(C1)CC(C2)C3)CC4)O\\n1071, caroxazone, 1, C1=CC=CC2=C1CN(CC(N)=O)C(O2)=O\\n1072, carperone, 1, C2=C(C(CCCN1CCC(OC(NC(C)C)=O)CC1)=O)C=CC(=C2)F\\n1073, carphenazine/carfenazine, 1, C1=C(C=CC3=C1N(CCCN2CCN(CC2)CCO)C4=C(S3)C=CC=C4)C(CC)=O\\n1074, carzenide, 1, C1=C(C=CC(=C1)[S](N)(=O)=O)C(O)=O\\n1075, cathinone, 1, [C@@H](C(C1=CC=CC=C1)=O)(C)N\\n1076, cebaracetam, 1, C1=CC(=CC=C1C3CN(CC(N2CC(NCC2)=O)=O)C(C3)=O)Cl\\n1077, celucloral, 1, OC1OC(COCCOC(O)C(Cl)(Cl)Cl)C(OC2OC(COCCOCCOC(O)C(Cl)(Cl)Cl)CC(OC(O)C(Cl)(Cl)Cl)C2O)C(OCCOCCOC(O)C(Cl)(Cl)Cl)C1O\\n1078, cerivastatin, 1, [C@@H](/C=C/C1=C(N=C(C(=C1C2=CC=C(C=C2)F)COC)C(C)C)C(C)C)(C[C@H](CC(O)=O)O)O\\n1079, cetohexazine, 1, CC1=CC(=NNC1=O)C\\n1080, cetophenicol/cetofenicol, 1, C1=C(C(O)C(CO)NC(C(Cl)Cl)=O)C=CC(=C1)C(=O)C\\n1081, cetotiamine, 1, C1=C(CN(C(=C(SC(OCC)=O)/CCOC(OCC)=O)/C)C=O)C(=NC(=N1)C)N\\n1082, chloralose, 1, [C@@H]2(OC1O[C@@H](O[C@@H]1[C@H]2O)C(Cl)(Cl)Cl)[C@H](O)CO\\n1083, chlorhexadol/chloralodol, 1, C(C(OC(C(Cl)(Cl)Cl)O)C)C(C)(C)O\\n1084, chlormezanone, 1, C1=CC(=CC=C1C2[S](CCC(N2C)=O)(=O)=O)Cl\\n1085, chloroform, 1, C(Cl)(Cl)Cl\\n1087, chromocarb, 1, C1=C2C(=CC=C1)C(C=C(O2)C(O)=O)=O\\n1088, ciclesonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)C[C@H]4O[C@H](O5)C6CCCCC6)C)C(COC(C(C)C)=O)=O\\n1089, ciclotizolam, 1, C1=C(SC2=C1C(=NCC3=NN=C([N]23)C4CCCCC4)C5=CC=CC=C5Cl)Br\\n1090, cicortonide, 1, [C@H]23C([C@@]1(OC(O[C@@H]1C2)(C)C)C(=O)COC(=O)C)(CC(O)[C@@]4(F)C3CC(=C5C4(CCC(=C5)OCCCl)C)C#N)C\\n1091, cilobamine, 1, [C@@]1([C@H](C2CCC1CC2)NC(C)C)(C3=CC(=C(C=C3)Cl)Cl)O\\n1092, cimemoxin, 1, C(C1CCCCC1)NN\\n1093, cimepanol, 1, CC(C(C1CCCCC1)O)C\\n1094, cimoxatone, 1, C1=CC(=CC=C1N2CC(COC)OC2=O)OCC3=CC(=CC=C3)C#N\\n1095, cinoctramide, 1, C1=C(C(=C(C=C1\\\\C=C\\\\C(N2CCCCCCC2)=O)OC)OC)OC\\n1096, cinolazepam, 1, C1=C(Cl)C=CC2=C1C(=NC(O)C(N2CCC#N)=O)C3=CC=CC=C3F\\n1097, ciprazafone, 1, C1=C(Cl)C=CC(=C1C(C2=C(C=CC=C2)Cl)=O)N(C(CNC3CC3)=O)C\\n1098, ciprocinonide, 1, [C@]34([C@H](C2[C@@](F)([C@@]1(C(=CC(=O)C=C1)[C@@H](F)C2)C)[C@@H](O)C3)C[C@H]5OC(O[C@@]45C(=O)COC(=O)C6CC6)(C)C)C\\n1099, citatepine, 1, C1=C(C#N)C=CC3=C1C4=C(C2=C(C=CC=C2)S3)CCN(CC4)C\\n1100, clebopride, 1, C1=C(C(=CC(=C1Cl)N)OC)C(NC3CCN(CC2=CC=CC=C2)CC3)=O\\n1101, clidafidine, 1, C1=C(C(=C(C=C1)Cl)NC2=NCCO2)Cl\\n1102, clobetasone, 1, [C@@]23(F)[C@H]([C@H]1[C@@]([C@@]([C@H](C1)C)(O)C(=O)CCl)(CC2=O)C)CCC4=CC(=O)C=C[C@]34C\\n1103, clobetasone-butyrate, 1, [C@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)C(C3)=O)C[C@@H]4C)C)(OC(CCC)=O)C(CCl)=O\\n1104, clocapramine, 1, C1=C(Cl)C=CC4=C1N(CCCN3CCC(N2CCCCC2)(CC3)C(N)=O)C5=C(CC4)C=CC=C5\\n1105, clocortolone, 1, [C@@]24(Cl)C1(C(=CC(=O)C=C1)[C@@H](F)CC2C3C(C([C@@H](C3)C)C(=O)CO)(CC4O)C)C\\n1106, clocortolone-acetate, 1, [C@H]2(C1=CC(C=C[C@@]1(C4([C@@H](C2)[C@@H]3C[C@H]([C@H](C(COC(=O)C)=O)[C@@]3(C)C[C@@H]4O)C)Cl)C)=O)F\\n1107, clocortolone-pivalate, 1, [C@@]24(Cl)C1(C(=CC(=O)C=C1)[C@@H](F)CC2C3C(C([C@@H](C3)C)C(=O)COC(=O)C(C)(C)C)(CC4O)C)C\\n1108, clodazon, 1, C1=C(Cl)C=CC3=C1N(C2=CC=CC=C2)C(=O)N3CCCN(C)C\\n1109, clofenciclan, 1, C2=C(C1(OCCN(CC)CC)CCCCC1)C=CC(=C2)Cl\\n1110, clofluperol, 1, C1=C(C(=CC=C1C3(CCN(CCCC(C2=CC=C(F)C=C2)=O)CC3)O)Cl)C(F)(F)F\\n1111, clopenthixol/zuclopenthixol, 1, C1=C(Cl)C=CC3=C1\\\\C(C2=C(C=CC=C2)S3)=C/CCN4CCN(CCO)CC4\\n1112, cloperidone, 1, C1=CC=CC4=C1C(N(CCCN3CCN(C2=CC(=CC=C2)Cl)CC3)C(N4)=O)=O\\n1113, cloprednol, 1, [C@H]24[C@H]1[C@@]([C@](C(CO)=O)(O)CC1)(C[C@@H]([C@@H]2[C@@]3(C(=CC(=O)C=C3)C(=C4)Cl)C)O)C\\n1114, clorethate/cloretate, 1, C(C(Cl)(Cl)Cl)OC(OCC(Cl)(Cl)Cl)=O\\n1115, clorgiline, 1, C1=C(C(=CC(=C1)Cl)Cl)OCCCN(CC#C)C\\n1116, cloroperone, 1, C3=C(C(C2CCN(CCCC(C1=CC=C(F)C=C1)=O)CC2)=O)C=CC(=C3)Cl\\n1117, clorotepine, 1, C1=C(Cl)C=CC4=C1C(N2CCN(C)CC2)CC3=C(C=CC=C3)S4\\n1118, clotiazepam, 1, C1=C(CC)SC2=C1C(=NCC(N2C)=O)C3=CC=CC=C3Cl\\n1119, cloticasone(cloticasone-propionate), 1, [C@]34([C@H](C2[C@@](F)([C@@]1(C(=CC(=O)C=C1)[C@@H](F)C2)C)[C@@H](O)C3)C[C@H]([C@]4(OC(=O)CC)C(SCCl)=O)C)C\\n1120, clotixamide(clotixamide), 1, CNC(=O)CCN1CCN(CCC=C2c3ccccc3Sc3ccc(cc23)Cl)CC1\\n1121, cogazocine, 1, C1=C(O)C=CC4=C1C3(C(C(N(CC2CCC2)CC3)C4)(C)C)CC\\n1122, conorphone(conorfone), 1, [C@H]12C5CC4=C3[C@]1([C@H](C(C[C@@H]2CC)=O)OC3=C(C=C4)OC)CCN5CC6CC6\\n1123, cormetasone, 1, CC1CC2C3CC(F)(F)C4=CC(=O)C=C[C@]4(C)C3(F)C(O)CC2(C)C1(O)C(=O)CO\\n1124, cortexolone(cortodoxonecortifen), 1, C(O)C(=O)C4(O)C3(C(C2C(C1(C(=CC(=O)CC1)CC2)C)CC3)CC4)C\\n1125, cortisone-acetate, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)C(C3)=O)CC[C@@]4(C(COC(C)=O)=O)O)C\\n1126, cortisuzol, 1, [C@H]37[C@H]2[C@@]([C@](C(COC(C1=CC(=CC=C1)[S](O)(=O)=O)=O)=O)(O)[C@@H](C2)C)(C[C@@H]([C@@H]3[C@@]4(C(=CC5=C(C4)C=N[N]5C6=CC=CC=C6)C(=C7)C)C)O)C\\n1127, cortivazol, 1, [C@H]26[C@H]1[C@@]([C@](C(COC(C)=O)=O)(O)[C@@H](C1)C)(C[C@@H]([C@@H]2[C@@]3(C(=CC4=C(C3)C=N[N]4C5=CC=CC=C5)C(=C6)C)C)O)C\\n1128, cotinine, 1, C1=CC=NC=C1C2N(C(=O)CC2)C\\n1129, cotriptyline, 1, C1=CC=CC2=C1C(C3=C(CC2)C=CC=C3)=CC(CN(C)C)=O\\n1130, crilvastatin, 1, CC2(CC(OC(C1NC(=O)CC1)=O)CC(C2)C)C\\n1131, crotetamide, 1, C(C(N(C(\\\\C=C\\\\C)=O)CC)C(N(C)C)=O)C\\n1132, cyclarbamate, 1, C1=CC=CC=C1NC(OCC3(COC(NC2=CC=CC=C2)=O)CCCC3)=O\\n1133, cyclindole(ciclindole), 1, C1=CC=CC2=C1C3=C([NH]2)CCC(C3)N(C)C\\n1134, cyclobarbital, 1, C(C2(C1=CCCCC1)C(NC(=O)NC2=O)=O)C\\n1135, cyclobenzaprine, 1, C1=CC=CC2=C1C(C3=C(C=C2)C=CC=C3)=CCCN(C)C\\n1136, cyclopregnol, 1, [C@]23(C1(C(C1)CC2)[C@H](O)C[C@@H]4[C@@H]3CC[C@]5([C@H]4CC[C@@H]5C(=O)C)C)C\\n1137, cycotiamine, 1, C1=NC(=NC(=C1CN(C(/C)=C2\\\\CCOC(=O)S2)C=O)N)C\\n1138, cyheptamide, 1, C1=CC=CC3=C1C(C2=C(C=CC=C2)CC3)C(N)=O\\n1139, cyprenorphine, 1, [C@@]135C67[C@@H](N(CC1)CC2CC2)CC4=CC=C(O)C(=C34)O[C@@H]5C(OC)(C(C6)C(O)(C)C)C=C7\\n1140, cyprodenate, 1, C(C1CCCCC1)CC(OCCN(C)C)=O\\n1141, cyprolidol, 1, C4=C(C(C2C(C1=CC=NC=C1)C2)(C3=CC=CC=C3)O)C=CC=C4\\n1142, cyproximide(ciproximide), 1, C1=CC(=CC=C1C23C(C(=O)NC2=O)C3)Cl\\n1143, dacemazine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)S3)C(CN(C)C)=O\\n1144, dazadrol, 1, C1=CC(=CC=C1C(C2=NCCN2)(C3=NC=CC=C3)O)Cl\\n1145, decimemide, 1, C1=C(C(N)=O)C=C(C(=C1OC)OCCCCCCCCCC)OC\\n1146, delfaprazine, 1, C1=C(C(=CC=C1C)CC2=CC=CC=C2)N3CCNCC3\\n1147, demoxepam, 1, C3=C(C1=C2C(=NC(=O)CN1O)C=CC(=C2)Cl)C=CC=C3\\n1148, denbufylline, 1, C1=NC2=C([N]1CC(C)=O)C(N(CCCC)C(N2CCCC)=O)=O\\n1149, deprodone, 1, CC(=O)C3(O)CCC4C2CCC1=CC(=O)C=CC1(C)C2C(O)CC34C\\n1150, desciclovir, 1, C1=NC(=NC2=C1N=C[N]2COCCO)N\\n1151, descinolone(descinolone-acetonide), 1, [C@H]34[C@H]2[C@@](F)([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@@H](O)C[C@@]3([C@](O)([C@H](O)C4)C(=O)C)C\\n1152, desonide, 1, [C@]12(OC(O[C@@H]1C[C@@H]3C2(CC(O)[C@H]4C3CCC5=CC(=O)C=CC45C)C)(C)C)C(=O)CO\\n1153, desoximetasone, 1, [C@H]23[C@@]([C@@]1(C(=CC(=O)C=C1)CC2)C)([C@H](C[C@]4([C@H]3C[C@@H](C)[C@@H]4C(CO)=O)C)O)F\\n1154, detomidine, 1, C2=C(CC1=CN=C[NH]1)C(=C(C=C2)C)C\\n1155, dexamethasone-acefurate, 1, [C@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@H]4C)C)(OC(C5=CC=CO5)=O)C(COC(C)=O)=O\\n1156, dexamisole, 1, [C@@H]3(C1=CC=CC=C1)CN2CCSC2=N3\\n1157, deximafen(imafen), 1, C3=C(C1N2C(=NC1)NCC2)C=CC=C3\\n1158, dexoxadrol(dioxadrolandlevoxadrol), 1, [C@@H]1(NCCCC1)[C@@H]2OC(OC2)(C3=CC=CC=C3)C4=CC=CC=C4\\n1159, dichlorisone, 1, [C@@]23(Cl)C(C1C([C@](O)(CC1)C(=O)CO)(CC2Cl)C)CCC4=CC(=O)C=CC34C\\n1160, dichlorisone-acetate, 1, [C@@]23(Cl)C(C1C([C@](O)(CC1)C(=O)COC(=O)C)(CC2Cl)C)CCC4=CC(=O)C=CC34C\\n1162, diclometide, 1, C1=C(Cl)C=C(C(=C1C(NCCN(CC)CC)=O)OC)Cl\\n1163, didrovaltrate, 1, [C@]23([C@@H]1[C@@H](C(=CO[C@H]1OC(=O)CC(C)C)COC(=O)CC(C)C)C[C@@H]2OC(=O)C)OC3\\n1164, difebarbamate, 1, C2=C(C1(C(N(C(=O)N(C1=O)CC(COCCCC)OC(N)=O)CC(COCCCC)OC(N)=O)=O)CC)C=CC=C2\\n1165, difluanine(difluanazine), 1, C4=C(C(C1=CC=C(C=C1)F)CCCN3CCN(CCNC2=CC=CC=C2)CC3)C=CC(=C4)F\\n1166, diflucortolone, 1, [C@@]24(F)C1(C(=CC(=O)C=C1)[C@@H](F)CC2C3C(C([C@@H](C3)C)C(=O)CO)(CC4O)C)C\\n1167, diflucortolone-pivalate, 1, [C@H]23[C@@]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)([C@H](C[C@]4([C@H]3C[C@@H](C)[C@@H]4C(COC(C(C)(C)C)=O)=O)C)O)F\\n1168, dimelazine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)S3)CC4(CN(C)CC4)C\\n1169, dimethadione, 1, CC1(C(NC(O1)=O)=O)C\\n1170, dimethazan, 1, C1=NC2=C([N]1CCN(C)C)C(N(C)C(N2C)=O)=O\\n1171, dimethylamphetamine(dimethamphetamine), 1, CN(C(C)CC1=CC=CC=C1)C\\n1172, dimiracetam, 1, O=C2N1C(NC(=O)C1)CC2\\n1173, dioxamate, 1, C(C1(OC(COC(N)=O)CO1)C)CCCCCCCC\\n1174, diphoxazide, 1, C2=C(C(C1=CC=CC=C1)(CC(NNC(C)=O)=O)O)C=CC=C2\\n1175, diprenorphine, 1, [C@]235[C@@]67[C@H](N(CC1CC1)CC2)CC4=C3C(=C(C=C4)O)O[C@H]5[C@](OC)([C@H](C6)C(C)(C)O)CC7\\n1176, divaplon, 1, C1=C(N=C2[N]1C(=C(CC)C(=N2)OC)C)C(C3=CC=CC=C3)=O\\n1177, dizocilpine(MK-801), 1, C1=CC=CC2=C1C3(C4=C(CC2N3)C=CC=C4)C\\n1178, domoprednate, 1, [C@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)CCC4)C)(OC(CCC)=O)C(C)=O\\n1179, donepezil, 1, [H+].C1=C(OC)C(=CC2=C1C(=O)C(C2)CC3CCN(CC3)CC4=CC=CC=C4)OC.[Cl-]\\n1180, dopamantine, 1, C4=C(CCNC(C12CC3CC(C1)CC(C2)C3)=O)C=CC(=C4O)O\\n1181, doxapram, 1, C1=CC=CC=C1C2(C(CN(C2=O)CC)CCN3CCOCC3)C4=CC=CC=C4\\n1182, drotebanol, 1, [C@@]123C4=C(C[C@H]([C@@]1(CC[C@H](C2)O)O)N(CC3)C)C=CC(=C4OC)OC\\n1183, duloxetine, 1, C1=CC=C3C(=C1OC(C2=CC=CS2)CCNC)C=CC=C3\\n1184, duoperone, 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN5CCC(C(C4=CC=C(F)C=C4)=O)CC5\\n1185, dupracetam, 1, C(N1C(CCC1)=O)C(NNC(CN2C(CCC2)=O)=O)=O\\n1186, eclanamine, 1, [C@H]2(N(C1=CC=C(C(=C1)Cl)Cl)C(CC)=O)[C@@H](CCC2)N(C)C\\n1187, edaravone, 1, C2=C(N1N=C(C)CC1=O)C=CC=C2\\n1188, efetozole, 1, C1=CC=CC=C1C(C)[N]2C=CN=C2C\\n1189, elanzepine, 1, C1=CC(=CC3=C1\\\\C(C2=CC=CC=C2N3)=C/CN(CC)CC)Cl\\n1190, elfazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(N2CC[S](CC)(=O)=O)=O)C3=CC=CC=C3F\\n1191, elopiprazole, 1, C1=C(C2=C(C=C1)C=CO2)N5CCN(CC3=CC=C([NH]3)C4=CC=C(C=C4)F)CC5\\n1192, eltoprazine, 1, C1=C(C2=C(C=C1)OCCO2)N3CCNCC3\\n1193, embutramide, 1, C1=C(C=CC=C1C(CNC(CCCO)=O)(CC)CC)OC\\n1194, enciprazine, 1, C3=C(N2CCN(CC(COC1=CC(=C(OC)C(=C1)OC)OC)O)CC2)C(=CC=C3)OC\\n1195, encyprate, 1, C1=CC=CC=C1CN(C2CC2)C(OCC)=O\\n1196, endixaprine, 1, C3=C(C2=CC=C(N1CCC(O)CC1)N=N2)C(=CC(=C3)Cl)Cl\\n1197, endomide, 1, [C@@H]1([C@@H]([C@H]2C=C[C@@H]1C2)C(N(CC)CC)=O)C(N(CC)CC)=O\\n1198, endrysone(endrisone), 1, CC3CC2C1CCC(C(C)=O)C1(C)CC(O)C2C4(C)C=CC(=O)C=C34\\n1199, enilospirone, 1, C3=C(OC2C1(NC(=O)C(O1)C)CCCC2)C=CC=C3Cl\\n1200, eptastigmine, 1, [C@@]12(C3=C(N(C)[C@H]1N(C)CC2)C=CC(=C3)OC(NCCCCCCC)=O)C\\n1201, erizepine, 1, C1=CC=CC3=C1C4=C(C2=C(C=CC=C2)N3C)CCN(CC4)C\\n1202, etacepride, 1, C1=C(C(C)=O)C=CC(=C1C(NCC2N(CCC2)CC)=O)OC\\n1203, etamiphyllin, 1, C1=NC2=C([N]1CCN(CC)CC)C(N(C)C(N2C)=O)=O\\n1204, etasuline, 1, C1=C(Cl)C=CC3=C1C(C2=CC=CC=C2)SC(=N3)NCC\\n1205, etazepine, 1, C1=CC=CC2=C1C(C3=C(N(C)C2=O)C=CC=C3)OCC\\n1206, ethomoxane, 1, [H+].C2=C1OCC(OC1=C(C=C2)OCC)CNCCCC.[Cl-]\\n1207, ethylene(ethene), 1, C(C)Cl\\n1208, ethyl-loflazepate, 1, C=C\\n1209, ethypicone, 1, C(C1(C(C(=CNC1=O)C)=O)CC)C\\n1210, eticyclidine, 1, C2=C(C1(CCCCC1)NCC)C=CC=C2\\n1211, etifoxine, 1, [H+].C2=C1C(OC(=NC1=CC=C2Cl)NCC)(C3=CC=CC=C3)C.[Cl-]\\n1212, etiracetam(levetiracetam), 1, C(C(N1C(CCC1)=O)C(N)=O)C\\n1213, etisulergine, 1, [C@@H]34C1=CC=CC2=C1C(=C[NH]2)C[C@H]3N(C[C@H](C4)N[S](N(CC)CC)(=O)=O)C\\n1214, etodroxizine, 1, C3=C(C(N1CCN(CC1)CCOCCOCCO)C2=CC=CC=C2)C=CC(=C3)Cl.O=C(O)\\\\C=C\\\\C(O)=O\\n1215, etoperidone, 1, C1=C(C=CC=C1N3CCN(CCCN2C(N(C(=N2)CC)CC)=O)CC3)Cl\\n1216, etorphine, 1, C2=CC(=C3OC5C14C6(C(N(CC1)C)CC2=C34)CC(C5(OC)C=C6)C(O)(CCC)C)O\\n1217, etoxadrol, 1, C3=C(C2(OC(C1CCCCN1)CO2)CC)C=CC=C3\\n1218, famciclovir, 1, C1=NC2=C([N]1CCC(COC(C)=O)COC(C)=O)N=C(N)N=C2\\n1219, fantridone, 1, C1=CC=CC2=C1C3=C(C(=O)N2CCCN(C)C)C=CC=C3\\n1220, fasiplon, 1, C1=C(N=C2[N]1C(=C(CC)C(=N2)OC)C)C3=NOC(=N3)C\\n1221, fasoracetam, 1, [C@H]1(NC(=O)CC1)C(=O)N2CCCCC2\\n1222, febarbamate, 1, C2=C(C1(C(N(CC(COCCCC)OC(N)=O)C(=O)NC1=O)=O)CC)C=CC=C2\\n1223, felbamate, 1, C1=C(C(COC(N)=O)COC(N)=O)C=CC=C1\\n1224, femoxetine, 1, C3=C(C2C(COC1=CC=C(OC)C=C1)CN(C)CC2)C=CC=C3\\n1225, fenaperone, 1, C1=CC(=CC=C1C(CCCN3CCN(C(OC2CCCCC2)=O)CC3)=O)F\\n1226, fenbutrazate, 1, C3=C(C1C(N(CCO1)CCOC(C(C2=CC=CC=C2)CC)=O)C)C=CC=C3\\n1227, fencamfamin, 1, C3=C(C1C(NCC)C2CC1CC2)C=CC=C3\\n1228, fengabine, 1, C2=C(\\\\C(NCCCC)=C\\\\1C=C(Cl)C=CC1=O)C(=CC=C2)Cl\\n1229, fenharmane, 1, C4=C3[NH]C1=C(CCNC1CC2=CC=CC=C2)C3=CC=C4\\n1230, fenimide, 1, C2=C(C1(C(C(=O)NC1=O)CC)C)C=CC=C2\\n1231, fenmetozole, 1, C1=C(C(=CC=C1OCC2=NCCN2)Cl)Cl\\n1232, fenmetramide, 1, C1=CC=CC=C1C2C(NC(=O)CO2)C\\n1233, fenobam, 1, C1=C(C=CC=C1NC(NC2=NC(=O)CN2C)=O)Cl\\n1234, fenozolone, 1, C1=CC=CC=C1C2C(N=C(O2)NCC)=O\\n1235, fenpentadiol, 1, C1=CC(=CC=C1C(CC(C)(C)O)(C)O)Cl\\n1236, fepitrizol, 1, C3=C(C1=NC(=N[N]1C)C2=CC=CN=C2)C(=CC=C3)CO\\n1237, fipexide, 1, C1=C4C(=CC(=C1)CN3CCN(C(COC2=CC=C(Cl)C=C2)=O)CC3)OCO4\\n1238, flerobuterol, 1, C1=C(C(CNC(C)(C)C)O)C(=CC=C1)F\\n1239, florfenicol, 1, [C@@H](C1=CC=C(C=C1)[S](C)(=O)=O)([C@H](NC(C(Cl)Cl)=O)CF)O\\n1240, fluacizine, 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=C(C=CC=C2)S3)C(CCN(CC)CC)=O\\n1241, fluazacort, 1, [C@@]34(F)C(C2C(C1(N=C(OC1C2)C)C(=O)COC(=O)C)(CC3O)C)CCC5=CC(=O)C=CC45C\\n1242, flubanilate, 1, CCOC(=O)N(CCN(C)C)c1cccc(c1)C(F)(F)F\\n1243, flubepride, 1, C1=C(C(=CC=C1[S](N)(=O)=O)OC)C(NCC3N(CC2=CC=C(F)C=C2)CCC3)=O\\n1244, flucloronide(fluclorolone-acetonide), 1, [C@@]23(Cl)C1(C(=CC(=O)C=C1)[C@@H](F)CC2C5C(CC3Cl)([C@]4(C(=O)CO)OC(O[C@@H]4C5)(C)C)C)C\\n1245, fludrocortisone, 1, [C@H]23[C@@]([C@@]1(C(=CC(=O)CC1)CC2)C)([C@H](C[C@]4([C@H]3CC[C@@]4(C(CO)=O)O)C)O)F\\n1246, fludrocortisone-acetate, 1, [C@H]23[C@@]([C@@]1(C(=CC(=O)CC1)CC2)C)([C@H](C[C@]4([C@H]3CC[C@@]4(C(COC(C)=O)=O)O)C)O)F\\n1247, flumezapine, 1, C1=C(F)C=CC2=C1NC(=C3C(=N2)SC(=C3)C)N4CCN(CC4)C\\n1248, flumoxonide, 1, [C@]12(OC(O[C@@H]1CC3C2(CC(O)[C@@]4(F)C3C[C@H](F)C5=CC(=O)C=CC45C)C)(C)C)C(=O)C(OC)OC\\n1249, fluocinonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)(F)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(COC(C)=O)=O\\n1250, fluocortin, 1, CCCCOC(=O)C(=O)C3C(C)CC4C2CC(F)C1=CC(=O)C=CC1(C)C2C(O)CC34C\\n1251, fluocortolone-caproate, 1, CCCCCC(=O)OCC(=O)C3C(C)CC4C2CC(F)C1=CC(=O)C=CC1(C)C2C(O)CC34C\\n1252, fluoresone, 1, C1=CC(=CC=C1[S](CC)(=O)=O)F\\n1253, fluorometholone-acetate, 1, [C@]23([C@H]([C@H]1[C@]([C@@](OC(C)=O)(C(C)=O)CC1)(C)C[C@@H]2O)C[C@H](C)C4=CC(=O)C=C[C@]34C)F\\n1254, fluotracen, 1, [C@@H]1(C3=C([C@H](C2=C1C=CC=C2)C)C=CC(=C3)C(F)(F)F)CCCN(C)C\\n1255, fluoxetine, 1, C2=C(C(OC1=CC=C(C=C1)C(F)(F)F)CCNC)C=CC=C2\\n1256, fluphenazine-enanthate, 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCN(CCOC(CCCCCC)=O)CC4\\n1257, flupimazine, 1, C1=CC=CC3=C1N(CCCN2CCC(OCCO)CC2)C4=C(S3)C=CC(=C4)C(F)(F)F\\n1258, fluprazine, 1, C1=C(C=CC=C1N2CCN(CCNC(N)=O)CC2)C(F)(F)F\\n1259, fluspirilene, 1, C1=CC=CC=C1N2C5(C(NC2)=O)CCN(CCCC(C3=CC=C(C=C3)F)C4=CC=C(C=C4)F)CC5\\n1260, flutazolam, 1, C1=C(Cl)C=CC2=C1C3(N(CC(=O)N2CCO)CCO3)C4=CC=CC=C4F\\n1261, flutemazepam, 1, CN2C(=O)C(O)N=C(c1ccccc1F)c3cc(Cl)ccc23\\n1262, fluticasone, 1, CC4CC3C2CC(F)C1=CC(=O)C=CC1(C)C2(F)C(O)CC3(C)C4(O)C(=O)SCF\\n1263, fluticasone-propionate, 1, [C@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)(F)[C@H](C3)O)C[C@H]4C)C)(OC(CC)=O)C(SCF)=O\\n1264, flutizenol, 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=C(SC=C2)S3)CCCN4CCN(CCO)CC4\\n1265, flutomidate, 1, C1=C(C(OCC)=O)[N](C=N1)C(C)C2=CC=C(C=C2)F\\n1266, flutroline, 1, C1=CC(=CC2=C1[N](C4=C2CN(CCCC(C3=CC=C(C=C3)F)O)CC4)C5=CC=C(F)C=C5)F\\n1267, fluvastatin, 1, [C@@H](O)(/C=C/C1=C(C2=C([N]1C(C)C)C=CC=C2)C3=CC=C(F)C=C3)C[C@H](O)CC(=O)O\\n1268, fluvoxamine, 1, C1=C(C(F)(F)F)C=CC(=C1)C(=N/OCCN)/CCCCOC.O=C(O)\\\\C=C/C(=O)O\\n1269, fluzinamide, 1, C2=C(OC1CN(C(NC)=O)C1)C=CC=C2C(F)(F)F\\n1270, folic-acid, 1, [C@H](NC(C3=CC=C(NCC1=NC2=C(N=C1)NC(=NC2=O)N)C=C3)=O)(CCC(O)=O)C(O)=O\\n1271, formocortal, 1, [C@]12(OC(O[C@@H]1CC3C2(CC(O)[C@@]4(F)C3CC(=C5C4(CCC(=C5)OCCCl)C)C=O)C)(C)C)C(=O)COC(=O)C\\n1272, fosazepam, 1, C1=CC(=CC2=C1N(C(CN=C2C3=CC=CC=C3)=O)C[P](C)(C)=O)Cl\\n1273, fosenazide, 1, C1=CC=CC=C1[P](C2=CC=CC=C2)(CC(NN)=O)=O\\n1274, fosfomycin, 1, [C@@H]1([P](O)(O)=O)[C@H](C)O1\\n1275, fosphenytoin, 1, C1=CC=CC=C1C3(C2=CC=CC=C2)C(N(CO[P](O)(O)=O)C(N3)=O)=O\\n1276, ftormetazine, 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=C(C=CC=C2)S3)C(CCN4CCN(C)CC4)=O\\n1277, ftorpropazine, 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=C(C=CC=C2)S3)C(CCN4CCN(CCO)CC4)=O\\n1278, furomazine, 1, C1=C(Cl)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN5CCC(C4C(OCC4)=O)(O)CC5\\n1279, fursultiamine, 1, C1=NC(=NC(=C1CN(C(=C(SSCC2CCCO2)\\\\CCO)/C)C=O)N)C\\n1280, gaboxadol, 1, O=C2C1=C(CNCC1)ON2\\n1281, ganaxolone, 1, [C@H]13[C@@H]([C@@]2([C@@H](CC1)C[C@](O)(CC2)C)C)CC[C@]4([C@H]3CC[C@@H]4C(=O)C)C\\n1282, gedocarnil, 1, C1=NC(=C(C2=C1[NH]C3=C2C(=CC=C3)OC4=CC=C(C=C4)Cl)COC)C(OC(C)C)=O\\n1283, gemazocine, 1, C1=C(O)C=CC4=C1C3(C(C(N(CC2CC2)CC3)C4)(C)C)CC\\n1284, gepirone, 1, [H+].C1=CC=NC(=N1)N2CCN(CC2)CCCCN3C(=O)CC(CC3=O)(C)C.[Cl-]\\n1285, gevotroline, 1, C1=C(F)C=CC2=C1C3=C([NH]2)CCN(C3)CCCC4=CN=CC=C4\\n1286, girisopam, 1, C1=C(OC)C(=CC2=C1C(=NN=C(C2)C)C3=CC=CC(=C3)Cl)OC\\n1287, halcinonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)CC1)CC2)C)(F)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(CCl)=O\\n1288, halocortolone, 1, [C@H]23C(Cl)([C@@]1(C(=CC(=O)C=C1)[C@@H](F)C2)C)[C@@H](F)C[C@]4([C@H]3C[C@H]([C@@H]4C(=O)CO)C)C\\n1289, halopemide, 1, C1=CC(=CC4=C1N(C3CCN(CCNC(C2=CC=C(F)C=C2)=O)CC3)C(N4)=O)Cl\\n1290, haloperidol-decanoate, 1, C1=CC(=CC=C1C3(CCN(CCCC(C2=CC=C(C=C2)F)=O)CC3)OC(CCCCCCCCC)=O)Cl\\n1291, halopredone(halopredone-acetate), 1, [C@]34([C@H]([C@H]2C([C@@]1(C(=CC(C(=C1)Br)=O)[C@@H](C2)F)C)([C@H](C3)O)F)CCC4(C(=O)CO)O)C\\n1292, haloxazolam, 1, C1=C(Br)C=CC3=C1C2(N(CCO2)CC(=O)N3)C4=C(C=CC=C4)F\\n1293, hexapradol, 1, C2=C(C(C1=CC=CC=C1)(C(CCCCC)N)O)C=CC=C2\\n1294, homopipramol, 1, C3=C2N(C1=CC=CC=C1C=CC2=CC=C3)CCCN4C=CN(CC=C4)CCO\\n1295, homprenorphine, 1, C1=CC(=C5C2=C1CC6C34C2(C(C(OC)(C(C3)C(CC)(C)O)C=C4)O5)CCN6CC7CC7)OC\\n1296, hopantenic-acid, 1, [C@H](C(NCCCC(O)=O)=O)(C(CO)(C)C)O\\n1297, hydrocortamate, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)[C@H](C3)O)CC[C@@]4(C(COC(CN(CC)CC)=O)=O)O)C\\n1298, hydrocortisone-aceponate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)[C@H](C3)O)CC4)C)(C(COC(C)=O)=O)OC(CC)=O\\n1299, hydrocortisonecypionate(hydrocortisone-cipionate), 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)[C@@H](O)C3)CC[C@]4(O)C(=O)COC(=O)CCC(O)=O)C.O\\n1300, hydrocortisone-hemisuccinate, 1, [C@@]4(OC(=O)CCCC)(C3(C(C2C(C1(C(=CC(=O)CC1)CC2)C)C(O)C3)CC4)C)C(=O)CO\\n1301, hydrocortisone-valerate, 1, CC14CCC(=O)C=C1CCC5C3CCC(O)(C(=O)COC(=O)CCC2CCCC2)C3(C)CC(O)C45\\n1302, hydromorphinol, 1, [C@@]125C3=C4C[C@H]([C@@]1(CC[C@@H]([C@@H]2OC3=C(C=C4)O)O)O)N(C)CC5\\n1303, hydroxindasol, 1, C1=C(O)C=CC2=C1C(=C(C)[N]2CC3=CC=C(OC)C=C3)CCN\\n1304, ibrotamide, 1, C(C(C(C)C)(C(N)=O)Br)C\\n1305, iclazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(N2CCOCC3CC3)=O)C4=CC=CC=C4\\n1306, ifoxetine, 1, [C@@H]2(OC1=C(C(=CC=C1)C)C)[C@@H](CNCC2)O\\n1307, ilepcimide, 1, C1=C3C(=CC=C1\\\\C=C\\\\C(N2CCCCC2)=O)OCO3\\n1308, iloperidone, 1, C1=CC(=CC2=C1C(=NO2)C4CCN(CCCOC3=C(C=C(C(C)=O)C=C3)OC)CC4)F\\n1309, imiclopazine, 1, C1=C(Cl)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN5CCN(CCN4C(N(C)CC4)=O)CC5\\n1310, imidoline, 1, C2=C(N1C(N(CCN(C)C)CC1)=O)C=CC=C2Cl\\n1311, imipraminoxide, 1, C1=CC=CC2=C1N(C3=C(CC2)C=CC=C3)CCC[N+](C)(C)[O-]\\n1312, indatraline, 1, [C@@H]2(C1=CC=CC=C1[C@H](NC)C2)C3=CC=C(Cl)C(=C3)Cl.[H+].[Cl-]\\n1313, indeloxazine, 1, C2=C(C1=C(C=CC1)C=C2)OCC3CNCCO3\\n1314, indomethacin(indometacin), 1, C1=CC(=CC2=C1[N](C(=C2CC(O)=O)C)C(C3=CC=C(Cl)C=C3)=O)OC\\n1315, indriline, 1, C1=CC=CC3=C1C(C2=CC=CC=C2)(C=C3)CCN(C)C\\n1316, iproclozide, 1, C1=CC(=CC=C1OCC(NNC(C)C)=O)Cl\\n1317, iprozilamine, 1, CSC1=C(N=C(N=C1N2CCN(C)CC2)NC(C)C)Cl\\n1318, ipsapirone, 1, [H+].C1=CC=CC2=C1[S](=O)(=O)N(C2=O)CCCCN4CCN(C3=NC=CC=N3)CC4.[Cl-]\\n1319, isocarboxazid, 1, C1=C(C)ON=C1C(NNCC2=CC=CC=C2)=O\\n1320, isoflupredone, 1, C(C(C1(CCC2C4C(C(CC12C)O)(F)C3(C=CC(C=C3CC4)=O)C)O)=O)O\\n1321, isoflupredone-acetate, 1, [C@H]23[C@@]([C@@]1(C(=CC(=O)C=C1)CC2)C)([C@H](C[C@]4([C@H]3CC[C@@]4(C(COC(C)=O)=O)O)C)O)F\\n1322, itrocinonide, 1, [C@]34([C@H]([C@H]2C(F)([C@@]1(C(=CC(=O)C=C1)[C@@H](F)C2)C)[C@@H](O)C3)C[C@H]5O[C@H](O[C@@]45C(O[C@@H](OC(OCC)=O)C)=O)CCC)C\\n1323, ivoqualine, 1, [C@H]3(CCCC1=C2C(=NC=C1)C=CC(=C2)OC)[C@@H](CNCC3)C=C\\n1324, ketazocine, 1, [C@H]12N(CC[C@](C1C)(C3=C(C2=O)C=CC(=C3)O)C)CC4CC4\\n1325, ketazolam, 1, C1=C(Cl)C=CC2=C1C3(N(CC(=O)N2C)C(C=C(O3)C)=O)C4=CC=CC=C4\\n1326, ketobemidone, 1, C1=C(C=CC=C1C2(CCN(C)CC2)C(CC)=O)O\\n1327, lesopitron, 1, C1=CC=NC(=N1)N3CCN(CCCC[N]2C=C(Cl)C=N2)CC3\\n1328, levophenacylmorphan, 1, C5=C(C(CN1C2C4C(CC1)(C3=C(C2)C=CC(=C3)O)CCCC4)=O)C=CC=C5\\n1329, linopirdine, 1, C1=CC=CC2=C1C(C(N2C3=CC=CC=C3)=O)(CC4=CC=NC=C4)CC5=CC=NC=C5\\n1330, lisuride, 1, [C@@H]14C(=C[C@H](NC(N(CC)CC)=O)CN1C)C2=CC=CC3=C2C(=C[NH]3)C4\\n1331, litracen, 1, C1=CC=CC3=C1C(C2=C(C=CC=C2)C3(C)C)=CCCNC\\n1332, lofendazam, 1, C1=C(Cl)C=CC3=C1N(C2=CC=CC=C2)C(CCN3)=O\\n1333, lofentanil, 1, [C@]3(N(C1=CC=CC=C1)C(=O)CC)(C(=O)OC)[C@H](CN(CCC2=CC=CC=C2)CC3)C\\n1334, lofepramine, 1, [H+].C1=CC=CC2=C1N(C3=C(CC2)C=CC=C3)CCCN(CC(=O)C4=CC=C(Cl)C=C4)C.[Cl-]\\n1335, lometraline, 1, C1=C(C2=C(C(=C1)OC)CCCC2N(C)C)Cl\\n1336, lopirazepam, 1, C3=C(C1=NC(C(=O)NC2=C1N=C(Cl)C=C2)O)C(=CC=C3)Cl\\n1337, loprazolam, 1, C1=CC(=CC4=C1N2C(=NC(/C2=O)=C\\\\N3CCN(C)CC3)CN=C4C5=C(C=CC=C5)Cl)[N+](=O)[O-]\\n1338, loratadine, 1, C1=CC(=CC2=C1C(C3=C(CC2)C=CC=N3)=C4CCN(C(OCC)=O)CC4)Cl\\n1339, lormetazepam, 1, C1=C(Cl)C=CC2=C1C(=NC(O)C(N2C)=O)C3=CC=CC=C3Cl\\n1340, lorpiprazole, 1, [C@H]23C1=NN=C([N]1C[C@H]2CCC3)CCN4CCN(CC4)C5=CC=CC(=C5)C(F)(F)F\\n1341, lortalamine, 1, [C@@]123OC4=C(C(C1CN(CC2)C)CC(=O)N3)C=C(Cl)C=C4\\n1342, lorzafone, 1, C1=CC(=CC(=C1N(C(=O)CNC(=O)CN)C)C(=O)C2=CC=CC=C2Cl)Cl.O\\n1343, losigamone, 1, C2=C(C(C1C(=CC(O1)=O)OC)O)C(=CC=C2)Cl\\n1344, loteprednol(loteprednol-etabonate), 1, CCOC(=O)OC3(CCC4C2CCC1=CC(=O)C=CC1(C)C2C(O)CC34C)C(=O)OCCl\\n1345, loxapine, 1, C1=C(Cl)C=CC2=C1C(=NC3=C(O2)C=CC=C3)N4CCN(CC4)C\\n1346, lufuradom, 1, C1=CC(=CC2=C1C(=NCC(N2C)CNC(C3=COC=C3)=O)C4=CC=CC=C4)F\\n1347, mazindol, 1, C1=CC=CC2=C1C(N3C2=NCC3)(C4=CC=C(C=C4)Cl)O\\n1348, mazipredone, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)CC[C@@]4(C(CN5CCN(C)CC5)=O)O)C\\n1349, mebanazine, 1, C1=CC=CC=C1C(NN)C\\n1350, mebutamate, 1, C(C(C(CC)C)(COC(N)=O)C)OC(N)=O\\n1351, meclonazepam, 1, [C@H]2(N=C(C1=C(C=CC(=C1)[N+](=O)[O-])NC2=O)C3=CC=CC=C3Cl)C\\n1352, mecloralurea, 1, CNC(NC(C(Cl)(Cl)Cl)O)=O\\n1353, meclorisone(meclorisone-dibutyrate), 1, [C@@]23(Cl)C(C1C([C@](O)([C@@H](C1)C)C(=O)CO)(CC2Cl)C)CCC4=CC(=O)C=CC34C\\n1354, medrysone, 1, [C@@H]23[C@H]([C@H]1[C@]([C@@H](C(C)=O)CC1)(C)C[C@@H]2O)C[C@H](C)C4=CC(=O)CC[C@]34C\\n1355, mefexamide, 1, [H+].C1=C(OCC(=O)NCCN(CC)CC)C=CC(=C1)OC.[Cl-]\\n1356, melitracen, 1, C1=CC=CC3=C1C(C2=C(C=CC=C2)C3(C)C)=CCCN(C)C\\n1357, melperone, 1, C1=CC(=CC=C1C(CCCN2CCC(C)CC2)=O)F\\n1358, memantine, 1, CC12CC3(CC(C1)(CC(C2)C3)N)C\\n1359, mephenoxalone, 1, C1=C(C(=CC=C1)OC)OCC2CNC(O2)=O\\n1360, meprotixol, 1, C1=C(OC)C=CC3=C1C(C2=C(C=CC=C2)S3)(CCCN(C)C)O\\n1361, mequitazine, 1, C1=CC=CC2=C1SC5=C(N2CC3C4CCN(C3)CC4)C=CC=C5\\n1362, mesulergine, 1, [C@@H]34C1=CC=CC2=C1C(=C[N]2C)C[C@H]3N(C[C@H](C4)N[S](N(C)C)(=O)=O)C\\n1363, metacetamol, 1, C1=C(C=CC=C1O)NC(C)=O\\n1364, metaglycodol, 1, C1=C(C(C(C)(C)O)(C)O)C=CC=C1Cl\\n1365, methapyrilene, 1, C2=C(N(CC1=CC=CS1)CCN(C)C)N=CC=C2\\n1366, metharbital, 1, C(C1(C(NC(N(C1=O)C)=O)=O)CC)C\\n1367, methitural, 1, C(C1(C(NC(=S)NC1=O)=O)C(CCC)C)CSC\\n1368, methyldesorphine, 1, C5=C3C1=C(OC4C12C(C(N(CC2)C)C3)CC=C4C)C(=C5)O\\n1369, methyldihydromorphine, 1, [C@@H]25CC1=CC=C(C4=C1[C@@]3([C@H]2CC[C@@H]([C@@H]3O4)OC)CCN5C)O\\n1370, methylprednisolone21hemisuccinate(methylprednisolone-hemisuccinate), 1, [C@@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)C)C)[C@H](C3)O)CC4)C)(C(COC(C)=O)=O)OC(CC)=O\\n1371, metiapine, 1, C1=C(C)C=CC3=C1C(=NC2=C(C=CC=C2)S3)N4CCN(C)CC4\\n1372, metofenazate, 1, C1=C(Cl)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN5CCN(CCOC(C4=CC(=C(OC)C(=C4)OC)OC)=O)CC5\\n1373, metopimazine, 1, C1=C([S](C)(=O)=O)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCC(C(N)=O)CC4\\n1374, metopon, 1, [C@@]125C3=C4C[C@H]([C@@H]1CCC([C@@]2(OC3=C(C=C4)O)C)=O)N(C)CC5\\n1375, metoserpate, 1, C5=C4C3=C(C2N(CC1CC(OC)C(C(C1C2)C(OC)=O)OC)CC3)[NH]C4=CC(=C5)OC\\n1376, metoxepin, 1, C1=C(OC)C=CC3=C1C(=CC2=C(C=CC=C2)O3)N4CCN(C)CC4\\n1377, metralindole, 1, C1=C(OC)C=CC2=C1C4=C3[N]2CCN(C3=NCC4)C\\n1378, mevastatin, 1, [C@@H]23[C@@H](CC[C@@H]1C[C@H](CC(O1)=O)O)[C@H](C=CC2=CCC[C@@H]3OC([C@H](CC)C)=O)C\\n1379, mexazolam, 1, C1=C(Cl)C=CC3=C1C2(N(C(C)CO2)CC(=O)N3)C4=CC=CC=C4Cl\\n1380, mezilamine, 1, CSC2=C(N1CCN(C)CC1)N=C(NC)N=C2Cl\\n1381, miconazole, 1, C1=C(C=CC(=C1Cl)COC(C[N]2C=NC=C2)C3=C(C=C(C=C3)Cl)Cl)Cl\\n1382, midaflur, 1, C1(=NC(NC1(C(F)(F)F)C(F)(F)F)(C(F)(F)F)C(F)(F)F)N\\n1383, milacemide, 1, C(NCC(N)=O)CCCC\\n1384, milameline, 1, CN1CC(=CCC1)\\\\C=N\\\\OC\\n1385, minaprine, 1, C1=C(C(=NN=C1C2=CC=CC=C2)NCCN3CCOCC3)C\\n1386, minaxolone, 1, [C@@H]23[C@H]([C@H]1[C@]([C@@H](C(C)=O)CC1)(C)C[C@H]2N(C)C)CC[C@@H]4[C@@]3(C[C@H](OCC)[C@H](C4)O)C\\n1387, mirisetron, 1, [C@@H]45N([C@@H](CC(NC(=O)C1=CN(C2=C(C1=O)C=CC=C2)C3CCCCC3)C4)CC5)C\\n1388, moclobemide, 1, C2=C(C(NCCN1CCOCC1)=O)C=CC(=C2)Cl\\n1389, modafinil, 1, C2=C(C(C1=CC=CC=C1)[S](CC(N)=O)=O)C=CC=C2\\n1390, mometasone(mometasone-furoate), 1, [C@H]34[C@H]2[C@@](Cl)([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@@H](O)C[C@@]3([C@](O)([C@@H](C4)C)C(=O)CCl)C\\n1391, monometacrine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)C3(C)C)CCCNC\\n1392, morsuximide, 1, C3=C(C2(C(N(CN1CCOCC1)C(C2)=O)=O)C)C=CC=C3\\n1393, mosapramine, 1, C1=C(Cl)C=CC5=C1N(CCCN4CCC3(N2C(CCCC2)NC3=O)CC4)C6=C(CC5)C=CC=C6\\n1394, motrazepam, 1, C1=C([N+](=O)[O-])C=CC2=C1C(=NCC(N2COC)=O)C3=CC=CC=C3\\n1395, moxadolen, 1, [C@@H]13C=C[C@@H]([C@@]2([C@H]1[C@H](OC2=O)OC(NC)=O)C)C3\\n1396, moxiraprine, 1, C(N1CCOCC1)CNC2=NNC(C=C2C)=C3C=CC(=O)C=C3\\n1397, myrophine, 1, [C@@]126C3=C4C[C@H]([C@@H]1C=C[C@@H]([C@@H]2OC3=C(C=C4)OCC5=CC=CC=C5)OC(CCCCCCCCCCCCC)=O)N(CC6)C\\n1398, nabazenil, 1, C1=C(C(C(CCCCC)C)C)C=C(C2=C1OC(C3=C2CC(C)CC3)(C)C)OC(CCCN4CCCCCC4)=O\\n1399, nabilone, 1, [C@H]23C1=C(C=C(C(CCCCCC)(C)C)C=C1O)OC([C@@H]2CCC(C3)=O)(C)C\\n1400, nafimidone, 1, C2=C(C(C[N]1C=CN=C1)=O)C=CC3=CC=CC=C23\\n1401, naflocort, 1, [C@]14([C@H]([C@H]3[C@](F)([C@@H](O)C1)[C@@]2(C(=CC(=O)C=C2)CC3)C)C[C@@H]6[C@]4(CC5=CC=CC=C5C6)C(=O)CO)C.O\\n1402, nalbuphine, 1, OC1CCC2(O)C3Cc5ccc(O)c6OC1C2(CCN3CC4CCC4)c56\\n1403, nalmefene, 1, [C@@]126C3=C4C[C@H]([C@@]1(CCC([C@@H]2OC3=C(C=C4)O)=C)O)N(CC5CC5)CC6\\n1404, nalmexone, 1, [C@@]125C3=C4C[C@H]([C@@]1(CCC([C@@H]2OC3=C(C=C4)O)=O)O)N(CC=C(C)C)CC5\\n1405, napactadine, 1, C1=C(C=CC2=C1C=CC=C2)CC([NH2+]C)=NC.[Cl-]\\n1406, napamezole, 1, C1=C2C(=CC=C1)CCC(=C2)CC3=NCNC3\\n1407, naprodoxime, 1, C2=C1C(=CC=CC1=CC=C2)OC(\\\\C(=N\\\\O)N)C\\n1408, nefazodone, 1, [H+].C4=C(N3CCN(CCCN1N=C(N(C1=O)CCOC2=CC=CC=C2)CC)CC3)C=CC=C4Cl.[Cl-]\\n1409, nefiracetam, 1, C1=CC=C(C(=C1C)NC(CN2C(CCC2)=O)=O)C\\n1410, neflumozide, 1, Fc5ccc4c(CCCN1CCC(CC1)n2c(=O)[nH]c3ccccc23)noc4c5\\n1411, nialamide, 1, C1=CN=CC=C1C(NNCCC(NCC2=CC=CC=C2)=O)=O\\n1412, niaprazine, 1, C1=CC(=CC=C1N3CCN(CCC(NC(C2=CC=CN=C2)=O)C)CC3)F\\n1413, nicocortonide, 1, [C@]125OC(O[C@@]([C@]1(C[C@H](O)[C@H]3[C@H]2CCC4=CC(=O)CC[C@]34C)C)(CC5)C(=O)COC(=O)C6=CC=NC=C6)\\\\C=C/C\\n1414, nicomorphine, 1, [C@@]127C3=C4C[C@H]([C@@H]1C=C[C@@H]([C@@H]2OC3=C(C=C4)OC(C5=CN=CC=C5)=O)OC(C6=CN=CC=C6)=O)N(C)CC7\\n1415, nicopholine, 1, C1=CC=NC=C1C(N2CCOCC2)=O\\n1416, nicotinamide, 1, C(C1=CN=CC=C1)(=O)N\\n1417, nikethamide, 1, C1=CC=NC=C1C(N(CC)CC)=O\\n1418, nimetazepam, 1, C1=C([N+]([O-])=O)C=CC2=C1C(=NCC(=O)N2C)C3=CC=CC=C3\\n1419, nisobamate, 1, C(C(COC(N)=O)(C(CC)C)C)OC(NC(C)C)=O\\n1420, nisoxetine, 1, C1=CC=CC=C1C(OC2=CC=CC=C2OC)CCNC\\n1421, nitrafudam, 1, NC(=N)c1ccc(o1)c2ccccc2N(=O)=O\\n1422, nivazol(nivacortol), 1, [C@H]12C([C@@](O)(CC1)C#C)(CC[C@H]3C2CCC4=CC5=C(CC34C)C=N[N]5C6=CC=C(F)C=C6)C\\n1423, nomelidine, 1, C1=CC(=CC=C1\\\\C(C2=CC=CN=C2)=C\\\\CNC)Br\\n1424, nomifensine, 1, C1=CC=C(C3=C1C(C2=CC=CC=C2)CN(C)C3)N\\n1425, nonapyrimine, 1, C1=C[NH]C2=C1C(=NC=N2)NCCCCCCCCC\\n1426, norflurane, 1, C(C(F)(F)F)F\\n1427, norlevorphanol, 1, C1=C(O)C=CC3=C1C24C(C(NCC2)C3)CCCC4\\n1428, normorphine, 1, C1=CC(=C3C2=C1CC5C4C2(C(O3)C(O)C=C4)CCN5)O\\n1429, nortetrazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(=O)N2)C3=CCCCC3\\n1430, noscapine, 1, [C@@H]3(C1=C(C=C2C(=C1OC)OCO2)CCN3C)[C@@H]5C4=C(C(=C(OC)C=C4)OC)C(O5)=O\\n1431, noxiptiline, 1, C1=CC=CC2=C1C(C3=C(CC2)C=CC=C3)=NOCCN(C)C\\n1432, nuclotixene, 1, C1=C(Cl)C=CC3=C1\\\\C(C2=C(C=CC=C2)S3)=C\\\\C4C5CCN(C4)CC5\\n1433, ocaperidone, 1, C1=CC(=CC2=C1C(=NO2)C5CCN(CCC3=C(N=C4N(C3=O)C=CC=C4C)C)CC5)F\\n1434, ocinaplon, 1, C1=N[N]3C(=C1C(C2=NC=CC=C2)=O)N=CC=C3C4=CC=NC=C4\\n1435, octamoxin, 1, C(C(NN)C)CCCCC\\n1436, octriptyline, 1, CNCCC=C3c1ccccc1C2CC2c4ccccc34\\n1437, omiloxetine, 1, [C@H]2([C@H](CN(CC(C1=CC=C(F)C=C1)=O)CC2)COC4=CC3=C(OCO3)C=C4)C5=CC=C(F)C=C5\\n1438, oxaflozane, 1, C2=C(C1CN(CCO1)C(C)C)C=CC(=C2)C(F)(F)F\\n1439, oxcarbazepine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)CC3=O)C(N)=O\\n1440, oxifentorex, 1, C1=CC=CC=C1C[N+](C(CC2=CC=CC=C2)C)(C)[O-]\\n1441, oxiracetam, 1, C(N1CC(O)CC1=O)C(N)=O\\n1442, oxisopred, 1, [C@H]34C1[C@@H](C2(C(C(=O)CC1)C(=O)C=C2)C)C(O)CC3([C@](O)(CC4)C(OC)=O)C\\n1443, oxitriptan, 1, C1=C(O)C=CC2=C1C(=C[NH]2)CC(C(O)=O)N\\n1444, oxitriptyline, 1, CN(C)C(=O)COC2c1ccccc1CCc3ccccc23\\n1445, pagoclone, 1, C4=C3C(N(C1=CC=C2C(=N1)N=C(Cl)C=C2)C(=O)C3=CC=C4)CC(CCC(C)C)=O\\n1446, panadiplon, 1, C1=NC(=C3[N]1C2=C(C=CC=C2)N(C3=O)C(C)C)C4=NOC(=N4)C5CC5\\n1447, panamesine, 1, [C@H]5(CN3CCC(C2=CC1=C(OCO1)C=C2)(CC3)O)CN(C4=CC=C(OC)C=C4)C(O5)=O\\n1448, pancopride, 1, C3=C(C(NC1C2CCN(C1)CC2)=O)C(=CC(=C3Cl)N)OCC4CC4\\n1449, panuramine, 1, C1=C4C(=CC=C1CN3CCC(NC(NC(C2=CC=CC=C2)=O)=O)CC3)C=CC=C4\\n1450, paraldehyde, 1, CC1OC(C)OC(O1)C\\n1451, paramethasone-acetate, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)[C@H](C3)O)C[C@H]([C@@]4(C(COC(C)=O)=O)O)C)C\\n1452, pargyline, 1, C1=CC=CC=C1CN(CC#C)C\\n1453, paroxetine, 1, [C@@H]3([C@H](COC1=CC=C2C(=C1)OCO2)CNCC3)C4=CC=C(C=C4)F\\n1454, pazinaclone, 1, C1=CC=CC2=C1C(N(C2=O)C3=NC4=C(C=C3)C=CC(=N4)Cl)CC(=O)N6CCC5(OCCO5)CC6\\n1455, penfluridol, 1, C4=C(C(CCCN1CCC(CC1)(C2=CC=C(C(=C2)C(F)(F)F)Cl)O)C3=CC=C(C=C3)F)C=CC(=C4)F\\n1456, pentabamate, 1, CC(C(C(OC(N)=O)C)C)OC(N)=O\\n1457, pentamorphone, 1, [C@]135[C@@]4([C@@H](CC2=C1C(=C(C=C2)O)O[C@H]3C(=O)C=C4)N(C)CC5)NCCCCC\\n1458, penthrichloral, 1, C(C1(COC(C(Cl)(Cl)Cl)OC1)CO)O\\n1459, pentiapine, 1, CN1CCN(CC1)C3=Nc2ccccc2Sc4nccn34\\n1460, perafensine, 1, C2=C1C(=NC(=CC1=CC=C2)N3CCNCC3)C4=CC=CC=C4\\n1461, perospirone, 1, [C@@H]45C(N(CCCCN3CCN(C1=NSC2=C1C=CC=C2)CC3)C([C@@H]4CCCC5)=O)=O\\n1462, perphenazine, 1, C1=C(Cl)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCN(CCO)CC4\\n1463, petrichloral, 1, C(C(COC(C(Cl)(Cl)Cl)O)(COC(C(Cl)(Cl)Cl)O)COC(C(Cl)(Cl)Cl)O)OC(C(Cl)(Cl)Cl)O\\n1464, phenacemide, 1, C1=CC=CC=C1CC(NC(N)=O)=O\\n1465, phenazocine, 1, C1=C(C=CC2=C1C3(C)C(C)C(C2)N(CC3)CCC4=CC=CC=C4)O\\n1466, pheneturide, 1, C1=CC=CC=C1C(C(NC(N)=O)=O)CC\\n1467, phenmetrazine, 1, C2=C(C1C(C)NCCO1)C=CC=C2\\n1468, phenoxypropazine(fenoxypropazine), 1, C1=C(OCC(NN)C)C=CC=C1\\n1469, phensuximide, 1, C1=CC=CC=C1C2C(N(C)C(C2)=O)=O\\n1470, phenyl-salicylate, 1, C1(=CC=CC=C1)OC2=CC=CC=C2C(=O)[O-]\\n1471, phenytoin, 1, C1=CC=CC=C1C3(C2=CC=CC=C2)NC(=O)NC3=O\\n1472, phetharbital, 1, C1=CC=CC=C1N2C(C(C(NC2=O)=O)(CC)CC)=O\\n1473, piberaline, 1, C3=C(C(N2CCN(CC1=CC=CC=C1)CC2)=O)N=CC=C3\\n1474, picenadol, 1, [C@]2(C1=CC(=CC=C1)O)([C@H](CN(C)CC2)C)CCC\\n1475, pidolacetamol, 1, [C@H]2(C(OC1=CC=C(NC(C)=O)C=C1)=O)NC(=O)CC2\\n1476, pipequaline, 1, C2=C(C1=CC=CC=C1)N=C4C(=C2CCC3CCNCC3)C=CC=C4\\n1477, piperacetazine, 1, C1=C(C(C)=O)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCC(CCO)CC4\\n1478, piperilate(pipethanate), 1, C3=C(C(C1=CC=CC=C1)(C(OCCN2CCCCC2)=O)O)C=CC=C3\\n1479, pipofezine, 1, C2=C(N1CCN(C)CC1)N=NC3=C2N(C4=C(O3)C=CC=C4)C\\n1480, pipotiazine-palmitate, 1, C1=C([S](N(C)C)(=O)=O)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCC(CCOC(CCCCCCCCCCCCCCC)=O)CC4\\n1481, piquindone, 1, [C@H]12C(C3=C(C[C@@H]1CCN(C2)C)[NH]C(=C3CC)C)=O\\n1482, piracetam, 1, C(N1C(CCC1)=O)C(N)=O\\n1483, pirandamine, 1, C1=CC=CC2=C1C3=C(C2)CCOC3(CCN(C)C)C\\n1484, piriqualone, 1, C1=CC=CC(=C1N2C(=NC3=C(C2=O)C=CC=C3)\\\\C=C\\\\C4=NC=CC=C4)C\\n1485, piritramide, 1, C1=CC=CC=C1C(C2=CC=CC=C2)(CCN4CCC(N3CCCCC3)(C(N)=O)CC4)C#N\\n1486, pirlindole, 1, [N]14C3=C(C2=C1C=CC(=C2)C)CCCC3NCC4.[H+].[Cl-]\\n1487, pivagabine, 1, C(NC(C(C)(C)C)=O)CCC(O)=O\\n1488, pivoxazepam, 1, C1=C(Cl)C=CC2=C1C(=NC(C(=O)N2)OC(C(C)(C)C)=O)C3=CC=CC=C3\\n1489, pramiracetam, 1, C(N1C(CCC1)=O)C(NCCN(C(C)C)C(C)C)=O\\n1490, prazitone, 1, C1=CC=CC=C1C3(CC2CCCCN2)C(NC(=O)NC3=O)=O\\n1491, prednicarbate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)CC4)C)(C(COC(CC)=O)=O)OC(OCC)=O\\n1492, prednisolamate, 1, C(C(C4(C3(C(C2C(C1(C(=CC(=O)C=C1)CC2)C)C(C3)O)CC4)C)O)=O)OC(CN(CC)CC)=O\\n1493, prednisolone-steaglate, 1, [C@H]34[C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@@H](O)C[C@@]3([C@](C(COC(COC(CCCCCCCCCCCCCCCCC)=O)=O)=O)(O)CC4)C\\n1494, prednisolone-tebutate, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)CC[C@@]4(C(COC(CC(C)(C)C)=O)=O)O)C\\n1495, prednylidene, 1, C(C(C3(C(CC4C2CCC1=CC(C=CC1(C)C2C(CC34C)O)=O)=C)O)=O)O\\n1496, pregabalin, 1, [C@@H](CC(=O)O)(CC(C)C)CN\\n1497, pregnenolone-succinate, 1, C(C(OC4CC3=CCC2C1C(C(CC1)C(C)=O)(CCC2C3(CC4)C)C)=O)CC(O)=O\\n1498, pridefine, 1, [H+].C3=C(C(=C1CCN(C1)CC)C2=CC=CC=C2)C=CC=C3.[Cl-]\\n1499, procarbazine, 1, C1=CC(=CC=C1C(NC(C)C)=O)CNNC\\n1500, procymate, 1, C(C(C1CCCCC1)OC(N)=O)C\\n1501, proflazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(N2CC(CO)O)=O)C3=CC=CC=C3F\\n1502, progabide, 1, C2=C(\\\\C(NCCCC(=O)N)=C/1C=C(F)C=CC1=O)C=CC(=C2)Cl\\n1503, prolintane, 1, C1=CC=CC=C1CC(N2CCCC2)CCC\\n1504, promoxolane, 1, C(C1OC(C(C)C)(C(C)C)OC1)O\\n1505, propacetamol, 1, C1=CC(=CC=C1OC(CN(CC)CC)=O)NC(C)=O\\n1506, propanidid, 1, [CH2-][CH-]C\\n1507, propizepine, 1, C1=CC=CC2=C1N(C(C3=C(N2)N=CC=C3)=O)CC(N(C)C)C\\n1508, propofol, 1, C1=CC=C(C(=C1C(C)C)O)C(C)C\\n1509, propyperone, 1, C1=CC(=CC=C1C(CCCN3CCC(N2CCCCC2)(CC3)C(CC)=O)=O)F\\n1510, prothixene, 1, C1=CC=CC3=C1C(C2=C(C=CC=C2)S3)=CCCN(C)C\\n1511, proxibarbal, 1, C(C1(C(NC(=O)NC1=O)=O)CC=C)C(C)O\\n1512, proxorphan, 1, Oc5ccc4CC2C1CCOCC1(CCN2CC3CC3)c4c5\\n1513, proxyphylline, 1, C1=NC2=C([N]1CC(C)O)C(N(C)C(N2C)=O)=O\\n1514, psilocybine, 1, C1=CC=C(C2=C1[NH]C=C2CCN(C)C)O[P](O)(O)=O\\n1515, pyridarone, 1, C2=C(C1=CC=NC=C1)OC3=C2C=CC=C3\\n1516, pyritinol, 1, C1=C(C(=C(C(=N1)C)O)CO)CSSCC2=CN=C(C(=C2CO)O)C\\n1517, quetiapine, 1, C1=CC=CC3=C1C(=NC2=CC=CC=C2S3)N4CCN(CCOCCO)CC4\\n1518, quilostigmine, 1, [C@@]45(C1=C(C=CC(=C1)OC(N3CC2=CC=CC=C2CC3)=O)N([C@H]4N(C)CC5)C)C\\n1519, quinagolide, 1, C1=CC=C(C2=C1CC3C(C2)CC(N[S](N(CC)CC)(=O)=O)CN3CCC)O\\n1520, ramciclane, 1, C1=CC=CC=C1CC2(C3(C(C)(C)C(C2)CC3)C)OCCN(C)C\\n1521, rasagiline, 1, [C@@H]1(C2=C(CC1)C=CC=C2)NCC#C\\n1522, reclazepam, 1, C1=CC(=CC3=C1N(C2=NC(=O)CO2)CCN=C3C4=CC=CC=C4Cl)Cl\\n1523, remacemide, 1, C1=CC=CC=C1C(CC2=CC=CC=C2)(NC(CN)=O)C\\n1524, remifentanil, 1, C1=CC=CC=C1N(C2(CCN(CCC(OC)=O)CC2)C(OC)=O)C(CC)=O\\n1525, resocortol, 1, CCC(=O)C1(O)CCC2C3CCC4=CC(=O)CCC4(C)C3C(O)CC21C\\n1526, retigabine, 1, C1=CC(=CC(=C1NC(OCC)=O)N)NCC2=CC=C(C=C2)F\\n1527, ricasetron, 1, C3(NC(N1C2=C(C(C)(C)C1)C=CC=C2)=O)C[C@@H]4N(C)[C@H](C3)CC4\\n1528, rilmazafone, 1, C2=C([N]1C(=NC(=N1)C(N(C)C)=O)CNC(CN)=O)C(=CC(=C2)Cl)C(C3=C(C=CC=C3)Cl)=O\\n1529, rimcazole, 1, CC4CN(CCCn2c1ccccc1c3ccccc23)CC(C)N4\\n1530, rimexolone, 1, [C@H]12[C@@]([C@](C(CC)=O)(C)[C@@H](C1)C)(C[C@H](O)[C@H]3[C@H]2CCC4=CC(=O)C=C[C@]34C)C\\n1531, ripazepam, 1, C1=CC=CC=C1C2=NCC(NC3=C2[N](N=C3C)CC)=O\\n1532, rivastigmine, 1, [C@H](C1=CC(=CC=C1)OC(N(CC)C)=O)(N(C)C)C\\n1533, robalzotan, 1, [C@@H]3(N(C1CCC1)C2CCC2)CC4=C(OC3)C(=CC=C4C(=O)N)F\\n1534, rociclovir, 1, C1=NC2=C([N]1COC(COC(C)C)COC(C)C)N=C(N)N=C2\\n1535, rofelodine, 1, C3=C(C2CN1C(CCN=C1C2)=O)C=CC=C3\\n1536, rofleponide, 1, [C@H]23[C@@]([C@@]1(O[C@@H](O[C@@H]1C2)CCC)C(=O)CO)(C[C@H](O)[C@@]4(F)[C@H]3C[C@H](F)C5=CC(=O)CC[C@]45C)C\\n1537, rolicyprine, 1, C3=C(C2C(NC(=O)C1NC(CC1)=O)C2)C=CC=C3\\n1538, rolipram, 1, C1=C(C(=CC=C1C2CNC(C2)=O)OC)OC3CCCC3\\n1539, romergoline, 1, [C@@H]25C(=C[C@@H](CN1CC(NC(C1)=O)=O)CN2C)C3=CC=CC4=C3C(=C[NH]4)C5\\n1540, romifidine, 1, C1=C(C(=C(C=C1)Br)NC2=NCCN2)F\\n1541, ropinirole, 1, C2=C(C1=C(NC(C1)=O)C=C2)CCN(CCC)CCC\\n1542, roxoperone, 1, C1=CC(=CC=C1C(CCCN3CCC2(C(N(C)C(C2)=O)=O)CC3)=O)F\\n1543, sabcomeline, 1, CON=C(C#N)C1CN2CCC1CC2\\n1544, salicin, 1, [C@@H]2(OC1=C(C=CC=C1)CO)O[C@@H]([C@@H](O)[C@@H]([C@H]2O)O)CO\\n1545, salicylamide, 1, C(C1=CC=CC=C1O)(=O)N\\n1546, salsalate(salsalic-acid), 1, C1=CC=CC(=C1OC(C2=C(C=CC=C2)O)=O)C(O)=O\\n1547, sameridine, 1, C1=CC=CC=C1C2(C(N(CC)C)=O)CCN(CCCCCC)CC2\\n1548, saripidem, 1, C3=C(C1=C([N]2C(=N1)C=CC=C2)CN(C(CCC)=O)C)C=CC(=C3)Cl\\n1549, savoxepin, 1, N#Cc5ccc4Oc1ccccc1C2=C(CCN(CC2)CC3CCCC3)c4c5\\n1550, selegiline, 1, [C@@H](CC1=CC=CC=C1)(N(CC#C)C)C\\n1551, semorphone, 1, [C@@]125C3=C4C[C@H]([C@@]1(CCC([C@@H]2OC3=C(C=C4)O)=O)O)N(CCOC)CC5\\n1552, sertindole, 1, C2=C(C1=C(C=CC(=C1)Cl)[N]2C3=CC=C(C=C3)F)C5CCN(CCN4C(NCC4)=O)CC5\\n1553, setiptiline, 1, C1=CC=CC3=C1C4=C(C2=C(C=CC=C2)C3)CN(C)CC4\\n1554, sibopirdine, 1, C1=CC=NC2=C1C(C3=C2N=CC=C3)(CC4=CC=NC=C4)CC5=CN=CC=C5.O\\n1555, sibutramine, 1, C1=CC(=CC=C1C2(C(CC(C)C)N(C)C)CCC2)Cl\\n1556, sifaprazine, 1, C1=CC=CC(=C1N2CCN(C)CC2)CC3=CC=CC=C3\\n1557, simvastatin, 1, [C@H]12[C@@H](OC(C(CC)(C)C)=O)C[C@@H](C)C=C1C=C[C@@H]([C@@H]2CC[C@@H]3C[C@H](CC(O3)=O)O)C\\n1558, soretolide, 1, C1=CC=C(C(=C1C)C(NC2=NOC(=C2)C)=O)C\\n1559, spiperone, 1, C1=CC=CC=C1N2C4(C(NC2)=O)CCN(CCCC(C3=CC=C(F)C=C3)=O)CC4\\n1560, spiradoline, 1, [C@]24(C[C@@H]([C@@H](N(C(CC1=CC(=C(Cl)C=C1)Cl)=O)C)CC2)N3CCCC3)CCCO4\\n1561, spiramide, 1, C4=C(N1C3(C(NC1)=O)CCN(CCCOC2=CC=C(C=C2)F)CC3)C=CC=C4\\n1562, spirilene, 1, C1=CC=CC=C1N2C4(C(NC2)=O)CCN(CC\\\\C=C(C3=CC=C(F)C=C3)\\\\C)CC4\\n1563, spiroxepin, 1, C1=CC=CC2=C1C4(C3=C(OC2)C=CC=C3)OC(CN(C)C)CO4\\n1564, stacofylline, 1, C(C1=NC2=C([N]1C)C(N(C)C(N2C)=O)=O)CCN3CCN(C(N(CC)CC)=O)CC3\\n1565, stiripentol, 1, C1=C2C(=CC=C1\\\\C=C\\\\C(C(C)(C)C)O)OCO2\\n1566, suclofenide, 1, C3=C(N2C(C(C1=CC=CC=C1)CC2=O)=O)C(=CC(=C3)[S](N)(=O)=O)Cl\\n1567, sulfamethoxazole(tiabendazole), 1, C1=CC(=CC=C1[S](NC2=NOC(=C2)C)(=O)=O)N\\n1568, sulfonmethane, 1, C([S](C(C)(C)[S](=O)(=O)CC)(=O)=O)C\\n1569, sulmepride, 1, C1=C(C(=CC=C1[S](N)(=O)=O)OC)C(NCC2N(CCC2)C)=O\\n1570, sulthiame(sultiame), 1, C1=CC(=CC=C1N2[S](CCCC2)(=O)=O)[S](N)(=O)=O\\n1571, sultopride, 1, C1=C([S](CC)(=O)=O)C=CC(=C1C(NCC2N(CCC2)CC)=O)OC\\n1572, sumacetamol, 1, [C@H](C(=O)OC1=CC=C(C=C1)NC(=O)C)(NC(=O)C)CCSC\\n1573, sunepitron, 1, O=C1CCC(=O)N1CC1CCC2CN(CCN2C1)c1ncccn1\\n1574, suproclone, 1, C4=C(N2C(C1=C(SCCS1)C2=O)OC(N3CCN(C(CC)=O)CC3)=O)N=C5C(=C4)C=CC(=N5)Cl\\n1575, suriclone, 1, C4=C(N2C(C1=C(SCCS1)C2=O)OC(N3CCN(C)CC3)=O)N=C5C(=C4)C=CC(=N5)Cl\\n1576, suritozole, 1, C1=C(C=CC=C1C2=NN(C)C(N2C)=S)F\\n1577, taglutimide(biglumide), 1, O=C3C1C(C2CCC1C2)C(=O)N3C4C(NC(=O)CC4)=O\\n1578, talopram, 1, C1=CC=CC3=C1C(C2=CC=CC=C2)(OC3(C)C)CCCNC\\n1579, talsupram, 1, C1=CC=CC3=C1C(C2=CC=CC=C2)(SC3(C)C)CCCNC\\n1580, taltrimide, 1, C1=CC=CC2=C1C(N(CC[S](NC(C)C)(=O)=O)C2=O)=O\\n1581, tameridone, 1, C1=NC5=C([N]1CCN4CCC(C2=C[NH]C3=C2C=CC=C3)CC4)C(N(C)C(N5C)=O)=O\\n1582, tametraline, 1, CNC2CCC(c1ccccc1)c3ccccc23\\n1583, tamitinol, 1, C1=C(C(=C(O)C(=N1)C)CNCC)CSC\\n1584, tamolarizine, 1, C4=C(C(N2CCN(CC(C1=CC=C(C(=C1)OC)OC)O)CC2)C3=CC=CC=C3)C=CC=C4\\n1585, tampramine, 1, CN(C)CCCN2c1ccccc1C(=Nc3cccnc23)c4ccccc4\\n1586, tandamine, 1, C1=CC=CC2=C1C3=C([N]2CC)C(SCC3)(CCN(C)C)C\\n1587, tandospirone, 1, [C@H]13[C@H]([C@H]2C[C@@H]1CC2)C(=O)N(C3=O)CCCCN4CCN(CC4)C5=NC=CC=N5.C(C(O)(C(=O)O)CC(=O)O)C(=O)O\\n1588, taniplon, 1, C2=C(C1=NOC(=N1)C)N=C3[N]2C4=C(C(=N3)OC)CCCC4\\n1589, tasosartan, 1, C3=C(CN2C1=NC(=NC(=C1CCC2=O)C)C)C=CC(=C3)C4=CC=CC=C4C5=N[NH]N=N5\\n1590, tazomeline, 1, C(SC1=NSN=C1C2=CCCN(C2)C)CCCCC\\n1591, tefludazine, 1, [C@@H]2(C3=C([C@H](C1=CC=C(C=C1)F)C2)C=CC(=C3)C(F)(F)F)N4CCN(CCO)CC4\\n1592, teflutixol, 1, C1=C(C(F)(F)F)C=CC3=C1C(C2=C(C=C(F)C=C2)S3)CCCN4CCN(CCO)CC4\\n1593, tenilsetam, 1, C2=C(C1C(NCCN1)=O)SC=C2\\n1594, tepirindole, 1, C3=C(C1=CCN(CCC)CC1)C2=CC(=CC=C2[NH]3)Cl\\n1595, terguride, 1, [C@@H]34C1=CC=CC2=C1C(=C[NH]2)C[C@H]3N(C[C@H](C4)NC(N(CC)CC)=O)C\\n1596, tetrabarbital, 1, C(C1(C(CCC)CC)C(NC(=O)NC1=O)=O)C\\n1597, tetrabenazine(tetrabenzine), 1, C1=C(OC)C(=CC3=C1C2N(CC(CC(C)C)C(C2)=O)CC3)OC\\n1598, tetrantoin, 1, C1=CC=CC2=C1CC3(CC2)NC(=O)NC3=O\\n1599, thalidomide, 1, C1=CC=CC3=C1C(N(C2C(NC(=O)CC2)=O)C3=O)=O\\n1600, thebacon, 1, C1=CC(=C4C2=C1CC5C3C2(C(C(=CC3)OC(C)=O)O4)CCN5C)OC\\n1601, thialbarbital, 1, C(C2(C1CCCC=C1)C(NC(=S)NC2=O)=O)C=C\\n1602, thiamylal, 1, C(C1(C(NC(=S)NC1=O)=O)C(CCC)C)C=C\\n1603, thiazesim(tiazesim), 1, C1=CC=CC2=C1N(C(CC(S2)C3=CC=CC=C3)=O)CCN(C)C\\n1604, thiopropazate, 1, C1=C(Cl)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCN(CCOC(C)=O)CC4\\n1605, thioproperazine, 1, C1=CC=CC2=C1N(C3=C(S2)C=CC(=C3)[S](N(C)C)(=O)=O)CCCN4CCN(CC4)C\\n1606, thozalinone(tozalinone), 1, C1=CC=CC=C1C2OC(=NC2=O)N(C)C\\n1607, tiagabine, 1, [C@@H]3(CN(CCC=C(C1=C(C=CS1)C)C2=C(C=CS2)C)CCC3)C(O)=O\\n1608, tianeptine, 1, C1=CC(=CC2=C1C(C3=C(N(C)[S]2(=O)=O)C=CC=C3)NCCCCCCC(O)=O)Cl\\n1609, tienocarbine, 1, C1=C4C(=C2C(=C1)SC=C2C)C3=C(CCN(C3)C)[NH]4\\n1610, tifemoxone, 1, C2=C(OCC1OC(NCC1)=S)C=CC=C2\\n1611, tilidine, 1, [H+].C2=C(C1(C(N(C)C)C=CCC1)C(OCC)=O)C=CC=C2.[Cl-]\\n1612, timelotem, 1, C1=C(F)C=CC2=C1N4C(CN=C2C3=CC=CS3)CN(C)CC4\\n1613, timiperone, 1, C1=CC=CC4=C1N(C3CCN(CCCC(C2=CC=C(F)C=C2)=O)CC3)C(N4)=S\\n1614, timirdine, 1, C2=C(N1C(SCC1)=N)C(=CC(=C2)Cl)N\\n1615, timobesone-acetate, 1, [C@H]12[C@@](C(OC(=O)C)([C@H](C1)C)C(SC)=O)(C[C@H](O)C3(F)[C@H]2CCC4=CC(=O)C=C[C@]34C)C\\n1616, tinidazole, 1, C1=C([N](CC[S](CC)(=O)=O)C(=N1)C)[N+](=O)[O-]\\n1617, tiomergine, 1, C5=C2C1=C([NH]C=C1CC4C2=CC(CSC3=NC=CC=C3)CN4C)C=C5\\n1618, tioperidone, 1, C4=C(N3CCN(CCCCN2C(C1=C(C=CC=C1)NC2=O)=O)CC3)C(=CC=C4)SCCC\\n1619, tiospirone, 1, C2=C1C(=NSC1=CC=C2)N5CCN(CCCCN3C(CC4(CC3=O)CCCC4)=O)CC5\\n1620, tipredane, 1, [C@]23([C@H]([C@H]1[C@]([C@@](SCC)(SC)CC1)(C)C[C@@H]2O)CCC4=CC(=O)C=C[C@]34C)F\\n1621, tisocromide, 1, C1=C(OC)C(=CC2=C1[S](OC(C2)C(NC(CC(N(C)C)(C)C)C)=O)(=O)=O)OC\\n1622, tixadil, 1, C1=CC=CC=C1CC(C)NCCC3C2=C(C=CC=C2)SC4=C3C=CC=C4\\n1623, tixocortol, 1, [C@H]34[C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)[C@@H](O)C[C@@]3([C@](C(CS)=O)(O)CC4)C\\n1624, tixocortol-pivalate, 1, [C@@H]23[C@H]([C@H]1[C@]([C@](C(CSC(C(C)(C)C)=O)=O)(O)CC1)(C)C[C@@H]2O)CCC4=CC(=O)CC[C@]34C\\n1625, tofenacin, 1, C2=C(C(C1=CC=CC=C1)OCCNC)C(=CC=C2)C\\n1626, tofisopam, 1, C1=C(OC)C(=CC2=C1C(=NN=C(C2CC)C)C3=CC=C(C(=C3)OC)OC)OC\\n1627, toloxychlorinol, 1, C1=CC=CC(=C1OCC(COC(C(Cl)(Cl)Cl)O)OC(C(Cl)(Cl)Cl)O)C\\n1628, tolpiprazole, 1, C1=C(C=CC=C1N3CCN(CCC2=N[NH]C(=C2)C)CC3)C\\n1629, tolquinzole, 1, C1=C(C)C=CC2=C1C3N(CC2)CCC(C3)(CC)O\\n1630, tolufazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(N2CC[S](C3=CC=C(C)C=C3)(=O)=O)=O)C4=CC=CC=C4Cl\\n1631, tonazocine, 1, [C@]1([C@]2(C3=C(C[C@@H]1N(CC2)C)C=CC(=C3)O)C)(CCC(=O)CCCCC)C\\n1632, topiramate, 1, [C@@]23([C@H]([C@H]1[C@H](OC(O1)(C)C)CO2)OC(O3)(C)C)CO[S](N)(=O)=O\\n1633, tracazolate, 1, C1=N[N](C2=C1C(=C(C(=N2)C)C(OCC)=O)NCCCC)CC\\n1634, tralonide, 1, [C@]12(OC(O[C@@H]1CC3C2(C[C@H](Cl)[C@@]4(Cl)C3C[C@H](F)C5=CC(=O)C=CC45C)C)(C)C)C(=O)CF\\n1635, tramadol, 1, [C@H]1([C@](O)(CCCC1)C2=CC(=CC=C2)OC)CN(C)C.[H+].[Cl-]\\n1636, trebenzomine, 1, C1=CC=CC2=C1CC(N(C)C)C(O2)C\\n1637, trefentanil, 1, [H+].C4=C(C2(N(C1=C(F)C=CC=C1)C(=O)CC)CCN(CC2)CCN3N=NN(C3=O)CC)C=CC=C4.[Cl-]\\n1638, triamcinolone-benetonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(COC(C(CNC(C6=CC=CC=C6)=O)C)=O)=O\\n1639, triamcinolone-furetonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(COC(C6=CC7=C(O6)C=CC=C7)=O)=O\\n1640, tribromoethanol, 1, BrCC(O)(Br)Br\\n1641, tricetamide, 1, C1=C(C(=C(C=C1C(NCC(N(CC)CC)=O)=O)OC)OC)OC\\n1642, triclodazol, 1, C3=C(C2(C1=CC=CC=C1)C(N(C(C(Cl)(Cl)Cl)O)CN2)=O)C=CC=C3\\n1643, triclofos, 1, O=[P](O)(OCC(Cl)(Cl)Cl)O\\n1644, triclonide, 1, [C@H]23C([C@@]1(OC(O[C@@H]1C2)(C)C)C(=O)CCl)(C[C@H](Cl)[C@@]4(Cl)C3C[C@H](F)C5=CC(=O)C=CC45C)C\\n1645, trimetrexate, 1, C1=C3C(=C(C(=C1)CNC2=CC(=C(C(=C2)OC)OC)OC)C)C(=NC(=N3)N)N\\n1646, trizoxime, 1, C3=C2N(CC1=CC=CC=C1)C(=O)\\\\C(NNC2=CC=C3)=C(\\\\N)N=O\\n1647, trocimine, 1, C2=C(C(N1CCCCCCC1)=O)C=C(OC)C(=C2OC)OC\\n1648, tryptophan(l-tryptophan), 1, N[C@@H](CC1=C[NH]C2=CC=CC=C12)C(=O)O\\n1649, uldazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(=N2)NOCC=C)C3=CC=CC=C3Cl\\n1650, umespirone, 1, C4=C(N3CCN(CCCCN1C(C2C(C(C1=O)C(=O)N(C2=O)CCCC)(C)C)=O)CC3)C(=CC=C4)OC\\n1651, urethane, 1, NC(=O)OCC\\n1652, valaciclovir, 1, [C@@H](N)(C(OCCOC[N]1C2=C(N=C1)C(=O)N=C(N2)N)=O)C(C)C\\n1653, valdipromide, 1, C(C(C(N)=O)(CCC)CCC)CC\\n1654, valganciclovir(RS790702), 1, [C@@H](N)(C(OCC(OC[N]1C2=C(N=C1)C(=O)N=C(N2)N)CO)=O)C(C)C\\n1655, valproatepivoxil, 1, C(C(C(OCOC(C(C)(C)C)=O)=O)CCC)CC\\n1656, valtrate, 1, [C@@]13([C@@H]2C(=C[C@@H]1OC(CC(C)C)=O)C(=CO[C@H]2OC(CC(C)C)=O)COC(C)=O)CO3\\n1657, velnacrine, 1, C1=CC=CC2=C1C(=C3C(=N2)CCCC3O)N\\n1658, venlafaxine, 1, C1=CC(=CC=C1C(C2(CCCCC2)O)CN(C)C)OC\\n1659, veralipride, 1, C1=C([S](N)(=O)=O)C=C(C(=C1C(NCC2N(CCC2)CC=C)=O)OC)OC\\n1660, vigabatrin, 1, C(C(C=C)N)CC(O)=O\\n1661, viloxazine, 1, C1=CC=CC(=C1OCC2CNCCO2)OCC\\n1662, vinyl-ether, 1, C(=C)OC=C\\n1663, vitamine(alpha-tocopherol), 1, [C@]2(OC1=C(C(=C(C(=C1CC2)C)O)C)C)(CCC[C@@H](CCC[C@@H](CCCC(C)C)C)C)C\\n1664, xaliproden, 1, C1=C(C(F)(F)F)C=CC=C1C2=CCN(CC2)CCC4=CC3=CC=CC=C3C=C4\\n1665, xanthiol, 1, OCCCN4CCN(CCCC2c1ccccc1Sc3ccc(Cl)cc23)CC4\\n1666, xorphanol, 1, [C@]135[C@H]([C@H](N(CC1)CC2CCC2)CC4=C3C=C(O)C=C4)[C@H](CC(C5)=C)C\\n1667, xylazine, 1, C1=C(C(=C(C=C1)C)NC2=NCCCS2)C\\n1668, zafuleptine, 1, C1=CC(=CC=C1CNC(C(C)C)CCCCCC(O)=O)F\\n1669, zalospirone, 1, [C@H]14[C@H]([C@@H]2[C@@H]3[C@H]([C@H]1C=C2)C=C3)C(=O)N(C4=O)CCCCN6CCN(C5=NC=CC=N5)CC6\\n1670, zetidoline, 1, C1=C(C=CC=C1N3C(N(CCN2CC(C)(C)C2)CC3)=O)Cl\\n1671, ziprasidone, 1, C1=CC=CC2=C1C(=NS2)N5CCN(CCC4=CC3=C(NC(C3)=O)C=C4Cl)CC5\\n1672, zolazepam, 1, C1=CC=CC(=C1C2=NCC(N(C3=C2C(=N[N]3C)C)C)=O)F\\n1673, zoloperone, 1, C4=C(C3=C(CCN2CCN(C1=C(C=CC=C1)OC)CC2)OC(N3)=O)C=CC(=C4)F\\n1674, zoniclezole, 1, C1=C(Cl)C=CC2=C1C(=NO2)C([N]3C=CN=C3)C\\n1675, zopiclone, 1, C1=CC(=CN=C1N2C(C3=C(C2=O)N=CC=N3)OC(N4CCN(C)CC4)=O)Cl\\n1676, zotepine, 1, C1=C(Cl)C=CC2=C1C(=CC3=C(S2)C=CC=C3)OCCN(C)C\\n1677, zylofuramine, 1, [C@@H]1(CCCO1)[C@@H](CC2=CC=CC=C2)NCC\\n1678, aceprometazine, 1, C1=CC2=C(C=C1)SC3=C(N2CC(C)N(C)C)C=C(C=C3)C(=O)C\\n1679, adinazolam, 1, C1=CC(=CC3=C1[N]2C(=NN=C2CN(C)C)CN=C3C4=CC=CC=C4)Cl\\n1680, albutoin, 1, C(N1C(C(CC(C)C)NC1=S)=O)C=C\\n1681, alclometasone, 1, [C@H]24[C@H]1[C@@]([C@](O)([C@@H](C1)C)C(=O)CO)(C[C@H](O)[C@@H]2[C@@]3(C(=CC(=O)C=C3)C[C@H]4Cl)C)C\\n1682, aletamine(alfetamine), 1, C1=C(CC(CC=C)N)C=CC=C1\\n1683, algestone, 1, [C@H]34[C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)CC[C@@]3([C@](O)([C@H](O)C4)C(=O)C)C\\n1684, alimemazine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)S3)CC(CN(C)C)C\\n1685, alpertine, 1, C1=C(OC)C(=CC2=C1C(=C(C(OCC)=O)[NH]2)CCN4CCN(C3=CC=CC=C3)CC4)OC\\n1686, alphaprodine(betaprodine), 1, C2=C(C1(C(CN(C)CC1)C)OC(CC)=O)C=CC=C2\\n1687, alpidem, 1, C3=C(C1=C([N]2C(=N1)C=CC(=C2)Cl)CC(N(CCC)CCC)=O)C=CC(=C3)Cl\\n1688, amebucort, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)[C@H](C2)C)C)[C@H](C3)O)CC4)C)(C(COC(C)=O)=O)OC(CCC)=O\\n1689, amedalin, 1, CNCCCC2(C)C(=O)N(c1ccccc1)c3ccccc23\\n1690, amfonelic-acid, 1, C1=CC(=NC2=C1C(C(=CN2CC)C(O)=O)=O)CC3=CC=CC=C3\\n1691, amiflamine, 1, [C@@H](O)([C@@H](O)C(=O)O)C(=O)O.[C@@H](N)(CC1=CC=C(N(C)C)C=C1C)C\\n1692, aminoglutethimide, 1, C1=CC(=CC=C1C2(C(NC(=O)CC2)=O)CC)N\\n1693, aminopromazine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)S3)CC(CN(C)C)N(C)C\\n1694, amiperone, 1, C1=CC(=CC=C1C3(CCN(CCCC(C2=CC=C(F)C=C2)=O)CC3)C(N(C)C)=O)Cl\\n1695, amoxapine, 1, C1=C(Cl)C=CC2=C1C(=NC3=C(O2)C=CC=C3)N4CCNCC4\\n1696, anisopirol, 1, C3=C(N2CCN(CCCC(C1=CC=C(F)C=C1)O)CC2)C(=CC=C3)OC\\n1697, aptazapine, 1, [N]12C(=CC=C1)C4N(C3=C(C2)C=CC=C3)CCN(C4)C.O=C(O)\\\\C=C/C(=O)O\\n1698, arfendazam, 1, C1=C(Cl)C=CC3=C1N(C2=CC=CC=C2)C(CCN3C(OCC)=O)=O\\n1699, asimadoline, 1, [C@H](N(C(C(C1=CC=CC=C1)C2=CC=CC=C2)=O)C)(C3=CC=CC=C3)CN4CC[C@@H](C4)O\\n1700, atibeprone, 1, C2=C(OCC1=NN=C(S1)C(C)C)C=CC3=C2OC(=O)C(=C3C)C\\n1701, atropine(hyoscyamine), 1, C1=CC=CC=C1C(C(OC2CC3N(C)C(C2)CC3)=O)CO\\n1702, bazinaprine, 1, C1=C(C(=NN=C1C2=CC=CC=C2)NCCN3CCOCC3)C#N\\n1703, beclomethasonedipropionate(beclometasone-dipropionate), 1, [C@@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(Cl)[C@H](C3)O)C[C@@H]4C)C)(C(COC(CC)=O)=O)OC(CC)=O\\n1704, befuraline, 1, C3=C(C(N2CCN(CC1=CC=CC=C1)CC2)=O)OC4=C3C=CC=C4\\n1705, beloxepin, 1, [C@@H]24[C@@](C1=CC=CC=C1OC3=C(C=CC=C23)C)(CCN(C4)C)O\\n1706, benmoxin, 1, C2=C(C(NNC(C1=CC=CC=C1)=O)C)C=CC=C2\\n1707, benzobarbital, 1, C1=CC=CC=C1C3(C(N(C(C2=CC=CC=C2)=O)C(=O)NC3=O)=O)CC\\n1708, benzoclidine, 1, C3=C(C(OC1C2CCN(C1)CC2)=O)C=CC=C3\\n1709, benzoctamine, 1, C1=CC=C2C(=C1)C3C4=C(C2(CNC)CC3)C=CC=C4\\n1710, benzphetamine(benzfetamine), 1, C1=CC(=CC=C1)CC(N(C)CC2=CC=CC=C2)C\\n1711, besipirdine, 1, [H+].C1=CC3=C([N]1N(C2=CC=NC=C2)CCC)C=CC=C3.[Cl-]\\n1712, betamethadol, 1, C2=C(C(CC(N(C)C)C)(C(CC)O)C1=CC=CC=C1)C=CC=C2\\n1713, betamethasone-acetate(dexamethasone-acetate), 1, [C@@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@@H]4C)C)(C(COC(C)=O)=O)OC(C(C)C)=O\\n1714, betamethasone-dipropionate(dexamethasone-dipropionate), 1, [C@H]34[C@H]2[C@@]([C@@]1(C(=CC(=O)C=C1)CC2)C)([C@@H](O)C[C@@]3([C@](C(COC(C)=O)=O)(O)[C@H](C4)C)C)F\\n1715, betamethasone-valerate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@@H]4C)C)(C(COC(CC)=O)=O)OC(CC)=O\\n1716, bifemelane, 1, [H+].C2=C(CC1=CC=CC=C1)C(=CC=C2)OCCCCNC.[Cl-]\\n1717, binedaline, 1, C2=C(C1=C(C=CC=C1)[N]2N(CCN(C)C)C)C3=CC=CC=C3\\n1718, brallobarbital, 1, C(C1(C(NC(=O)NC1=O)=O)CC=C)C(=C)Br\\n1719, brifentanil, 1, [C@@H]3(N(C1=C(C=CC=C1)F)C(COC)=O)[C@@H](CN(CCN2C(N(CC)N=N2)=O)CC3)C\\n1720, brofaromine, 1, C2=C(C1CCNCC1)OC3=C2C=C(OC)C=C3Br\\n1721, bromamid, 1, C1=CC(=CC=C1NCCC(N(C)C)=O)Br\\n1722, bromodiphenhydramine(bromazine), 1, C2=C(C(OCCN(C)C)C1=CC=CC=C1)C=CC(=C2)Br\\n1723, bromopride, 1, C1=C(Br)C(=CC(=C1C(NCCN(CC)CC)=O)OC)N\\n1724, bromperidol-decanoate, 1, C1=CC(=CC=C1C3(CCN(CCCC(C2=CC=C(C=C2)F)=O)CC3)OC(CCCCCCCCC)=O)Br\\n1725, budesonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)C[C@H]4OC(O5)CCC)C)C(CO)=O\\n1726, budipine, 1, C1=CC=CC=C1C3(C2=CC=CC=C2)CCN(C(C)(C)C)CC3\\n1727, bufenadrine, 1, C2=C(C(C1=CC=CC=C1)OCCN(C)C)C(=CC=C2)C(C)(C)C\\n1728, buspirone, 1, C1=NC(=NC=C1)N4CCN(CCCCN2C(CC3(CC2=O)CCCC3)=O)CC4\\n1729, butalbital, 1, C(C1(C(NC(=O)NC1=O)=O)CC=C)C(C)C\\n1730, butamoxane, 1, C1=CC=CC2=C1OC(CNCCCC)CO2\\n1731, butriptyline, 1, C1=CC=CC3=C1C(C2=CC=CC=C2CC3)CC(CN(C)C)C\\n1732, carbenzide, 1, C1=CC=CC=C1C(NNC(OCC)=O)C\\n1733, carbocloral, 1, C(OC(NC(C(Cl)(Cl)Cl)O)=O)C\\n1734, carbromal, 1, C(C(C(NC(N)=O)=O)(CC)Br)C\\n1735, carfentanil, 1, C1=CC=CC=C1N(C3(CCN(CCC2=CC=CC=C2)CC3)C(OC)=O)C(CC)=O\\n1736, carpipramine, 1, C1=CC=CC4=C1N(CCCN3CCC(N2CCCCC2)(CC3)C(N)=O)C5=C(CC4)C=CC=C5\\n1737, carvotroline, 1, C1=C(F)C=CC2=C1C3=C([NH]2)CCN(C3)CCC4=CC=NC=C4\\n1738, cericlamine, 1, [H+].C1=C(CC(N(C)C)(CO)C)C=CC(=C1Cl)Cl.[Cl-]\\n1739, chlorcyclizine, 1, C3=C(C(N1CCN(C)CC1)C2=CC=CC=C2)C=CC(=C3)Cl\\n1740, chlorobutanol, 1, ClC(CCC)O\\n1741, chloroprednisone, 1, [C@H]24[C@H]1[C@@]([C@](C(COC(C)=O)=O)(O)CC1)(CC([C@@H]2[C@@]3(C(=CC(=O)C=C3)[C@H](C4)Cl)C)=O)C\\n1742, chloroserpidine, 1, [C@@H]36C2=C(C1=CC(=CC=C1[NH]2)Cl)CCN3C[C@H]5C[C@@H](OC(C4=CC(=C(C(=C4)OC)OC)OC)=O)[C@@H]([C@H]([C@H]5C6)C(OC)=O)OC\\n1743, chlorphenesin-carbamate, 1, C1=CC(=CC=C1OCC(COC(N)=O)O)Cl\\n1744, chlorphenoxamine, 1, C2=C(C(C1=CC=CC=C1)(OCCN(C)C)C)C=CC(=C2)Cl\\n1745, chlorthenoxazine, 1, C1=C2C(=CC=C1)OC(CCCl)NC2=O\\n1746, cianopramine, 1, C1=C(C#N)C=CC3=C1N(C2=C(C=CC=C2)CC3)CCCN(C)C\\n1747, ciclazindol, 1, C1=CC=CC2=C1C(C3=NCCCN23)(C4=CC=CC(=C4)Cl)O\\n1748, ciclopramine, 1, C4=C2C1=C(CCC3=C(N1CCC2NC)C=CC=C3)C=C4\\n1749, cidoxepin, 1, CN(C)CCC=C2c1ccccc1COc3ccccc23\\n1750, ciladopa, 1, [C@@H](CN2CCN(C1=CC=CC=CC1=O)CC2)(C3=CC(=C(OC)C=C3)OC)O\\n1751, cinfenine, 1, C1=CC=CC=C1C(C2=CC=CC=C2)OCCN(C\\\\C=C\\\\C3=CC=CC=C3)C\\n1752, cinitapride, 1, C1=C(C(=CC(=C1[N+](=O)[O-])N)OCC)C(NC3CCN(CC2CCC=CC2)CC3)=O\\n1753, cinperene, 1, C1=CC=CC=C1C4(C3CCN(C\\\\C=C\\\\C2=CC=CC=C2)CC3)C(NC(=O)CC4)=O\\n1754, cinromide, 1, C1=C(C=CC=C1Br)\\\\C=C\\\\C(NCC)=O\\n1755, cintriamide(cintramide), 1, C1=C(C(=C(C=C1\\\\C=C\\\\C(N)=O)OC)OC)OC\\n1756, cinuperone, 1, C3=C(N2CCN(CCCC(C1=CC=C(F)C=C1)=O)CC2)N=CC4=CC=CC=C34\\n1757, citalopram, 1, C1=CC(=CC3=C1C(C2=CC=C(F)C=C2)(OC3)CCCN(C)C)C#N\\n1758, citenamide, 1, C1=CC=CC3=C1C(C2=C(C=CC=C2)C=C3)C(N)=O\\n1759, clazolam, 1, C1=C(Cl)C=CC4=C1C3C2=CC=CC=C2CCN3CC(=O)N4C\\n1760, clemeprol, 1, C1=CC=CC(=C1)C(C(CN(C)C)O)C2=CC=C(C=C2)Cl\\n1761, climazolam, 1, C1=CC(=CC3=C1[N]2C(=CN=C2C)CN=C3C4=CC=CC=C4Cl)Cl\\n1762, clobetasol, 1, [C@@]23(F)C(C1C([C@](O)(C(C1)C)C(=O)CCl)(CC2O)C)CCC4=CC(=O)C=CC34C\\n1763, clobetasol-propionate, 1, [C@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@@H]4C)C)(OC(CC)=O)C(CCl)=O\\n1764, clofexamide, 1, C1=CC(=CC=C1OCC(NCCN(CC)CC)=O)Cl\\n1765, clomacran, 1, C1=C(Cl)C=CC3=C1C(C2=C(C=CC=C2)N3)CCCN(C)C\\n1766, clopipazan, 1, CN1CCC(CC1)=C3c2ccccc2Oc4ccc(Cl)cc34\\n1767, clothiapine(clotiapine), 1, C1=C(Cl)C=CC2=C1C(=NC3=C(S2)C=CC=C3)N4CCN(CC4)C\\n1768, clovoxamine, 1, C1=C(C(=N/OCCN)/CCCCOC)C=CC(=C1)Cl\\n1769, cloxazolam, 1, C1=C(Cl)C=CC3=C1C2(OCCN2CC(=O)N3)C4=C(Cl)C=CC=C4\\n1770, cloxypendyl, 1, C1=C(Cl)C=NC2=C1SC4=C(N2CCCN3CCN(CCO)CC3)C=CC=C4\\n1771, cormethasoneacetate(cormetasone-acetate), 1, CC4CC3C2CC(F)(F)C1=CC(=O)C=CC1(C)C2(F)C(O)CC3(C)C4(O)C(=O)COC(C)=O\\n1772, cropropamide, 1, C(C(N(C(\\\\C=C\\\\C)=O)CCC)C(N(C)C)=O)C\\n1773, cyamemazine, 1, C1=C(C#N)C=CC3=C1N(C2=C(C=CC=C2)S3)CC(CN(C)C)C\\n1774, cyclazodone, 1, C3=C(C1OC(=NC1=O)NC2CC2)C=CC=C3\\n1775, cypenamine, 1, C2=C(C1C(CCC1)N)C=CC=C2\\n1776, cyprazepam, 1, C4=C(C2=C3C(=NC(=NCC1CC1)CN2O)C=CC(=C3)Cl)C=CC=C4\\n1777, dabelotine, 1, C1=CC=C3C(=C1OCC2CNCCO2)N(CCC3)C\\n1778, daledalin, 1, C1=CC=CC3=C1N(C2=CC=CC=C2)CC3(CCCNC)C\\n1779, damotepine, 1, C1=CC=CC3=C1SC2=C(C=CC=C2)C=C3CN(C)C\\n1780, danitracen, 1, C1=CC=CC3=C1C(C2=C(C=CC=C2)C3O)=C4CCN(C)CC4\\n1781, dapoxetine, 1, [C@H](CCOC1=CC=CC2=CC=CC=C12)(C3=CC=CC=C3)N(C)C\\n1782, darvon(dextropropoxypheneandlevopropoxyphene), 1, [C@](OC(=O)CC)([C@@H](CN(C)C)C)(CC1=CC=CC=C1)C2=CC=CC=C2.[H+].[Cl-]\\n1783, dazepinil, 1, CN2C(Cc1ccccc1N=C2C)c3ccccc3\\n1784, declenperone, 1, C1=CC=CC2=C1N(C(=O)N2)CCCN4CCC(C(C3=CC=C(F)C=C3)=O)CC4\\n1785, deflazacort, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)C[C@H]4OC(=N5)C)C)C(COC(C)=O)=O\\n1786, delergotrile, 1, [C@@H]34CC1=C[NH]C2=CC=CC(=C12)C3C[C@@H](CN4C)CC#N\\n1787, delorazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(N2)=O)C3=CC=CC=C3Cl\\n1788, demexiptiline, 1, C1=CC=CC2=C1C(C3=C(C=C2)C=CC=C3)=NOCCNC\\n1789, denzimol, 1, C1=CC(=CC=C1C(C[N]2C=CN=C2)O)CCC3=CC=CC=C3\\n1790, depramine, 1, C1=CC=CC2=C1N(C3=C(C=C2)C=CC=C3)CCCN(C)C\\n1791, desflurane, 1, O(C(C(F)(F)F)F)C(F)F\\n1792, desomorphine, 1, C1=CC(=C3C2=C1CC5C4C2(C(O3)CCC4)CCN5C)O\\n1793, dexclamol, 1, [C@H]15C3=C(CCC2=C1C=CC=C2)C=CC=C3[C@H]4C[C@](C(C)C)(O)CCN4C5\\n1794, dextromoramide(levomoramideandracemoramide), 1, C4=C(C(C(=O)N1CCCC1)(C2=CC=CC=C2)C(CN3CCOCC3)C)C=CC=C4\\n1795, dextrorphan, 1, [C@]134[C@@H]([C@H](CC2=C1C=C(O)C=C2)N(C)CC3)CCCC4\\n1796, dezinamide, 1, C1=C(C=CC=C1OC2CN(C(N)=O)C2)C(F)(F)F\\n1797, dezocine, 1, CC13CCCCCC(Cc2ccc(O)cc12)C3N\\n1798, diacetylnalorphine, 1, C1=CC(=C3C2=C1CC5C4C2(C(O3)C(C=C4)OC(C)=O)CCN5CC=C)OC(C)=O\\n1799, dibenzepin, 1, C1=CC=CC2=C1N(C(C3=C(N2C)C=CC=C3)=O)CCN(C)C\\n1800, dicarbine, 1, C1=C(C)C=CC3=C1C2C(CCN(C2)C)N3\\n1801, dichlormezanone, 1, C2=C(C1[S](CCC(N1C)=O)(=O)=O)C=CC(=C2Cl)Cl\\n1802, diclofensine, 1, C1=CC(=CC3=C1C(C2=CC(=C(Cl)C=C2)Cl)CN(C)C3)OC\\n1803, diethazine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)S3)CCN(CC)CC\\n1804, difencloxazine, 1, Clc1ccc(cc1)C(OCCN2CCOCC2)c3ccccc3\\n1805, diflorasone-diacetate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)(F)[C@H](C3)O)C[C@@H]4C)C)(C(COC(C)=O)=O)OC(C)=O\\n1806, difluprednate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)(F)[C@H](C3)O)CC4)C)(C(COC(C)=O)=O)OC(CCC)=O\\n1807, dimemorfan, 1, [C@]134[C@@H]([C@H](CC2=C1C=C(C)C=C2)N(C)CC3)CCCC4\\n1808, dimeprozan, 1, C3=CC1=C(C(/C2=C(O1)C=CC=C2)=C/CCN(C)C)C=C3OC\\n1809, dimesone, 1, [C@]23([C@H]([C@H]1[C@]([C@](C(CO)=O)(C)[C@@H](C1)C)(C)C[C@@H]2O)CCC4=CC(=O)C=C[C@]34C)F\\n1810, dimetacrine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)C3(C)C)CCCN(C)C\\n1811, dinazafone, 1, C1=C(C(=CC=C1Cl)N(C(CNCC(C)=C)=O)C)C(C2=CC=CC=C2)=O\\n1812, doliracetam, 1, C1=CC=CC2=C1C(C(N2CC(N)=O)=O)C3=CC=CC=C3\\n1813, dosulepin, 1, [H+].C3=C2\\\\C(C1=CC=CC=C1SCC2=CC=C3)=C\\\\CCN(C)C.[Cl-]\\n1814, doxefazepam, 1, C1=C(Cl)C=CC2=C1C(=NC(O)C(N2CCO)=O)C3=CC=CC=C3F\\n1815, doxenitoin, 1, C3=C(C2(C1=CC=CC=C1)C(NCN2)=O)C=CC=C3\\n1816, doxibetasol, 1, [C@H]34[C@H]2[C@@](F)([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@@H](O)C[C@@]3([C@](O)([C@H](C4)C)C(=O)C)C\\n1817, drocinonide, 1, [C@H]23C([C@@]1(OC(O[C@@H]1C2)(C)C)C(=O)CO)(C[C@H](O)[C@@]4(F)C3CCC5C4(CCC(=O)C5)C)C\\n1818, dulozafone, 1, C1=C(C(=CC=C1Cl)N(C(CN(CCO)CCO)=O)C)C(C2=C(C=CC=C2)Cl)=O\\n1819, enpiprazole, 1, C1=CC=CC(=C1N3CCN(CCC2=C[N](C)N=C2)CC3)Cl\\n1820, enprazepine, 1, C1=CC=CC2=C1N(CC3=C(C2=C)C=CC=C3)CCCN(C)C\\n1821, eprobemide, 1, C1=CC(=CC=C1C(NCCCN2CCOCC2)=O)Cl\\n1822, eptazocine, 1, C1=C(O)C=CC3=C1C2(CC(CN(CC2)C)C3)C\\n1823, eterobarb, 1, C2=C(C1(C(N(C(=O)N(C1=O)COC)COC)=O)CC)C=CC=C2\\n1824, ethotoin, 1, C1=CC=CC=C1C2C(N(CC)C(N2)=O)=O\\n1825, ethyl-chloride, 1, C1=C(Cl)C=CC2=C1C(=NC(C(=O)N2)C(OCC)=O)C3=CC=CC=C3F\\n1826, etizolam, 1, C1=C(SC2=C1C(=NCC3=NN=C([N]23)C)C4=CC=CC=C4Cl)CC\\n1827, etrabamine, 1, C2=NC1=C(CC(NC)CC1)S2\\n1828, etryptamine, 1, C1=CC=CC2=C1C(=C[NH]2)CC(CC)N\\n1829, fenaclon, 1, C1=CC=CC=C1CCNC(CCCl)=O\\n1830, fenadiazole, 1, O=C2\\\\C(=C1\\\\OC=NN1)C=CC=C2\\n1831, fezolamine, 1, C1=C(C(=N[N]1CCCN(C)C)C2=CC=CC=C2)C3=CC=CC=C3\\n1832, fletazepam, 1, C1=C(Cl)C=CC2=C1C(=NCCN2CC(F)(F)F)C3=CC=CC=C3F\\n1833, fluanisone, 1, C1=CC=CC(=C1N3CCN(CCCC(C2=CC=C(F)C=C2)=O)CC3)OC\\n1834, flucindol, 1, CN(C)C3CCc2[nH]c1c(F)cc(F)cc1c2C3\\n1835, fluciprazine, 1, C1=CC(=CC=C1N3CCN(CC(O)COC2(CCCCC2)C#C)CC3)F\\n1836, fludiazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(=O)N2C)C3=CC=CC=C3F\\n1837, flumethasone(flumetasoneanddiflorasone), 1, [C@H]23[C@@]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)([C@H](C[C@]4([C@H]3C[C@@H](C)[C@@]4(C(CO)=O)O)C)O)F\\n1838, flumethasone21pivalate(flumetasone-pivalate), 1, [C@]12(F)[C@H](CC4([C@H](C1C[C@@H](C3=CC(=O)C=CC23C)F)C[C@H]([C@]4(O)C(=O)COC(=O)C(C)(C)C)C)C)O\\n1839, flunisolide, 1, [C@]14(C([C@H]3[C@H]([C@@H](O)C1)C2(C(=CC(=O)C=C2)[C@@H](F)C3)C)C[C@H]5OC(O[C@@]45C(=O)CO)(C)C)C.[C@]69(C([C@H]8[C@H]([C@@H](O)C6)C7(C(=CC(=O)C=C7)[C@@H](F)C8)C)C[C@H]%10OC(O[C@@]9%10C(=O)CO)(C)C)C.O\\n1840, flunisolide-acetate, 1, [C@]12(OC(O[C@@H]1CC3C2(CC(O)[C@H]4C3C[C@H](F)C5=CC(=O)C=CC45C)C)(C)C)C(=O)COC(=O)C\\n1841, fluocinolone, 1, [C@H]2(C1=CC(=O)C=C[C@@]1([C@@]4([C@@H](C2)[C@@H]3C[C@@H](O)[C@]([C@@]3(C)C[C@@H]4O)(O)C(=O)CO)F)C)F\\n1842, fluocinolone-acetonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)[C@H](C2)F)C)(F)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(CO)=O\\n1843, fluocortin-butyl, 1, [C@H]24[C@H]1[C@@]([C@@H](C(C(OCCCC)=O)=O)[C@@H](C1)C)(C[C@@H]([C@@H]2[C@@]3(C(=CC(=O)C=C3)[C@H](C4)F)C)O)C\\n1844, fluocortolone, 1, [C@H]24[C@H]1[C@@]([C@@H](C(CO)=O)[C@@H](C1)C)(C[C@@H]([C@@H]2[C@@]3(C(=CC(=O)C=C3)[C@H](C4)F)C)O)C\\n1845, fluorometholone, 1, [C@]23([C@H]([C@H]1[C@]([C@](C(C)=O)(O)CC1)(C)C[C@@H]2O)C[C@H](C)C4=CC(=O)C=C[C@]34C)F\\n1846, fluperlapine, 1, C1=CC=CC3=C1C(=NC2=C(C=CC(=C2)F)C3)N4CCN(C)CC4\\n1847, fluperolone(fluperolone-acetate), 1, [C@H]23[C@@]([C@@]1(C(=CC(=O)C=C1)CC2)C)([C@H](C[C@]4([C@H]3CC[C@@]4(C([C@@H](OC(C)=O)C)=O)O)C)O)F\\n1848, fluprednidene, 1, [C@]34([C@H]([C@H]2[C@@]([C@@]1(C(=CC(C=C1)=O)CC2)C)(C(C3)O)F)CC([C@@]4(C(=O)CO)O)=C)C\\n1849, fluprednisolone, 1, [C@H]24[C@H]1[C@@]([C@](C(CO)=O)(O)CC1)(C[C@@H]([C@@H]2[C@@]3(C(=CC(=O)C=C3)[C@H](C4)F)C)O)C\\n1850, fluprednisolone-valerate, 1, [C@@]4(OC(=O)CCCC)(C3(C(C2C(C1(C(=CC(=O)C=C1)C(F)C2)C)C(O)C3)CC4)C)C(=O)CO\\n1851, fluradoline, 1, C1=C(F)C=CC3=C1C=C(SCCNC)C2=CC=CC=C2O3\\n1852, flurandrenolide(fludroxycortide), 1, [C@@]45([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)[C@H](C2)F)C)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(CO)=O\\n1853, fluspiperone, 1, C1=CC(=CC=C1N2C4(C(NC2)=O)CCN(CCCC(C3=CC=C(F)C=C3)=O)CC4)F\\n1854, flutoprazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(N2CC3CC3)=O)C4=CC=CC=C4F\\n1855, furfenorex, 1, C1=CC=CC=C1CC(N(CC2=CC=CO2)C)C\\n1856, galantamine, 1, [C@]234C1=C(C=CC(=C1O[C@H]2C[C@@H](O)C=C3)OC)CN(CC4)C\\n1857, gamfexine, 1, C2=C(C(C1CCCCC1)CCN(C)C)C=CC=C2\\n1858, guanoxyfen, 1, NC(=N)NCCCOc1ccccc1\\n1859, halazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(=O)N2CC(F)(F)F)C3=CC=CC=C3\\n1860, halometasone, 1, [C@H]23[C@@]([C@@]1(C(=CC(C(=C1)Cl)=O)[C@H](C2)F)C)([C@H](C[C@]4(C3C[C@H]([C@@]4(C(CO)=O)O)C)C)O)F\\n1861, heptabarb, 1, C(C2(C1=CCCCCC1)C(NC(=O)NC2=O)=O)C\\n1862, hepzidine, 1, C1=CC=CC2=C1C(C3=C(CC2)C=CC=C3)OC4CCN(C)CC4\\n1863, homochlorcyclizine, 1, C1=CC(=CC=C1C(N2CCN(C)CCC2)C3=CC=CC=C3)Cl\\n1864, homofenazine(homophenazine), 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCN(CCO)CCC4\\n1865, hydrocortisone-acetate, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)[C@H](C3)O)CC[C@@]4(C(COC(C)=O)=O)O)C\\n1866, hydrocortisone-buteprate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)[C@H](C3)O)CC4)C)(C(COC(CC)=O)=O)OC(CCC)=O\\n1867, hydrocortisone-butyrate, 1, [C@]4([C@@]3([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)CC1)CC2)C)[C@H](C3)O)CC4)C)(OC(CCC)=O)C(CO)=O\\n1868, hydromorphone, 1, [C@@]125C3=C4C[C@H]([C@@H]1CCC([C@@H]2OC3=C(C=C4)O)=O)N(C)CC5\\n1869, hydroxypethidine, 1, C2=C(C1(C(OCC)=O)CCN(C)CC1)C=CC=C2O\\n1870, icometasone-enbutate, 1, [C@@]4([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(Cl)[C@H](C3)O)C[C@H]4C)C)(C(COC(C)=O)=O)OC(CCC)=O\\n1871, iminophenimide, 1, C2=C(C1(C(NC(=O)CN1)=O)CC)C=CC=C2\\n1872, indalpine, 1, C1=CC=CC2=C1C(=C[NH]2)CCC3CCNCC3\\n1873, intriptyline, 1, CN(C)CC#CC=C2c1ccccc1C=Cc3ccccc23\\n1874, iprindole, 1, C1=CC=CC3=C1C2=C(CCCCCC2)[N]3CCCN(C)C\\n1875, iproniazid, 1, C1=CN=CC=C1C(NNC(C)C)=O\\n1876, isomethadone, 1, C1=CC=CC=C1C(C2=CC=CC=C2)(C(CN(C)C)C)C(CC)=O\\n1877, isoprednidene, 1, [C@@]4(O)(C3(C(C2C(C1(C(=CC(=O)CC1)C=C2)C)C(O)C3)CC4=C)C)C(=O)CO\\n1878, isothamine(profenamine), 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)S3)CC(N(CC)CC)C\\n1879, ketipramine(ketimipramine), 1, C1=CC=CC2=C1N(C3=C(CC2=O)C=CC=C3)CCCN(C)C\\n1880, ketorfanol, 1, [C@]235[C@H]([C@H](N(CC1CC1)CC2)CC4=C3C(=CC=C4)O)CCC(C5)=O\\n1881, levodopa, 1, [C@H](CC1=CC(=C(C=C1)O)O)(C(O)=O)N\\n1882, levomepromazine, 1, [C@H](CN1C3=C(SC2=C1C=CC=C2)C=CC(=C3)OC)(CN(C)C)C\\n1883, levoprotiline(oxaprotiline), 1, C13(C4=C(C(C2=C1C=CC=C2)CC3)C=CC=C4)C[C@H](CNC)O\\n1884, lirequinil, 1, [C@H]5(CN(C(C3=C2C1=CC(=CC=C1CCN2C(=O)C(=C3)C4=CC=CC=C4)Cl)=O)CC5)OCC\\n1885, litoxetine, 1, C2=C(COC1CCNCC1)C=CC3=C2C=CC=C3\\n1886, lobeline, 1, C3=C(C(CC2N(C(CC(C1=CC=CC=C1)=O)CCC2)C)O)C=CC=C3\\n1887, lodaxaprine, 1, C1=CC(=NN=C1C2=C(Cl)C=CC=C2)N3CCC(O)CC3\\n1888, lorcinadol, 1, C3=C(N2CCN(C\\\\C=C\\\\C1=CC=CC=C1)CC2)N=NC(=C3)Cl\\n1889, loreclezole, 1, C1=CC(=CC(=C1\\\\C(=C\\\\[N]2C=NC=N2)Cl)Cl)Cl\\n1890, losindole, 1, [C@@H]1([C@H]3[C@@H](CC2=C1C=C(Cl)C=C2)CN(C3)C)C4=CC=CC=C4\\n1891, maroxepin, 1, C1=CC=CC3=C1C4=C(C2=C(C=CC=C2)O3)CCN(CC4)C\\n1892, mazapertine, 1, C3=C(CN2CCN(C1=C(OC(C)C)C=CC=C1)CC2)C=CC=C3C(=O)N4CCCCC4.C(C(=O)O)CC(=O)O\\n1893, medifoxamine, 1, C2=C(OC(OC1=CC=CC=C1)CN(C)C)C=CC=C2\\n1894, mefeclorazine, 1, C1=CC(=CC(=C1OC)OC)CCN2CCN(CC2)C3=CC(=CC=C3)Cl\\n1895, menitrazepam, 1, C1=C([N+](=O)[O-])C=CC2=C1C(=NCC(N2C)=O)C3=CCCCC3\\n1896, mephenytoin, 1, C2=C(C1(C(N(C)C(N1)=O)=O)CC)C=CC=C2\\n1897, mephobarbital(methylphenobarbital), 1, C2=C(C1(C(N(C(NC1=O)=O)C)=O)CC)C=CC=C2\\n1898, mepiprazole, 1, C1=C(C=CC=C1N3CCN(CCC2=N[NH]C(=C2)C)CC3)Cl\\n1899, meprednisone, 1, [C@H]12[C@@]([C@](C(CO)=O)(O)[C@H](C1)C)(CC(=O)[C@H]3[C@H]2CCC4=CC(=O)C=C[C@]34C)C\\n1900, metaclazepam, 1, C1=C(Br)C=CC2=C1C(=NCC(N2C)COC)C3=CC=CC=C3Cl\\n1901, metamfepramone, 1, C1=CC=CC(=C1)C(=O)C(C)N(C)C\\n1902, metapramine, 1, C1=CC=CC2=C1C(CC3=C(N2C)C=CC=C3)NC\\n1903, metaxalone, 1, C1=C(C=C(C=C1OCC2CNC(O2)=O)C)C\\n1904, metazocine, 1, C1=CC(=CC2=C1CC3C(C2(CCN3C)C)C)O\\n1905, metetoin, 1, C1=CC=CC=C1C2(N(C(=O)NC2=O)C)CC\\n1906, methadone(levomethadone), 1, C1=CC=CC=C1C(C2=CC=CC=C2)(C(CC)=O)CC(N(C)C)C\\n1907, methastyridone, 1, C1=CC=CC=C1/C=C/C2C(NC(O2)(C)C)=O\\n1908, methiomeprazine(levometiomeprazine), 1, C1=C(SC)C=CC3=C1N(C2=C(C=CC=C2)S3)CC(CN(C)C)C\\n1909, methsuximide(mesuximide), 1, C2=C(C1(C(N(C)C(C1)=O)=O)C)C=CC=C2\\n1910, methylprednisolone, 1, [C@H]24[C@H]1[C@@]([C@](C(CO)=O)(O)CC1)(C[C@@H]([C@@H]2[C@@]3(C(=CC(=O)C=C3)[C@H](C4)C)C)O)C\\n1911, methylprednisolone-aceponate, 1, [C@H]24[C@H]1[C@@]([C@](C(COC(C)=O)=O)(O)CC1)(C[C@@H]([C@@H]2[C@@]3(C(=CC(=O)C=C3)[C@H](C4)C)C)O)C\\n1912, methylprednisolone-acetate, 1, [C@H]24[C@H]1[C@@]([C@](C(COC(CCC(O)=O)=O)=O)(O)CC1)(C[C@@H]([C@@H]2[C@@]3(C(=CC(=O)C=C3)[C@H](C4)C)C)O)C\\n1913, metitepine(methiothepin), 1, C1=C(SC)C=CC4=C1C(N2CCN(C)CC2)CC3=C(C=CC=C3)S4\\n1914, metomidate, 1, C1=NC=C([N]1C(C2=CC=CC=C2)C)C(=O)OC\\n1915, mezepine, 1, C1=CC=CC2=C1N(CC3=C(C2)C=CC=C3)CCCNC\\n1916, midazolam(midazolammaleate), 1, C1=CC(=CC3=C1[N]2C(=CN=C2C)CN=C3C4=CC=CC=C4F)Cl\\n1917, milenperone, 1, C1=CC(=CC2=C1N(C(=O)N2)CCCN4CCC(C(C3=CC=C(F)C=C3)=O)CC4)Cl\\n1918, milipertine, 1, C1=C(OC)C(=CC2=C1C(=C(C)[NH]2)CCN4CCN(C3=CC=CC=C3OC)CC4)OC\\n1919, milnacipran, 1, [C@]1([C@H](C1)CN)(C(=O)N(CC)CC)C2=CC=CC=C2.[H+].[Cl-]\\n1920, mindoperone, 1, C1=CC(=CC2=C1C(=C([NH]2)C)C4CCN(CCCC(C3=CC=C(F)C=C3)=O)CC4)OC\\n1921, mirfentanil, 1, C1=NC=CN=C1N(C3CCN(CCC2=CC=CC=C2)CC3)C(C4=CC=CO4)=O\\n1922, modaline, 1, C1=CN=C(C(=N1)N2CCCCC2)C\\n1923, mofegiline, 1, C1=C(CC\\\\C(CN)=C/F)C=CC(=C1)F\\n1924, molindone, 1, C(C1=C(C)[NH]C3=C1C(C(CN2CCOCC2)CC3)=O)C\\n1925, nafenodone(dexnafenodone), 1, [C@]3(C1=CC=CC=C1)(C(C2=C(C=CC=C2)CC3)=O)CCN(C)C\\n1926, napitane, 1, C5CC(CN1CCC(C1)c2ccccc2)c4ccc3OCOc3c4C5\\n1927, naranol, 1, C1=C4C(=C2C(=C1)C=CC=C2)CC3CN(C)CC(C3(O4)O)C\\n1928, nealbarbital, 1, C(C1(C(NC(=O)NC1=O)=O)CC=C)C(C)(C)C\\n1929, nebracetam, 1, C1=CC=CC=C1CN2C(CC(C2)CN)=O\\n1930, nefopam, 1, C1=CC=CC3=C1C(C2=CC=CC=C2)OCCN(C3)C\\n1931, nemonapride, 1, [C@H]2(NC(=O)C1=C(OC)C=C(NC)C(=C1)Cl)[C@H](N(CC2)CC3=CC=CC=C3)C\\n1932, nerisopam, 1, C2=C1C(=NN=C(CC1=CC(=C2OC)OC)C)C3=CC=C(N)C=C3\\n1933, nicodicodine, 1, C1=CC(=C3C2=C1CC6C4C2(C(O3)C(CC4)OC(C5=CN=CC=C5)=O)CCN6C)OC\\n1934, nonaperone, 1, C1=CC(=CC=C1C(CCCN2CC3CCC(C2)CC3)=O)F\\n1935, norcodeine, 1, C1=CC(=C3C2=C1CC5C4C2(C(O3)C(O)C=C4)CCN5)OC\\n1936, normethadone, 1, C2=C(C(C1=CC=CC=C1)(CCN(C)C)C(CC)=O)C=CC=C2\\n1937, ocfentanil, 1, C3=C(N(C2CCN(CCC1=CC=CC=C1)CC2)C(COC)=O)C(=CC=C3)F\\n1938, odapipam, 1, [C@@H]4(C1=CC=CC2=C1OCC2)C3=CC(=C(Cl)C=C3CCN(C4)C)O.CC(=O)O\\n1939, olanzapine, 1, C1=C3C(=CC=C1)N=C2SC(=CC2=C(N3)N4CCN(CC4)C)C\\n1940, opipramol, 1, C1=CC=CC2=C1N(C3=C(C=C2)C=CC=C3)CCCN4CCN(CCO)CC4\\n1941, oxanamide, 1, O1CCCCC1C(=O)N\\n1942, oxazolam, 1, [C@@]12(C3=C(NC(CN1C[C@H](O2)C)=O)C=CC(=C3)Cl)C4=CC=CC=C4\\n1943, oxilorphan, 1, [C@]235[C@]([C@H](N(CC1CC1)CC2)CC4=C3C=C(O)C=C4)(CCCC5)O\\n1944, oxomemazine, 1, C1=CC=CC3=C1N(C2=C(C=CC=C2)[S]3(=O)=O)CC(CN(C)C)C\\n1945, oxypertine, 1, C1=C(OC)C(=CC2=C1C(=C(C)[NH]2)CCN4CCN(C3=CC=CC=C3)CC4)OC\\n1946, oxyridazine, 1, C1=C(OC)C=CC3=C1N(C2=C(C=CC=C2)S3)CCC4N(CCCC4)C\\n1947, paracetamol(acetaminophen), 1, N(C(=O)C)C1=CC=C(O)C=C1\\n1948, paramethadione, 1, C(C)C1(C)C(=O)N(C(O1)=O)C\\n1949, paramethasone, 1, [C@]12(C(=CC(C=C1)=O)[C@H](C[C@H]3[C@H]4[C@](C[C@@H]([C@H]23)O)([C@](C(=O)CO)(O)[C@@H](C4)C)C)F)C\\n1950, parapropamol, 1, C1=C(NC(CC)=O)C=CC(=C1)O\\n1951, paraxazone, 1, C1=CC=CC2=C1N(C(CO2)=O)CC(N)=O\\n1952, penciclovir, 1, C1=NC2=C([N]1CCC(CO)CO)NC(=NC2=O)N\\n1953, pentamoxane, 1, CC(C)CCNCC2COc1ccccc1O2\\n1954, pentylenetetrazole(pentetrazol), 1, C2C1=NN=N[N]1CCCC2\\n1955, periciazine, 1, C1=C(C=CC3=C1N(CCCN2CCC(CC2)O)C4=C(S3)C=CC=C4)C#N\\n1956, perimetazine, 1, C1=C(OC)C=CC3=C1N(C2=C(C=CC=C2)S3)CC(CN4CCC(O)CC4)C\\n1957, perisoxal, 1, C2=C(C1=CC=CC=C1)ON=C2C(CN3CCCCC3)O\\n1958, phacetoperane, 1, CC(=O)OC(C1CCCCN1)c2ccccc2\\n1959, phenaglycodol, 1, C1=C(C(C(C)(C)O)(C)O)C=CC(=C1)Cl\\n1960, phendimetrazine, 1, [C@H]2(C1=CC=CC=C1)[C@@H](N(CCO2)C)C\\n1961, phenelzine, 1, [Cl].C1=CC=CC=C1CCNN\\n1962, pheniprazine, 1, C1=CC=CC=C1CC(NN)C\\n1963, phenomorphan, 1, C1=C(O)C=CC4=C1C35C(C(N(CCC2=CC=CC=C2)CC3)C4)CCCC5\\n1964, pimethixene, 1, C1=CC=CC3=C1C(C2=C(C=CC=C2)S3)=C4CCN(C)CC4\\n1965, piminodine, 1, C3=C(C1(C(OCC)=O)CCN(CC1)CCCNC2=CC=CC=C2)C=CC=C3\\n1966, pinazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(N2CC#C)=O)C3=CC=CC=C3\\n1967, pipotiazine, 1, C1=C([S](N(C)C)(=O)=O)C=CC3=C1N(C2=C(C=CC=C2)S3)CCCN4CCC(CCO)CC4\\n1968, pipradrol, 1, C1=C(C=CC=C1)C(C2=CC=CC=C2)(C3NCCCC3)O\\n1969, piribedil, 1, C4Oc3ccc(CN1CCN(CC1)c2ncccn2)cc3O4\\n1970, pizotifen, 1, C1=CSC2=C1C(C3=C(CC2)C=CC=C3)=C4CCN(CC4)C\\n1971, prazepine, 1, C1=CC=CC2=C1N(C3=C(CC2)C=CC=C3)CCCN(C)C\\n1972, prednisolone-acetate, 1, [C@]34([C@H]([C@H]2[C@@H]([C@@]1(C(=CC(=O)C=C1)CC2)C)[C@H](C3)O)CC[C@@]4(C(COC(C)=O)=O)O)C\\n1973, prednisolone-hemisuccinate, 1, [C@]14([C@@](C(=O)COC(CCC(=O)[O-])=O)(CC[C@H]1[C@@H]3CCC2=CC(C=C[C@@]2([C@H]3[C@H](C4)O)C)=O)O)C.[Na+]\\n1974, prednival, 1, C(C)CCC(=O)OC3(CCC4C2CCC1=CC(=O)C=CC1(C)C2C(O)CC34C)C(=O)CO\\n1975, premazepam, 1, C1=C2C(=C([N]1C)C)C(=NCC(N2)=O)C3=CC=CC=C3\\n1976, prideperone, 1, C1=C(C(=CC=C1C#N)OC)C(NCCN3CCC(C(C2=CC=C(F)C=C2)=O)CC3)=O\\n1977, procinonide, 1, [C@@]23(F)C1(C(=CC(=O)C=C1)[C@@H](F)CC2C5C(CC3O)([C@]4(C(=O)COC(=O)CC)OC(O[C@@H]4C5)(C)C)C)C\\n1978, profexalone, 1, C1=CC=CC=C1C2CN(C(NCCC)=O)C(O2)=O\\n1979, promethazine, 1, C1=CC=CC2=C1N(C3=C(S2)C=CC=C3)CC(C)N(C)C\\n1980, propoxate, 1, C1=NC=C([N]1C(C2=CC=CC=C2)C)C(OCCC)=O\\n1981, prosulpride, 1, C1=C([S](N)(=O)=O)C=CC(=C1C(NCC2N(CCC2)CCC)=O)OC\\n1982, prothipendyl, 1, C1=CC=CC2=C1N(C3=C(S2)C=CC=N3)CCCN(C)C\\n1983, pyrithyldione, 1, C(C1(C(C=CNC1=O)=O)CC)C\\n1984, pyrovalerone, 1, C2=C(C(C(N1CCCC1)CCC)=O)C=CC(=C2)C\\n1985, quinupramine, 1, C1=CC=CC4=C1N(C2C3CCN(C2)CC3)C5=C(CC4)C=CC=C5\\n1986, ralitoline, 1, C1=C(Cl)C(=C(C=C1)C)NC(=O)/C=C/2SCC(=O)N2C\\n1987, razobazam, 1, C3=C(N2C1=C([NH]N=C1N(C(=O)CC2=O)C)C)C=CC=C3\\n1988, reboxetine, 1, [C@H](C1=CC=CC=C1)([C@@H]2CNCCO2)OC3=CC=CC=C3OCC\\n1989, remoxipride, 1, [C@H]2(CNC(C1=C(C=CC(=C1OC)Br)OC)=O)N(CCC2)CC\\n1990, revospirone, 1, C1=CC=CC4=C1C(N(CCCN3CCN(C2=NC=CC=N2)CC3)[S]4(=O)=O)=O\\n1991, rilapine, 1, C1=C(Cl)C=CC\\\\3=C1C=C(N2CCN(C)CC2)C4=C(C3=C/C#N)C=CC=C4\\n1992, riluzole, 1, [H+].C1=C(OC(F)(F)F)C=CC2=C1SC(=N2)N.[Cl-]\\n1993, roflurane, 1, COC(C(F)Br)(F)F\\n1994, roletamide, 1, C1=C(C(=C(C=C1C(\\\\C=C\\\\N2CC=CC2)=O)OC)OC)OC\\n1995, rolziracetam, 1, O=C1N2C(CC1)CCC2=O\\n1996, ropizine, 1, C4=C(C(N2CCN(\\\\N=C\\\\C1=NC(=CC=C1)C)CC2)C3=CC=CC=C3)C=CC=C4\\n1997, rufinamide, 1, C1=C(C(N)=O)N=N[N]1CC2=C(C=CC=C2F)F\\n1998, sabeluzole, 1, C1=CC=CC2=C1N=C(S2)N(C4CCN(CC(COC3=CC=C(C=C3)F)O)CC4)C\\n1999, seproxetine, 1, NCCC(Oc1ccc(cc1)C(F)(F)F)c2ccccc2\\n2000, setoperone, 1, C4=C(C(C3CCN(CCC1=C(N=C2N(C1=O)CCS2)C)CC3)=O)C=CC(=C4)F\\n2001, sipatrigine, 1, C3=C(C2=C(N=C(N1CCN(C)CC1)N=C2)N)C(=C(C=C3Cl)Cl)Cl\\n2002, spiclamine, 1, [C@@H]3(C1=CC=C(Cl)C=C1)[C@H]2CC[C@@H](C2)C34CCC(=N4)N5CCOCC5\\n2003, spiroxatrine, 1, C1=CC=CC=C1N2C5(C(NC2)=O)CCN(CC4OC3=C(C=CC=C3)OC4)CC5\\n2004, sulazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(=S)N2C)C3=CC=CC=C3\\n2005, sulfonethylmethane, 1, C(C([S](CC)(=O)=O)([S](CC)(=O)=O)C)C\\n2006, sulpiride, 1, C1=C([S](=O)(=O)N)C=CC(=C1C(NCC2N(CC)CCC2)=O)OC\\n2007, supidimide, 1, C1=CC=CC3=C1C(N(C2C(NCCC2)=O)[S]3(=O)=O)=O\\n2008, taclamine, 1, C5=C2C1=C(CCC4=C(C1CN3CCCCC23)C=CC=C4)C=C5\\n2009, talbutal, 1, C(C(C)C1(CC=C)C(=O)NC(NC1=O)=O)C\\n2010, tenilapine, 1, C3=C2C(=NC1=CSC=C1C(/C2=CS3)=C/C#N)N4CCN(C)CC4\\n2011, teniloxazine, 1, C1=CC=CC(=C1CC2=CC=CS2)OCC3OCCNC3.O=C(O)\\\\C=C/C(=O)O\\n2012, tetrazepam, 1, C1=C(Cl)C=CC2=C1C(=NCC(=O)N2C)C3=CCCCC3\\n2013, thiotetrabarbital, 1, C(C1(C(CCC)CC)C(NC(=S)NC1=O)=O)C\\n2014, tiapride, 1, C1=C([S](C)(=O)=O)C=CC(=C1C(NCCN(CC)CC)=O)OC\\n2015, ticabesone(ticabesone-propionate), 1, [C@]14([C@](OC(=O)CC)([C@@H](CC1C3[C@@](F)(C2(C(=CC(=O)C=C2)[C@@H](F)C3)C)[C@@H](O)C4)C)C(SC)=O)C\\n2016, tienopramine, 1, C1=CSC2=C1N(C3=C(C=C2)C=CC=C3)CCCN(C)C\\n2017, tiflucarbine, 1, C1=C4C(=C2C(=C1F)SC=C2C)C3=C(CCN(C3)CC)[NH]4\\n2018, tiletamine, 1, C1=C(SC=C1)C2(C(CCCC2)=O)NCC\\n2019, timobesone, 1, CSC(=O)C3(O)C(C)CC4C2CCC1=CC(=O)C=CC1(C)C2(F)C(O)CC34C\\n2020, tolboxane, 1, C1=CC(=CC=C1B2OCC(CCC)(C)CO2)C\\n2021, toloxatone, 1, C2=C(N1CC(CO)OC1=O)C=CC=C2C\\n2022, tomoxetine, 1, [C@H](OC1=C(C=CC=C1)C)(C2=CC=CC=C2)CCNC\\n2023, toprilidine, 1, C1=CC=CC(=C1N3CCN(CCCOC2=CC=CC=N2)CC3)C\\n2024, trepipam(trimopam), 1, [C@H]2(C1=CC(=C(OC)C=C1CCN(C2)C)OC)C3=CC=CC=C3\\n2025, triamcinolone-acetonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(CO)=O\\n2026, triamcinolone-diacetate, 1, [C@]34([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@H]([C@@]4(C(COC(C)=O)=O)O)OC(C)=O)C\\n2027, triamcinolone-hexacetonide, 1, [C@@]45([C@@]3([C@H]([C@H]2[C@]([C@@]1(C(=CC(=O)C=C1)CC2)C)(F)[C@H](C3)O)C[C@H]4OC(O5)(C)C)C)C(COC(CC(C)(C)C)=O)=O\\n2028, triclacetamol, 1, C1=C(NC(C(Cl)(Cl)Cl)=O)C=CC(=C1)O\\n2029, triflubazam, 1, C1=C(C(F)(F)F)C=CC3=C1N(C2=CC=CC=C2)C(CC(N3C)=O)=O\\n2030, trifluomeprazine, 1, C1=CC=C2C(=C1)SC3=C(N2CC(C)CN(C)C)C=C(C=C3)C(F)(F)F\\n2031, trifluoperazine, 1, C1=C(C=CC3=C1N(CCCN2CCN(CC2)C)C4=C(S3)C=CC=C4)C(F)(F)F\\n2032, trimethadione, 1, CC1(C(N(C(O1)=O)C)=O)C\\n2033, tropapride, 1, [C@H]13N([C@H](CC(C1)NC(C2=CC=CC(=C2OC)OC)=O)CC3)CC4=CC=CC=C4\\n2034, tuclazepam, 1, C1=C2C(=CC=C1)N(C(Cl)(CN=C2C3=CC=CC=C3Cl)CO)C\\n2035, ulobetasol, 1, CC4CC3C2CC(F)C1=CC(=O)C=CC1(C)C2(F)C(O)CC3(C)C4(O)C(=O)CCl\\n2036, valdetamide, 1, C(C(C(N)=O)(CC)CC)C=C\\n2037, valnoctamide, 1, C(C(C(CC)C)C(N)=O)C\\n2038, valperinol, 1, [C@@H]1(C[C@H]3[C@H]2[C@@]1(O[C@H](O[C@H]2OC)[C@@H]3C)CN4CCCCC4)O\\n2039, valpromide, 1, C(C(CCC)C(N)=O)CC\\n2040, vinbarbital, 1, C(C1(C(NC(=O)NC1=O)=O)C(=C/CC)/C)C\\n2041, vinylbital, 1, C(C(C1(C(NC(=O)NC1=O)=O)C=C)C)CC\\n2042, viqualine, 1, C1=C(OC)C=CC2=NC=CC(=C12)CCCC3C(CNCC3)C=C\\n2043, xanomeline, 1, C(OC1=NSN=C1C2=CCCN(C2)C)CCCCC\\n2044, zaleplon, 1, C2=N[N]1C(=CC=NC1=C2C#N)C3=CC(=CC=C3)N(C(C)=O)CC\\n2045, zimeldine, 1, C2=C(\\\\C(C1=CC=CN=C1)=C\\\\CN(C)C)C=CC(=C2)Br\\n2046, zomebazam, 1, C3=C(N2C1=C([N](C)N=C1C)N(C(=O)CC2=O)C)C=CC=C3\\n2047, zometapine, 1, C3=C(C1=NCCN=C2N(NC(=C12)C)C)C=CC=C3Cl\\n2048, nitrous-oxide, 1, [N+](=[N-])=O\\n2049, licostinel, 1, C1=C(Cl)C(=C(C2=C1NC(=O)C(N2)=O)[N+](=O)[O-])Cl\\n2050, ademetionine(adenosyl-methionine), 1, [C@H]3([N]2C1=C(C(=NC=N1)N)N=C2)[C@@H]([C@@H]([C@H](O3)C[S+](CC[C@@H](C([O-])=O)N)C)O)O\\n2051, mesocarb, 1, [O+]1=N[N](C=C1[N-]C(NC2=CC=CC=C2)=O)C(CC3=CC=CC=C3)C\\n2052, tofisoline, 1, C1=C(OC)C(=CC2=C1C(=[N+](C(=C2CC)C)[NH-])C3=CC=C(C(=C3)OC)OC)OC\\n2053, azidamfenicol, 1, [N+](=NCC(=O)N[C@@H]([C@H](O)C1=CC=C([N+]([O-])=O)C=C1)CO)=[N-]', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='85403800-668f-47f0-990a-090943c28702', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport os\\n# Tymczasowe wyciszenie logów TensorFlow\\nos.environ[\"TF_CPP_MIN_LOG_LEVEL\"] = \"3\"\\n\\nimport torch\\nimport numpy as np\\nimport pandas as pd\\nimport matplotlib.pyplot as plt\\nimport warnings\\nfrom rdkit import Chem\\nfrom rdkit import RDLogger\\n\\n# Tymczasowe wyciszenie ostrzeżeń i logów RDKit\\nwarnings.filterwarnings(\"ignore\")\\nRDLogger.DisableLog(\"rdApp.*\")\\n\\n# Ustawienie ziarna losowości dla powtarzalności wyników\\nnp.random.seed(42)\\ntorch.manual_seed(42)\\nimport requests\\n\\n# Ścieżka do pliku CSV\\ncsv_url = \"https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/BBBP.csv\"\\n\\n# Pobranie pliku CSV\\nresponse = requests.get(csv_url)\\ncsv_path = \\'BBBP.csv\\'\\n\\n# Zapisanie pliku lokalnie\\nwith open(csv_path, \\'wb\\') as f:\\n    f.write(response.content)\\n\\n# Wczytanie danych z CSV\\ndf = pd.read_csv(csv_path, usecols=[1, 2, 3])\\n\\n# Wyświetlenie wierszy od 96 do 104\\nprint(df.iloc[96:104])\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='56951f06-449f-4176-9979-328ee846f7e6', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport numpy as np\\n\\nclass Featurizer:\\n    def __init__(self, allowable_sets):\\n        self.dim = 0\\n        self.features_mapping = {}\\n        for k, s in allowable_sets.items():\\n            s = sorted(list(s))\\n            self.features_mapping[k] = dict(zip(s, range(self.dim, len(s) + self.dim)))\\n            self.dim += len(s)\\n\\n    def encode(self, inputs):\\n        output = np.zeros((self.dim,))\\n        for name_feature, feature_mapping in self.features_mapping.items():\\n            feature = getattr(self, name_feature)(inputs)\\n            if feature not in feature_mapping:\\n                continue\\n            output[feature_mapping[feature]] = 1.0\\n        return output\\n\\n\\nclass AtomFeaturizer(Featurizer):\\n    def __init__(self, allowable_sets):\\n        super().__init__(allowable_sets)\\n\\n    def symbol(self, atom):\\n        return atom.GetSymbol()\\n\\n    def n_valence(self, atom):\\n        return atom.GetTotalValence()\\n\\n    def n_hydrogens(self, atom):\\n        return atom.GetTotalNumHs()\\n\\n    def hybridization(self, atom):\\n        return atom.GetHybridization().name.lower()\\n\\n\\nclass BondFeaturizer(Featurizer):\\n    def __init__(self, allowable_sets):\\n        super().__init__(allowable_sets)\\n        self.dim += 1  # Dodatkowy wymiar dla bond=None\\n\\n    def encode(self, bond):\\n        output = np.zeros((self.dim,))\\n        if bond is None:\\n            output[-1] = 1.0  # Oznaczanie braku wiązania\\n            return output\\n        output = super().encode(bond)\\n        return output\\n\\n    def bond_type(self, bond):\\n        return bond.GetBondType().name.lower()\\n\\n    def conjugated(self, bond):\\n        return bond.GetIsConjugated()\\n\\n\\n# Przykładowe użycie\\natom_featurizer = AtomFeaturizer(\\n    allowable_sets={\\n        \"symbol\": {\"B\", \"Br\", \"C\", \"Ca\", \"Cl\", \"F\", \"H\", \"I\", \"N\", \"Na\", \"O\", \"P\", \"S\"},\\n        \"n_valence\": {0, 1, 2, 3, 4, 5, 6},\\n        \"n_hydrogens\": {0, 1, 2, 3, 4},\\n        \"hybridization\": {\"s\", \"sp\", \"sp2\", \"sp3\"},\\n    }\\n)\\n\\nbond_featurizer = BondFeaturizer(\\n    allowable_sets={\\n        \"bond_type\": {\"single\", \"double\", \"triple\", \"aromatic\"},\\n        \"conjugated\": {True, False},\\n    }\\n)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='065b4552-0edf-4b83-a29b-54b00a989af9', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\ndef molecule_from_smiles(smiles):\\n    # MolFromSmiles(m, sanitize=True) should be equivalent to\\n    # MolFromSmiles(m, sanitize=False) -> SanitizeMol(m) -> AssignStereochemistry(m, ...)\\n    molecule = Chem.MolFromSmiles(smiles, sanitize=False)\\n\\n    # If sanitization is unsuccessful, catch the error, and try again without\\n    # the sanitization step that caused the error\\n    flag = Chem.SanitizeMol(molecule, catchErrors=True)\\n    if flag != Chem.SanitizeFlags.SANITIZE_NONE:\\n        Chem.SanitizeMol(molecule, sanitizeOps=Chem.SanitizeFlags.SANITIZE_ALL ^ flag)\\n\\n    Chem.AssignStereochemistry(molecule, cleanIt=True, force=True)\\n    return molecule\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='f1395b99-8812-414a-aa99-a449275b5a3c', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(f\"Name:\\\\t{df.name[10]}\\\\nSMILES:\\\\t{df.smiles[10]}\\\\nBBBP:\\\\t{df.p_np[10]}\")\\nmolecule = molecule_from_smiles(df.iloc[10].smiles)\\nprint(\"Molecule:\")\\nmolecule\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='63084aa2-75ea-4fbd-89d9-b8150b912849', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport py3Dmol\\nfrom rdkit import Chem\\n\\n# Przykładowy SMILES\\nsmiles = \"OCC(C)(O)c1onc(c2ncn3c2CN(C)C(c4c3cccc4Cl)=O)n1\"  # Etanol\\nmolecule = Chem.MolFromSmiles(smiles)\\n\\n# Konwersja molekuły do formatu 3D\\nblock = Chem.MolToMolBlock(molecule)\\n\\n# Wizualizacja w 3D\\nview = py3Dmol.view(width=800, height=400)\\nview.addModel(block, \\'mol\\')\\nview.setStyle({\\'stick\\': {}})\\nview.zoomTo()\\nview.show()\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='c13494c6-fd12-4fdb-9d0a-3faf7f40c9ac', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\n# graph = graph_from_molecule(molecule)\\n# print(\"Graph (including self-loops):\")\\n# print(\"\\\\tatom features\\\\t\", graph[0].shape)\\n# print(\"\\\\tbond features\\\\t\", graph[1].shape)\\n# print(\"\\\\tpair indices\\\\t\", graph[2].shape)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='98abbe19-cb81-472e-ab47-5e67b30cce61', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport torch\\nfrom torch.utils.data import Dataset, DataLoader\\n\\ndef prepare_batch(x_batch, y_batch):\\n    \"\"\"Merges (sub)graphs of batch into a single global (disconnected) graph.\\n    \"\"\"\\n\\n    atom_features, bond_features, pair_indices = x_batch\\n\\n    # Obtain number of atoms and bonds for each graph (molecule)\\n    num_atoms = [features.size(0) for features in atom_features]\\n    num_bonds = [features.size(0) for features in bond_features]\\n\\n    # Obtain partition indices (molecule_indicator)\\n    molecule_indices = torch.arange(len(num_atoms))\\n    molecule_indicator = molecule_indices.repeat_interleave(num_atoms)\\n\\n    # Merge (sub)graphs into a global (disconnected) graph\\n    gather_indices = torch.repeat_interleave(molecule_indices[:-1], num_bonds[1:])\\n    increment = torch.cumsum(torch.tensor(num_atoms[:-1]), dim=0)\\n    \\n    # Pad increment to match the size of the bonds\\n    increment = torch.cat((torch.zeros(num_bonds[0]), increment[gather_indices]))\\n    \\n    # Flatten and adjust pair_indices\\n    pair_indices = torch.cat([p + increment.unsqueeze(1) for p in pair_indices])\\n    \\n    # Flatten atom_features and bond_features\\n    atom_features = torch.cat(atom_features)\\n    bond_features = torch.cat(bond_features)\\n\\n    return (atom_features, bond_features, pair_indices, molecule_indicator), y_batch\\n\\nclass MPNNDataset(Dataset):\\n    def __init__(self, X, y):\\n        self.X = X\\n        self.y = y\\n\\n    def __len__(self):\\n        return len(self.y)\\n\\n    def __getitem__(self, index):\\n        x_item = self.X[index]\\n        y_item = self.y[index]\\n        return prepare_batch(x_item, y_item)\\n\\ndef create_dataloader(X, y, batch_size=32, shuffle=False):\\n    dataset = MPNNDataset(X, y)\\n    return DataLoader(dataset, batch_size=batch_size, shuffle=shuffle)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='12328585-953b-4990-8ab6-be00cfd6af26', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport torch\\nimport torch.nn as nn\\n\\nclass EdgeNetwork(nn.Module):\\n    def __init__(self):\\n        super(EdgeNetwork, self).__init__()\\n\\n    def build(self, input_shape):\\n        self.atom_dim = input_shape[0][-1]\\n        self.bond_dim = input_shape[1][-1]\\n        self.kernel = nn.Parameter(torch.Tensor(self.bond_dim, self.atom_dim * self.atom_dim))\\n        self.bias = nn.Parameter(torch.zeros(self.atom_dim * self.atom_dim))\\n        nn.init.xavier_uniform_(self.kernel)  # Inicjalizacja zgodna z glorot_uniform\\n\\n    def forward(self, inputs):\\n        atom_features, bond_features, pair_indices = inputs\\n\\n        # Apply linear transformation to bond features\\n        bond_features = torch.matmul(bond_features, self.kernel) + self.bias\\n\\n        # Reshape for neighborhood aggregation later\\n        bond_features = bond_features.view(-1, self.atom_dim, self.atom_dim)\\n\\n        # Obtain atom features of neighbors\\n        atom_features_neighbors = atom_features[pair_indices[:, 1]]\\n        atom_features_neighbors = atom_features_neighbors.unsqueeze(-1)\\n\\n        # Apply neighborhood aggregation\\n        transformed_features = torch.matmul(bond_features, atom_features_neighbors)\\n        transformed_features = transformed_features.squeeze(-1)\\n\\n        # Aggregate features using scatter_add\\n        aggregated_features = torch.zeros(atom_features.size(0), self.atom_dim).to(atom_features.device)\\n        aggregated_features.index_add_(0, pair_indices[:, 0], transformed_features)\\n\\n        return aggregated_features\\n\\n\\nclass MessagePassing(nn.Module):\\n    def __init__(self, units, steps=4):\\n        super(MessagePassing, self).__init__()\\n        self.units = units\\n        self.steps = steps\\n        self.message_step = EdgeNetwork()\\n        self.update_step = nn.GRUCell(self.units, self.units)\\n\\n    def forward(self, inputs):\\n        atom_features, bond_features, pair_indices = inputs\\n\\n        # Pad atom features if number of desired units exceeds atom_features dim.\\n        pad_length = max(0, self.units - atom_features.size(1))\\n        \\n        if pad_length > 0:\\n            padding = torch.zeros(atom_features.size(0), pad_length).to(atom_features.device)\\n            atom_features_updated = torch.cat([atom_features, padding], dim=1)\\n        else:\\n            atom_features_updated = atom_features\\n\\n        # Perform a number of steps of message passing\\n        for _ in range(self.steps):\\n            # Aggregate information from neighbors\\n            atom_features_aggregated = self.message_step(\\n                (atom_features_updated, bond_features, pair_indices)\\n            )\\n\\n            # Update node state via a step of GRU\\n            atom_features_updated = self.update_step(\\n                atom_features_aggregated,\\n                atom_features_updated\\n            )\\n        \\n        return atom_features_updated\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='92816b37-904b-4564-ac21-afe2e46bae2d', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport torch\\nimport torch.nn as nn\\nimport torch.nn.functional as F\\n\\nclass PartitionPadding(nn.Module):\\n    def __init__(self, batch_size):\\n        super(PartitionPadding, self).__init__()\\n        self.batch_size = batch_size\\n\\n    def forward(self, inputs):\\n        atom_features, molecule_indicator = inputs\\n\\n        # Obtain subgraphs\\n        atom_features_partitioned = [atom_features[molecule_indicator == i] for i in range(self.batch_size)]\\n\\n        # Pad and stack subgraphs\\n        num_atoms = [f.size(0) for f in atom_features_partitioned]\\n        max_num_atoms = max(num_atoms)\\n\\n        atom_features_stacked = []\\n        for f, n in zip(atom_features_partitioned, num_atoms):\\n            padded = F.pad(f, (0, 0, 0, max_num_atoms - n))  # Pad to max_num_atoms\\n            atom_features_stacked.append(padded)\\n\\n        atom_features_stacked = torch.stack(atom_features_stacked)\\n\\n        # Remove empty subgraphs (usually for last batch in dataset)\\n        non_empty_indices = (atom_features_stacked.sum(dim=(1, 2)) != 0).nonzero(as_tuple=True)[0]\\n        return atom_features_stacked[non_empty_indices]\\n\\nclass TransformerEncoderReadout(nn.Module):\\n    def __init__(self, num_heads=8, embed_dim=64, dense_dim=512, batch_size=32):\\n        super(TransformerEncoderReadout, self).__init__()\\n\\n        self.partition_padding = PartitionPadding(batch_size)\\n        self.attention = nn.MultiheadAttention(embed_dim=embed_dim, num_heads=num_heads)\\n        self.dense_proj = nn.Sequential(\\n            nn.Linear(embed_dim, dense_dim),\\n            nn.ReLU(),\\n            nn.Linear(dense_dim, embed_dim),\\n        )\\n        self.layernorm_1 = nn.LayerNorm(embed_dim)\\n        self.layernorm_2 = nn.LayerNorm(embed_dim)\\n\\n    def forward(self, inputs):\\n        x = self.partition_padding(inputs)\\n\\n        # Create padding mask\\n        padding_mask = (x != 0).any(dim=-1)\\n        padding_mask = padding_mask.unsqueeze(1).unsqueeze(2)  # Shape: (batch_size, 1, 1, seq_len)\\n\\n        attention_output, _ = self.attention(x, x, x, key_padding_mask=~padding_mask.squeeze(-1))\\n        \\n        proj_input = self.layernorm_1(x + attention_output)\\n        proj_output = self.layernorm_2(proj_input + self.dense_proj(proj_input))\\n\\n        return proj_output.mean(dim=1)  # Global average pooling\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='6d29d802-e2d5-41e9-a248-bf25a125cd63', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport torch\\nimport torch.nn as nn\\nimport torch.nn.functional as F\\n\\nclass MPNNModel(nn.Module):\\n    def __init__(self, atom_dim, bond_dim, message_units=64, message_steps=4, num_attention_heads=8, dense_units=512):\\n        super(MPNNModel, self).__init__()\\n        \\n        self.message_passing = MessagePassing(message_units, message_steps)\\n        self.transformer_readout = TransformerEncoderReadout(num_heads=num_attention_heads, embed_dim=message_units, dense_dim=dense_units)\\n        \\n        self.dense1 = nn.Linear(dense_units, dense_units)\\n        self.dense2 = nn.Linear(dense_units, 1)\\n\\n    def forward(self, atom_features, bond_features, pair_indices, molecule_indicator):\\n        x = self.message_passing((atom_features, bond_features, pair_indices))\\n        x = self.transformer_readout((x, molecule_indicator))\\n        \\n        x = F.relu(self.dense1(x))\\n        x = torch.sigmoid(self.dense2(x))\\n        \\n        return x\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='d342c1be-24e2-414d-b629-7d3e3698378d', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nfrom rdkit import Chem\\n\\ndef molecule_from_smiles(smiles):\\n    \"\"\"Funkcja pomocnicza do konwersji SMILES na cząsteczkę RDKit.\"\"\"\\n    molecule = Chem.MolFromSmiles(smiles)\\n    if molecule is None:\\n        raise ValueError(f\"Invalid SMILES string: {smiles}\")\\n    Chem.SanitizeMol(molecule)  # Opcjonalnie: sanitizacja cząsteczki\\n    return molecule\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='1cc38529-c0bd-4536-a1c0-361e482c83d7', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\n# def graphs_from_smiles(smiles_list):\\n#     atom_features_list = []\\n#     bond_features_list = []\\n#     pair_indices_list = []\\n\\n#     for smiles in smiles_list:\\n#         try:\\n#             molecule = molecule_from_smiles(smiles)\\n#             atom_features, bond_features, pair_indices = graph_from_molecule(molecule)\\n\\n#             atom_features_list.append(atom_features)\\n#             bond_features_list.append(bond_features)\\n#             pair_indices_list.append(pair_indices)\\n#         except ValueError as e:\\n#             print(e)  # Wyświetl błąd dla niepoprawnych SMILES\\n\\n#     return (\\n#         [torch.tensor(features, dtype=torch.float32) for features in atom_features_list],\\n#         [torch.tensor(features, dtype=torch.float32) for features in bond_features_list],\\n#         [torch.tensor(indices, dtype=torch.int64) for indices in pair_indices_list],\\n#     )\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='06fd2308-06f4-4d89-842d-a94e2f7bb04e', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport torch\\nimport torch.nn as nn\\nimport torch.optim as optim\\n\\n# Definiowanie modelu MPNN\\nmpnn = MPNNModel(\\n    atom_dim=x_train[0][0][0].shape,  # Upewnij się, że używasz drugiego wymiaru\\n    bond_dim=x_train[1][0][0].shape,  # Upewnij się, że używasz drugiego wymiaru\\n)\\n\\n# Ustawienie optymalizatora i funkcji straty\\noptimizer = optim.Adam(mpnn.parameters(), lr=5e-4)\\n\\n# Funkcja straty\\ndef binary_crossentropy(predictions, targets):\\n    return nn.BCELoss()(predictions.view(-1), targets.view(-1))\\n\\n# Przykładowa pętla treningowa\\ndef train(model, data_loader):\\n    model.train()\\n    for atom_features, bond_features, pair_indices, molecule_indicator, targets in data_loader:\\n        optimizer.zero_grad()\\n        outputs = model(atom_features, bond_features, pair_indices, molecule_indicator)\\n        loss = binary_crossentropy(outputs, targets)\\n        loss.backward()\\n        optimizer.step()\\n        print(f\"Loss: {loss.item()}\")\\n\\n# Przykład użycia w pętli treningowej\\n# Zakładając `data_loader` jest instancją DataLoader z odpowiednimi danymi.\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='a5f15c98-ce54-464f-8404-1212d9afd875', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nfrom torch.utils.data import Dataset\\n\\nclass MPNNDataset(Dataset):\\n    def __init__(self, X, y):\\n        self.X = X\\n        self.y = y\\n\\n    def __len__(self):\\n        return len(self.y)\\n\\n    def __getitem__(self, idx):\\n        return self.X[idx], self.y[idx]\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='4572862a-21f9-430a-b81c-cae700f2e735', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport numpy as np\\nimport pandas as pd\\nimport torch\\nfrom torch.utils.data import DataLoader, Dataset\\nimport matplotlib.pyplot as plt\\nfrom sklearn.metrics import roc_auc_score\\n\\n# Zakładam, że df jest już wczytany jako DataFrame\\n# df = pd.read_csv(\\'path_to_your_data.csv\\')  # Przykład wczytania danych\\n\\nclass MPNNDataset(Dataset):\\n    def __init__(self, X, y):\\n        self.X = X\\n        self.y = y\\n\\n    def __len__(self):\\n        return len(self.y)\\n\\n    def __getitem__(self, idx):\\n        return self.X[idx], self.y[idx]\\n\\n# Funkcja do przekształcania SMILES na grafy\\ndef graphs_from_smiles(smiles_list):\\n    atom_features_list = []\\n    bond_features_list = []\\n    pair_indices_list = []\\n\\n    for smiles in smiles_list:\\n        molecule = molecule_from_smiles(smiles)\\n        if molecule is None:\\n            continue  # Skip invalid SMILES\\n        atom_features, bond_features, pair_indices = graph_from_molecule(molecule)\\n        \\n        atom_features_list.append(atom_features)\\n        bond_features_list.append(bond_features)\\n        pair_indices_list.append(pair_indices)\\n\\n    return (\\n        [torch.tensor(features, dtype=torch.float32) for features in atom_features_list],\\n        [torch.tensor(features, dtype=torch.float32) for features in bond_features_list],\\n        [torch.tensor(indices, dtype=torch.int64) for indices in pair_indices_list],\\n    )\\n\\n# Shuffle array of indices ranging from 0 to the number of samples in df\\npermuted_indices = np.random.permutation(np.arange(df.shape[0]))\\n\\n# Train set: 80 % of data\\ntrain_index = permuted_indices[: int(df.shape[0] * 0.8)]\\nx_train = graphs_from_smiles(df.iloc[train_index].smiles)\\ny_train = torch.tensor(df.iloc[train_index].p_np.values, dtype=torch.float32)\\n\\n# Valid set: 19 % of data\\nvalid_index = permuted_indices[int(df.shape[0] * 0.8): int(df.shape[0] * 0.99)]\\nx_valid = graphs_from_smiles(df.iloc[valid_index].smiles)\\ny_valid = torch.tensor(df.iloc[valid_index].p_np.values, dtype=torch.float32)\\n\\n# Test set: 1 % of data\\ntest_index = permuted_indices[int(df.shape[0] * 0.99):]\\nx_test = graphs_from_smiles(df.iloc[test_index].smiles)\\ny_test = torch.tensor(df.iloc[test_index].p_np.values, dtype=torch.float32)\\n\\n# Tworzenie datasetów\\ntrain_dataset = MPNNDataset(x_train, y_train)\\nvalid_dataset = MPNNDataset(x_valid, y_valid)\\n\\n# Tworzenie dataloaderów\\nbatch_size = 32\\ntrain_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)\\nvalid_loader = DataLoader(valid_dataset, batch_size=batch_size, shuffle=False)\\n\\n# Definiowanie modelu MPNN\\nmpnn_model = MPNNModel(\\n    atom_dim=x_train[0][0][0].shape,  # Upewnij się, że używasz drugiego wymiaru\\n    bond_dim=x_train[1][0][0].shape,  # Upewnij się, że używasz drugiego wymiaru\\n)\\n\\n# Ustawienie optymalizatora\\noptimizer = optim.Adam(mpnn_model.parameters(), lr=5e-4)\\n\\n# Funkcja straty\\ndef binary_crossentropy(predictions, targets):\\n    return nn.BCELoss()(predictions.view(-1), targets.view(-1))\\n\\n# Pętla treningowa\\nnum_epochs = 40\\ntrain_auc_history = []\\nvalid_auc_history = []\\n\\nfor epoch in range(num_epochs):\\n    mpnn_model.train()  # Ustaw model w trybie treningowym\\n    total_loss = 0.0\\n    all_targets = []\\n    all_predictions = []\\n\\n    # Trening\\n    for atom_features, bond_features, pair_indices, molecule_indicator, targets in train_loader:\\n        optimizer.zero_grad()  # Zerowanie gradientów\\n        outputs = mpnn_model(atom_features, bond_features, pair_indices, molecule_indicator)\\n        \\n        loss = binary_crossentropy(outputs.squeeze(), targets)  # Squeeze to match dimensions if necessary\\n        loss.backward()  # Obliczanie gradientów\\n        optimizer.step()  # Aktualizacja wag\\n        \\n        total_loss += loss.item()\\n        \\n        all_targets.extend(targets.cpu().numpy())\\n        all_predictions.extend(outputs.detach().cpu().numpy())\\n\\n    # Obliczanie AUC dla zbioru treningowego\\n    train_auc = roc_auc_score(all_targets, all_predictions)\\n    train_auc_history.append(train_auc)\\n\\n    # Walidacja\\n    mpnn_model.eval()  # Ustaw model w trybie ewaluacji\\n    valid_targets = []\\n    valid_predictions = []\\n\\n    with torch.no_grad():  # Wyłącz obliczanie gradientów dla walidacji\\n        for atom_features, bond_features, pair_indices, molecule_indicator, targets in valid_loader:\\n            outputs = mpnn_model(atom_features, bond_features, pair_indices, molecule_indicator)\\n            valid_targets.extend(targets.cpu().numpy())\\n            valid_predictions.extend(outputs.detach().cpu().numpy())\\n\\n    # Obliczanie AUC dla zbioru walidacyjnego\\n    valid_auc = roc_auc_score(valid_targets, valid_predictions)\\n    valid_auc_history.append(valid_auc)\\n\\n    print(f\"Epoch {epoch + 1}/{num_epochs}, Loss: {total_loss / len(train_loader):.4f}, Train AUC: {train_auc:.4f}, Valid AUC: {valid_auc:.4f}\")\\n\\n# Wizualizacja wyników AUC\\nplt.figure(figsize=(10, 6))\\nplt.plot(train_auc_history, label=\"Train AUC\")\\nplt.plot(valid_auc_history, label=\"Valid AUC\")\\nplt.xlabel(\"Epochs\", fontsize=16)\\nplt.ylabel(\"AUC\", fontsize=16)\\nplt.legend(fontsize=16)\\nplt.title(\"Training and Validation AUC over Epochs\", fontsize=18)\\nplt.show()\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='752cc789-a8e3-4227-b93e-6a99d7b6c176', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\ntrain_dataset[0]\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='3992127b-5ad5-471f-b908-68a2bb958ac8', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\blood.ipynb', 'file_name': 'blood.ipynb', 'file_size': 3270677, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\ntrain_loader[0]\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='e2fc691a-af77-4910-9933-b9030de8c4c0', embedding=None, metadata={'page_label': '1', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='1Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdata\\na dataset of laymen olfactory \\nperception for 74 mono-molecular \\nodors\\nAntonie Louise Bierling  1,2,3\\u2009✉, Alexander Croy  4, Tim Jesgarzewsky1, Maria Rommel2,5, \\nGianaurelio Cuniberti  3, Thomas Hummel5 & Ilona Croy1,2,6\\nThe molecular structure of an odor determines whether and how it is perceived by humans. However, \\nthe principles of how odorant chemistry links to perceptual patterns remain largely unknown and are \\nprimarily studied using odor rating datasets from highly trained olfactory experts, such as perfumers. \\nThis limits our knowledge of typical odor perception and its variability over individuals. We provide a \\ndataset featuring free descriptions, evaluative ratings, and qualitative labels for 74 chemically diverse \\nmono-molecular odorants, rated by a large sample of young adults. A total of 1,227 participants \\ndescribed and rated the odors, and completed questionnaires covering their demographic background, \\npersonality traits, and the role of olfaction in their daily lives. The dataset offers a valuable foundation \\nfor research aimed at understanding the fundamentals of olfactory perception.\\nBackground & Summary\\nFor decades, scientists have been trying to answer the question of why an odor smells the way it does 1. Since \\nthen, research has led to the decoding of the olfactory receptor genome and outlined that only approximately \\n400 human olfactory receptor genes are responsible for the recognition of more than one million different olfac-\\ntory percepts through olfactory pattern recognition at the level of the olfactory bulb2. However, the question of \\nhow exactly chemical structure relates to smell remains unsolved. Some associations have been discovered. For \\nexample, a few specific chemical groups are associated with characteristic odors, e.g., esters typically smell fruity \\nand floral3, and larger and more complex molecules tend to smell more pleasant than smaller and less complex \\nones4,5. However, only very few of such general rules exist. Odors posit, besides size and complexity, hundreds or \\neven thousands of molecular properties and all can be relevant to olfactory perception.\\nUnraveling the structure-percept relation hence calls for large datasets in which many participants evalu -\\nate many odors. Generating datasets of odor perception is quite time-consuming. Olfactory stimuli must be \\ncarefully prepared and stored to maintain odor quality, the presentation is only possible in an analogue fashion \\n(compared to images or sound, which can easily be distributed digitally) and odor adaptation sets in fast with \\ncross-effects between different odors6, limiting the number of odor presentation per trial. This makes large-scale \\napproaches challenging. As a result, almost all available databases originate from trained expert panels or the \\nperfume industry, such as those from Dravnieks7, Leffingwell8, or The Good Scent Company (dataset available \\nfrom https://github.com/pyrfume, where a well-curated collection of currently available databases is stored). \\nWhile such expert databases are a valuable resource for evaluations of olfactory quality and convince by high \\nreliability, they are somewhat limited in their validity and objectivity. Most of those use predefined labels (such \\nas “pleasant” , “sweet” , “bitter” , “musky”) which prime the smellers’ perception. The labels vary between data-\\nbases, which makes comparisons challenging. Furthermore, for many of these databases, the data generation \\nprocess lacks transparency, particularly concerning how perceptual labels were chosen, the number of expert or \\nnon-expert individuals involved in the evaluations, and how the evaluations of multiple experts were aggregated. \\n1Department of Clinical Psychology, Institute of Psychology, Friedrich-Schiller-University Jena, Jena, 07743, \\nGermany. 2Department of Psychotherapy and Psychosomatic Medicine, Faculty of Medicine and University Hospital \\nCarl Gustav Carus, TUD Dresden University of Technology, Dresden, 01307, Germany. 3institute for Materials Science \\nand Max Bergmann Center for Biomaterials, TU Dresden, Dresden, 01069, Germany. 4institute of Physical chemistry, \\nFriedrich-Schiller-University Jena, Jena, 07743, Germany. 5Faculty of Medicine, TU Dresden, Smell and Taste Clinic, \\nDresden, 01307, Germany. 6German Centre for Mental Health (DZPG), site Halle-Jena-Magdeburg, Halle-Jena-\\nMagdeburg, Germany. ✉e-mail: antonie.bierling@uni-jena.de\\nDATA DeSCrip Tor\\nopeN\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='9496d38a-b5c6-4afb-9391-79a88ad7f62b', embedding=None, metadata={'page_label': '2', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='2Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nFinally, ratings which are obtained from a small sample of trained experts, are not necessarily representative of \\nthe general population’s (laypeople) perception9. One important dataset investigating laypeople was published \\nin 2016 by Keller & Vosshall10, in whose study over 450 substances were rated in several odor concentrations by \\n55 healthy laypeople. With their dataset, some of the previous findings in experts were replicated, e.g., a positive \\nassociation between pleasantness and molecular size10. However, the study also revealed substantial variance \\nin olfactory perception. This is not surprising - the perception of smells is dependent on numerous factors, \\nfrom person-related factors like gender11, age12,13, and personality13,14, or cultural background15,16, to context and \\nexperience related influences on smelling17,18. Thus, in order to quantify and conduct more fine-tuned analyses \\non inter-individual variation in olfactory perception, larger datasets are necessary, ideally, with a magnitude of \\n10 or 20 times the number of participants as studied in Keller & Vosshall. Such data would enable a more precise \\ninvestigation of the relationship between chemical structure and perception, offering a significantly improved \\nsignal-to-noise ratio by better control over person-related variability.\\nWith this dataset, we aim to provide a comprehensive database of olfactory perception based on a large-scale \\napproach. This study involved the participation of 1,227 individuals who evaluated 74 mono-molecular odors \\nby free description and rating scales. Additionally, the participants completed questionnaires regarding their \\nsocio-demographic background, personality, and the importance of olfaction in their everyday lives (see sche-\\nmatic Fig.\\xa01). The odorants were chosen to span on a broad range of chemical composition. Furthermore, we \\nchecked for an overlap of our chosen molecules and the perceptual dimensions investigated with previous data-\\nsets such as the one from Keller & Vosshall10 to enable pooling and enlarging the current database of olfactory \\nperception. With the dataset described here, we thus expect to provide a valuable tool for further investigations \\non the relation of the “structure-percept problem” as well as a useful database for olfactory research.\\nMethods\\nStudy design and sample.  Study design and rationale . The aim of this study was to create a database \\ncontaining the odor perception of as many laypersons as possible for as many mono-molecular odorants as pos-\\nsible. To achieve this goal, we followed a nested design by collecting odor ratings from 1,080 healthy individuals, \\nwith each odorant being rated by at least 120 individuals, thus, resulting in nine sampling groups. Each sampling \\ngroup received an odor set consisting of 10 mono-molecular odorants: eight unique odorants and two “anchor \\nodorants” , which were used in all odor sets. Thus, a total of 74 odorants has been investigated in this study (9 odor \\nsets \\xa0× 8 unique odorants \\xa0+ 2 anchor odorants \\xa0= 74 odorants). The study was primarily run at the participants’ \\nhomes (“@home” group). Odor sets 1–7 (recruitment goal n = 840) were provided to the participants as test kits \\ncontaining 10 odorant samples in plastic containers and a link to an online survey, which enabled the participants \\nto conduct the study at home. To ensure the validity of this approach, we aimed to invite 240 participants for \\nthe odor sets 8 and 9 (“@lab” group) to our laboratory facilities. This enabled us to compare anchor odor ratings \\nbetween “@home” and “@lab” . For testing reliability, we invited 60 additional participants (“retest” group), who \\nperformed the study with a subset of odorants at two appointments separated by approximately one week. This \\ndesign enables analyses of olfactory perception within-subject, between-subject, and temporal within-subject sta-\\nbility. The 72 unique odorants were randomly assigned to the nine odor sets with one exception: Some odorants \\nwere identified as being too dangerous (e.g., irritating for the skin if touched) for an uncomplicated testing at the \\nhomes of the participants without supervision. These odorants were assigned to the odor sets 8 and 9, which were \\ntested under supervision at our laboratories. For the retest sample, a selection of six odorants out of the 74 was \\nmade based on interim results of the main study (details given below). Finally, in addition to the 1,080 healthy \\nindividuals, we aimed to recruit 120 patients with reduced olfactory performance (patient group) to compare \\nhealthy and pathologically altered olfactory perception. The patients received odor set 9 to enable direct compar-\\nisons between healthy participants and patients. An overview of the study design with the sampling groups and \\nassigned odor sets is given in Figs.\\xa01 and 2.\\nInclusion and exclusion criteria. We included healthy individuals within an age range of 16 to 45 years who \\nwere proficient in German and reported normal olfactory ability, as well as patients with a reduced olfactory \\nperformance. Olfactory ability was additionally tested with the Sniffin Sticks ®(Burghart Messtechnik, Holm, \\nGermany) test battery in all @lab participants and in all patients. Patients were excluded if they fell outside of the \\nrange for hyposmia, i.e., values below 16 or above 30. Further exclusion criteria were pregnancy as well as - for \\nthe healthy participants - significant health impairments associated with interference with olfactory function \\n(e.g., diabetes mellitus, renal insufficiency); acute or pronounced chronic inflammation of the nose (including \\nacute COVID-19 infection) and paranasal sinuses; other relevant pre-existing or concomitant diseases in the ear, \\nnose and throat area or underlying brain-organic diseases.\\nParticipants and recruitment. In total, we recruited 1,227 participants, thus, over-achieving our recruitment \\ngoal of 1,080 healthy individuals. The minimum group occupancy rate for healthy participants was 95.8% \\n(Fig.\\xa01). Participants were recruited through a study website, via public notices, posts in forums and social net-\\nworks, announcements in lectures at the universities of Dresden and Jena, and notices in the outpatient clinic \\nof the Interdisciplinary Center for Smell and Taste. Participants received moderate financial compensation for \\nparticipation (@lab, retest), or a little present (@home; sticker, keychain, and possibility to participate in a lottery \\nto win vouchers with a value of 50€). Patients received no financial compensation. For each of the nine odor \\nsets, we stopped the recruitment as soon as n = 120 participants had finished taking part in the whole study. \\nHowever, as on average one third of the recruited @home participants dropped out, we handed out more @\\nhome test kits than necessary to compensate dropouts. Similarly, for the same reason we arranged more @lab \\nappointments than necessary to compensate “no show” participants. This resulted in a slight over-recruitment \\nfor some odor sets (+35 for @home and +4 for @lab). On the other hand, although the @home participants ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='9ff9142a-ee15-418a-acba-7f22df2ab14d', embedding=None, metadata={'page_label': '3', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='3Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nwere informed beforehand about inclusion and exclusion criteria (see section below), some of the participants \\nstill named exclusion criteria after receiving the test kit when conducting the study at home. Thus, for some \\nodor sets, fewer than 120 participants fulfilled the inclusion-/exclusion criteria. 87 participants across all @\\nhome participants reported one or several exclusion criteria: 76 reported a current or chronic infection of the \\nnose or sinuses, five a general health impairment related to olfaction, three an ENT (ear-nose-throat) related \\nhealth impairment and three were pregnant. The data for the 87 excluded participants are also accessible via the \\nprovided files, but not counted or reported here in the validation section. For the @lab participants, inclusion \\nand exclusion criteria were double-checked before invitation to the on site appointment and checked again on \\nFig. 1 Overview of the study design and procedures. The study has been conducted in Germany and in the \\nGerman language only, instructions and scales are presented in English here for better readability. First, all \\nparticipants filled in questionnaires using an online survey implemented via LimeSurvey. Then, participants \\nwere assigned to one out of nine odor sets, containing a subset of ten out of the total 74 mono-molecular odors. \\nThe odor samples were provided in small plastic containers. Two “anchor odors” were used in all odor sets \\nfor comparison between groups, the other eight odors differed between groups. The participants first gave a \\nfree description for each odor (in German) and then evaluated them again using eight visual analogue scales \\nand 16 further qualitative labels obtained from Keller & Vosshall10. The labels from Keller & Vosshall were \\ntranslated into German for the study, but are displayed in English in this figure. 875 participants for odor sets 1 \\nto 7 (@home) performed the questionnaires and odor evaluation at home using a prepared test kit. 244 healthy \\nparticipants and 48 patients for odor sets 8 and 9 (@lab) performed the odor evaluation on site at the laboratory \\nfacilities. Another 60 healthy participants were recruited for a retest study: Each smelled at a selection of six \\nodors chosen based on interim results of data collection from the nine odor sets on two measurement occasions \\nseparated by approximately one week.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='912bdc9b-b3a7-426f-a18a-977966ae4ca9', embedding=None, metadata={'page_label': '4', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='4Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\narrival. Thus, no participants were excluded for @lab. In the end, we included 875 participants (+87 excluded) \\nfor @home and 244 participants for @lab. Concerning the patient sample, recruitment was more challenging. In \\norder to enable better comparability with the healthy group, the age range was also set at 16-45 years. However, \\npatients presenting themselves in the clinic with olfactory disorders are typically older. As the recruitment for \\nthis study was during different phases of the ongoing COVID-19 pandemic (2021–2023), fluctuating access to \\nyoung, hyposmic patients was given. Further complicating recruitment, not all patients in the respective age \\nwere eligible. Anosmic individuals, i.e., patients who were diagnosed having none or a very reduced sense of \\nsmell19, were not included for ethical reasons. For them, the task of giving proper descriptions and ratings of \\nodors, which they cannot perceive at all is very frustrating and discouraging. The final sample size for this group \\nwas n=48 patients.\\nFig. 2 Odors selected from the chemical space. Odors were chosen from a list of commonly used odors in \\nindustry and research provided in Khan et al.4 and used for conducting a principal component analysis on a \\nrange of physical and chemical properties as described in Bierling et al.26. The position of the molecules chosen \\nfor this study in the physico-chemical odor space is indicated as “x” . The red “x” show the two anchor odors used \\nacross all groups. Two-dimensional illustrations of the molecules are shown for each group. Molecule names are \\ngiven as abbreviative forms, see Data Record 2, together with the corresponding CAS number in brackets. Note: \\nThe odor tributyrin (set 1, odor number 9) has later been replaced by 2-phenyl ethanol (PEA) due to chemical \\nreactivity with the plastic containers. For the same reason, PEA was used in odor set 3 (odor number 23) instead \\nof the originally chosen odor cuminol (4Isoprop). For details see section “Odor selection” .', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='1689ebbc-3722-4823-9bd8-3acb68383048', embedding=None, metadata={'page_label': '5', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='5Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nprocedure. Ethical approval and informed consent. This study has been approved by the respective Ethics \\nCommittees of the TU Dresden (process number 361082020) and the Friedrich-Schiller-Universität Jena (process \\nnumber FSV 23/019). At the beginning, the subjects were informed online about the aim and procedures of the \\nstudy as well as the data protection measures taken. For all participants younger than 18, written informed con-\\nsent from the parents had been obtained beforehand in addition to the informed consent given by the under-aged \\nparticipants themselves. Participants have been explicitly informed and asked for consent of publishing the data-\\nset in an open access publication. To protect the participants’ anonymity, we checked the sociodemographic varia-\\nbles (age, gender, nationality, mother tongue) for rare combinations, which might identify individuals. Moreover, \\nthe free descriptions of odors were screened for potentially identifying information during the standardization \\nprocess described below.\\nOnline survey. All recruited @home and @lab participants were given access to an online survey implemented \\nvia the TU Dresden hosted LimeSurvey platform before the actual experiment. All questionnaires and odor \\nevaluations have been conducted in German. Participants were first asked to complete a short questionnaire to \\nrecord demographic data (age, gender, nationality, native language, education level, professional level, smoking \\nhabits, and subjective smelling ability at the testing day on a scale from 0 = very low to 100 = very good olfac-\\ntory ability). In addition, a screening of mental health was performed using the Patient Health Questionnaire 420 \\nand the presence of a (recovered) COVID-19 infection or any exclusion criteria was asked. Subsequently, several \\nother questionnaires were completed online: The participants’ personality was assessed with the long 60-item \\nform of the Big Five Inventory-221, the role of olfaction in the everyday lives of the participants was evaluated \\nfrom a questionnaire on olfactory dreams22,23 and a questionnaire on the individual odor significance 18. After \\ncompletion of the online questionnaires, the olfactory evaluation took place either at the participants’ home (@\\nhome) or in our laboratory facilities (@lab). All obtained data (socio-demographic data, questionnaires) are \\ndescribed in Data Record 1.\\nOdor evaluation. The @home participants received a set of ten labeled plastic containers (diameter of 3cm) \\nloaded with 0.5ml of the different odors on a gauze swab and a detailed instruction including a link to an online \\nsurvey. The participants were instructed to conduct the odor evaluation in an undisturbed place, which is as \\nfree as possible from odors in the surroundings. They were further instructed not to smoke, eat or drink coffee \\nfor at least 30 minutes before conducting the experiment. To ensure variability in the order of presentation of \\nodors, we created four randomization groups (A, B, C, D) for each of the nine odor sets. Hence benzyl acetate, \\nfor example, could appear as the first odor in one randomization group and the fifth in another. Participants \\nreceived the 10 labeled containers and were instructed to evaluate them in chronological order. To resolve the \\norder of appearance of the odors later, the participants’ randomization group was included in their participant \\ncode. The odor evaluation was implemented via an online survey (LimeSurvey) and the instructions given in \\nthe survey also structured the evaluation procedure. In a first block, the participants were instructed to describe \\nthe smell of the odors freely. For this purpose, the participants were asked to open the labeled plastic container \\nwith the respective number from 1 to 10, smell it for a few seconds, and describe the odor of the sample as pre-\\ncisely as possible in a free-text field before closing the container and opening the next one. In the second block, \\nthe participants were asked to smell each odor again and rate it on eight visual analogue scales (pleasantness, \\nfamiliarity, intensity, irritability, edibility, disgust, warmness, coldness; German scales used: angenehm, bekannt, \\nintensiv, reizend, essbar, Ekel erregend, warm, kalt) and 16 qualitative descriptors, before closing the container \\nand opening the next one. For the analogue scales, the slider was set exactly in the middle between the anchors \\n“not at all” to “very” , but had to be clicked once, if the middle position was to be targeted. For the sixteen qualita-\\ntive descriptors, participants were asked to mark “yes” or “no” (sweet, sour, fruity, spices, bakery, garlic, fish, burnt, \\ndecayed, grass, wood, chemical, flower, musky, sweaty, ammonia/urinous; German scales used: süß, sauer, fruchtig, \\nwürzig, nach Backware, nach Knoblauch, nach Fisch, verbrannt, verwest, nach Gras, nach Holz, chemisch, blumig, \\nmoschusartig, nach Ammoniak/Urin). The choice of the visual analogue scales was made in alignment with pre-\\nvious studies in the field 24,25 and represents a mixture of very commonly used scales to characterize olfactory \\nquality (pleasantness, intensity, familiarity, edibility, disgust) and trigeminal quality (warm, cold, irritating). The \\nchoice of qualitative labels was made in alignment with the qualitative descriptors used in Keller & Vosshall10, \\nhowever, to reduce the strain of the participants, labels were asked binary (yes/no) instead of using visual ana-\\nlogue scales as in Keller & Vosshall10. The @lab participants and patients performed the odor evaluations in the \\nlaboratory facilities of the University Hospital in Dresden. Participants were instructed by the investigator on \\nsite and then performed exactly the same procedure as the @home participants with the only difference, that \\nthe investigator remained passively in the room (e.g., to answer questions if participants had any) and that a test \\nof olfactory ability was conducted after the odor evaluation (described in next section). The retest participants \\nconducted the same procedure as the @lab participants and the patients, but without the test of olfactory ability \\nand on a subset of six chosen odors, on two consecutive appointments separated by approximately a week. All \\nobtained data (odor evaluations main study and retest) are described in Data Record 1.\\nTest of olfactory ability. The Sniffin’ Sticks test is a standard procedure that uses felt-tip pens loaded with odors, \\neach of which is briefly opened and presented to the subject for smelling 19. The total sum score is made up of \\nthree subtests for odor threshold, discrimination, and identification. In each subtest, a maximum of 16 points \\ncan be achieved, i.e. the total scale ranges from 0 to 48 points. A sum score of at least 31 points indicates normal \\nolfactory functioning (normosmia), values between 16 and 30 refer to a reduced olfactory functioning (hypos-\\nmia) and a score below 16 refers to a very reduced or absent olfactory functioning (anosmia).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='f50a1e5b-3998-4eca-99b5-b73900bec1d4', embedding=None, metadata={'page_label': '6', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='6Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nodor molecule selection. In total, 74 mono-molecular odors were chosen, which differ in their chemical \\ncomposition and overlap with other studies conducted on olfactory perception (see Fig.\\xa0 2). In order to achieve \\nsampling group @home @lab patients retest\\nodor set 1 2 3 4 5 6 7 8 9 9 retest\\nAge\\nMean 26.99 26.77 23.99 24.78 25.38 25.99 22.95 27.74 28.88 33.95 27.17\\nStd 6.73 7.59 7.5 7.77 8.02 7.85 4.53 7.42 7.16 7.34 5.32\\nGender (%)\\nFemale\\n71 86 92 88 87 82 103 60 83 29 41\\n62% 66% 72% 72% 65% 69% 80% 50% 67% 60% 68%\\nMale\\n43 44 35 35 45 34 24 50 33 14 18\\n37% 34% 28% 28% 34% 29% 19% 41% 27% 29% 30%\\nOther\\n1 2 1 1 1 1 1\\n1% 2% 1% 1% 1% 2% 2%\\nHighest education (%)\\nnone\\n1 2 1 1 3\\n1% 2% 1% 1% 3%\\nlower secondary\\n1 1 2 1 1 2\\n1% 1% 2% 1% 1% 2%\\nmiddle school\\n6 26 27 36 37 9 2 20 21 3 3\\n5% 20% 21% 29% 28% 8% 2% 17% 17% 5% 5%\\nhigher secondary\\n104 104 96 86 92 108 123 87 93 57 57\\n90% 79% 76% 70% 69% 92% 96% 72% 76% 95% 95%\\nother\\n1 1 1 2 1 1 1\\n1% 1% 1% 2% 1% 1% 1%\\nHighest professional education (%)\\nnone\\n39 38 76 70 67 60 86 30 32 18 18\\n34% 29% 60% 57% 50% 51% 67% 25% 26% 30% 30%\\nvocational training\\n11 26 15 22 24 15 18 35 32 4 4\\n10% 20% 12% 18% 18% 13% 14% 29% 26% 7% 7%\\nmaster craftsman\\n5 1 2 1 3 1 7 2 1 1\\n4% 1% 2% 1% 3% 1% 6% 2% 2% 2%\\nbachelor\\n23 18 13 8 11 18 11 19 18 16 16\\n20% 14% 10% 7% 8% 15% 9% 16% 15% 27% 27%\\nmaster\\n35 41 19 19 28 20 11 18 28 19 19\\n30% 31% 15% 15% 21% 17% 9% 15% 23% 32% 32%\\ndoctorate\\n4 3 3 3 3 3 1 1 4 2 2\\n3% 2% 2% 2% 2% 3% 1% 1% 3% 3% 3%\\nSmoking behavior (%)\\nnon-smoker\\n99 100 103 96 115 99 100 83 87 50 50\\n86% 76% 81% 78% 86% 84% 78% 69% 71% 83% 83%\\nsometimes\\n11 14 15 15 10 12 21 17 17 7 7\\n10% 11% 12% 12% 8% 10% 16% 14% 14% 12% 12%\\nseveral times a week\\n1 1 2 1 5 3 3 3\\n1% 1% 2% 1% 4% 2% 5% 5%\\nonce a day\\n2 1 2 3 1\\n2% 1% 2% 2% 1%\\nseveral times a day\\n3 12 7 8 7 6 3 10 9\\n3% 9% 6% 7% 5% 5% 2% 8% 7%\\nCOVID-19 infection (%)\\nYes\\n22 29 72 77 81 98 104 25 69 35 47\\n19% 22% 57% 63% 61% 83% 81% 21% 56% 73% 78%\\nNo\\n93 102 55 46 52 20 24 85 47 8 13\\n81% 78% 43% 37% 39% 17% 19% 70% 38% 17% 22%\\nTable 1. Overview of socio-demographic background. Mean age and distribution of gender, highest education, \\nhighest professional education, smoking behavior and the prevalence of a (recovered) COVID-19 infection are \\ngiven as frequencies and in percentage split by group.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='fd1a6b42-7d4a-45f4-bf7d-c700a491d676', embedding=None, metadata={'page_label': '7', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='7Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\na list of chemically diverse molecules, we used the physico-chemical “odor space” described in Bierling et al .26, \\nwhich was based on chemical and physical descriptors for 1389 odor molecules obtained from Khan et al.4. For \\nthe choice of molecules in this study, we first calculated an average distance vector between the 1389 odor mol-\\necules by calculating the distance of each molecule to all other molecules within the three-dimensional space \\nderived from the first three principal components of the odor space4,26. The resulting list of molecules was then \\nsorted by the distance vector, i.e., from the highest averaged distance of a molecule to all other molecules in the \\nodor space to the lowest averaged distance to all other molecules. From this list, we successively chose odors \\n(starting from the molecule with the highest average distance vector), which showed an overlap with the mole-\\ncules studied in Bierling et al.26, Khan, et al.4, Keller and Vosshall10 and Lötsch, et al.24 in order to enable as many \\ndirect comparisons of olfactory perception with previous studies as possible. As a last step, odors were checked \\nfor their availability, price and safety. As odors for odor sets 1 to 7 were intended for testing at the participants’ \\nhomes, we omitted odors, which were in any way hazardous, i.e., corrosive.\\nFor two specific odor molecules, tributyrin and cuminol, a chemical reaction with the plastic container was \\nobserved after some time of storage. This led to some participants being unable to open the plastic containers \\nand the danger of liquid dripping from the containers. In case of tributyrin, 83 of the 115 participants still \\ngave descriptions on the odor and these can be found in the dataset, whereas for cuminol, the problem already \\nbecame apparent after a few days of storage during the preparation of the test kits, thus, the odor was exchanged \\nbefore any examinations took place. Both tributyrin and cuminol were replaced by 2-phenylethanol, which is an \\noften used odor in olfactory research and is also used in groups 9 and 10.\\nBenzyl acetate and 4-decanolide were chosen as “anchor” odors, which remained the same for all partici -\\npants. Those molecules were selected, because both have been shown to work in experiments with a large variety \\nof participants, including participants with reduced olfactory abilities24.\\nFor the retest study, we chose six odors based on their perception measures pleasantness, familiarity, irrita-\\nbility and disgust after approximately 80% of the data collection for @home and @lab testings had been finished. \\nBased on the interim results, we chose to include the (at that time) most and least pleasant, familiar, irritating \\nand disgusting odors, which resulted in the four odors maltol (most pleasant, least irritating), (R)-(-)-carvone \\n(most familiar, least disgusting), allyl caproate (second-most disgusting, least familiar) and p-tolyl acetate (least \\npleasant, most irritating and most disgusting). Furthermore, we included the two anchor odors 4-decanolide \\nand benzyl acetate, which ranked medium for the four perceptual measures with a high variability. The final \\nodor selection, as well as further information on the used substances is made available (see Data Record 2).\\nConcentration of the odors. All 74 odors were piloted to be similar in their intensity and were then used in the \\nobtained ideal concentrations (e.g., lower concentration for very intense odors, higher concentration for more \\nsubtle odors). First, the two anchor odors benzyl acetate and 4-decanolide were piloted to have an average inten-\\nsity of around 50 on a scale from 1 = not perceivable to 100 = extremely intense in n=18 piloting participants. \\nAfterwards, all other odors were piloted group-wise with at least n = 10 participants per group to match the \\nintensity of benzyl acetate and 4-decanolide. As a starting concentration for piloting, if available, the suggestion \\nprovided by the database of The Good Scent Company was used as a first estimate of ideal intensity. If none was \\navailable, a moderate intensity was first agreed upon between the two study investigators A.L.B. and M.R. If the \\npiloting showed that an odor was very intense or very subtle compared to the anchor odors, the odor was used \\nin a more diluted or more concentrated form in the following steps “undiluted” , “1:1” , “1:10” , “1:100” to “1:1000” . \\nResults from intensity piloting are provided in Data record 3.\\nData Records\\nThe dataset is available at zenodo (14727277)27. In total, the record comprises six files: One main data file (data.\\nxslx) containing the data (i.e., socio-demographic data, questionnaires, odor evaluations) from the main study \\nand the retest study, another file containing information on the odor stimuli (odors.xlsx), a file for the intensity \\npiloting results (intensity_piloting.xlsx), a file containing a detailed list of all variables (variables-dictionary.\\nxlsx), a file containing the most frequent free descriptions with an English translation (free_descriptions_trans-\\nlated.xlsx) and the jupyter notebook file (analyses.ipynb), which contains the analyses and code for creating the \\nillustrations shown in this manuscript.\\nData record 1 - data.xlsx. The dataset is provided in an excel spreadsheet (XLSX) and is structured in long \\nformat, i.e., each row represents one participant’s ratings of one odor (for retest: at one measurement occasion), \\nwith a total of ten rows per participants for the main study (10 odors \\xa0× 1 measurement) and twelve rows per par-\\nticipant in the retest study (6 odors \\xa0× 2 measurement occasions). The dataset is structured in four types of data, \\nwhich are explained in detail in the variables dictionary: 1) identifiers of participants, odors and study groups, 2) \\nsocio-demographic variables, 3) results of the odor evaluation and 4) the questionnaire results as raw items and \\nas aggregated scores.\\nAs identifiers of the participants, the study (main study/retest), dataset type (home/lab/pat/retest), a code \\nfor each participant and the belonging to the different odor groups and randomization groups are given. For the \\nstimuli, the molecule and order of appearance of the molecules are given, as well as the time of storage of the @\\nhome odors in the main study and the time between test and retest appointment for the retest study.\\nFor the socio-demographic background, we asked for the participants’ age, gender, highest school and pro-\\nfessional education, as well as smoking habits, and if participants ever suffered from a COVID-19 infection. \\nNationality and mother tongue were also asked, however, as only a very small minority of participants did not \\ngrow up in Germany (3.3%) or speaking German as a native language (2.3%), these two variables have been \\nomitted from the dataset to ensure anonymity of the participants.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='9881e45e-a5a0-4293-acf4-c87fdef2f76a', embedding=None, metadata={'page_label': '8', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='8Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nFor the odor evaluation, each participant was asked to give a free description for each odor. The free descrip-\\ntions are provided in a standardized way for better comparability. Standardization was achieved by the following \\nprocedure. For each free description entry, all relevant odor description terms were identified. For example, for \\nan entry “This odor smells like lemon, a little fruity, sour and remembers me of a cleaning agent. ” the following \\nterms were identified: “lemon” , “fruity” , “sour” and “cleaning agent” . Furthermore, for standardization, the terms \\nwere transferred to their so-called “lemmas” , i.e., the dictionary form of words sharing the same word stem and \\nsame meaning. For example, “sweetish” and “sweet” would both be coded as “sweet” . If a word is both used as \\nan adjective and noun (e.g., “lemon” and “lemony”), the more frequent term was used. Negations were coded \\nas “NEG-” for both lexical (“not”) and morphological (“un-”) negations. Modifiers indicating intensity, location, \\ntemporality or similar (e.g. “a little” , “very” , “at home”) were not included. For complex descriptions, which would \\nchange their meaning if split in several terms (e.g. “old people perfume”) were kept as one term. Specific brand \\nnames (e.g. “Colgate toothpaste” , “Mucosolvan cough syrup”) were also coded, but the more general term (“tooth-\\npaste” , “cough syrup”) without brand name was coded as an additional term to facilitate comparability between \\ndescriptions. Finally, all free standardized descriptions were checked again for consistency and redundancies again \\nbefore storage in this data file. The raw unprocessed data for the free descriptions are not available in this dataset \\nin order to ensure that no person identifying information is published, but can be obtained upon request from the \\nauthors. A file containing English translations of the most frequent German descriptions is given in a separate file  \\n(see Data Record 4).\\nFurthermore, the ratings of the odors on eight visual analogue scales (pleasant, intense, familiar, warm, \\ncold, edible, disgusting, irritating) as well as on 16 qualitative labels (sweet, sour, fruit, spices, bakery, garlic, \\nfish, burnt, decayed, grass, wood, chemical, flower, musky, sweaty, ammonia/urinous) obtained from Keller & \\nVosshall are presented. In addition, we also present the results of the Sniffin Sticks test batteries (threshold, dis-\\ncrimination, identification, total score) as well as the subjective olfactory performance.\\nFinally, the responses of the questionnaires  on mental health, personality, individual odor significance, \\n(self-)smelling behavior and olfactory dreams are provided both as the raw items and in an aggregated form. \\nDetails on the coding of answers are given in the variables dictionary.\\nData record 2 - odors.xlsx. A list of all substances used as odor stimuli is given in an excel spreadsheet \\n(“odors.xlsx”) and provides one row per odor used in one group, together with further information on concen-\\ntration, identifiers and catalogue numbers. The abbreviated code of the molecules is given for matching with the \\nperception dataset.\\nData record 3 - intensity_piloting.xlsx. The data of the intensity piloting is provided in another excel \\nspreadsheet (“intensity_piloting.xlsx”) in the long format, i.e., each row indicates one participant rating one odor \\nsubstance’s intensity from 1 = not perceivable to 100 = very intensive. The starting concentration (odor:solvent) \\nand volume used in the piloting is given, as well as the abbreviative code of the molecules for matching with the \\nperception data.\\nData record 4 - free_descriptions_translated.xlsx. This dataset (“free_descriptions_translated.xlsx”) \\ncontains the top-ranked free descriptions together with an English translation. To provide this file, the German \\nfree descriptions from the main dataset (see Data Record 1) were grouped and sorted by frequency of nam-\\ning across all odor sets and participants. From that, we identified the descriptions ranked 1 to 100, including \\nties, and translated them to English using back-and-forth translation. Discrepancies between translation and \\nback-translation were resolved by discussion. Within this dataset, one row represents one description (column \\n“free_description”) given by one participant (“code”) for one molecule (“molcode”) as well as the English trans-\\nlation (“translation”). There can be several rows per participant and molecule as participants typically used sev-\\neral descriptors in their free descriptions. Using these top 100 ranked descriptions, the translated dataset covers \\n26,115 out of the original 37,041 free descriptions, which represents 70.5% of all free descriptions.\\nSample composition. In total, the dataset comprises data from n = 1,227 individuals, who participated in \\nthe different study groups as outlined above. The sample is composed of young healthy individuals and patients \\nwith a reduced olfactory function. Across all study groups, the mean age was 26.18 years (SD = 7.61) with a \\ngender distribution of 68.27% females, 31.21% males and 0.52% who identified other. The education level of \\nthe participants was generally high, with 81.11% having completed the highest secondary education track in \\nGermany; 16.71% holding a degree from a German middle school, and 1.50% having another or (until that time) \\nno completed educational degree. With regard to professional education, the majority of 45.23% of participants \\nhad no completed professional education, which is probably attributed to a high percentage of undergraduate \\nuniversity students in the study. A total of 34.79% of participants held university degrees, with 12.62% having \\nobtained a Bachelor’s degree, 19.90% a Master’s degree and 2.27% a doctoral degree. Additionally, 18.00% had \\ncompleted a vocational training diploma, and 2.00% were master craftsmen. Across all study groups, the majority \\nof participants were non-smokers (80.12%), followed by occasional smokers (“sometimes” , 12.00%), and regular \\nor frequent smokers (“several times a week” , 0.82%, “once a day” , 1.18%, “several times a day” , 5.90%). The preva-\\nlence of a (recovered) COVID-19 infection varied largely between groups. The recruitment for this study started \\nwith the groups 1, 2 and 8 in autumn 2021, when only a small percentage of the population had suffered from a \\nCOVID-19 infection. Accordingly, in those groups only around 20% of the participants had a COVID-19 infec-\\ntion in the past (see Table\\xa01), whereas this percentage continuously increased with the recruitment period until ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='aabfd522-939a-41db-b95c-ccfdc3a4fd9d', embedding=None, metadata={'page_label': '9', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='9Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nlate summer and autumn 2023, where groups 6, 7 and the retest group showed prevalences of around 70–80%.  \\nA detailed overview of the named socio-demographic variables within each study group is given in Table\\xa01.\\nTechnical Validation\\nIn order to validate this dataset, we investigated the validity, objectivity and reliability of the odor evaluations. \\nTo this end, we compared our dataset to the results of another laymen perception dataset (Keller & Vosshall, \\n201610). Furthermore, we checked for systematic differences of sampling group or condition using the anchor \\nratings. Finally, we tested the temporal stability of olfactory perception by correlating the retest study’s percep-\\ntion measures between the two measurement occasions. More details on each validation check are outlined in \\nthe following sections.\\nValidity: Comparison with another laymen dataset. The file containing the laymen perception data \\nof n = 55 participants of Keller & Vosshall can be downloaded via10. In total, n = 56 odors overlapped between \\nthe studies. As Keller & Vosshall investigated different concentrations of the odors, the highest available con-\\ncentration was used for comparisons, as this was closest to our concentrations in this study. We calculated the \\npercentage of our participants who rated “yes” for 19 qualitative perception measures, which were available in \\nboth studies: chemical, sweet, cold, warm, spices, fruit, edible, flower, sour, wood, musky, decayed, ammonia/\\nurinuos, grass, bakery, sweaty, burnt, garlic and fish. For edible, warm and cold, we transformed our data to val-\\nues between 0 and 1. For the data from Keller & Vosshall, the data were also transformed into values between 0 \\ndimension\\n4-decanolide benzyl acetate\\ndf1 df2 F p-unc np2 ddof1 ddof2 F p-unc np2\\npleasant 8 446.7 3.390 0.001*** 0.024 8 447.3 1.829 0.070 0.013\\nintensive 8 447.3 8.603 0.000*** 0.063 8 447.3 6.509 0.000*** 0.048\\nfamiliar 8 447.1 1.621 0.116 0.012 8 447.0 2.966 0.003** 0.022\\nwarm 8 447.0 2.510 0.011* 0.017 8 447.7 1.507 0.152 0.011\\ncold 8 447.1 3.348 0.001*** 0.023 8 447.6 1.710 0.094 0.013\\nirritating 8 447.1 1.402 0.193 0.010 8 447.3 2.458 0.013* 0.018\\nedible 8 447.1 4.034 0.000*** 0.027 8 447.2 2.280 0.021* 0.016\\ndisgusting 8 446.1 3.535 0.001*** 0.021 8 447.1 1.088 0.370 0.007\\nTable 2. Results from robust Welch one-way ANOV A testing the influence of odor set on each of the eight \\nperception measures separately for the two anchor odors 4-decanolide and benzyl acetate. df = degrees of \\nfreedom, p-unc = uncorrected p-value, np2 = partial η2 Significance marked according to *p < 0.05,  \\n**p < 0.01, ***p < 0.001.\\nFig. 3 Comparison of the eight perception measures between the participants for the nine odor sets plus the \\nretest group for the two anchor odors 4-decanolide and benzyl acetate. The black dotted lines indicate the \\naverage mean across all groups.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='2c238711-34fe-4490-bf1a-92a05ae4b3a3', embedding=None, metadata={'page_label': '10', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='10Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nFig. 4 Comparison of qualitative descriptor usage with Keller & Vosshall (2016). a. For each molecule \\n(abbreviations see Table\\xa01), the proportion in percent of using each of the 19 qualitative labels is compared \\nbetween the dataset presented here as well as the dataset by Keller & Vosshall10. In total, 55 odors were suitable \\nfor comparison. The two anchor odors benzyl acetate (Benzyl) and 4-decanolide (Decan) are highlighted in \\nthe left panel. Red coloured areas indicate a higher descriptor usage by participants in the dataset of Keller & \\nVosshall, blue colored areas indicate higher percentage of usage in this dataset. Grey points and grey shaded \\nareas indicate a difference smaller than 10%. b. Overall frequencies of using each descriptor in both datasets, \\naveraged across all molecules, which overlap between the two datasets. Red indicates frequencies in the Keller \\ndataset, blue in this dataset. The order of descriptors is sorted by frequency (from top to bottom) of descriptor \\nusage in this dataset.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='cf5e48f3-f328-4432-b319-5b62b53d8e97', embedding=None, metadata={'page_label': '11', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='11Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nand 1. In order to evaluate differences between the two datasets, we the percentage of “yes” per measure and per \\nmolecule between the participants in Keller & Vosshall and our dataset. Overall, very similar patterns of ratings \\nemerged, the results for each odor are visualized in Fig.\\xa04. Furthermore, in order to evaluate whether participants \\nfrom both datasets showed similar rank values for the 56 molecules, we calculated Spearman correlation coef-\\nficients for the associations between the visual analogue scale ratings pleasantness, intensity, edibility, warmth, \\ncoldness and familiarity which were present in both studies. As a result, we found significant positive associations \\ndimension\\n4-decanolide\\nA B mean(A) mean(B) diff se T df p-corr hedges\\npleasant\\n2 5 49.59 61.79 −12.21 3.61 −3.39 246.39 0.023* −0.42\\n2 7 49.59 61.79 −12.20 3.57 −3.42 241.03 0.021* −0.43\\nintensse\\n1 3 59.26 72.29 −13.03 3.04 −4.29 217.33 <0.001*** −0.56\\n1 6 59.26 69.73 −10.47 3.08 −3.41 216.08 0.022* −0.45\\n1 8 59.26 76.37 −17.11 2.97 −5.75 207.18 <0.001*** −0.76\\n1 9 59.26 74.74 −15.48 3.03 −5.10 214.01 <0.001*** −0.67\\n2 5 68.80 59.05 9.74 3.03 3.22 254.63 0.038* 0.40\\n3 5 72.29 59.05 13.23 2.88 4.59 246.21 <0.001*** 0.57\\n4 5 68.44 59.05 9.39 2.93 3.20 247.41 0.041* 0.40\\n5 6 59.05 69.73 −10.68 2.92 −3.66 241.11 0.009** −0.46\\n5 8 59.05 76.37 −17.31 2.81 −6.16 236.16 <0.001*** −0.78\\n5 9 59.05 74.74 −15.68 2.88 −5.45 241.42 <0.001*** −0.69\\n7 8 66.01 76.37 −10.36 2.66 −3.90 234.90 0.004** −0.50\\n7 9 66.01 74.74 −8.73 2.73 −3.20 238.82 0.041* −0.41\\nwarm\\n1 3 43.65 32.56 11.09 3.34 3.32 233.96 0.029* 0.43\\n1 7 43.65 33.52 10.12 3.19 3.18 231.95 0.044* 0.41\\ncold\\n1 3 43.43 55.17 −11.74 3.43 −3.42 233.91 0.02* −0.44\\n1 6 43.43 55.90 −12.47 3.36 −3.71 225.94 0.008** −0.49\\n4 6 45.03 55.90 −10.87 3.45 −3.15 235.00 0.047* −0.41\\nedible\\n2 3 24.76 43.28 −18.52 4.05 −4.58 241.40 <0.001*** −0.58\\n2 4 24.76 38.32 −13.56 3.92 −3.46 244.07 0.018* −0.44\\n2 5 24.76 40.06 −15.30 3.93 −3.90 252.33 0.004** −0.48\\n2 7 24.76 42.30 −17.54 4.08 −4.30 241.55 <0.001*** −0.54\\n2 9 24.76 37.64 −12.88 4.03 −3.20 234.12 0.041* −0.41\\ndisgust\\n2 7 33.56 20.11 13.45 3.37 3.99 223.28 0.003** 0.50\\n4 7 30.48 20.11 10.38 3.06 3.39 231.58 0.023* 0.43\\n6 7 31.59 20.11 11.49 3.42 3.36 202.13 0.025* 0.44\\n7 8 20.11 31.21 −11.11 3.36 −3.30 203.41 0.03* −0.43\\nbenzyl acetate\\ndimension A B mean(A) mean(B) diff se T df p-corr hedges\\nintense\\n1 2 54.425 66 −11.575 3.382 −3.422 214.378 0.021* −0.446\\n1 6 54.425 66.661 −12.236 3.465 −3.531 215.189 0.015* −0.467\\n1 8 54.425 71.36 −16.935 3.268 −5.182 197.787 <0.001*** −0.687\\n1 9 54.425 65.256 −10.832 3.329 −3.253 205.11 0.035* −0.43\\n2 4 66 53.639 12.361 3.377 3.661 227.864 0.009** 0.465\\n3 8 61.153 71.36 −10.206 2.803 −3.641 233.814 0.01** −0.467\\n4 6 53.639 66.661 −13.022 3.46 −3.764 227.384 0.007** −0.484\\n4 8 53.639 71.36 −17.72 3.262 −5.433 211 <0.001*** −0.696\\n4 9 53.639 65.256 −11.617 3.323 −3.495 218.26 0.016* −0.448\\n7 8 59.556 71.36 −11.803 2.706 −4.361 235.728 <0.001*** −0.561\\nfamiliar 1 6 58.354 70.304 −11.95 3.73 −3.204 225.993 0.041* −0.423\\nirritating 4 6 38.59 52.33 −13.74 3.633 −3.782 234.833 0.006** −0.49\\nTable 3. Results from Games-Howell post-hoc comparisons for the two anchor odors 4-decanolide and benzyl \\nacetate. For the sake of readability, only significant comparisons are given. All results can be obtained from \\nthe analyses notebook provided. A, B = odor sets for comparison, mean(A), mean(B) = mean value of this \\ndimension for this odor set, diff = difference between means, df = degrees of freedom, p-corr = corrected \\np-values, hedges = hedges effect size; Significance marked according to *p < 0.05, **p < 0.01, ***p < 0.001.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='54809c05-1289-47cb-8bd3-db9669f4b477', embedding=None, metadata={'page_label': '12', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='12Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nfor pleasantness (r = 0.68, p < 0.001), familiarity (r = 0.58, p < 0.001), edibility (r = 0.37, p < 0.01) and intensity  \\n(r = 0.28, p < 0.05), but none for warmth (r = −0.07, p = 0.59) and coldness (r = −0.20, p = 0.11).\\nobjectivity: Context stability. When comparing the perception measures for the two anchor odors \\nbetween the nine odor sets in all healthy participants, significant differences in the ratings were seen between \\ngroups for both anchor odors (see Table\\xa02 and Fig.\\xa03). For 4-decanolide, significant differences in the pleasantness, \\nintensity, warmth, coldness, edibility and disgust ratings were found, with overall small effect sizes (all partial η2 \\n< 0.06), except for a moderate effect for intensity (partial η2 = 0.063). For benzyl acetate, significant differences \\nin the ratings were found for intensity, familiarity, irritability and edibility, with small effect sizes for all (all partial \\nη2 < 0.06), however, intensity showing the highest effect size again (partial η2 = 0.048).\\nGames-Howell post hoc comparisons showed significant differences in the perception of the two anchor \\nodors 4-decanolide and benzyl acetate between the nine odor sets; mostly with low effect sizes of Hedges’ g < 0.5 \\n(see Table\\xa03). For both anchor odors, intensity showed the most and largest differences between the odor sets, \\nwith moderate effect sizes for a few comparisons. For all significant comparisons refer to Table\\xa03).\\nConcerning the experimental setting, no significant difference in mean or variance were found for the per-\\nception measures pleasantness, warmness, irritability, edibility, and disgust between @home and @lab measure-\\nments. However, significant differences in the mean values were found for intensity (t = −6.77, p < 0.001, d = \\n−0.37) and coldness (t =  −3.01, p < 0.01, d = −0.16) with low effect sizes. Thus, participants in the @home \\nsetting rated odors as slightly less intense and less cold. Furthermore, a significant difference in the variance was \\nfound for intensity (t = 19.78, p < 0.001). See Fig.\\xa05.\\nreliability: Temporal stability. Retest reliability was estimated from n = 60 participants, who were asked \\nto rate and describe six of the odors (benzyl acetate, 4-decanolide, maltol, allyl caproate, (R)-(-)-carvone and \\np-tolyl acetate) at two appointments within a time span of a week (M = 7.37 days, SD = 1.37). Spearman corre-\\nlation coefficients showed significant associations of all perception measures with the retest measurement at the \\nfollow-up appointment (see Fig.\\xa06 and Table\\xa04) for all chosen odors. The highest correlation coefficients were \\nfound for pleasantness (overall correlation across odors between test and retest r =  0.81, p < 0.001), edibility \\n(r = 0.79, p < 0.001) and disgust (r = 0.77, p < 0.001). The least stability was seen for warmness (r = 0.60, p < \\n0.001), coldness (r =  0.63, p < 0.001) and intensity (r =  0.63, p < 0.001) ratings. Furthermore, the two odors \\n4-decanolide and allyl caproate showed on average a less stable pattern of retest reliability compared to the other \\nodors.\\nUsage Notes\\nThe code needed to perform the analyses and illustrations shown in this manuscript is provided as a Jupyter \\nNotebook (analyses.ipynb), operating with Python 3.7 and using the packages pandas, numpy, matplotlib, stats-\\nmodels, scipy, pingouin, seaborn, PIL and rdkit. For further adaptation of the illustrations in this manuscript, \\nFig. 5 Comparison between the two conditions @home and @lab. Each plot illustrates the distribution of one \\nof the eight perception dimensions’ ratings in boxenplots for the two anchor odors used across all groups for the \\ntwo conditions @home and @lab. A lineplot connecting the mean values is given in each subplot.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='47d91808-2078-4114-bc03-f5ad5835bd66', embedding=None, metadata={'page_label': '13', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='13Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nAdobe Illustrator 2024 and Adobe InDesign 2024 were used. In order to reproduce the analyses, download the \\njupyter notebook file (analyses.ipynb) and the three files data.xlsx, odors.xlsx and intensity_piloting.xlsx in the \\nsame directory. The files are loaded as dataframes from the original excel files in the notebook. For the compar-\\nisons with the dataset by Keller & Vosshall, the original dataset can be obtained via10.\\npleasant intensivef amiliar warmc olde dible irritating disgusting\\n0\\n50\\n100\\nBenzyl\\n0\\n50\\n100\\nCarvon\\n0\\n50\\n100\\nDecan\\n0\\n50\\n100\\nMaltol\\n0\\n50\\n100\\nTolyl\\nr=.72\\nr=.69\\nr=.54\\nr=.69\\nr=.80\\ntest retest\\nr=.43\\nr=.62\\nr=.54\\nr=.55\\nr=.42\\ntest retest\\nr=.56\\nr=.43\\nr=.51\\nr=.73\\nr=.64\\ntest retest\\nr=.46\\nr=.59\\nr=.43\\nr=.60\\nr=.50\\ntestr etest\\nr=.65\\nr=.47\\nr=.31\\nr=.62\\nr=.75\\ntest retest\\nr=.64\\nr=.86\\nr=.60\\nr=.64\\nr=.73\\ntest retest\\nr=.70\\nr=.57\\nr=0.37\\nr=0.49\\nr=0.67\\ntest retest\\n0\\n50\\n100\\nAllycap\\nr=.60 r=.68 r=.30 r=.38 r=.38 r=.60 r=.52 r=.76\\nr=.65\\nr=.72\\nr=.31\\nr=.52\\nr=.80\\ntest retest\\n0\\n50\\n100Average\\nr=.82 r=.63 r=.68 r=.60 r=.63 r=.81 r=.64 r=.76\\ntest retest test retest test retest test retestt estr etestt est retestt estr etestt est retest\\nFig. 6 Temporal stability of odor perception. The top panel illustrates the average ratings of the eight visual \\nanalogue scales at test and retest appointment as connected lineplots. Each connected line indicates ratings \\nfrom one participant. The dark green color indicates that no substantial change has occurred (absolute change \\n<10%), light green lines indicate increases or decreases by at least 10%. The bold black lines indicate the average \\nratings across participants. Below the top panel, the ratings are shown for each odor, thus, in each plot one line \\nrepresents the rating of this dimension for this odor by one participant at the two measurement occasions. On \\ntop of each plot, the correlation coefficient is given. A dotted line indicates a value of 50, which is exactly the \\nmiddle position indicating a neutral opinion.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='3ed12063-0c5f-4e0b-960d-6ca4843b7551', embedding=None, metadata={'page_label': '14', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='14Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nCode availability\\nThe code created to perform the analyses and creation of illustrations in this manuscript is available at27.\\nReceived: 16 September 2024; Accepted: 7 February 2025;\\nPublished: xx xx xxxx\\nReferences\\n 1. Ohloff, G. Chemistry of odor stimuli. Experientia 42, 271–279, https://doi.org/10.1007/BF01942507 (1986).\\n 2. Buck, L. & Axel, R. A novel multigene family may encode odorant receptors: a molecular basis for odor recognition. Cell 65, 175–187 \\n(1991).\\n 3. Genva, M., Kenne Kemene, T., Deleu, M., Lins, L. & Fauconnier, M.-L. Is it possible to predict the odor of a molecule on the basis of \\nits structure? International Journal of Molecular Sciences 20, https://doi.org/10.3390/ijms20123018 (2019).\\n 4. Khan, R. M. et al. Predicting odor pleasantness from odorant structure: pleasantness as a reflection of the physical world. J Neurosci \\n27, 10015–23, https://doi.org/10.1523/JNEUROSCI.1158-07.2007 (2007).\\n 5. Kermen, F . et al. Molecular complexity determines the number of olfactory notes and the pleasantness of smells. Scientific reports 1, \\n206, https://doi.org/10.1038/srep00206 (2011).\\n 6. Pellegrino, R., Sinding, C., de Wijk, R. & Hummel, T. Habituation and adaptation to odors in humans. Physiology & Behavior 177, \\n13–19, https://doi.org/10.1016/j.physbeh.2017.04.006 (2017).\\n 7. Dravnieks, A. Atlas of odor character profiles. (ASTM, Philadelphia, PA, 1985).\\n 8. Sanchez-Lengeling, B. et al. Machine learning for scent: Learning generalizable perceptual representations of small molecules. arXiv \\npreprint arXiv:1910.10685 (2019).\\n 9. Croijmans, I. & Majid, A. Not all flavor expertise is equal: The language of wine and coffee experts. PLoS One 11,\\xa0https://doi.\\norg/10.1371/journal.pone.0155845 (2016).\\n 10. Keller, A. & Vosshall, L. B. Olfactory perception of chemically diverse molecules. BMC Neurosci 17, 55, https://doi.org/10.1186/\\ns12868-016-0287-2 (2016).\\n 11. Sorokowski, P . et al . Sex differences in human olfaction: A meta-analysis. Front Psychol  10, 242, https://doi.org/10.3389/\\nfpsyg.2019.00242 (2019).\\n 12. Olofsson, J. K., Ekström, I., Larsson, M. & Nordin, S. Olfaction and aging: A review of the current state of research and future \\ndirections. i-Perception 12, 20416695211020331, https://doi.org/10.1177/20416695211020331 (2021).\\n 13. Larsson, M., Finkel, D. & Pedersen, N. L. Odor identification: influences of age, gender, cognition, and personality. The Journals \\nof Gerontology Series B: Psychological Sciences and Social Sciences 55, P304–P310,\\xa0 https://doi.org/10.1093/geronb/55.5.P304  \\n(2000).\\n 14. Shepherd, D., Hautus, M. J. & Urale, P . W . Personality and perceptions of common odors. Chemosensory Perception 10, 23–30, \\nhttps://doi.org/10.1007/s12078-016-9220-4 (2017).\\n 15. Chrea, C. et al. Culture and odor categorization: agreement between cultures depends upon the odors. Food quality and preference \\n15, 669–679,\\xa0https://doi.org/10.1016/j.foodqual.2003.10.005 (2004).\\n 16. Ayabe-Kanamura, S. et al. Differences in perception of everyday odors: a japanese-german cross-cultural study. Chemical senses 23, \\n31–38,\\xa0https://doi.org/10.1093/chemse/23.1.31 (1998).\\n 17. Kaeppler, K. & Mueller, F . Odor classification: a review of factors influencing perception-based odor arrangements. Chemical senses \\n38, 189–209, https://doi.org/10.1093/chemse/bjs141 (2013).\\n 18. Croy, I., Buschhüter, D., Seo, H.-S., Negoias, S. & Hummel, T. Individual significance of olfaction: development of a questionnaire. \\nEuropean Archives of Oto-Rhino-Laryngology 267, 67–71,\\xa0https://doi.org/10.1007/s00405-009-1054-0 (2010).\\n 19. Hummel, T., Sekinger, B., Wolf, S. R., Pauli, E. & Kobal, G. ‘sniffin’sticks’: olfactory performance assessed by the combined testing of \\nodor identification, odor discrimination and olfactory threshold. Chemical senses 22, 39–52 (1997).\\n 20. Kroenke, K., Spitzer, R. L., Williams, J. B. & Löwe, B. An ultra-brief screening scale for anxiety and depression: the phq–4. \\nPsychosomatics 50, 613–621 (2009).\\n 21. Danner, D. et al. Die deutsche Version des Big Five Inventory 2 (BFI-2). Open Access Repositorium für Messinstrumente https://doi.\\norg/10.6102/zis247 (2016).\\n 22. Stevenson, R. J. & Case, T. I. Olfactory dreams: phenomenology, relationship to volitional imagery and odor identification. \\nImagination, Cognition and Personality 24, 69–90,\\xa0https://doi.org/10.2190/R97G-RM98-B58N-DUYA (2004).\\n 23. Weitz, H., Croy, I., Seo, H., Negoias, S. & Hummel, T. Studies on olfactory dreaming. Chemosensory Perception 3, 129–134, https://\\ndoi.org/10.1007/s12078-010-9074-0 (2010).\\n 24. Lötsch, J. et al. Data-science based analysis of perceptual spaces of odors in olfactory loss. Scientific Reports 11, 10595, https://doi.\\norg/10.1038/s41598-021-89969-9 (2021).\\n 25. Licon, C. C., Manesse, C., Dantec, M., Fournel, A. & Bensafi, M. Pleasantness and trigeminal sensations as salient dimensions  \\nin organizing the semantic and physiological spaces of odors. Scientific reports  8, https://doi.org/10.1038/s41598-018-26510-5  \\n(2018).\\n 26. Bierling, A. L., Croy, I., Hummel, T., Cuniberti, G. & Croy, A. Olfactory perception in relation to the physicochemical odor space. \\nBrain Sciences 11,\\xa0https://doi.org/10.3390/brainsci11050563 (2021).\\n 27. Bierling, A. L. et al. A dataset of laymen olfactory perception for 74 monomolecular odors, https://doi.org/10.5281/zenodo.14727277 \\n(2025).\\nOdor pleasant intensive familiar warm cold edible irritating disgusting Mean\\nbenzyl acetate 0.74*** 0.45*** 0.58*** 0.44*** 0.64*** 0.64*** 0.69*** 0.67*** 0.61\\n4-decanolide 0.51*** 0.51*** 0.48*** 0.42*** 0.31* 0.59*** 0.39** 0.37** 0.45\\nmaltol 0.69*** 0.57*** 0.73*** 0.54*** 0.58*** 0.70*** 0.54*** 0.58*** 0.62\\nallyl caproate 0.65*** 0.68*** 0.27* 0.38** 0.43*** 0.48*** 0.53*** 0.76*** 0.53\\n(R)-(-)-carvone 0.74*** 0.58*** 0.61*** 0.55*** 0.40** 0.83*** 0.57*** 0.78*** 0.63\\np-tolyl acetate 0.72*** 0.46*** 0.63*** 0.49*** 0.76*** 0.56*** 0.68*** 0.81*** 0.64\\nAcross odors 0.81*** 0.63*** 0.70*** 0.60*** 0.63*** 0.79*** 0.65*** 0.77***\\nTable 4. Correlations (Spearman) of perception measures with follow-up appointment (n = 60). Significance \\nmarked according to *p < 0.05, **p < 0.01, ***p < 0.001.\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='3c54c7cf-b877-4a3f-8459-bb330f14e6f1', embedding=None, metadata={'page_label': '15', 'file_name': 'dataset_laymen.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\dataset_laymen.pdf', 'file_type': 'application/pdf', 'file_size': 14551934, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='15Scientific  Data |          (2025) 12:347  | https://doi.org/10.1038/s41597-025-04644-2\\nwww.nature.com/scientificdatawww.nature.com/scientificdata/\\nacknowledgements\\nThis research was funded within the project “Olfactorial Perceptronics” by the VolkswagenStiftung (Grant Nos. \\n96632 and 9B396). We acknowledge Florian Anders for helpful comments and Regine Klappauf for helping with \\nrecruitment and testing.\\nAuthor contributions\\nA.L.B., A.C., T.H. and I.C. conceptualized the study, A.L.B., M.R. and T.J. conducted the experiments. A.L.B. \\nanalysed the results and wrote the first draft of the manuscript. G.C. provided laboratory resources. T.H., I.C., \\nA.C. and G.C. acquired financial support leading to this publication. All authors reviewed the manuscript.\\nFunding\\nOpen Access funding enabled and organized by Projekt DEAL.\\nCompeting interests\\nThe authors declare no competing interests.\\nadditional information\\nCorrespondence and requests for materials should be addressed to A.L.B.\\nReprints and permissions information is available at www.nature.com/reprints.\\nPublisher’s note Springer Nature remains neutral with regard to jurisdictional claims in published maps and \\ninstitutional affiliations.\\nOpen Access This article is licensed under a Creative Commons Attribution 4.0 International \\nLicense, which permits use, sharing, adaptation, distribution and reproduction in any medium or \\nformat, as long as you give appropriate credit to the original author(s) and the source, provide a link to the Cre-\\native Commons licence, and indicate if changes were made. The images or other third party material in this \\narticle are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line to the \\nmaterial. If material is not included in the article’s Creative Commons licence and your intended use is not per-\\nmitted by statutory regulation or exceeds the permitted use, you will need to obtain permission directly from the \\ncopyright holder. To view a copy of this licence, visit http://creativecommons.org/licenses/by/4.0/.\\n \\n© The Author(s) 2025', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='6a851d57-2075-4965-891d-bc08d4796d72', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\graph_utils.py', 'file_name': 'graph_utils.py', 'file_type': 'text/x-python', 'file_size': 12155, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='# graph_utils.py\\r\\n\\r\\nimport torch\\r\\nimport torch.nn as nn\\r\\nimport torch.nn.functional as F\\r\\n\\r\\nimport numpy as np\\r\\nfrom rdkit import Chem\\r\\n\\r\\nclass Featurizer:\\r\\n    def __init__(self, allowable_sets):\\r\\n        self.dim = 0\\r\\n        self.features_mapping = {}\\r\\n        for k, s in allowable_sets.items():\\r\\n            s = sorted(list(s))\\r\\n            self.features_mapping[k] = dict(zip(s, range(self.dim, len(s) + self.dim)))\\r\\n            self.dim += len(s)\\r\\n\\r\\n    def encode(self, inputs):\\r\\n        output = np.zeros((self.dim,))\\r\\n        for name_feature, feature_mapping in self.features_mapping.items():\\r\\n            feature = getattr(self, name_feature)(inputs)\\r\\n            if feature not in feature_mapping:\\r\\n                continue\\r\\n            output[feature_mapping[feature]] = 1.0\\r\\n        return output\\r\\n\\r\\nclass AtomFeaturizer(Featurizer):\\r\\n    def __init__(self, allowable_sets):\\r\\n        super().__init__(allowable_sets)\\r\\n\\r\\n    def symbol(self, atom):\\r\\n        return atom.GetSymbol()\\r\\n\\r\\n    def n_valence(self, atom):\\r\\n        return atom.GetTotalValence()\\r\\n\\r\\n    def n_hydrogens(self, atom):\\r\\n        return atom.GetTotalNumHs()\\r\\n\\r\\n    def hybridization(self, atom):\\r\\n        return atom.GetHybridization().name.lower()\\r\\n\\r\\nclass BondFeaturizer(Featurizer):\\r\\n    def __init__(self, allowable_sets):\\r\\n        super().__init__(allowable_sets)\\r\\n        self.dim += 1\\r\\n\\r\\n    def encode(self, bond):\\r\\n        output = np.zeros((self.dim,))\\r\\n        if bond is None:\\r\\n            output[-1] = 1.0\\r\\n            return output\\r\\n        output = super().encode(bond)\\r\\n        return output\\r\\n\\r\\n    def bond_type(self, bond):\\r\\n        return bond.GetBondType().name.lower()\\r\\n\\r\\n    def conjugated(self, bond):\\r\\n        return bond.GetIsConjugated()\\r\\n\\r\\n\\r\\n\\r\\n# Definiowanie featurizerów\\r\\natom_featurizer = AtomFeaturizer(\\r\\n    allowable_sets={\\r\\n        \"symbol\": {\"B\", \"Br\", \"C\", \"Ca\", \"Cl\", \"F\", \"H\", \"I\", \"N\", \"Na\", \"O\", \"P\", \"S\"},\\r\\n        \"n_valence\": {0, 1, 2, 3, 4, 5, 6},\\r\\n        \"n_hydrogens\": {0, 1, 2, 3, 4},\\r\\n        \"hybridization\": {\"s\", \"sp\", \"sp2\", \"sp3\"},\\r\\n    }\\r\\n)\\r\\n\\r\\nbond_featurizer = BondFeaturizer(\\r\\n    allowable_sets={\\r\\n        \"bond_type\": {\"single\", \"double\", \"triple\", \"aromatic\"},\\r\\n        \"conjugated\": {True, False},\\r\\n    }\\r\\n)\\r\\n\\r\\n\\r\\ndef molecule_from_smiles(smiles):\\r\\n    molecule = Chem.MolFromSmiles(smiles)\\r\\n    if molecule is None:\\r\\n        raise ValueError(f\"Invalid SMILES string: {smiles}\")\\r\\n    Chem.SanitizeMol(molecule)\\r\\n    return molecule\\r\\n\\r\\ndef graph_from_molecule(molecule):\\r\\n    atom_features = []\\r\\n    bond_features = []\\r\\n    pair_indices = []\\r\\n\\r\\n    for atom in molecule.GetAtoms():\\r\\n        atom_features.append(atom_featurizer.encode(atom))\\r\\n        \\r\\n        # Dodawanie pętli własnych\\r\\n        pair_indices.append([atom.GetIdx(), atom.GetIdx()])\\r\\n        bond_features.append(bond_featurizer.encode(None))\\r\\n\\r\\n        for neighbor in atom.GetNeighbors():\\r\\n            bond = molecule.GetBondBetweenAtoms(atom.GetIdx(), neighbor.GetIdx())\\r\\n            pair_indices.append([atom.GetIdx(), neighbor.GetIdx()])\\r\\n            bond_features.append(bond_featurizer.encode(bond))\\r\\n\\r\\n    return np.array(atom_features), np.array(bond_features), np.array(pair_indices)\\r\\n\\r\\ndef graphs_from_smiles(smiles_list):\\r\\n    atom_features_list = []\\r\\n    bond_features_list = []\\r\\n    pair_indices_list = []\\r\\n\\r\\n    for smiles in smiles_list:\\r\\n        try:\\r\\n            molecule = molecule_from_smiles(smiles)\\r\\n            atom_features, bond_features, pair_indices = graph_from_molecule(molecule)\\r\\n\\r\\n            atom_features_list.append(atom_features)\\r\\n            bond_features_list.append(bond_features)\\r\\n            pair_indices_list.append(pair_indices)\\r\\n        except ValueError as e:\\r\\n            print(e)  # Wyświetl błąd dla niepoprawnych SMILES\\r\\n\\r\\n    return (\\r\\n        [torch.tensor(features, dtype=torch.float32) for features in atom_features_list],\\r\\n        [torch.tensor(features, dtype=torch.float32) for features in bond_features_list],\\r\\n        [torch.tensor(indices, dtype=torch.int64) for indices in pair_indices_list],\\r\\n    )\\r\\n\\r\\nfrom torch.utils.data import Dataset, DataLoader\\r\\n\\r\\ndef prepare_batch(x_batch, y_batch):\\r\\n    \"\"\"Merges (sub)graphs of batch into a single global (disconnected) graph.\\r\\n    \"\"\"\\r\\n\\r\\n    atom_features, bond_features, pair_indices = x_batch\\r\\n\\r\\n    # Obtain number of atoms and bonds for each graph (molecule)\\r\\n    num_atoms = [features.size(0) for features in atom_features]\\r\\n    num_bonds = [features.size(0) for features in bond_features]\\r\\n\\r\\n    # Obtain partition indices (molecule_indicator)\\r\\n    molecule_indices = torch.arange(len(num_atoms))\\r\\n    molecule_indicator = molecule_indices.repeat_interleave(num_atoms)\\r\\n\\r\\n    # Merge (sub)graphs into a global (disconnected) graph\\r\\n    gather_indices = torch.repeat_interleave(molecule_indices[:-1], num_bonds[1:])\\r\\n    increment = torch.cumsum(torch.tensor(num_atoms[:-1]), dim=0)\\r\\n    \\r\\n    # Pad increment to match the size of the bonds\\r\\n    increment = torch.cat((torch.zeros(num_bonds[0]), increment[gather_indices]))\\r\\n    \\r\\n    # Flatten and adjust pair_indices\\r\\n    pair_indices = torch.cat([p + increment.unsqueeze(1) for p in pair_indices])\\r\\n    \\r\\n    # Flatten atom_features and bond_features\\r\\n    atom_features = torch.cat(atom_features)\\r\\n    bond_features = torch.cat(bond_features)\\r\\n\\r\\n    return (atom_features, bond_features, pair_indices, molecule_indicator), y_batch\\r\\n\\r\\nclass MPNNDataset(Dataset):\\r\\n    def __init__(self, X, y):\\r\\n        self.X = X\\r\\n        self.y = y\\r\\n\\r\\n    def __len__(self):\\r\\n        return len(self.y)  # Powinno być zgodne z długością X\\r\\n\\r\\n    def __getitem__(self, index):\\r\\n        if index >= len(self.X) or index >= len(self.y):\\r\\n            print(f\"Index out of range: {index}, Lengths - X: {len(self.X)}, y: {len(self.y)}\")\\r\\n        \\r\\n        x_item = self.X[index]\\r\\n        y_item = self.y[index]\\r\\n        return prepare_batch(x_item, y_item)\\r\\n\\r\\ndef create_dataloader(X, y, batch_size=32, shuffle=False):\\r\\n    dataset = MPNNDataset(X, y)\\r\\n    return DataLoader(dataset, batch_size=batch_size, shuffle=shuffle)\\r\\n\\r\\n\\r\\n\\r\\n\\r\\n#### model #####\\r\\nclass EdgeNetwork(nn.Module):\\r\\n    def __init__(self):\\r\\n        super(EdgeNetwork, self).__init__()\\r\\n\\r\\n    def build(self, input_shape):\\r\\n        self.atom_dim = input_shape[0][-1]\\r\\n        self.bond_dim = input_shape[1][-1]\\r\\n        self.kernel = nn.Parameter(torch.Tensor(self.bond_dim, self.atom_dim * self.atom_dim))\\r\\n        self.bias = nn.Parameter(torch.zeros(self.atom_dim * self.atom_dim))\\r\\n        nn.init.xavier_uniform_(self.kernel)  # Inicjalizacja zgodna z glorot_uniform\\r\\n\\r\\n    def forward(self, inputs):\\r\\n        atom_features, bond_features, pair_indices = inputs\\r\\n\\r\\n        # Apply linear transformation to bond features\\r\\n        bond_features = torch.matmul(bond_features, self.kernel) + self.bias\\r\\n\\r\\n        # Reshape for neighborhood aggregation later\\r\\n        bond_features = bond_features.view(-1, self.atom_dim, self.atom_dim)\\r\\n\\r\\n        # Obtain atom features of neighbors\\r\\n        atom_features_neighbors = atom_features[pair_indices[:, 1]]\\r\\n        atom_features_neighbors = atom_features_neighbors.unsqueeze(-1)\\r\\n\\r\\n        # Apply neighborhood aggregation\\r\\n        transformed_features = torch.matmul(bond_features, atom_features_neighbors)\\r\\n        transformed_features = transformed_features.squeeze(-1)\\r\\n\\r\\n        # Aggregate features using scatter_add\\r\\n        aggregated_features = torch.zeros(atom_features.size(0), self.atom_dim).to(atom_features.device)\\r\\n        aggregated_features.index_add_(0, pair_indices[:, 0], transformed_features)\\r\\n\\r\\n        return aggregated_features\\r\\n\\r\\n\\r\\nclass MessagePassing(nn.Module):\\r\\n    def __init__(self, units, steps=4):\\r\\n        super(MessagePassing, self).__init__()\\r\\n        self.units = units\\r\\n        self.steps = steps\\r\\n        self.message_step = EdgeNetwork()\\r\\n        self.update_step = nn.GRUCell(self.units, self.units)\\r\\n\\r\\n    def forward(self, inputs):\\r\\n        atom_features, bond_features, pair_indices = inputs\\r\\n\\r\\n        # Pad atom features if number of desired units exceeds atom_features dim.\\r\\n        pad_length = max(0, self.units - atom_features.size(1))\\r\\n        \\r\\n        if pad_length > 0:\\r\\n            padding = torch.zeros(atom_features.size(0), pad_length).to(atom_features.device)\\r\\n            atom_features_updated = torch.cat([atom_features, padding], dim=1)\\r\\n        else:\\r\\n            atom_features_updated = atom_features\\r\\n\\r\\n        # Perform a number of steps of message passing\\r\\n        for _ in range(self.steps):\\r\\n            # Aggregate information from neighbors\\r\\n            atom_features_aggregated = self.message_step(\\r\\n                (atom_features_updated, bond_features, pair_indices)\\r\\n            )\\r\\n\\r\\n            # Update node state via a step of GRU\\r\\n            atom_features_updated = self.update_step(\\r\\n                atom_features_aggregated,\\r\\n                atom_features_updated\\r\\n            )\\r\\n        \\r\\n        return atom_features_updated\\r\\n        \\r\\n\\r\\n\\r\\nclass PartitionPadding(nn.Module):\\r\\n    def __init__(self, batch_size):\\r\\n        super(PartitionPadding, self).__init__()\\r\\n        self.batch_size = batch_size\\r\\n\\r\\n    def forward(self, inputs):\\r\\n        atom_features, molecule_indicator = inputs\\r\\n\\r\\n        # Obtain subgraphs\\r\\n        atom_features_partitioned = [atom_features[molecule_indicator == i] for i in range(self.batch_size)]\\r\\n\\r\\n        # Pad and stack subgraphs\\r\\n        num_atoms = [f.size(0) for f in atom_features_partitioned]\\r\\n        max_num_atoms = max(num_atoms)\\r\\n\\r\\n        atom_features_stacked = []\\r\\n        for f, n in zip(atom_features_partitioned, num_atoms):\\r\\n            padded = F.pad(f, (0, 0, 0, max_num_atoms - n))  # Pad to max_num_atoms\\r\\n            atom_features_stacked.append(padded)\\r\\n\\r\\n        atom_features_stacked = torch.stack(atom_features_stacked)\\r\\n\\r\\n        # Remove empty subgraphs (usually for last batch in dataset)\\r\\n        non_empty_indices = (atom_features_stacked.sum(dim=(1, 2)) != 0).nonzero(as_tuple=True)[0]\\r\\n        return atom_features_stacked[non_empty_indices]\\r\\n\\r\\nclass TransformerEncoderReadout(nn.Module):\\r\\n    def __init__(self, num_heads=8, embed_dim=64, dense_dim=512, batch_size=32):\\r\\n        super(TransformerEncoderReadout, self).__init__()\\r\\n\\r\\n        self.partition_padding = PartitionPadding(batch_size)\\r\\n        self.attention = nn.MultiheadAttention(embed_dim=embed_dim, num_heads=num_heads)\\r\\n        self.dense_proj = nn.Sequential(\\r\\n            nn.Linear(embed_dim, dense_dim),\\r\\n            nn.ReLU(),\\r\\n            nn.Linear(dense_dim, embed_dim),\\r\\n        )\\r\\n        self.layernorm_1 = nn.LayerNorm(embed_dim)\\r\\n        self.layernorm_2 = nn.LayerNorm(embed_dim)\\r\\n\\r\\n    def forward(self, inputs):\\r\\n        x = self.partition_padding(inputs)\\r\\n\\r\\n        # Create padding mask\\r\\n        padding_mask = (x != 0).any(dim=-1)\\r\\n        padding_mask = padding_mask.unsqueeze(1).unsqueeze(2)  # Shape: (batch_size, 1, 1, seq_len)\\r\\n\\r\\n        attention_output, _ = self.attention(x, x, x, key_padding_mask=~padding_mask.squeeze(-1))\\r\\n        \\r\\n        proj_input = self.layernorm_1(x + attention_output)\\r\\n        proj_output = self.layernorm_2(proj_input + self.dense_proj(proj_input))\\r\\n\\r\\n        return proj_output.mean(dim=1)  # Global average pooling\\r\\n    \\r\\n\\r\\n\\r\\nclass MPNNModel(nn.Module):\\r\\n    def __init__(self, atom_dim, bond_dim, message_units=64, message_steps=4, num_attention_heads=8, dense_units=512):\\r\\n        super(MPNNModel, self).__init__()\\r\\n        \\r\\n        self.message_passing = MessagePassing(message_units, message_steps)\\r\\n        self.transformer_readout = TransformerEncoderReadout(num_heads=num_attention_heads, embed_dim=message_units, dense_dim=dense_units)\\r\\n        \\r\\n        self.dense1 = nn.Linear(dense_units, dense_units)\\r\\n        self.dense2 = nn.Linear(dense_units, 1)\\r\\n\\r\\n    def forward(self, atom_features, bond_features, pair_indices, molecule_indicator):\\r\\n        x = self.message_passing((atom_features, bond_features, pair_indices))\\r\\n        x = self.transformer_readout((x, molecule_indicator))\\r\\n        \\r\\n        x = F.relu(self.dense1(x))\\r\\n        x = torch.sigmoid(self.dense2(x))\\r\\n        \\r\\n        return x\\r\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='317d94a2-b956-4881-9649-ca68d122d740', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\index.py', 'file_name': 'index.py', 'file_type': 'text/x-python', 'file_size': 1186, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='import os\\r\\nimport chromadb\\r\\nfrom llama_index.core import SimpleDirectoryReader\\r\\nfrom llama_index.vector_stores.chroma import ChromaVectorStore\\r\\nfrom llama_index.embeddings.huggingface import HuggingFaceEmbedding\\r\\nfrom llama_index.core import VectorStoreIndex\\r\\nfrom llama_index.llms.ollama import Ollama\\r\\n\\r\\ndef create_index(\\r\\n    chroma_path: str = \"./OSMO\",\\r\\n    collection_name: str = \"dataset_laymen\",\\r\\n    embedding_model: str = \"BAAI/bge-small-en-v1.5\",\\\\\\r\\n    \\r\\n\\r\\n\\r\\n    \\r\\n):\\r\\n    \"\"\"Tworzy i zwraca query engine dla podanych parametrów.\"\"\"\\r\\n    \\r\\n    # Inicjalizacja ChromaDB\\r\\n    db = chromadb.PersistentClient(path=chroma_path)\\r\\n    chroma_collection = db.get_or_create_collection(collection_name)\\r\\n    vector_store = ChromaVectorStore(chroma_collection=chroma_collection)\\r\\n    \\r\\n    # Model embeddingowy\\r\\n    embed_model = HuggingFaceEmbedding(model_name=embedding_model)\\r\\n    \\r\\n    # Indeks\\r\\n    index = VectorStoreIndex.from_vector_store(\\r\\n        vector_store, \\r\\n        embed_model=embed_model\\r\\n    )   \\r\\n    # Query engine\\r\\n    return index\\r\\n\\r\\n\\r\\ndef create_llm(llm_model: str = \"qwen2:7b\"):\\r\\n    \"\"\"Tworzy wybrany LLM\"\"\"\\r\\n    llm = Ollama(model=llm_model)\\r\\n    return llm\\r\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='4a10c36d-e10d-48d8-8008-63b0bc425b47', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inf.ipynb', 'file_name': 'inf.ipynb', 'file_size': 5066130, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\n# main.py (lub jakikolwiek inny plik)\\n\\nimport torch\\nfrom torch.utils.data import Dataset, DataLoader\\nfrom graph_utils import graphs_from_smiles, MPNNDataset, create_dataloader, prepare_batch # Importuj funkcje z pliku graph_utils.py\\nimport numpy as np \\nimport pandas as pd\\nimport matplotlib.pyplot as plt\\n\\n# Pobieranie danych\\ncsv_path = \"BBBP.csv\"  # Ścieżka do pliku CSV z danymi\\ndf = pd.read_csv(csv_path, usecols=[1, 2, 3])\\n\\ndf\\n\\n# Zakładam, że masz już DataFrame df z danymi SMILES\\npermuted_indices = np.random.permutation(np.arange(df.shape[0]))\\n\\n# Train set: 80 % of data\\ntrain_index = permuted_indices[: int(df.shape[0] * 0.8)]\\nx_train = graphs_from_smiles(df.iloc[train_index].smiles)\\ny_train = torch.tensor(df.iloc[train_index].p_np.values, dtype=torch.float32)\\n\\n# Valid set: 19 % of data\\nvalid_index = permuted_indices[int(df.shape[0] * 0.8): int(df.shape[0] * 0.99)]\\nx_valid = graphs_from_smiles(df.iloc[valid_index].smiles)\\ny_valid = torch.tensor(df.iloc[valid_index].p_np.values, dtype=torch.float32)\\n\\n# Test set: 1 % of data\\ntest_index = permuted_indices[int(df.shape[0] * 0.99):]\\nx_test = graphs_from_smiles(df.iloc[test_index].smiles)\\ny_test = torch.tensor(df.iloc[test_index].p_np.values, dtype=torch.float32)\\n\\n#Kontynuuj z tworzeniem datasetów i dataloaderów...\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='5420d50c-e28c-4e17-839c-70f0771ad8e3', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inf.ipynb', 'file_name': 'inf.ipynb', 'file_size': 5066130, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\ny_train.shape\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='67efdb42-71e0-4d5f-87aa-b8d5f92f906d', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inf.ipynb', 'file_name': 'inf.ipynb', 'file_size': 5066130, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\ntrain_dataset = MPNNDataset(x_train, y_train)\\nvalid_dataset = MPNNDataset(x_valid, y_valid)\\n\\nbatch_size = 32\\ntrain_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)\\nvalid_loader = DataLoader(valid_dataset, batch_size=batch_size, shuffle=False)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='8db21344-bd5b-4a1e-80f1-f68b801be527', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inf.ipynb', 'file_name': 'inf.ipynb', 'file_size': 5066130, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(f\"x_train length: {len(x_train)}\")\\nprint(f\"y_train length: {len(y_train)}\")\\n\\n# Sprawdź pierwszy element\\nif len(x_train) > 0:\\n    print(f\"First element in x_train: {x_train[0]}\")\\nprint(f\"First element in y_train: {y_train[0]}\")\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='4112f3d6-447b-48ab-82bf-5f4504c318c7', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inf.ipynb', 'file_name': 'inf.ipynb', 'file_size': 5066130, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nfrom graph_utils import MPNNModel\\n\\nimport torch.optim as optim\\n\\n\\n# Definiowanie modelu MPNN\\nmpnn_model = MPNNModel(\\n    atom_dim=x_train[0][0][0].shape,  # Upewnij się, że używasz drugiego wymiaru\\n    bond_dim=x_train[1][0][0].shape,  # Upewnij się, że używasz drugiego wymiaru\\n)\\n\\n# Ustawienie optymalizatora\\noptimizer = optim.Adam(mpnn_model.parameters(), lr=5e-4)\\n\\n# Funkcja straty\\ndef binary_crossentropy(predictions, targets):\\n    return nn.BCELoss()(predictions.view(-1), targets.view(-1))\\n\\n# Pętla treningowa\\nnum_epochs = 40\\ntrain_auc_history = []\\nvalid_auc_history = []\\n\\nfor epoch in range(num_epochs):\\n    mpnn_model.train()  # Ustaw model w trybie treningowym\\n    total_loss = 0.0\\n    all_targets = []\\n    all_predictions = []\\n\\n    # Trening\\n    for atom_features, bond_features, pair_indices, molecule_indicator, targets in train_loader:\\n        optimizer.zero_grad()  # Zerowanie gradientów\\n        outputs = mpnn_model(atom_features, bond_features, pair_indices, molecule_indicator)\\n        \\n        loss = binary_crossentropy(outputs.squeeze(), targets)  # Squeeze to match dimensions if necessary\\n        loss.backward()  # Obliczanie gradientów\\n        optimizer.step()  # Aktualizacja wag\\n        \\n        total_loss += loss.item()\\n        \\n        all_targets.extend(targets.cpu().numpy())\\n        all_predictions.extend(outputs.detach().cpu().numpy())\\n\\n    # Obliczanie AUC dla zbioru treningowego\\n    train_auc = roc_auc_score(all_targets, all_predictions)\\n    train_auc_history.append(train_auc)\\n\\n    # Walidacja\\n    mpnn_model.eval()  # Ustaw model w trybie ewaluacji\\n    valid_targets = []\\n    valid_predictions = []\\n\\n    with torch.no_grad():  # Wyłącz obliczanie gradientów dla walidacji\\n        for atom_features, bond_features, pair_indices, molecule_indicator, targets in valid_loader:\\n            outputs = mpnn_model(atom_features, bond_features, pair_indices, molecule_indicator)\\n            valid_targets.extend(targets.cpu().numpy())\\n            valid_predictions.extend(outputs.detach().cpu().numpy())\\n\\n    # Obliczanie AUC dla zbioru walidacyjnego\\n    valid_auc = roc_auc_score(valid_targets, valid_predictions)\\n    valid_auc_history.append(valid_auc)\\n\\n    print(f\"Epoch {epoch + 1}/{num_epochs}, Loss: {total_loss / len(train_loader):.4f}, Train AUC: {train_auc:.4f}, Valid AUC: {valid_auc:.4f}\")\\n\\n# Wizualizacja wyników AUC\\nplt.figure(figsize=(10, 6))\\nplt.plot(train_auc_history, label=\"Train AUC\")\\nplt.plot(valid_auc_history, label=\"Valid AUC\")\\nplt.xlabel(\"Epochs\", fontsize=16)\\nplt.ylabel(\"AUC\", fontsize=16)\\nplt.legend(fontsize=16)\\nplt.title(\"Training and Validation AUC over Epochs\", fontsize=18)\\nplt.show()\\n\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='7f82bb87-778f-419b-bd72-1db43c334c7f', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nfrom llama_index.embeddings.huggingface import HuggingFaceEmbedding\\nfrom transformers import AutoTokenizer\\n\\nembed_model = HuggingFaceEmbedding(model_name=\"BAAI/bge-small-en-v1.5\")\\n\\nfrom transformers import AutoTokenizer\\ntokenizer = AutoTokenizer.from_pretrained(\"BAAI/bge-small-en-v1.5\")\\ntokenizer\\n\\n\\n# In[ ]:\\n\\n\\nimport os, chromadb\\nfrom llama_index.vector_stores.chroma import ChromaVectorStore\\nfrom llama_index.embeddings.huggingface import HuggingFaceEmbedding\\nfrom llama_index.core import VectorStoreIndex\\nfrom llama_index.llms.huggingface import HuggingFaceLLM\\nfrom llama_index.llms.ollama import Ollama\\n\\ndb = chromadb.PersistentClient(path=\"./OSMO\")\\nchroma_collection = db.get_or_create_collection(\"dataset-laymen\")\\nvector_store = ChromaVectorStore(chroma_collection=chroma_collection)\\nembed_model = HuggingFaceEmbedding(model_name=\"BAAI/bge-small-en-v1.5\")\\n\\n\\nindex = VectorStoreIndex.from_vector_store(vector_store, embed_model=embed_model)\\nllm = Ollama(model=\"qwen2:7b\")\\nquery_engine = index.as_query_engine(llm=llm)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='875340bb-bfba-4492-aecd-dedf47ee9ac5', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nresponse = query_engine.query(\"Co określa ustawa?\")\\nprint(response)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='53cba88c-07c9-463e-8fbb-1693c2261f86', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(\"\\\\nŹródła:\")\\nfor node in response.source_nodes:\\n    print(f\"  * Dokument: {node.node.ref_doc_id}\\\\n\")\\n    print(f\"    Fragment tekstu: {node.node.get_content()}\\\\n\")\\n    print(f\"    Podobieństwo: {node.score}\\\\n\")\\n    print(\"-\" * 40)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='b5f551f0-373c-469c-880d-1bc3a222e8e8', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport os, chromadb\\nfrom llama_index.vector_stores.chroma import ChromaVectorStore\\nfrom llama_index.embeddings.huggingface import HuggingFaceEmbedding\\nfrom llama_index.core import VectorStoreIndex\\nfrom llama_index.llms.huggingface import HuggingFaceLLM\\nfrom llama_index.llms.ollama import Ollama\\n\\ndb2 = chromadb.PersistentClient(path=\"./ustawy_BAAI\")\\nchroma_collection2 = db2.get_or_create_collection(\"pomoc_ukrainie\")\\nvector_store2 = ChromaVectorStore(chroma_collection=chroma_collection2)\\nembed_model2 = HuggingFaceEmbedding(model_name=\"BAAI/bge-small-en-v1.5\")\\n\\nindex2 = VectorStoreIndex.from_vector_store(vector_store2, embed_model=embed_model2)\\nllm = Ollama(model=\"qwen2:7b\")\\nquery_engine2 = index2.as_query_engine(llm=llm)\\n\\nresponse2 = query_engine2.query(\"O czym mówi ustawa?\")\\nprint(response2)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='158d7c4f-ebb0-4f8d-9a3c-39aae92e2a89', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(\"\\\\nŹródła:\")\\nfor node in response2.source_nodes:\\n    print(f\"  * Dokument: {node.node.ref_doc_id}\\\\n\")\\n    print(f\"    Fragment tekstu: {node.node.get_content()}\\\\n\")\\n    print(f\"    Podobieństwo: {node.score}\\\\n\")\\n    print(\"-\" * 40)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='f01f163b-a1c0-4f8c-834e-b2f34d534071', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery = \"Co określa ustawa?\"\\nresponse = query_engine.query(query)\\nresponse2 = query_engine2.query(query)\\n\\nprint(response)\\nprint(response2)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='cf7cb1c5-d2dd-42fa-a1d4-ba6c08947f3f', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(\"\\\\nŹródła:\")\\nfor node in response2.source_nodes:\\n    print(f\"  * Dokument: {node.node.ref_doc_id}\\\\n\")\\n    print(f\"    Fragment tekstu: {node.node.get_content()}\\\\n\")\\n    print(f\"    Podobieństwo: {node.score}\\\\n\")\\n    print(\"-\" * 40)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='5881acbf-8c00-4239-b6bc-050778f5ab61', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\inspect.ipynb', 'file_name': 'inspect.ipynb', 'file_size': 25943, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-29'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(\"\\\\nŹródła:\")\\nfor node in response.source_nodes:\\n    print(f\"  * Dokument: {node.node.ref_doc_id}\\\\n\")\\n    print(f\"    Fragment tekstu: {node.node.get_content()}\\\\n\")\\n    print(f\"    Podobieństwo: {node.score}\\\\n\")\\n    print(\"-\" * 40)\\n\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='7c0e552d-2ff4-4ba4-9f2c-d65f9c24f0b6', embedding=None, metadata={'page_label': '1', 'file_name': 'ISOEN.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\ISOEN.pdf', 'file_type': 'application/pdf', 'file_size': 518357, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='See discussions, stats, and author profiles for this publication at: https://www.researchgate.net/publication/333530395\\nCo-occurrence-based clustering of odor descriptors for predicting structure-\\nodor relationship\\nConference Paper · May 2019\\nDOI: 10.1109/ISOEN.2019.8823446\\nCITATIONS\\n11\\nREADS\\n362\\n3 authors:\\nChuanjun Liu\\nKyushu University\\n91 PUBLICATIONS\\xa0\\xa0\\xa01,661 CITATIONS\\xa0\\xa0\\xa0\\nSEE PROFILE\\nLiang Shang\\nChinese Academy of Sciences\\n22 PUBLICATIONS\\xa0\\xa0\\xa0477 CITATIONS\\xa0\\xa0\\xa0\\nSEE PROFILE\\nKenshi Hayashi\\nKyushu University\\n239 PUBLICATIONS\\xa0\\xa0\\xa02,877 CITATIONS\\xa0\\xa0\\xa0\\nSEE PROFILE\\nAll content following this page was uploaded by Chuanjun Liu on 28 September 2019.\\nThe user has requested enhancement of the downloaded file.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='579202b5-4fdf-4bb0-b074-3c22c9f14266', embedding=None, metadata={'page_label': '2', 'file_name': 'ISOEN.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\ISOEN.pdf', 'file_type': 'application/pdf', 'file_size': 518357, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='CO-OCCURRENCE-BASED CLUSTERING OF ODOR DESCRIPTORS FOR PREDICTING STRUCTURE-ODOR RELATIONSHIP  Chuanjun Liu1,2, Liang Shang2, Kenshi Hayashi2 1 Research Laboratory, U.S.E. Co., Ltd., 22-10 Ebisu 4-Chome, Shibuya-ku, Tokyo 150-0013, Japan 2 Graduate School of Information Science and Electrical Engineering, Kyushu University,  744,  Motooka, Nishi-ku, Fukuoka, 819-0395, Japan  ABSTRACT  One problem of machine-learning-based prediction of structure-odor relationship is that odorant molecules are usually labeled with ambiguous descriptors when they are collected from different sources. This study focused on the clustering of the odor descriptors by text mining approaches as well as the prediction of newly established labels from physicochemical parameters of the classified odorant molecules. An odor database was established by web scraping and transferred to a document-term matrix including 4011 odorants and 100 odor descriptors. The clustering of the odor descriptors was carried out by using different co-occurrence matrix and clustering approaches. A hierarchical cluster analysis combined with a co-occurrence probability distribution matrix has shown good results in the descriptor clustering. The attribute labels of each class were established and then predicted from physicochemical parameters of the classified odorants by using random forest model. An average accuracy higher than 82.42% was obtained, indicating the effectiveness of the proposed approaches for predicting structure-odor relationship.   Index Terms— structure-odor relationship, text mining, machine learning, odor descriptors, physicochemical parameters, clustering, prediction  1. INTRODUCTION  Although understanding the relationship between the molecular structure of odorants and their olfactory perception has long remained a major challenge, the data-driven approaches rapidly developed in recent years has made it possible to predict the olfactory perception from their physicochemical properties [1,2,]. Such kind of predictions may not only deepen our understanding on the olfactory perception of odorant molecules, but also explore new applications that cannot be realized before. For example, in our previous work we have introduced a machine-learning-based gas chromatography-olfactometry (GCO), in which the olfactometric detection can be done by a machine classifier, and thus the human panelist might be no longer needed [3,4].      The previous study was carried out using a dataset with limited odorants and odor descriptors. For the practical \\napplication of structure-odor relationship prediction, the concept should be proofed by data on a large scale. The problem of sample size can be solved by collecting data from open or commercial sources as many researches have done [5,6]. However, there exists an phenomenon in the collected database that the odorant molecules are usually noted by descriptors with ambiguous, vague and unintelligible meaning. This may make it difficult to correctly label the odorant molecules for the machine learning, and thus decreases the prediction accuracy of models.  In this study, we extend the perception prediction of odorant molecules by using an enlarged database that is collected from different sources. Differently from the single descriptor-labeled prediction in the previous work, the odor descriptors are firstly clustered based on their co-occurrence relationship in a document term matrix by various classification approaches. The clustered odor descriptors with semantic similarity are endowed with new attribute labels, which is used to group odorant molecules again. The predict of the group labels is carried out by a random forest model from  physicochemical parameters of the odorant molecules.   2. MATERIALS AND METHODS  Raw data including the CAS number and odor descriptors was obtained by web scraping from different websites [7]. R software was used in data process  as well as model prediction. Ambiguous, lengthy  and irrelevant expressions were sorted and integrated by both semi-automatic and manual methods. Compound descriptor words (or sentences) were split and only single adjective word was used to describe the odorant molecules (for example, “coffee-like” to “coffee”, “roasted meat” to “roasted” and “meaty”). The olfaction-unrelated vocabularies were also excluded (for example, “sharp”, “rich” and “odorless”). The raw data included  ~ 5000 odorants and ~ 500 descriptors. In order to ensure enough sample size for co-occurrence analysis as well as the model prediction, we only counted the descriptors with top 100 frequency. Therefore, the final document-term matrix consisted of  4011 molecules and 100 descriptors.  A co-occurrence matrix (100×100 dimensions) of the 100 descriptors was established from the document-term matrix. Two major approaches were used in the descriptor clustering: pairwise-based clustering and similarity-based clustering. ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='921f4406-a784-489c-a78b-7ca315c92f25', embedding=None, metadata={'page_label': '3', 'file_name': 'ISOEN.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\ISOEN.pdf', 'file_type': 'application/pdf', 'file_size': 518357, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='The principle of pairwise-based clustering is that if terms ω1 and ω2  co-occur frequently, they are considered to be in the same cluster. The principle of similarity-based clustering is that if term ω1 and ω2 have similar distribution of co-occurrence with other terms, they are considered to be in the same cluster [8,9]. The pairwise-based clustering was carried out by using Jaccard index matrix (we called it J-matrix) and the similarity-based clustering was carried out by using co-occurrence probability distribution matrix (we called it D-matrix). Both hierarchical (hclust) and non-hierarchical (k-means) approaches were applied in the descriptor clustering based on the above two matrix. Odorant molecules labeled with odor descriptors belonging to the same cluster were grouped together with new attribute labels for the next prediction.   The physicochemical parameters of the odorant molecules were calculated by Dragon cheminformation software (Ver 7.0, Kode, Italy). Parameters with “NA” (not applicable), as well as  those zero values that cannot be used in data scaling, were removed from the dataset, which finally afforded a matrix with 4011 odorants and 1780 parameters. These parameters include molecular weight, number of atoms, aromatic ratio, topological charge index, and so on (for detailed information please refer to [3]). Principal component  analysis (PCA) was used in the feature extraction from the   molecular parameters. To avoid loss of characteristic  information from the  original dataset,  we selected PCs with accumulative contributions of 99.99%, which included approximately 200 PCs.    Since the data of the labeled classes is a typical imbalanced dataset, synthetic minority oversampling technique (SMOTE) [10] was used to balance both the minor and major samples according to actual needs. After the sample balancing, The SMOTE samples were then divided  into training and test sets with a ratio of 4:1 by using the createDataPartition function of the caret package. The predict of classified attribute labels was carried out by using the package ‘randomForest’, in which parameters such as the number of trees and the number of variables are optimized automatically. A 5-fold cross-validation approach was applied in the sample splitting to confirm the model generalization. The prediction accuracy was evaluated by a mean value of 5 validations with the standard deviation.  \\n3. RESULTS AND DISCUSSION  Figure 1 shows the frequency distribution of odor descriptors  by the number of odorant molecules in the raw database. Table 1 shows the number of odorants by the top 100 descriptors. It can be seen that the top one descriptor (fruity) is almost associated by one third of the odorant molecules (~1400). An odorant molecule (herein methyl eugenol) is noted by descriptors with the maximum number of 29. This phenomenon that one odorant is represented by so many descriptors and one descriptor is shared by so many odorants makes it difficult to precisely label the odorant molecules. This will influence the feature extraction from the molecular structure as well as the following machine learning. Therefore, it is necessary to cluster the odor descriptors  to obtain more precise labeling information. Moreover, The minimum number of odorant molecules for the top 100 descriptors is larger than 58, which may ensure the sample size of the prediction models.  We firstly compared the clustering effect of the descriptors by PCA when different data matrix (J-matrix and D-matrix) and algorithms (hclust and k-means) were applied. Figure 2 shows the visualized results of four combinations: J-hclust, J-kmean, D-hlcust and D-kmean, respectively. It is noticed that the cumulative contribution rates of the two principle components (calculated by the sum of Dim 1 and Dim 2) in the case of D-matrix (24.5%) is much higher than that of J-matrix (8.0%). These results indicate that the co-occurrence probability distribution matrix (D-matrix) may has much better effect than the Jaccard index matrix to extract the similarity of odor descriptors. Additionally, in the case of D-matrix, the hierarchical classification (hclust) shows less overlapping between different descriptor clusters. It is then suggested that the D-hclust approach is much better for the descriptor clustering.  \\nTable 1 Number of odorants by the top 100 descriptors  fruity1402roasted195cheese111 lily 72sweet1244 dry 191tobacco111natural72green1235honey186 onion110smoky72floral999creamy185terpenic105 soap 70woody896 wine176 anise103 dairy 69herbal802camphor175 clean103grassy69spicy606 berry171 warm103alliaceous65fresh492pineapple166pungent101muguet65fatty 486caramel165coconutty100cortex64citrus431ethereal158 musk100 plum 63balsamic429phenolic154 burnt98 bitter 62waxy417orange149lemon97cinnamon62rose 376aldehydic145peach96geranium62earthy332 leaf 142pepper95 tea 62nutty328 grape137almond94 milk 61sulfurous279 amber130mushroom92 orris 61oily 274 vanilla130cherry87 rummy61tropical269metallic122chocolate87 hyacinth60minty244 melon121 violet87 lavender60pine 237 pear117apricot86 clove59apple216 coffee115 cedar83coumarin59meaty216 animal114medicinal83 peel 59vegetable208jasmine112 garlic79raspberry58musty195banana111cocoa76strawberry58powdery195 butter111 hay 75 weedy58\\nFigure 1 Number distribution of odorants by descriptors. 0100200300400500600700\\n123456789101112131415161718192021222324252629Number of descriptors  \\nNumber of odorants', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='7c9fa8c1-abdc-4b3d-9795-7880b52d83a0', embedding=None, metadata={'page_label': '4', 'file_name': 'ISOEN.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\ISOEN.pdf', 'file_type': 'application/pdf', 'file_size': 518357, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 3 shows the dendrogram of the 100 descriptors based on the D-hclust approach. The number of classes was tentatively fixed as ten in this plot. It can be seen that the descriptors with semantic similarity are almost divided into the same class. For example, alliaceous, sulfurous, onion and garlic are clustered into C2. The descriptors related to tropic fruit are clustered together in C7. Therefore, it is possible to annotate the odorant communities with new attribute labels (this is another topic and not discussed in this work). The rearrangement of odorant molecules may hold more semantic commonality, which will be helpful for the feature extraction from physicochemical parameters.  Table 2 lists the prediction results of the above 10 classes by using a random forest model for the test subset (20% of \\nthe SMOTE samples). The highest mean prediction accuracy (96.24%) is observed for the class of C10, while the minimum accuracy (69.66%) is observed for the class of C6. The average accuracy of the 10 classes is about 82.42%. Relatively low accuracy was observed for classes with diversified descriptors and more branches in the dendrogram. It is found that if the class number increases (that can be done by adjusting the cut-off distance), both the minimum and average accuracies will increase further. For example, when the class number is fixed as 16, the obtained minimum and average accuracies are 74.22% and 85.32%, respectively (data not shown). This result suggests the determination of appropriate class number might be an important issue for the prediction. \\nfruity\\nsweet\\ngreen\\nfloral\\nwoody\\nherbal\\nspicy\\nfresh\\nfatty\\ncitrus\\nbalsamic\\nwaxy\\nrose\\nearthy\\nnutty\\nsulfurous\\noily\\ntropical\\nminty\\npine\\napple\\nmeatyvegetable\\nmusty\\npowdery\\nroasted\\ndry\\nhoney\\ncreamy\\nwine\\ncamphor\\nberry\\npineapple\\ncaramel\\nethereal\\nphenolic\\norangealdehydicleaf\\ngrape\\namber\\nvanilla\\nmetallic\\nmelon\\npear\\ncoffee\\nanimal\\njasmine\\nbanana butter\\ncheese\\ntobacco\\nonion\\nterpenic\\naniseclean\\nwarm\\npungent\\ncoconut\\nmusk\\nburnt\\nlemon\\npeach\\npepper\\nalmond\\nmushroom\\ncherry\\nchocolate\\nviolet\\napricot\\ncedar\\nmedicinal\\ngarlic\\ncocoa\\nhaylily\\nnatural\\nsmoky\\nsoap\\ndairy\\ngrassy\\nalliaceous\\nmuguet\\ncortex\\nplum\\nbitter\\ncinnamon\\ngeranium\\ntea\\nmilk\\norris\\nrummy\\nhyacinthlavender\\nclovecoumarin\\npeel\\nraspberry\\nstrawberry\\nweedy\\n−4\\n−2\\n0\\n2\\n−5.0 −2.5 0.0 2.5 5.0\\nDim1 (4.5%)\\nDim2 (3.5%)\\ncluster\\na\\na\\na\\na\\na\\na\\na\\na\\na\\na\\n1\\n2\\n3\\n4\\n5\\n6\\n7\\n8\\n9\\n10\\n(a1): J−hclust\\nfruity\\nsweet\\ngreen\\nfloral\\nwoody\\nherbal\\nspicy\\nfresh\\nfatty\\ncitrus\\nbalsamic\\nwaxy\\nrose\\nearthy\\nnutty\\nsulfurous\\noily\\ntropical\\nminty\\npine\\napple\\nmeatyvegetable\\nmusty\\npowdery\\nroasted\\ndry\\nhoney\\ncreamy\\nwine\\ncamphor\\nberry\\npineapple\\ncaramel\\nethereal\\nphenolic\\norangealdehydicleaf\\ngrape\\namber\\nvanilla\\nmetallic\\nmelon\\npear\\ncoffee\\nanimal\\njasmine\\nbanana butter\\ncheese\\ntobacco\\nonion\\nterpenic\\naniseclean\\nwarm\\npungent\\ncoconut\\nmusk\\nburnt\\nlemon\\npeach\\npepper\\nalmond\\nmushroom\\ncherry\\nchocolate\\nviolet\\napricot\\ncedar\\nmedicinal\\ngarlic\\ncocoa\\nhaylily\\nnatural\\nsmoky\\nsoap\\ndairy\\ngrassy\\nalliaceous\\nmuguet\\ncortex\\nplum\\nbitter\\ncinnamon\\ngeranium\\ntea\\nmilk\\norris\\nrummy\\nhyacinthlavender\\nclovecoumarin\\npeel\\nraspberry\\nstrawberry\\nweedy\\n−4\\n−2\\n0\\n2\\n−5.0 −2.5 0.0 2.5 5.0\\nDim1 (4.5%)\\nDim2 (3.5%)\\ncluster\\na\\na\\na\\na\\na\\na\\na\\na\\na\\na\\n1\\n2\\n3\\n4\\n5\\n6\\n7\\n8\\n9\\n10\\n(a2): J−kmean\\nfruity\\nsweet\\ngreen\\nfloral\\nwoody\\nherbal\\nspicy\\nfresh\\nfatty\\ncitrus\\nbalsamic\\nwaxy\\nrose\\nearthy\\nnutty\\nsulfurous\\noily\\ntropical\\nminty\\npine\\napple\\nmeaty\\nvegetable\\nmusty\\npowdery\\nroasteddry\\nhoney\\ncreamy\\nwine\\ncamphor\\nberry\\npineapple\\ncaramel\\nethereal\\nphenolic\\norange\\naldehydic\\nleaf\\ngrape\\namber\\nvanilla\\nmetallic\\nmelon\\npear\\ncoffeeanimal\\njasmine\\nbanana\\nbuttercheese\\ntobacco\\nonion\\nterpenic\\nanise\\nclean\\nwarm\\npungent\\ncoconut\\nmusk\\nburnt\\nlemon\\npeach\\npepper almond\\nmushroom\\ncherrychocolate\\nviolet\\napricot\\ncedarmedicinal\\ngarlic\\ncocoa\\nhay\\nlilynatural\\nsmoky\\nsoap\\ndairy\\ngrassy\\nalliaceous\\nmuguetcortex\\nplum\\nbitter\\ncinnamon\\ngeranium\\ntea\\nmilk\\norris\\nrummy\\nhyacinthlavender\\nclovecoumarin\\npeel raspberry\\nstrawberry\\nweedy\\n−4\\n0\\n4\\n−5 0 5 10\\nDim1 (12.9%)\\nDim2 (11.6%)\\ncluster\\na\\na\\na\\na\\na\\na\\na\\na\\na\\na\\n1\\n2\\n3\\n4\\n5\\n6\\n7\\n8\\n9\\n10\\n(b1): D−hclust\\nfruity\\nsweet\\ngreen\\nfloral\\nwoody\\nherbal\\nspicy\\nfresh\\nfatty\\ncitrus\\nbalsamic\\nwaxy\\nrose\\nearthy\\nnutty\\nsulfurous\\noily\\ntropical\\nminty\\npine\\napple\\nmeaty\\nvegetable\\nmusty\\npowdery\\nroasteddry\\nhoney\\ncreamy\\nwine\\ncamphor\\nberry\\npineapple\\ncaramel\\nethereal\\nphenolic\\norange\\naldehydic\\nleaf\\ngrape\\namber\\nvanilla\\nmetallic\\nmelon\\npear\\ncoffeeanimal\\njasmine\\nbanana\\nbuttercheese\\ntobacco\\nonion\\nterpenic\\nanise\\nclean\\nwarm\\npungent\\ncoconut\\nmusk\\nburnt\\nlemon\\npeach\\npepper almond\\nmushroom\\ncherrychocolate\\nviolet\\napricot\\ncedarmedicinal\\ngarlic\\ncocoa\\nhay\\nlilynatural\\nsmoky\\nsoap\\ndairy\\ngrassy\\nalliaceous\\nmuguetcortex\\nplum\\nbitter\\ncinnamon\\ngeranium\\ntea\\nmilk\\norris\\nrummy\\nhyacinthlavender\\nclovecoumarin\\npeel raspberry\\nstrawberry\\nweedy\\n−4\\n0\\n4\\n−5 0 5 10\\nDim1 (12.9%)\\nDim2 (11.6%)\\ncluster\\na\\na\\na\\na\\na\\na\\na\\na\\na\\na\\n1\\n2\\n3\\n4\\n5\\n6\\n7\\n8\\n9\\n10\\n(b2): D−kmean\\nFigure 2 Visualized results of odor descriptors by different data matrix (J-matrix and D-matrix) and different clustering approaches (hierarchical and k-means). ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='b66f6c8c-05db-4e77-8ee3-20df2647350c', embedding=None, metadata={'page_label': '5', 'file_name': 'ISOEN.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\ISOEN.pdf', 'file_type': 'application/pdf', 'file_size': 518357, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='  \\n  4. CONCLUSION  Text mining approaches were introduced in the classification of odor descriptors and odorant molecules for predicting structure-odor relationship. Compared with our previous prediction based on single descriptor label, the descriptor classification enlarged the range of prediction because more descriptors and odorants were considered. We suggested that the classification of odor descriptors might help to the feature extraction from both the odor descriptors and the molecular structures, and thus to improve the prediction accuracy of machine-learning-based olfactory perception.   \\n 5. REFERENCES  [1] K. Kaeppler, F. Mueller, “Odor classification: a review of factors influencing perception-based odor arrangement”, Chemical Senses, 38, 189-209 (2013) [2] B. Auffarth, “Understanding smell – The olfactory stimulus problem”, Neuroscience and Biobehavioral Reviews, 37, 1667-1669 (2013) [3] L. Shang, C. Liu, Y. Tomiura, K. Hayashi, “Machine-learning- based olfactometer: prediction of odor perception from physicochemical features of odorant molecules” Analytical Chemistry, 89,11999-12005 (2017)  [4] L. Shang, C. Liu, Y. Tomiura, K. Hayashi, “Odorant clustering based on molecular parameter-feature extraction and imaging analysis of olfactory bulb odor maps”, Sensors and Actuators B: Chemcial, 255, 508-518 (2018) [5] R. Kumar, R. Kaur, B. Auffarth, A.P. Bhondekar, “Understanding the odor spaces: A step towards solving olfactory stimulus-percept problem”, PLOS ONE, 10(10):e0141263(2015) [6] A. Tromelin, C. Chabanet, K. Audouze, F. Koensgen, “Multivariate statisrical analysis of a large odorants database aimed at revealing similarities and links between odorants and odors” Flavour Fragrance Jouranl, 33,106-126 (2018)  [7]  OdorMapOB: https://senselab.med.yale.edu/odormapdb/; SuperScent: http://bioinf-applied.charite.de/superscent/; GoodScents:http://www.thegoodscentscompany.com; Flavornet:http://www.flavornet.org/index.html;  Sigam-Aldrich: https://www.sigmaaldrich.com/industries/flavors-and-fragrances.html [8] Y. Matsuo, M. Ishizuka, “Keyword extraction from a document using word co-occurrence statistical information” International Journal on Artificial Intelligence, 13 (1), 157-169 (2002) [9] N. Takahashi, A. Okabe, “Word clustering based on co- occurrence information in English GIS textbooks”, Theory and Application of GIS, 15(2), 129-136 (2007)  [10] N. Chawla, K. Bowyer, L. Hall, W. Kegelmeyer, “SMOTE: Synthetic minority over-sampling technique”, Journal of Artificial Intelligence Research, 16, 321-357 (2002)  \\nTable 2 Model parameters and prediction accuracies of 10 odorant classes.  C1 C2 C3 C4 C5 C6 C7 C8 C9 C10\\nP 356 283 1293309 1319169111633199 951 89N 3655 3728 27183702269223202848 812 3060 3922\\nover 400 400 100 400 100 25 100 200 200 1000under 120 120 200 120 200 500 200 150 150 120\\nP_SMOTE1780 1415 258615452638211323262436 2853 1068N_SMOTE1708 1358 258614832638211023262436 2853 979\\nTP* 328 256 375 273 365 308 360 382 461 186FP* 28 27 142 36 162 114 105 105 109 9FN* 33 26 118 30 102 142 106 83 106 6TN* 308 245 399 266 425 280 359 404 464 207\\nAcc% 89.31 90.7075.4288.7774.9269.6676.4081.2081.5596.24SD% 0.89 0.66 2.46 0.620.63 1.47 0.96 1.07 0.38 0.50P/Nmeanspositiveandnegativesampleofindividualclass;over/under:parametersusedtoexpandtheminoritysampleandreducethemajoritysamplesinSMOTE;P_SMOTE/N_SMOTE:positiveandnegativesamplesafterbalancedbySMOTE.TN:Truenegative,FP:Falsepositive,FN:Falsenegative,TP:Truepositive(*:thevalueisanexampleofonecross-validation);Accisthemeanpredictionaccuracyof5-foldcross-validation.SDisthestandarddeviation.\\nfruity sweetgreenfloral\\nwoodyherbal\\nspicy\\nfresh\\nfatty\\ncitrus\\nbalsamic\\nwaxy\\nrose\\nearthynutty\\nsulfurous\\noily\\ntropical\\nminty\\npine\\napple\\nmeatyvegetablemusty\\npowdery\\nroasted\\ndry\\nhoney\\ncreamy\\nwine\\ncamphor\\nberry\\npineapple\\ncaramel\\nethereal\\nphenolic\\norangealdehydic\\nleaf\\ngrape\\namber\\nvanilla metallic\\nmelon\\npear\\ncoffee\\nanimal\\njasmine\\nbanana\\nbutter\\ncheese\\ntobacco\\nonion\\nterpenic\\nanise\\nclean\\nwarm\\npungent\\ncoconut\\nmuskburnt\\nlemon\\npeach\\npepper\\nalmond\\nmushroom\\ncherry\\nchocolate\\nviolet\\napricot\\ncedar\\nmedicinal\\ngarlic\\ncocoahay\\nlily\\nnatural\\nsmoky\\nsoap\\ndairy\\ngrassy\\nalliaceous\\nmuguet\\ncortex\\nplum\\nbittercinnamon\\ngeranium\\ntea\\nmilk\\norrisrummy\\nhyacinth\\nlavender\\nclove\\ncoumarin\\npeel\\nraspberry\\nstrawberry\\nweedy\\nC1\\nC2\\nC3\\nC4C5\\nC6\\nC7\\nC8\\nC9\\nC10\\nFigure 3 Dendrogram of odor descriptors by D-hclust. \\nView publication stats', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='2cdfc8f9-a5dc-40f2-9644-df65a4e7d331', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\links.txt', 'file_name': 'links.txt', 'file_type': 'text/plain', 'file_size': 124, 'creation_date': '2025-01-03', 'last_modified_date': '2025-01-03'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='https://en.wikipedia.org/wiki/Chemical_sensor_array\\r\\nhttps://huggingface.co/datasets/seyonec/goodscents_leffingwell?row=28\\r\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='475df377-870d-463c-9117-b04b3ada94be', embedding=None, metadata={'page_label': '1', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\ue929ViewOnline\\n\\ue92dExportCitation\\nRESEARCH ARTICLE |  MAY 15 2023\\nMachine learning-enabled graphene-based electronic\\nolfaction sensors and their olfactory performance\\nassessment \\nSpecial Collection: Volatile Organic Compounds and their Applications\\nShirong Huang \\n  ; Alexander Croy\\ue923 \\n  ; Antonie Louise Bierling \\n  ; Vyacheslav Khavrus \\n  ;\\nLuis Antonio Panes-Ruiz \\n  ; Arezoo Dianat \\n  ; Bergoi Ibarlucea\\ue923 \\n  ; Gianaurelio Cuniberti\\ue923 \\nAppl. Phys. Rev. 10, 021406 (2023)\\nhttps://doi.org/10.1063/5.0132177\\nArticles You May Be Interested In\\nE-noses: Smells like the future of sensors\\nScilight (May 2023)\\nQuantum origins of molecular recognition and olfaction in drosophila\\nJ. Chem. Phys. (November 2012)\\nAn explicit electron-vibron model for olfactory inelastic electron transfer spectroscopy\\nJ. Appl. Phys. (April 2019)\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='61a3fc2f-b7bf-4437-ad65-bd7fd219f266', embedding=None, metadata={'page_label': '2', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Machine learning-enabled graphene-based\\nelectronic olfaction sensors and their olfactory\\nperformance assessment\\nCite as: Appl. Phys. Rev.10, 021406 (2023);doi: 10.1063/5.0132177\\nSubmitted: 27 October 2022. Accepted: 20 April 2023.\\nPublished Online: 15 May 2023\\nShirong Huang,1\\n Alexander Croy,2,a)\\n Antonie LouiseBierling,1,3\\n Vyacheslav Khavrus,4\\nLuis AntonioPanes-Ruiz,1\\n Arezoo Dianat,1\\n Bergoi Ibarlucea,1,5,a)\\n and GianaurelioCuniberti1,5,a)\\nAFFILIATIONS\\n1Institute for Materials Science and Max Bergmann Center for Biomaterials, TU Dresden, 01062 Dresden, Germany\\n2Institute of Physical Chemistry, Friedrich Schiller University Jena, Helmholtzweg 4, 07743 Jena, Germany\\n3Department of Psychotherapy and Psychosomatic Medicine, Faculty of Medicine, TU Dresden, 01307 Dresden, Germany\\n4SmartNanotubes Technologies GmbH, Dresdner Str. 172, 01705 Freital, Germany\\n5Dresden Center for Computational Materials Science (DCMS), TU Dresden, 01062 Dresden, Germany\\nNote: This paper is part of the APR Special Topic on Volatile Organic Compounds and their Applications.\\na)Authors to whom correspondence should be addressed:alexander.croy@uni-jena.de; bergoi.ibarlucea@tu-dresden.de;\\nand gianaurelio.cuniberti@tu-dresden.de\\nABSTRACT\\nOlfaction is an evolutionary old sensory system, which provides sophisticated access to information about our surroundings. In particular,\\ndetecting the volatile organic compounds (VOCs) emitted during natural and artiﬁcial processes can be used as characteristic ﬁngerprints\\nand help to identify their source. Inspired by the biological example, artiﬁcial olfaction aims at achieving similar performance and thus\\ndigitizing the sense of smell. Here, we present the development of machine learning-enabled graphene-based electronic olfaction sensors and\\npropose an approach to assess their olfactory performance toward VOCs. Eleven transient kinetic features extracted from the sensing\\nresponse proﬁle are utilized as their ﬁngerprint information. By mimicking the Snifﬁn’ Sticks test, electronic olfaction sensors exhibit high\\nolfactory performance toward four VOC odors (clove, eucalyptus, lemon, and rose scent) in terms of odor detection threshold, odor discrim-\\nination, and identiﬁcation performance. Upon exposure to binary odor mixtures, response features of electronic olfaction sensors behave\\nmore similarly to that of an individual odor, with a tendency that correlates with their ratio, resembling the overshadowing effect in human\\nolfactory perception. Molecular dynamics simulations and density functional theory calculations results reveal competing adsorption mecha-\\nnisms between odorant molecules and sensing materials. This may facilitate electronic olfaction sensor applications in some emerging ﬁelds,\\nsuch as environmental monitoring or public security.\\nVC 2023 Author(s). All article content, except where otherwise noted, is licensed under a Creative Commons Attribution (CC BY) license (http://\\ncreativecommons.org/licenses/by/4.0/). https://doi.org/10.1063/5.0132177\\nI. INTRODUCTION\\nOlfaction, as one of the oldest senses in terms of evolution, pro-\\nvides a highly adapted way of gathering information about the\\nchemical environment.1 Detection and identiﬁcation of volatile\\norganic compounds (VOCs) are important tasks fulﬁlled by biologi-\\ncal olfactory systems for navigation and survival.2,3 In the context of\\nmodern-day applications, digitizing the sense of smell using micro-\\nand nanoelectronic devices to identify VOCs and their mixtures\\nresults in bioinspired artiﬁcial olfaction.4,5 The effectiveness of artiﬁ-\\ncial olfaction, i.e., electronic noses (e-noses), relies on the one hand\\non the sensor hardware and on the other hand on the processing of\\nthe obtained signals. Herein, e-noses refer to instrumental replicas of\\nthe human sense of smell, comprising an array of electronic chemi-\\ncal sensors with partial speciﬁcity and a suitable pattern-recognition\\nsystem, capable of detecting simple or complex odors.6,7 Over the\\nyears, artiﬁcial olfaction has led to developments in biological\\nmodeling,8 sensor technology,9–11 and bioinspired technologies.12,13\\nIn this context, machine learning techniques are crucial to extract\\ncharacteristic information and to provide means of VOC discrimi-\\nnation and identiﬁcation.14\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-1\\nVC Author(s) 2023\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='fdf8e1ab-4d64-4ca7-8f65-c90ef3dd09ff', embedding=None, metadata={'page_label': '3', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Over the past decades, there has been notable progress in the\\ndevelopment of gas sensors for e-nose systems, including the PEN3\\nt y p ep o r t a b l ee l e c t r o n i cn o s eb yA I RS E N S EA n a l y t i c sG m b H\\n(Germany) and Mk3.3 by E-Nose Pty Ltd. (Australia). These sensors\\nshow great potential in a wide range of applications.15–19\\nDespite this signiﬁcant progress, the compactness of e-noses\\nremains a challenge due to the complex layout design of sensor arrays\\nand the need for high working temperatures. Metal-oxide-semicon-\\nductor (MOS-type) gas sensors are a prominent type due to their low\\ncost, feasible fabrication, and sensitivity.20,21 However, they require\\nhigh operating temperatures in the range of 200–500/C14 C since thermal\\nenergy is essential to activate the adsorption of ionized oxygen species\\nand to overcome the barriers of sensing reactions. For example, the\\nBME688 product by Bosch Sensortec (Germany) is a four-in-one gas\\nsensor that includes artiﬁcial intelligence and is housed in a rugged yet\\ncompact 3.0/C2 3.0 /C2 0.9 mm3 package, with an operating temperature\\nof up to 320/C14 C. Accordingly, a heater has to be taken into consider-\\na t i o ni nt h ed e s i g no ft h es e n s o rd e v i c e .22\\nCarbon nanomaterials, such as carbon nanotubes and graphene,\\noffer a promising alternative sensing material, due to their exceptional\\nproperties.23,24 For example, SmartNanotubes GmbH (Germany) has\\nproduced carbon nanotubes-based electronic noses consisting of\\nmulti-channel sensors for various gases and VOC discrimination.23,25\\nGraphene possesses remarkably high electron mobilities and an ultra-\\nhigh surface-to-volume ratio,26 and has already demonstrated out-\\nstanding performance in gas sensing applications compared to\\ntraditional materials, including high sensitivity, excellent reversibility,\\nand low operation temperature (usually room temperature).24,27,28\\nTherefore, the development of compact and miniaturized e-noses\\noperated at room temperature is of signiﬁcant importance.\\nHuman olfactory performance can be evaluated using a specialized\\npsychophysical test, such as the Snifﬁn’ Sticks test (Burghardt\\nVR\\n, Wedel,\\nGermany), which was introduced by Hummelet al. in 1997 and has\\nsince been validated in multiple European countries.29 This provides a\\nsemi-objective measure of olfactory function by presenting odor-ﬁlled\\nfelt pens to the subject. The assessment comprises three subtests: odor\\nthreshold (OT), odor discrimination (OD), and odor identiﬁcation\\n(OI).30 To conduct the odor threshold (OT) test, the threshold concen-\\ntration at which the subject can identify the odor (either 2-\\nphenylethanol or n-butanol) is determined using a single-staircase\\nbased on three-alternative forced choice (3-AFC) procedure, beginning\\nwith a 4% solution (with scores ranging from 1 to 16). For the odor dis-\\ncrimination (OD) test, the subject is asked to identify the odor pen that\\nhas a different scent from a triplet that contains two identical odors and\\na different one, using a 3-AFC paradigm (with scores ranging from 0 to\\n16). In the odor identiﬁcation (OI) test, the subject is required to iden-\\ntify the odor pen and indicate its name from a list of four written\\ndescriptors (with scores ranging from 0 to 16).31 An overall score [TDI\\n(Threshold score, Discrimination score, Identiﬁcation score)] is then\\ncalculated by adding up the scores from the OT, OD, and OI tests. A\\ns c o r eo f3 0 . 5p o i n t so rh i g h e ri n d i c a t e sn o r m o s m i a ,w h i l eas c o r e\\nbetween 16.5 and 30 points indicates hyposmia, and a score lower than\\n16.5 points indicates functional anosmia.32 However, to the best of our\\nknowledge, there have been no attempts to develop an olfactory perfor-\\nmance assessment methodology for artiﬁcial olfaction.\\nIn this contribution, we present the development of machine\\nlearning-enabled graphene-based single-channel electronic olfaction\\n(e-olfaction) sensors and propose a methodology to evaluate their\\nolfactory performance. We selected four VOC-based odors, namely,\\neucalyptol (Euca, eucalyptus scent), 2-nonanone (2Nona, lemon\\nscent), eugenol (Euge, clove scent), and 2-phenylethanol (2Phe, rose\\nscent), which are widely used in human olfactory performance assess-\\nment.30 We achieved a low odor detection limit of 4.4 ppm (for 2Phe),\\nand our e-olfaction platform showed high odor discrimination\\n(83.3%) and identiﬁcation (97.5%) accuracies. Moreover, we investi-\\ngated the similarity of perceived features between binary odor mixtures\\nand individual odors using e-olfaction. We also used molecular\\ndynamics simulations (MDS) and density functional theory (DFT) cal-\\nculations to elucidate the adsorption interaction between odorant mol-\\necules and sensing materials. Finally, we provide insights on the effect\\nof humidity in the performance of the method and validate it by iden-\\ntifying additional VOCs more commonly tested in gas sensing applica-\\ntions using humid air as carrier gas. Our results demonstrate that the\\ndeveloped e-olfaction exhibits high olfactory performance in snifﬁng\\nout VOC-based odors. This technology could be applied in various\\ndomains, such as environmental monitoring and public security.\\nII. RESULTS AND DISCUSSION\\nA. E-olfaction platform\\nThe workﬂow of our developed e-olfaction platform is illustrated\\nin Fig. 1, which consists of three steps: olfactory performance measure-\\nment, odor feature extraction, and odor TDI performance analysis.\\nThe essential component of this platform is the sensor device, which is\\nbuilt using a graphene dispersion prepared by exfoliation of graphite\\nassisted with an 8-aminopyrene-1,3,6-trisulphonic acid trisodium salt\\n(APTS) in an aqueous solution. While the usual approach for gra-\\nphene sensor functionalization with pyrene derivatives relies on its\\nmodiﬁcation after the deposition step,33 here the exfoliation and func-\\ntionalization take place at the same time.34 APTS, therefore, plays a\\ndual role as a stabilizer to preserve graphene dispersion and as active\\nadsorption sites for analyte molecules.24 The microelectrodes are\\nbridged through graphene ﬂakes via dielectrophoresis (DEP) of drop-\\ncast graphene dispersion under an alternating current (AC), resulting\\nin a chemiresistor-type gas sensor. The odor sensing measurement\\nsetup consists of three modules: an odor vapor generation module, an\\nodor sensing measurement module, and an external signal displaying\\nmodule. Individual odor tests mimicking Snifﬁn’ sticks tests and\\nbinary odor mixture tests are conducted as shown inFig. 1(a).\\nThe raw data acquired from the odor sensing setup were the sen-\\nsor current evolution of a single graphene-based gas sensor as a func-\\ntion of the operation time, comprising 24 repetitions for each odor\\n(Euca, Euge, 2Phe, and 2Nona). The repetitions were performed to\\ncollect as much raw data as possible for machine learning in the subse-\\nquent steps. In a practical application, one repetition is sufﬁcient for\\ndetection analysis once the machine learning classiﬁer model has been\\nestablished and deployed on the e-olfaction platform. In this work,\\ninstead of steady-state features, eleven transient features (a1,b 1,c 1,a 2,\\nb2,c 2,S ,k max,k min,a min, and Area) extracted from sensing response\\nproﬁle are considered for machine learning analysis as shown inFig.\\n1(b). The exponential features (a1,b 1,c 1,a 2,b 2,a n dc2)a r eﬁ t t i n g\\nparameters extracted from sensing response proﬁles and sensing\\nrecovery proﬁles following exponential ﬁtting, respectively. Transient\\nfeatureS is the transient-state sensing response in the odor exposure\\nphase, while featureArea is the area under the sensing response curve\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-2\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='c2539640-9afe-4118-aa75-d87aa2621a2a', embedding=None, metadata={'page_label': '4', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='in both odor exposure phase and odor ﬂushing periods. Transient fea-\\ntures kmax and kmin represent the maximum and minimum values\\nexisting in the sensing response proﬁle following the ﬁrst-derivative\\nﬁtting transformation, whileamin represents the minimum value in the\\ntransformed sensing response proﬁle following the second-derivative\\nﬁtting transformation. The details on the feature extraction method\\nare described in the supplementary material. Following features extrac-\\ntion, both unsupervised machine learning analysis and supervised\\nmachine learning analysis are conducted by mimicking the Snifﬁn’\\nsticks test. Finally, the olfactory performance of the developed\\ne-olfaction platform, i.e., odor threshold performance, odor discrimi-\\nnation performance, and odor identiﬁcation performance, is evaluated\\nas shown inFig. 1(c).\\nB. Odor threshold characterization\\nTo conduct the odor threshold test, the e-olfaction sensor device\\nwas exposed to odor 2Phe at decreasing concentrations ranging from\\n19.0 to 4.4 ppm. The same measurement was performed for pure N2\\nunder identical conditions to establish as an odorless control group.\\nOdor clustering results by unsupervised machine learning (principal\\ncomponent analysis, PCA) are shown inFig. 2. At high concentrations\\nof odor 2Phe, such as 19.0 ppm, the odor 2Phe cluster and the refer-\\nence cluster are clearly separated with a relatively large centroid\\ndistance. However, as the concentration of odor 2Phe decreased to\\n15.8 ppm, its cluster overlapped with the reference cluster in several\\npoints. At the lowest concentration (4.4 ppm, the minimum value\\nachieved by our setup), these two clusters exhibit moderate overlap.\\nThese results suggest that the developed e-olfaction platform can dis-\\ncriminate odor 2Phe from the odorless reference even at low concen-\\ntrations, down to 4.4 ppm.\\nC. Odor discrimination characterization\\nIn order to conduct the odor discrimination test, the e-olfaction\\nsensor was exposed to individual odors, involving Euca, 2Nona, Euge,\\nand 2Phe (pure N2 as odorless reference). As deﬁned by Snifﬁn’ Sticks\\ntest, for odor discrimination, the subject has to indicate the different\\nodor pen in a triplet without odor label information. Therefore, unsu-\\npervised machine learning (e.g., PCA, K-Means) was employed to\\ndiscriminate different odors without odor label feeding. PCA is a non-\\nparametric statistical technique primarily utilized for dimensionality\\nreduction or compression of a high-dimensional dataset onto a lower-\\ndimensional feature subspace to maintain most of the relevant infor-\\nmation.35 Following feature extraction and feature processing, the ﬁrst\\ntwo principal components of the odor features were plotted (PC1 vs\\nPC2) as illustrated inFig. 3(a). The ﬁrst principal component explains\\n50.63% of the variance, and the second principal component explains\\nFIG. 1. E-olfaction platform. (a) Olfactory performance measurement. Odorant molecules are vaporized by the carrier gas (N2) via a bubbling approach and delivered to the\\ngas sensing test chamber for olfactory performance evaluation by mimicking the Snifﬁn’ Sticks test. Four odorant molecules are involved, including2-Nonanone (2Nona),\\nEugenol (Euge), 2-Phenylethanol(2Phe), and Eucalyptol(Euca), which are widely utilized to evaluate the olfactory performance of Anosmia patients in Snifﬁn’ Stick tests.\\nOdorant molecules reach the sensor device surface and interact with the active sensing materials (i.e., functionalized graphene) via physical/chemical adsorption. Charge\\ntransfer occurs between odorant molecules and sensing materials, altering the electrical conductivity of sensing materials. (b) Odor feature extraction. The electrical signal for\\neach odor is then processed, including drift correction, signal normalization, feature extraction, etc. Each odor is represented by a multivariatefeature vector. (c) Odor TDI\\nresults analysis. Finally, the olfactory performance of the e-olfaction platform is evaluated in terms of odor threshold (OT), discrimination (OD), and identiﬁcation (OI) perfor-\\nmance by machine learning.\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-3\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='d896a570-199a-4779-86d6-ea3d75868c2a', embedding=None, metadata={'page_label': '5', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='22.76%. Together, the ﬁrst two principal components explain 73.39%\\nof the variance. It is noticed that the 2Nona cluster demonstrates excel-\\nlent separation from the other odor clusters, whereas the 2Phe, Euge,\\nand Euca clusters exhibit some overlapping.\\nMeanwhile, odor clustering analysis by the K-Means algorithm\\n(n_clusters¼5) was performed as illustrated inFig. 3(b). Clustering\\nanalysis attempts to group similar results based on a similarity mea-\\nsure, among which distance functions, such as Euclidean distance,36\\nare commonly utilized. Clusters are organized in such a fashion that\\nany two samples within the same cluster have a minimal distance and\\nany two samples across different clusters exhibit maximal distance.37\\nHerein, the most popular clustering algorithm, i.e., the K-means\\nalgorithm, is demonstrated as an example, and the results are shown\\nin Fig. 3(b). The odor confusion matrix achieved by the K-Means algo-\\nrithm is shown inFig. 3(c). Based on the confusion matrix results, the\\ndiscriminating precision result for each odor is analyzed.38 As shown\\nin Fig. 3(d), it is revealed that 2Nona exhibits the highest discrimina-\\ntive precision (95.8%) while Euca exhibits the lowest one (67.7%).\\nTherefore, implementing the K-Means clustering algorithm, the aver-\\nage discriminative precision for all odors approaches 83.3%.\\nD. Odor identification characterization\\nThe Snifﬁn’ Sticks test requires the subject to identify an odor\\npen and indicate its name from a list of written odor labels, which cor-\\nresponds to supervised machine learning analysis. In this section, the\\nmeasurement data are the same as the one used in Sec.IV E,b u tw i t h\\nthe addition of odor category labels. The supervised learning technique\\n(e.g., linear discriminant analysis) utilizes labeled datasets to train clas-\\nsiﬁer algorithms that can classify or predict outcomes for unknown\\ndatasets.39 Odor feature datasets as well as their labels are processed by\\nthe classiﬁer model, such as linear discriminant analysis (LDA), which\\nis a linear transformation technique for dimensionality reduction and\\na well-known classiﬁer.40 In contrast to PCA, LDA attempts to ﬁnd a\\nfeature subspace that optimizes class separability. Odor classiﬁcation\\nresults are illustrated inFig. 4(a). The ﬁrst three linear discriminants\\naccount for the total variance (LD1, LD2, and LD3 explain 82.42%,\\n14.36%, and 2.67%, respectively). These results indicate that the four\\nodor clusters, as well as the reference cluster, were classiﬁed accurately\\nwithout confusion. With the 10-fold cross-validation approach,41 the\\naverage classiﬁcation accuracy by LDA classiﬁer reaches 97.5% as\\nshown in the supplementary material.\\nTo assess the contribution of odor features to odor identiﬁcation,\\nthe importance weight of these eleven feature parameters (a1,b 1,c 1,a 2,\\nb2,c 2,S ,k max,k min,a min, and Area) were analyzed using the\\nRandomForest classiﬁer by scikit-learn software.42 As depicted inFig.\\n4(b), the four most important features (S, kmax,k min, and amin)m a k e u p\\nto 57.9% of total feature importance. The sole featurekmin contributes\\nto 21.4% of total feature importance. Hence, these transient derivative\\nfeatures (S, kmax,k min, and amin) are discriminative for odors, con-\\nstructing unique ﬁngerprints (see supplementary material). The con-\\ntribution of features whose importance weight is less than 5% (e.g., a2,\\nc1,a n dc2) is also essential to the effective odor identiﬁcation as we\\ncould observe slight odor clusters overlapping following removing\\nthese small weight features (see supplementary material).\\nE. Electronic perception to binary odor mixture\\nIn addition to the individual odors, the response toward binary\\nodor mixtures is investigated as well. Details on the sampling protocols\\nand measurements are described in the supplementary material.\\nFollowing feature extraction, the feature data of odor mixtures are ana-\\nlyzed by PCA. As shown inFig. 5(a), the ﬁrst six principal components\\nof a binary odor mixture could explain nearly 100% variance, while\\nthe ﬁrst two principal components could explain more than 75%. The\\nPCA score plot of a binary odor mixture, such as odor Euca-2Nona, is\\nshown inFig. 5(b). We found that the pure 2Nona cluster is separated\\nvery well from the pure Euca cluster without any overlapping. With\\nthe increasing Euca:2Nona ratio, the odor mixture cluster tends to\\nmove toward the pure Euca cluster and vice versa.\\nFIG. 2. Odor threshold performance. Discrimination of odor 2Phe from the odorless\\nreference (pure N2) in principal component analysis (PCA) space with descending\\nconcentrations of odor 2Phe (from 19.0 to 4.4 ppm).\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-4\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='aee10b8c-009b-4b0d-a68e-f50ac1152748', embedding=None, metadata={'page_label': '6', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Furthermore, a more quantitative analysis of the odor mixture\\nfeatures is carried out as schemed inFig. 5(d). The centroid points of\\nindividual odor clusters and odor mixture clusters are obtained ﬁrst,\\nand the Euclidean distance between the centroid point of an odor-\\nmixture cluster and the centroid point of a pure-odor cluster are\\nmeasured in PCA space. Consequently, each odor mixture has two dis-\\ntance parameters,d1 and d2, and they are represented in a 2D coordi-\\nnate system (d1 vs d2), as illustrated inFig. 5(e). A statistical result\\nbased on the introduced concept of “probability,” derived from the\\nequations at the bottom panel ofFig. 5(d),i ss h o w ni nFig. 5(g).T h e s e\\nresults suggest that the features of the odor mixture Euca-2Nona\\nbehave closer to Euca, with a higher probability and certainty of ﬁnd-\\ning Euca molecules in the mixture when its proportion in the mixture\\nincreases. A similar observation is found for the binary odor mixture\\nEuge-2Phe, as shown inFigs. 5(c), 5(f),a n d5(h), while odor Euge pre-\\nv a i l si nt h ep e r c e i v e df e a t u r e so v e ro d o r2 P h e .T os u mu p ,E u c a\\nFIG. 3. Odor discrimination performance.\\n(a) First principal component (PC1) vs\\nsecond principal component (PC2) plot of\\nodor clusters (ground truth in legend). (b)\\nOdor clustering results implemented with\\nthe K-Means clustering algorithm (pre-\\ndicted label in legend). (c) Confusion\\nmatrix of odor clustering implemented with\\nK-Means clustering algorithms. (d)\\nDiscrimination precision results of odor\\nclustering implemented with the K-Means\\nclustering algorithm. Color code: Euca\\n(green), Euge (red), 2Phe (dark grey),\\n2Nona (blue), and reference (light gray).\\nFIG. 4. Odor identiﬁcation performance.\\n(a) Odor classiﬁcation results imple-\\nmented with a Linear Discriminant\\nAnalysis (LDA) classiﬁer algorithm, the\\nﬁrst three discriminants (LD1, LD2, LD3)\\nexplain over 99% variance. (b) Relative\\nimportance loadings of 11 features for the\\nodor identiﬁcation implemented with the\\nrandom forest algorithm.\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-5\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='a736bd20-8138-4050-aac6-f31c9577f2cb', embedding=None, metadata={'page_label': '7', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='dominates the feature perception in the mixture Euca-2Nona, while\\nEuge dominates the feature perception in the mixture Euge-2Phe as\\nsummarized inFig. 5(i).\\nPrevious human psychophysical studies reported that the percep-\\ntion of a binary odor mixture could give rise to one of these three out-\\ncomes: an elemental perception, a conﬁgural perception, or an\\novershadowing effect.43,44 An elemental perception implies that both\\ncomponents of the odor mixture are perceived in the odor mixture,\\nw h e r e a sac o n ﬁ g u r a lp e r c e p t i o ni n f e r st h a tan e wo d o ri sp e r c e i v e d .45\\nOvershadowing refers to recognizing only one odor component in a\\nbinary mixture.46 Hence, our developed e-olfaction sensor perceives\\nfeatures closer to one of the odor components in the mixture upon\\nexposure to binary odor mixtures, behaving similarly to the oversha-\\ndowing effect in human olfaction perception.\\nF. Odorant molecules interaction with sensing\\nmaterials\\nTo provide insights on the underlying mechanism of the\\n“overshadowing effect” perceived by the e-olfaction sensor, the inter-\\naction between odor molecules and APTS-functionalized graphene\\nwas investigated via classical molecular dynamics simulations (MDS)\\nand density functional theory (DFT) calculations (see Sec.IV for\\ndetails). Pure odor models consisting of Euca, 2Nona, Euge, 2Phe, and\\nFIG. 5. Feature similarity of binary odor mixture to individual odor. (a) Explained variance of the ﬁrst 6 principal components of a binary odor mixture (Eucaand 2Nona) fea-\\ntures and their cumulative explained variance. PCA score plots of binary odor mixtures (b) Euca and 2Nona, (c) Euge and 2Phe. (d) Schematic diagram of centroid distance\\ncomponent and similarity formula of binary odor mixture to odor component.d1, centroid distance component 1, denotes centroid distance of odor mixture cluster and odor\\ncomponent 1 cluster.d2, centroid distance component 2, denotes centroid distance of odor mixture cluster and odor component 2 cluster. Centroid distance component analy-\\nsis plot of binary odor mixtures (e) Euca and 2Nona, (f) Euge and 2Phe. (g) Similarity of binary odor mixture Euca and 2Nona to odor component Euca. (h) Similarity of binary\\nodor mixture Euge and 2Phe to odor component Euge. (i) Feature perception of binary odor mixture generated by the e-olfaction platform.\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-6\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='05752b5b-8e58-4621-98fe-1b171da86d11', embedding=None, metadata={'page_label': '8', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='odor mixture models, including Euca-2Nona, Euge-2Phe, Euca-Euge,\\nEuca-2Phe, Euge-2Nona, and 2Nona-2Phe, were simulated. All mod-\\nels contained the same amount of odor molecules. The morphology of\\nodor molecules on the graphene surface is demonstrated in\\nFigs. 6(a)–6(f). The center-of-mass (COM) distance between odor\\nmolecules and APTS molecules is measured and shown inFig. 6(g).\\nBinding energy and charge transfer amount between odor molecules\\nand APTS molecules were calculated by DFT and are illustrated in\\nFigs. 6(h)and 6(i), respectively.\\nIn the individual odor case shown inFigs. 6(a)and 6(b),o d o r\\nmolecules of both Euca and 2Nona favor surrounding APTS. The\\nheads of odor molecules, which contain oxygen functional groups, pre-\\nfer to point into APTS, which is mainly governed by traditional dipo-\\nle–dipole interaction.47 In the odor mixture model shown inFig. 6(c),\\nwe observed that Euca molecules approach APTS much closer than\\n2Nona, which is supported by the COM distance to APTS shown in\\nFig. 6(g). Comparing the odor mixture model with the pure odor\\nmodel, the COM distance to APTS of odor Euca decreases while the\\nCOM distance to APTS of 2Nona increases, indicating competitive\\nadsorption of Euca and 2Nona on APTS. The binding energy of Euca\\nis slightly weaker than that of 2Nona, whereas the Euca molecule is\\nsmaller and more compact than 2Nona, which might be more beneﬁ-\\ncial for Euca to approach APTS closely.\\nOdor Euge and 2Phe exhibit very similar structures, consisting of\\na benzene ring and oxygen-containing functional groups. In the pure\\nodor case shown inFigs. 6(d)and 6(e), similar to Euca and 2Nona,\\nEuge and 2Phe molecules are found to surround APTS. The odor\\nmolecules’ heads are observed to point into the APTS head, which is\\nmainly affected by hydrogen bonding interaction between the head of\\nodor molecules and the head of APTS.48 Hydrogen bonding interac-\\ntion is believed to be stronger than conventional dipole–dipole interac-\\ntion.49,50 Apart from hydrogen bonding interaction,p–p stacking\\nexists between Euge or 2Phe and APTS as well, which does not exist in\\nthe Euca or 2Nona model. For this reason, some Euge and 2Phe mole-\\ncules are observed to stack onto APTS inFig. 6(f). In the odor mixture\\nmodel, the COM distance to APTS is demonstrated and compared\\nwith the pure odor model shown inFig. 6(g). In the odor mixture\\nmodel, the COM distance to APTS of Euge decreases a lot whilst the\\nCOM distance to APTS of 2Phe increases, which can be interpreted as\\ncompetitive adsorption of Euge and 2Phe on APTS. This competitive\\nadsorption originates from different binding energies to APTS, as\\ndepicted inFig. 6(h). Euge’s binding energies are larger than 2Phe’s\\nleading to solid binding between Euge and APTS. Consequently, Euge\\noccupies more space near APTS than 2Phe.\\nOdor molecules interact with APTS-functionalized graphene\\ninvolving two events: odor molecules adsorption on APTS and subse-\\nquently charge transfer between odor molecules and APTS-\\nfunctionalized graphene. Both synergistically yield a characteristic\\nsensing response.51 The charge transfer amount between odor mole-\\ncules and APTS-functionalized graphene is calculated by DFT, as\\nshown inFig. 6(i). Under the same circumstances, odor Euca transfers\\n18.4% more charge to APTS than 2Nona, while Euge transfers 83.8%\\nmore charge to APTS than 2Phe. With both odor adsorption and\\ncharge transfer in mind, odor molecules exhibiting stronger\\nFIG. 6. Odorant molecule interaction with\\nAPTS functionalized graphene. Snapshot\\nof odorant molecules interacting with\\nAPTS-functionalized graphene via molec-\\nular dynamics simulations, (a) Euca, (b)\\n2Nona, (c) Euca and 2Nona odor mixture,\\n(d) Euge, (e) 2Phe, (f) Euge and 2Phe\\nodor mixture. Color code: graphene\\n(gray), Euca (lime), 2Nona (blue), Euge\\n(orange), 2Phe (purple), carbon (cyan),\\nnitrogen (blue), sulfur (yellow), oxygen\\n(red). (g) Center of mass (COM) distance\\nbetween odor molecules and APTS mole-\\ncules in the context of individual odor and\\nbinary odor mixture, respectively. Binding\\nenergy (h) and charge transfer (i) between\\nodorant molecules and APTS from DFT\\ncalculations.\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-7\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='56654426-ac98-444b-bdb4-55037e8aa49d', embedding=None, metadata={'page_label': '9', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='interaction with sensing element materials embrace more chances to\\napproach the functionalized graphene and dominate the signal charac-\\nteristic in the odor mixture, which subsequently suppresses the signal\\nof the other odor. These results demonstrate that owing to the distinct\\nbinding energies and charge transfer characteristics of odors, competi-\\ntive adsorption to the sensing material occurs between two odor com-\\nponents in the mixture.\\nG. Impact of air and humidity\\nHydrogen bonding interaction induced by humidity plays an\\nimportant role in the gas sensing activities, particularly for graphene-\\nbased chemiresistive-type gas sensors, as it was veriﬁed in our previous\\nwork.24 Although the humidity and presence of air may affect the\\nintrinsic response, i.e., the interaction between the sensor and the tar-\\nget VOC, the effective response induced by such interaction is still dis-\\ntinct and, therefore, detectable by machine learning classiﬁer\\nalgorithms. This could be veriﬁed by performing experiments using air\\nas carrier gas and detecting VOCs and their binary mixture with water\\nvapor. The experiment details, response results, as well as classiﬁcation\\nresults are shown in the supplementary material.\\nFirst, ethanol (ETH) was chosen as a frequently found VOC in\\ngas sensor applications. The sensor was exposed to 48 cycles of pure\\nwater (alcohol by volume 0 or ABV0), pure ethanol (ABV100), and a\\nbinary mixture of 80% ethanol and 20% water (ABV80). The pure eth-\\nanol cluster is very well separated from the other clusters. The water\\nvapor and binary mixture have a very small degree of overlapping,\\nallowing for their classiﬁcation. The overall classiﬁcation accuracy to\\nidentify these three vapors could reach 91.7%. From these results, we\\ncan conclude that the extracted features (ﬁngerprint) from the effective\\nsensing response proﬁle are distinct. In combination with machine\\nlearning algorithms, these multiple transient features could contribute\\nto the VOC identiﬁcation at high accuracy.\\nWe further proceeded to the exposure of four additional com-\\nmon VOCs, namely, ethanol (ETH), methanol (MTH), 2-propanol\\n(PRL), and water vapor. In the PCA analysis, it could be found that\\nthe ﬁrst three components could explain around 87.7% variance. The\\nwater vapor cluster and methanol vapor cluster were well separated\\nfrom the other two, while the ethanol cluster and 2-propanol cluster\\nshowed some overlapping. With the LDA, the separability of these\\nfour clusters was better except for a small overlapping between ethanol\\nand 2-propanol. With the excellent separability in LDA space, the con-\\nfusion matrix of these gases’ classiﬁcation exhibited high accuracy (see\\nsupplementary material). The overall classiﬁcation accuracy was\\n95.8% implemented by the hold-out strategy, in which 75% of features\\nwere employed to train classiﬁer models and the rest features were\\nemployed to validate the trained classiﬁer models. These experiments\\nsuggest that the discrimination of VOCs using multiple transient fea-\\ntures is effective in ambient air and not just using nitrogen as carrier\\ngas.\\nAn aging test by exposing the sensor to various humidity levels\\nfor 36 h (720 cycles) did not evidence sensor deterioration due to water\\nvapor effects.\\nIII. CONCLUSION\\nIn this work, we introduce the development of machine learning-\\nenabled graphene-based single-channel electronic olfaction (e-\\nolfaction) sensors, along with an olfactory performance assessment\\nmethodology for characterizing their performance. The e-olfaction\\nsensor device comprises a single-channel nanosensor, which distin-\\nguishes it from conventional e-nose systems that utilize sensor arrays,\\nand operates at room temperature. This unique design shows great\\npotential for miniaturization and portability. We extract eleven kineti-\\ncally transient-state features are extracted from odor sensing response\\nproﬁles, which takes advantage of characteristic response proﬁles of\\nodor sensing. Our e-olfaction concept exhibits excellent performance\\nunder controlled lab conditions, not limited by the use of air as the\\ncarrier gas, in terms of odor threshold, odor discrimination and odor\\nidentiﬁcation. Speciﬁcally, we achieve a low odor threshold toward\\n2Phe (4.4 ppm) and an excellent odor discrimination performance\\n(83.3%). Furthermore, the incorporation of supervised machine learn-\\ning classiﬁer algorithms (LDA) enables high odor identiﬁcation accu-\\nracy (97.5%). Simultaneously, the e-olfaction device manifests efﬁcient\\nprocessing capability toward binary odor mixtures. The response to\\nbinary odor mixtures is shown to behave close to an individual odor\\nwhile the response of the other odor is partially suppressed. This phe-\\nnomenon is analogous to the overshadowing effect in human olfaction\\nperception upon processing binary odor mixtures.\\nThe relatively long sampling data acquisition period is a limita-\\ntion of this work, which could be addressed by optimizing the sam-\\npling protocols and developing faster responding sensing materials.\\nFuture investigations could explore this topic, which is out of the scope\\nof the present work.\\nThe potential compact size of our e-olfaction platform meets the\\nminiaturization requirements for seamless integration into various\\ndevices. It leverages arrays of highly sensitive nanomaterials function-\\nalized in diverse ways, thereby enabling the detection and discrimina-\\ntion of a much larger amount of target odor molecules and their\\ncomplex mixtures. When coupled with mobile devices for data analy-\\nsis, it holds great promise for assisting individuals with olfactory disor-\\nders in the near future. In addition, it has the potential to be applied in\\nnumerous emerging ﬁelds, such as environmental monitoring or pub-\\nlic security.\\nIV. METHODS\\nA. Graphene dispersion preparation\\nGraphite powder (crystalline,/C0 20 þ84 mesh, 99.9%, metal basis,\\nCAS: 7782-42-5) was ordered from Alfa Aesar Company. 8-\\nAminopyrene-1,3,6-trisulphonic acid trisodium salt (APTS,\\nC16H8NNa3O9S3, CAS: 196504-57-1) was bought from VWR\\nInternational Company. The optimized recipe for graphene dispersion\\npreparation was as below:5 ml APTS aqueous solution (1 mg/ml) was\\nmixed with 5 ml graphite powder aqueous solution (30 mg/ml), the\\nmixture was then sonicated for 2 h by horn-type sonicator (50%\\namplitude, Model 250-D, Branson Digital Soniﬁer, U.S.) in an ice\\nbath. Afterward, the obtained solution was left to stand overnight. To\\nfurther remove unexfoliated graphite particles, the dispersions were\\ncentrifuged at 4000 rpm for 2 min (Eppendorf, MiniSpin plus,\\nGermany), and the top 50% supernatant was pipetted out to another\\ncentrifugation tube. The rest was discarded. Following feeding de-\\nionized (DI) water to the original volume, the supernatant dispersion\\nwas then re-dispersed by a mild sonication setup (VWR ultrasonic\\ncleaner, type: USC 300 TH, HF 45 kHz, 80 W, U.S.) for 10 min in an\\nice bath. Thereafter, the above centrifugation and re-dispersing\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-8\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='7dffdaf4-0da1-47fc-90f0-75814cc61cf4', embedding=None, metadata={'page_label': '10', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='procedure were performed twice more. Finally, the aqueous dispersion\\nof graphene functionalized by APTS was achieved.\\nB. Sensor device fabrication\\nThe sensor device was fabricated by dielectrophoretic alignment\\nof graphene ﬂakes on gold interdigitated electrodes (IDEs) fabricated\\non silicon wafers. Electrode fabrication was done utilizing a standard\\nmicrofabrication process comprising photolithography, gold thermal\\nevaporation and liftoff, as introduced in our previous published\\nwork.25 First, the IDEs pattern were fabricated on p-type Si substrate\\n(4-in.) with 300 nm of thermally grown SiO2 using photolithography\\ntechniques (MJB4, S€USS MicroTec, Germany). 15 nm chromium and\\n100 nm gold were then deposited on the substrate via thermal evapora-\\ntion and liftoff process. After that, the bare devices were cleaned with\\nacetone and isopropanol, respectively. Finally, the devices were heated\\nto 100/C14 C for 5 min in order to remove residual solvents and naturally\\ncooled down to room temperature. IDE structure on the device fea-\\ntures a gap size of 3lm and a ﬁnger width of 4lm, respectively.\\nAlternating current DEP with a signal generator (AFG320 signal gen-\\nerator, Sony-Tektronix, Japan) was applied to deposit graphene on the\\ndevice, which bridged neighboring electrodes precisely and achieved a\\nhomogeneous graphene network on the device. DEP parameters,\\nincluding frequency of the applied signal (f), peak-to-peak voltage\\n(Vpp), and processing time (t), play a major role in the alignment of\\ngraphene ﬂakes.52 The optimized parameters were as follows:\\nVpp¼10 V, f¼200 kHz, t¼30 s. In a typical DEP procedure, a drop-\\nlet of 10ll dispersion was pipetted onto the IDE, which was ﬁxed on a\\nprobe station (Karl S€uss, Garching, Germany), then alternating current\\nwith speciﬁc frequency was applied on the sensor device for 30 s, fol-\\nlowed by DI water rinsing and N2 ﬂow drying. Sensor device fabrication\\nwas completed and stored under an inert atmosphere (N2)u n t i lu s e .\\nC. Odor vapor preparation\\nFour basic odors, eucalyptol (CAS number: 470-82-6, item num-\\nber: C80601–500ML, purity /C21 99.0%, Sigma-Aldrich, U.S.), 2-\\nnonanone (CAS number: 821–55-6, item number: W278505-250G-K,\\npurity/C21 99.0%, Sigma-Aldrich, U.S.), eugenol (CAS number: 97-53-0,\\nitem number: W246700-1 KG-K, purity /C21 98.0%, Sigma-Aldrich,\\nU.S.), and 2-phenylethanol (CAS number: 60-12-8, item number:\\n77861-1L, purity/C21 99.0%, Sigma-Aldrich, U.S.), were investigated in\\nthis work. A bubbler evaporation system was developed to generate\\nodor vapor as well as deliver odor vapor to the gas chamber.53 The\\nﬂow rates of dry nitrogen, acting as carrier gas and dilution gas, were\\ntuned by mass ﬂow controllers (MFC, type number: GF040, Brooks\\nInstruments Company). The source of both carrier gas and dilution\\ngas was nitrogen, whose ﬂow rate was precisely controlled by a mass\\nﬂow controller (MFC). Odor concentration could be ﬁnely tuned by\\ncombining the ﬂow rate of both carrier and dilution gas. For the analy-\\nsis of the impact of humidity on the results, we substituted N2 for\\nambient air as carrier gas. The bubbling approach using air through\\nwater and ethanol was used to generate their vapors at different ratios.\\nD. Odor sensing measurement\\nThe sensing performance of e-olfaction toward both individual\\nodor and odor mixture were evaluated at room temperature (20/C14 C),\\nrespectively. The gas chamber was a homemade gas sensing setup and\\napplied to measure the electrical property of the sensor upon exposure\\nto individual odor vapor or odor mixture. 25 Upon odor vapor\\nadsorbed by functionalized graphene on the sensor, the electrical con-\\nductivity of the sensor shifted due to the occurrence of charge carrier\\ntransfer between odor molecules and functionalized graphene. A con-\\nstant voltage (V ¼0.1 V) was applied to the sensor and the sensor cur-\\nrent (I) was continuously recorded by a source meter (Keithley 2602,\\nTektronix GmbH, Germany) during exposure to odor vapors and\\nrecovery steps under pure nitrogen ﬂow. Upon odor vapor adsorption\\nby functionalized graphene on the sensor, the electrical conductivity of\\nthe sensor shifted due to the occurrence of charge carrier transfer\\nbetween odor molecules and functionalized graphene. Following that,\\nthe path of carrier gas was switched off while the ﬂow rate of dilution\\ngas was increased to desorb and ﬂush odor molecules from the gra-\\nphene and released sensor recovery. In this work, the period for odor\\nexposure and odor ﬂushing is 15 and 10 min, respectively. To produce\\nmore data for machine learning analysis, 24 repetition tests were carried\\nout for each odor test on the same condition. Before the initial cycle test,\\na stabilization process was performed to remove the adsorbed water or\\nother contaminants was performed, in which pure nitrogen ﬂow was\\nactive until the resistance of the sensor device approached a plateau\\nstate. In this work, a set of visual basic (VB) scripts were developed to\\nautomatically manage the whole measurement procedure,54 including,\\nstart time and end time of the odor exposure step as well as odor ﬂush-\\ning step, the ﬂow rate of the carrier gas, and dilution gas in a different\\nphase, as well as acquire current information, etc.\\nE. Odor discrimination\\nOdor discrimination aims to classify odor scatters with similar\\nfeatures into the same class while odor scatters with distinctive proper-\\nties into different groups. This goal could be achieved via unsupervised\\nmachine learning. In this session, 4 odors, as well as an odorless refer-\\nence (pure N2), were discriminated. Before feeding into clustering\\nalgorithms, feature data of all odors had transformation preprocessing\\nusing StandardScaler, MinMaxScaler, and L2 normalization algo-\\nrithms. The transformed features were further applied principal com-\\nponent analysis (PCA) for dimensionality reduction and analyzed\\nwith diverse clustering algorithms.\\nF. Odor identification\\nOdor identiﬁcation is to predict the label information for the new\\nodor from known labels after algorithm training. This could be\\nachieved via supervised machine learning. In this session, 4 odors, as\\nwell as odorless reference (pure N2), were identiﬁed with labels. In\\ncontrast to unsupervised machine learning, odor labels target had to\\nbe predicted as well.39 All the features together with additional target\\n(odor labels) information were split up into two sets, 75% data were\\nused to train classiﬁer algorithms while the rest 25% data were used to\\ntest classiﬁer algorithms. Before the training data were fed into classi-\\nﬁer algorithm, for example, linear discriminant analysis (LDA) algo-\\nrithm, the features of the training dataset were transformed with\\nStandardScaler algorithm. Afterward, the transformation parameters\\ngenerated from training data were applied upon the test data set\\nto generate transformed test data set. The prediction accuracy\\nachieved by the LDA algorithm could be obtained. Meanwhile, the\\nperformance of LDA classiﬁer algorithms was achieved using K-Fold\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-9\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='01bf8309-1eaf-4087-b6e6-180acdedca3b', embedding=None, metadata={'page_label': '11', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='cross-validation (K ¼10) approach. The above data processing of both\\nunsupervised and supervised machine learning was conducted via a\\nset of Python programs.\\nSUPPLEMENTARY MATERIAL\\nSee the supplementary material for additional results, supporting\\ndata, and data analysis procedures.\\nACKNOWLEDGMENTS\\nThis research was funded by VolkswagenStiftung (Grant Nos.\\n96632 and 9B396) as well as EU H2020-MSCA-RISE-2016 project\\nCARBO IMmap (No. 734381), EU project “Smart Electronic\\nOlfaction for Body Odor Diagnostics” (SMELLODI, Grant No.\\n101046369), 6G-life project (Federal Ministry of Education and\\nResearch of Germany in the programme of “Souver €an. Digital.\\nVernetzt.,” Project Identiﬁcation No. 16KISK001K), China\\nScholarship Council (CSC). Odor samples were kindly provided by\\nProfessor Thomas Hummel (Smell and Taste Clinic, Department of\\nOtorhinolaryngology, TU Dresden, Germany). We are grateful to\\nViktor Bezugly, Ilona Croy, Thomas Hummel, for the discussion\\nduring the course of this research. We also appreciate Markus\\nL€ofﬂer (Dresden Center for Nanoanalysis, Center for Advancing\\nElectronics Dresden (CfAED), TU Dresden, Germany) for technical\\nassistance on SEM imaging. We acknowledge the computational\\nresources provided by the Centre of Information Services and\\nHigh-Performance Computing (ZIH) at the Technische Universit€at\\nDresden, as well as the use of the state-of-the-art facilities in the\\nDresden Center for Nanoanalysis at the TU Dresden.\\nAUTHOR DECLARATIONS\\nConflict of Interest\\nThe authors have no conﬂicts to disclose.\\nAuthor Contributions\\nShirong Huang: Formal analysis (equal); Investigation (equal);\\nMethodology (equal); Software (equal); Writing – original draft\\n(equal). Alexander Croy: Conceptualization (equal); Data curation\\n(equal); Funding acquisition (equal); Project administration (equal);\\nSupervision (equal); Writing – review & editing (equal).Antonie\\nLouise Bierling:Methodology (equal); Visualization (equal); Writing\\n– review & editing (equal).Vyacheslav Khavrus:Investigation (equal);\\nMethodology (equal). Luis Antonio Panes-Ruiz: Methodology\\n(equal). Arezoo Dianat:Investigation (equal); Methodology (equal).\\nBergoi Ibarlucea:Conceptualization (equal); Data curation (equal);\\nFunding acquisition (equal); Methodology (equal); Project administra-\\ntion (equal); Supervision (equal); Writing – review & editing (equal).\\nGianaurelio Cuniberti:Conceptualization (equal); Funding acquisi-\\ntion (equal); Project administration (equal); Supervision (equal);\\nWriting – review & editing (equal).\\nDATA AVAILABILITY\\nThe data that support the ﬁndings of this study are available\\nfrom the corresponding authors upon reasonable request.\\nREFERENCES\\n1C. Sarafoleanu, C. Mella, M. Georgescu, and C. Perederco, J. Med. Life2(2),\\n196–198 (2009).\\n2N. Zannoni, M. Wikelski, A. Gagliardo, A. Raza, S. Kramer, C. Seghetti, N.\\nWang, A. Edtbauer, and J. Williams,Sci. Rep.10(1), 15879 (2020).\\n3T.-M. Koski, T. Laaksonen, E. M€antyl€a, S. Ruuskanen, T. Li, P. S. Gir/C19on-Calva,\\nL. Huttunen, J. D. Blande, J. K. Holopainen, and T. Klemola,Ethology 121(12),\\n1131–1144 (2015).\\n4A. J. M. Barbosa, A. R. Oliveira, and A. C. A. Roque, Trends Biotechnol.\\n36(12), 1244–1258 (2018).\\n5C. Ban, X. Min, J. Xu, F. Xiu, Y. Nie, Y. Hu, H. Zhang, M. Eginligil, J. Liu, W.\\nZhang, and W. Huang,Adv. Mater. Technol.6(11), 2100366 (2021).\\n6J. W. Gardner and P. N. Bartlett,Sens. Actuators, B18(1–3), 210–211 (1994).\\n7J. A. Covington, S. Marco, K. C. Persaud, S. S. Schiffman, and H. T. Nagle,\\nIEEE Sens. J.21(11), 12969–12990 (2021).\\n8A. Sharma, R. Kumar, I. Aier, R. Semwal, P. Tyagi, and P. Varadwaj,Curr.\\nNeuropharmacol. 17(9), 891–911 (2019).\\n9J. Gutierrez and M. C. Horrillo,Talanta 124, 95–105 (2014).\\n10S. Y. Park, Y. Kim, T. Kim, T. H. Eom, S. Y. Kim, and H. W. Jang,InfoMat\\n1(3), 289–316 (2019).\\n11F. Rock, N. Barsan, and U. Weimar,Chem. Rev.108(2), 705–725 (2008).\\n12Y. A. Jodat, K. Kiaee, D. V. Jarquin, R. L. De la Garza Hernandez, T. Wang, S.\\nJoshi, Z. Rezaei, B. A. G. de Melo, D. Ge, M. S. Mannoor, and S. R. Shin,Adv.\\nSci. (Weinheim, Ger.)7(5), 1901878 (2020).\\n13H. J. Ko and T. H. Park,J. Biol. Eng.10, 17 (2016).\\n14S. Huang, A. Croy, L. A. Panes-Ruiz, V. Khavrus, V. Bezugly, B. Ibarlucea, and\\nG. Cuniberti,Adv. Intell. Syst.4(4), 2200016 (2022).\\n15F. S. Fedorov, A. Yaqin, D. V. Krasnikov, V. A. Kondrashov, G. Ovchinnikov,\\nY. Kostyukevich, S. Osipenko, and A. G. Nasibulin,Food Chem. 345, 128747\\n(2021).\\n16H. Rottiers, D. A. T. Sosa, L. Van de Vyver, M. Hinneh, H. Everaert, J. De\\nWever, K. Messens, and K. Dewettinck,Food Anal. Methods 12(2), 475–488\\n(2018).\\n17J. Zhang, Y. Xue, T. Zhang, Y. Chen, X. Wei, H. Wan, and P. Wang,\\nJ. Electrochem. Soc.167(14), 147519 (2020).\\n18A. Lichtenstein, E. Havivi, R. Shacham, E. Hahamy, R. Leibovich, A. Pevzner,\\nV. Krivitsky, G. Davivi, I. Presman, R. Elnathan, Y. Engel, E. Flaxer, and F.\\nPatolsky, Nat. Commun.5, 4195 (2014).\\n19B .S h a n ,Y .Y .B r o z a ,W .L i ,Y .W a n g ,S .W u ,Z .L i u ,J .W a n g ,S .G u i ,L .\\nW a n g ,Z .Z h a n g ,W .L i u ,S .Z h o u ,W .J i n ,Q .Z h a n g ,D .H u ,L .L i n ,Q .Z h a n g ,\\nW. Li, J. Wang, H. Liu, Y. Pan, and H. Haick,ACS Nano 14(9), 12125–12132\\n(2020).\\n20M. M. Macias, A. G. Manso, C. J. Orellana, H. M. Velasco, R. G. Caballero, and\\nJ. C. Chamizo,Sensors (Basel)13(1), 208–220 (2012).\\n21E. Bonah, X. Huang, R. Yi, J. H. Aheto, R. Osae, and M. Golly,J. Food Process\\nEng. 42(6), e13236 (2019).\\n22R. Kumar, X. Liu, J. Zhang, and M. Kumar, Nano-Micro Lett. 12(1), 164\\n(2020).\\n23L. A. Panes-Ruiz, L. Riemenschneider, M. M. Al Chawa, M. Lofﬂer, B.\\nRellinghaus, R. Tetzlaff, V. Bezugly, B. Ibarlucea, and G. Cuniberti,Nano Res.\\n15, 2512–2521 (2021).\\n24S. Huang, L. A. Panes-Ruiz, A. Croy, M. L€ofﬂer, V. Khavrus, V. Bezugly, and\\nG. Cuniberti,Carbon 173, 262–270 (2021).\\n25L. A. Panes-Ruiz, M. Shaygan, Y. Fu, Y. Liu, V. Khavrus, S. Oswald, T. Gemming,\\nL. Baraban, V. Bezugly, and G. Cuniberti,ACS Sens.3(1), 79–86 (2018).\\n26F. Yavari and N. Koratkar,J. Phys. Chem. Lett.3(13), 1746–1753 (2012).\\n27H. Hu, X. Yang, X. Guo, K. Khaliji, S. R. Biswas, F. J. G. de Abajo, T. Low, Z.\\nSun, and Q. Dai,Nat. Commun.10(1), 1131 (2019).\\n28W. Li, X. Geng, Y. Guo, J. Rong, Y. Gong, L. Wu, X. Zhang, P. Li, J. Xu, G.\\nCheng, M. Sun, and L. Liu,ACS Nano5(9), 6955–6961 (2011).\\n29T. Hummel, B. Sekinger, S. R. Wolf, E. Pauli, and G. Kobal,Chem. Sens.22(1),\\n39–52 (1997).\\n30T. Hummel, K. Rissom, J. Reden, A. Hahner, M. Weidenbecher, and K. B.\\nHuttenbrink, Laryngoscope 119(3), 496–499 (2009).\\n31M. Wolfensberger, I. Schnieper, and A. Welge-Lussen, Acta Oto-Laryngol.\\n120(2), 303–306 (2000).\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-10\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='3de5e677-323d-4f01-9500-a2ca8235487e', embedding=None, metadata={'page_label': '12', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='32G. Kobal, L. Klimek, M. Wolfensberger, H. Gudziol, A. Temmel, C. M. Owen,\\nH. Seeber, E. Pauli, and T. Hummel,Eur. Arch. Oto-Rhino-Laryngol.257(4),\\n205–211 (2000).\\n33D. K. H. Tsang, T. J. Lieberthal, C. Watts, I. E. Dunlop, S. Ramadan, A. E. del\\nRio Hernandez, and N. Klein,Sci. Rep.9(1), 13946 (2019).\\n34E. Y. L. Teo, G. A. M. Ali, H. Algarni, W. Cheewasedtham, T. Rujiralai, and K.\\nF. Chong,Mater. Chem. Phys.231, 286–291 (2019).\\n35H. Abdi and L. J. Williams, Wiley Interdiscip. Rev.: Comput. Stat. 2(4),\\n433–459 (2010).\\n36N. Bouhmala, paper presented at the 5th IIAI International Congress on\\nAdvanced Applied Informatics (IIAI-AAI), 2016.\\n37J. Irani, N. Pise, and M. Phatak,Int. J. Comput. Appl.134, 9–14 (2016).\\n38M. Hossin and M. N. Sulaiman,Int. J. Data Min. Knowl. Manage. Process.\\n5(2), 11 (2015).\\n39J. Lotsch, D. Kringel, and T. Hummel,Chem. Sens.44(1), 11–22 (2019).\\n40A. J. Izenman, in Modern Multivariate Statistical Techniques: Regression,\\nClassiﬁcation, and Manifold Learning, edited by A. J. Izenman (Springer, New\\nYork, NY, 2008), pp. 237–280.\\n41T. Fushiki,Stat. Comput.21(2), 137–146 (2011).\\n42B. H. Menze, B. M. Kelm, R. Masuch, U. Himmelreich, P. Bachert, W. Petrich,\\nand F. A. Hamprecht,BMC Bioinf.10(1), 213 (2009).\\n43C. Sinding, G. Coureaud, C. Chabanet, A. Chambault, N. B/C19eno, T. Dosne, B. Schaal,\\nand T. Thomas-Danguin,Flavour Science(Academic Press 2014), pp. 27–31.\\n44M. E. Frank, D. B. Fletcher, and T. P. Hettinger,Chem. Sens. 42(7), 537–546\\n(2017).\\n45T. Thomas-Danguin, C. Sinding, S. Romagny, F. El Mountassir, B. Atanasova,\\nE. Le Berre, A. M. Le Bon, and G. Coureaud,Front. Psychol.5, 504 (2014).\\n46L. M. Kay, T. Crk, and J. Thorngate,Behav. Neurosci.119(3), 726–733 (2005).\\n47Y. El-Sayed and T. J. Bandosz,Langmuir 21(4), 1282–1289 (2005).\\n48M. Franz, H. A. Arafat, and N. G. Pinto,Carbon 38(13), 1807–1819 (2000).\\n49A. Rajak, A. K. Singh, L. Roy, and A. Das,ChemNanoMat 8(6), e202200082\\n(2022).\\n50E. M. D. Keegstra, A. L. Spek, J. W. Zwikker, and L. W. Jenneskens,J. Chem.\\nSoc., Chem. Commun.1994(14), 1633–1634.\\n51Q. Yuan, Y.-P. Zhao, L. Li, and T. Wang,J. Phys. Chem. C113(15), 6107–6113\\n(2009).\\n52J. Wang, S. Rathi, B. Singh, I. Lee, H.-I. Joh, and G.-H. Kim,ACS Appl. Mater.\\nInterfaces 7(25), 13768–13775 (2015).\\n53H. Qi, J. Liu, J. Pionteck, P. P€otschke, and E. M€ader, Sens. Actuators, B213,\\n20–26 (2015).\\n54F. Balena and J. Fawcette,Programming Microsoft Visual Basic 6.0(Microsoft\\nPress, Washington, 1999).\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-11\\nVC Author(s) 2023\\n 02 January 2025 11:54:10\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='aff1e2e8-dde5-4cc9-a522-be69af998bea', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\main.ipynb', 'file_name': 'main.ipynb', 'file_size': 59464, 'creation_date': '2024-12-28', 'last_modified_date': '2025-04-25'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nfrom datasets import load_dataset\\n\\nds = load_dataset(\"seyonec/goodscents_leffingwell\")\\nds\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='570ddf15-6d6a-4938-9fd1-c519ee6078d4', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\main.ipynb', 'file_name': 'main.ipynb', 'file_size': 59464, 'creation_date': '2024-12-28', 'last_modified_date': '2025-04-25'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nimport networkx as nx\\nimport matplotlib.pyplot as plt\\n\\n# Assuming ds is already defined and contains the graph data\\ngraph = ds[\"train\"][123]\\n\\nedges = graph[\"edge_index\"]\\nnum_edges = len(edges[0])\\nnum_nodes = graph[\"num_nodes\"]\\n\\n# Conversion to NetworkX format\\nG = nx.Graph()\\nG.add_nodes_from(range(num_nodes))\\nG.add_edges_from([(edges[0][i], edges[1][i]) for i in range(num_edges)])\\n\\n# Set up the plot\\nplt.figure(figsize=(10, 8))  # Set figure size\\n\\n# Define a layout for our nodes \\npos = nx.spring_layout(G)  # You can also try nx.circular_layout(G) or nx.kamada_kaway_layout(G)\\n\\n# Draw the nodes and edges\\nnx.draw_networkx_nodes(G, pos, node_size=700, node_color=\\'lightblue\\', alpha=0.7)\\nnx.draw_networkx_edges(G, pos, width=2, alpha=0.5, edge_color=\\'gray\\')\\n\\n# Draw labels for the nodes\\nlabels = {i: str(i) for i in range(num_nodes)}  # Label nodes with their indices\\nnx.draw_networkx_labels(G, pos, labels, font_size=12)\\n\\n# Title and axis settings\\nplt.title(\"Graph Visualization\", fontsize=16)\\nplt.axis(\\'off\\')  # Turn off the axis\\n\\n# Show the plot\\nplt.show()\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='583fc5bc-47ad-4d80-989c-134fb8a7ba27', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\main.ipynb', 'file_name': 'main.ipynb', 'file_size': 59464, 'creation_date': '2024-12-28', 'last_modified_date': '2025-04-25'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\"\\n\\n\\ndata = ds['train'].to_pandas()\\ndata\\n\\n\\n# \", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='b6a00767-ac74-4449-929a-7b8b415299a0', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\main.ipynb', 'file_name': 'main.ipynb', 'file_size': 59464, 'creation_date': '2024-12-28', 'last_modified_date': '2025-04-25'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\"\\n\\n\\n# import plotly.express as px\\n# from sklearn.decomposition import PCA\\n\\n# df = px.data.iris()\\n# X = df[['sepal_length', 'sepal_width', 'petal_length', 'petal_width']]\\n\\n# pca = PCA(n_components=3)\\n# components = pca.fit_transform(X)\\n\\n# total_var = pca.explained_variance_ratio_.sum() * 100\\n\\n# fig = px.scatter_3d(\\n#     components, x=0, y=1, z=2, color=df['species'],\\n#     title=f'Total Explained Variance: {total_var:.2f}%',\\n#     labels={'0': 'PC 1', '1': 'PC 2', '2': 'PC 3'}\\n# )\\n# fig.show()\\n\\n\", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='7641ce22-b01f-4886-a45c-54275bf9c16c', embedding=None, metadata={'page_label': '1', 'file_name': 'NeurIPS_ML4PS_2019_39.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\NeurIPS_ML4PS_2019_39.pdf', 'file_type': 'application/pdf', 'file_size': 702741, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='The Chemistry of Smell: Learning Generalizable\\nPerceptual Representations of Small Molecules\\nBenjamin Sanchez-Lengeling1,*, Jennifer N Wei 1,*, Brian K Lee 1, Richard C Gerkin 2, Alán\\nAspuru-Guzik3, and Alexander B Wiltschko 1\\n1Google Research, Brain Team\\n2Arizona State University\\n3Department of Chemistry, University of Toronto\\n3Department of Computer Science, University of Toronto\\n3Vector Institute for Artiﬁcial Intelligence, Toronto, Ontario, Canada\\n3 Canadian Institute for Advanced Research, Toronto, Ontario, Canada\\n*Contributed equally\\nAbstract\\nIn machine learning for domains in physics and chemistry, an important prob-\\nlem is learning meaningful and useful representations of physical phenomena that\\nare both predictive and generalize to new tasks. One such problem is predicting\\nquantitative structure-odor relationships (QSOR). A solution would impact hu-\\nman nutrition, manufacture of synthetic fragrance, the environment, and sensory\\nneuroscience. In this paper, we propose the use of graph neural networks for this\\ndecades-old task, and show results that signiﬁcantly out-perform prior methods on\\na novel data set labeled by olfactory experts. Additional analysis shows that the\\nlearned embeddings from graph neural networks capture a meaningful represen-\\ntation of the underlying relationship between structure and odor, as demonstrated\\nby strong performance on two challenging transfer learning tasks. Based on these\\nearly results with graph neural networks for molecular properties, we hope the\\nﬁeld can build towards doing for olfaction what machine learning has already\\ndone for vision.\\n1 Introduction\\nPredicting analytical properties of molecules is an area of growing research in machine learning, par-\\nticularly as models for learning from graph-valued inputs improve in sophistication and robustness.\\nA molecular property prediction problem that has received comparatively little attention during this\\nsurge in research activity is the prediction of Quantitative Structure-Odor Relationships (QSOR) 1.\\nThis is a 70+ year old problem straddling chemistry, physics, sensory neuroscience and machine\\nlearning. This has remained an open problem for so long due to its difﬁculty—very small changes\\nin a molecule’s structure can have dramatic effects on its odor, a phenomenon known in medicinal\\nchemistry as an “activity cliff” ( 1; 2). A classic example is Lyral, which is a commercially suc-\\ncessful muguet molecule (a ﬂoral scent often used in dryer sheets). Its structural neighbors are not\\nalways perceptual neighbors, and vice versa (Figure 1).\\nAdvances in deep learning for vision and audition suggest that we might be able to directly pre-\\ndict the end sensory result of an input stimulus, even without detailed knowledge of the systems\\nand circuits linking the physical world and our internal sensations. Advances in deep learning for\\nolfaction would aid in the discovery of new synthetic odorants, thereby reducing ecological im-\\npact of harvesting natural products, and would advance our understanding of sensory perception in\\n1As opposed to Quantitative Structure-Activity Relationships (QSAR), a term from medicinal chemistry\\nSecond Workshop on Machine Learning and the Physical Sciences (NeurIPS 2019), Vancouver, Canada.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='e917597e-d666-4558-b761-24fb414ceea1', embedding=None, metadata={'page_label': '2', 'file_name': 'NeurIPS_ML4PS_2019_39.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\NeurIPS_ML4PS_2019_39.pdf', 'file_type': 'application/pdf', 'file_size': 702741, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 1: Structurally similar molecules do not necessarily have similar odor descriptors. A.\\nLyral, the reference molecule. B. Molecules with similar structure can share similar odor descriptors.\\nC. However, a small structural change can render the molecule odorless. D. Further, large structural\\nchanges can leave the odor of the molecule largely unchanged. Example from Ohloff, Pickenhagen\\nand Kraft ( 5).\\nthe brain by offering new ways of representing and processing olfactory data. Here, we curated a\\ndataset of molecules associated with expert-labeled odor descriptors 2, and train Graph Neural Net-\\nworks (GNN) to predict these odor descriptors using a molecule’s graph structure alone. Further,\\nwe show that the embeddings learned to solve this task generalize well to downstream tasks, a rare\\noccurrence in machine learning applications in chemistry ( 3; 4). This indicates we have captured a\\ngeneral-purpose representation of a molecule’s odor properties, which might be useful in the future\\nfor rational molecular design or screening.\\n2 Olfactory Dataset\\nWe assembled an expert-labeled set of n = 5030molecules from two separate sources: GoodScents\\nperfume materials database, n = 3786(6) and Lefﬁngwell PMP 2001 database, n = 3561(7). The\\ndatasets share 2317 overlapping molecules. Molecules are labeled with one or more odor descrip-\\ntors by olfactory experts (usually a practicing perfumer), creating a multi-label prediction problem.\\nGoodScents describes a list of 1–15 odor descriptors for each molecule (Figure 2A), whereas Leff-\\ningwell uses free-form text. Odor descriptors were canonicalized using GoodScents’ ontology, and\\noverlapping molecules inherited the union of both datasets’ odor descriptors. After ﬁltering for odor\\ndescriptors with at least 30 representative molecules, 138 odor descriptors remained, including an\\nodorless descriptor (Figure 2B). Some odor descriptors are extremely common, like fruity or green,\\nwhile others are rare, like radish or bready. This bias may be due to our dataset being a collection of\\nmaterials for perfumery and on the degree of speciﬁcity of the odor descriptors. Note that there is an\\nextremely strong co-occurrence structure among odor descriptors that reﬂects a common-sense intu-\\nition of which odor descriptors are similar and dissimilar (Figure 2C). For example, there is a dairy\\ncluster that includes the dairy, yogurt, milk, and cheese descriptors. There is also a fruity cluster\\nwith apple, pear, pineapple, pear, etc., and a bakery cluster that includes toasted, nutty, and cocoa.\\nPrevious approaches in QSOR often train one model per odor descriptor. Here, we apply GNNs to\\nall odor descriptor tasks at once, allowing us to take advantage of this correlation structure.\\n3 Results\\n3.1 Classiﬁcation Performance Comparison\\nWe consider two types of GNNs: Message Passing Neural Networks (MPNN) ( 8) and Graph Con-\\nvolution Networks (GCN) ( 9), and compare against baselines of random forests (RF) and k-nearest\\nneighbors (kNN). For baseline featurizations, we used bit-based RDKitFingerprints (bFP), count-\\nbased Morgan ﬁngerprints (cFP), and Mordred features ( 10). We report several metrics (Table 1),\\nas each metric can highlight different performance characteristics, however we primarily compare\\nmodels on mean AUROC (averaged across odor descriptors). We found that MPNNs and GCNs\\nperform similarly. Both MPNNs and GCNs signiﬁcantly outperform all baseline models.\\n2In QSAR terminology, descriptors are used for input features in a model. The term odor descriptors is\\nused in QSOR for odorant properties to predict.\\n2', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='089afc38-673b-4e54-b7e3-aa1924411500', embedding=None, metadata={'page_label': '3', 'file_name': 'NeurIPS_ML4PS_2019_39.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\NeurIPS_ML4PS_2019_39.pdf', 'file_type': 'application/pdf', 'file_size': 702741, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 2: Dataset overview. A. Distribution of odor descriptor frequencies. B. Distribution of label\\ndensity. C. Co-ocurrence matrix for odor descriptors. The 10 most frequent descriptors are removed\\nfor visual clarity, and remaining descriptors re-ordered using spectral clustering. Main odor groups\\nwith examples are highlighted.\\nAUROC Precision Recall F1\\nMPNN 0.890 [0.882, 0.898] 0.379 [0.352, 0.399] 0.387 [0.366, 0.408] 0.362 [0.335, 0.375]\\nGCN 0.894 [0.888, 0.902] 0.379 [0.351, 0.398] 0.390 [0.365, 0.412] 0.360 [0.337, 0.372]\\nRF-Mordred 0.850 [0.838, 0.860] 0.311 [0.288, 0.333] 0.393 [0.372, 0.417] 0.306 [0.283, 0.319]\\nRF-bFP 0.832 [0.821, 0.842] 0.321 [0.293, 0.339] 0.371 [0.350, 0.390] 0.295 [0.272, 0.308]\\nRF-cFP 0.845 [0.835, 0.854] 0.315 [0.280, 0.332] 0.375 [0.354, 0.398] 0.295 [0.272, 0.311]\\nKNN-bFP 0.791 [0.778, 0.803] 0.328 [0.305, 0.347] 0.390 [0.366, 0.411] 0.323 [0.299, 0.335]\\nKNN-cFP 0.796 [0.785, 0.809] 0.333 [0.307, 0.351] 0.365 [0.342, 0.389] 0.316 [0.292, 0.327]\\nTable 1: Odor descriptor prediction results. mean, 95% CI [lower, upper] bootstrap bounds\\nreported. Numbers reported are an unweighted mean across all 138 odor descriptors. Precision/recall\\ndecision thresholds are optimized for F1 score. Best values for each metric are in bold; recall had\\nno clear winner.\\n3.2 Evaluating Embedding Performance in a Lookup Task\\nWe wished to evaluate whether embeddings extracted from our trained GCN space reﬂected odor per-\\nceptual space. Speciﬁcally, we compare whether molecules with small cosine distances in our GCN\\nembeddings were perceptually similar, as compared to using Tanimoto distances on bFP features.\\nWhile the latter strategy is commonly used in cheminformatics for molecule retrieval, molecules\\nwith similar structural features do not always smell the same (Figure 1), so we anticipated that it\\nwould not perform as well at retrieving similar-smelling molecules.\\nTo test this idea, we trained a k-nearest neighbors classiﬁer ( k = 20) using cosine distance on GCN\\nembeddings, and using Tanimoto distance on bit-based Morgan ﬁngerprints. GCN embeddings\\n(AUROC = 0.818, 95% CI [0.806, 0.830] ) outperformed bFP (AUROC = 0.782, 95% CI [0.773,\\n0.797]). By inspecting the nearest neighbors found by each method (Figure 3), we can see that\\nboth methods yield molecules with similar structural features, but retrieval using GCN embeddings\\nproduces molecules that are more perceptually similar to the source molecule. We conclude that\\nrelative to bit-based ﬁngerprints, GCN embeddings emphasize similarity of smell over similarity in\\nstructure.\\n3', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='016f4502-e06e-456a-9c68-667b6b0653d5', embedding=None, metadata={'page_label': '4', 'file_name': 'NeurIPS_ML4PS_2019_39.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\NeurIPS_ML4PS_2019_39.pdf', 'file_type': 'application/pdf', 'file_size': 702741, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 3: Nearest neighbor retrieval. k=5 nearest neighbors are shown for cosine similarity on\\nGCN embeddings and for Tanimoto distance on bit-based Morgan ﬁngerprints. Reported AUROCs\\nare averaged over all odor descriptors, with k = 20.\\n4 Evaluating Generalization Performance of Odor Embeddings\\nWe have shown that GNNs signiﬁcantly outperform existing methods on predicting odor descriptors,\\nand that our embedding space has useful local structure that can be used to search for similarly-\\nsmelling molecules. We now explore whether these odor embeddings generalize to other odor pre-\\ndiction tasks.\\n4.1 Transfer Learning to Previously-Unseen Odor Descriptors\\nAn odor descriptor may be newly invented or reﬁned (e.g., molecules with the pear descriptor might\\nbe later attributed a more speciﬁc pear skin, pear stem, pear ﬂesh, pear core descriptor). A useful\\nodor embedding would be able to perform transfer learning ( 11) to this new descriptor, using only\\nlimited data. To approximate this scenario, we ablated one odor descriptor at a time from our dataset.\\nUsing the embeddings trained from (N \\x001) odor descriptors as a featurization, we trained a random\\nforest to predict the previously held-out odor descriptor. We used cFP and Mordred features as\\na baseline for comparison. The results are shown in Figure 4. GNN embeddings signiﬁcantly\\noutperform Morgan ﬁngerprints and Mordred features on this task, but as expected, still perform\\nslightly worse than a GNN trained on the target odor. This indicates that GNN-based embeddings\\nmay generalize to predict new, but related, odors.\\nFigure 4: Mean AUROC on a previously held-out odor. Average AUROC scores across all labels\\non the single label ablation task. The error bars denote 95% conﬁdence intervals. The top bar denotes\\nthe performance of the model trained on all of the labels. The middle bar denotes the performance\\nof a random forest model trained using the GNN embeddings from a model trained on (N \\x001) odor\\nlabels. The bottom bar denotes a random forest trained on counting Morgan ﬁngerprints.\\n4.2 Transfer Learning to the DREAM Olfactory Prediction Challenge\\nThe DREAM Olfaction Prediction Challenge ( 12) was an open competition to build QSOR models\\non a dataset collected from untrained panelists. The DREAM dataset has several differences from\\n4', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='e70ca98c-b929-4cd5-bce4-b3f1295fda5c', embedding=None, metadata={'page_label': '5', 'file_name': 'NeurIPS_ML4PS_2019_39.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\NeurIPS_ML4PS_2019_39.pdf', 'file_type': 'application/pdf', 'file_size': 702741, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='our own. First, it was a regression problem – panelists rated the amount that a molecule smelled of\\na particular odor descriptor on a scale from 1 to 100. Second, it had 476 molecules compared to\\nour \\x18 5k (although our dataset contains nearly all of the DREAM molecules). Third, the ratings\\nwere provided by a large panel of untrained individuals over a short period of time, whereas ours\\nwere gleaned from a small set of experts over many years. The DREAM challenge measured model\\nperformance as the Pearson’s r correlation of model predictions with the mean reported intensity of\\neach odor descriptor.\\nThe winning DREAM model used random forest models with a combination of several sources of\\nfeatures, primarily Dragon and Morgan ﬁngerprints, among other sources of information ( 12). Using\\nonly our embedding with a random forest model, we achieve a mean Pearson’s r = 0:55, and the\\nstate-of-the-art model described above achieved a mean Pearson’s r = 0:54. While we can have\\nbetter average performance in 13 tasks, after taking into account conﬁdence intervals, we ﬁnd the\\nperformance is indistinguishable between the two models for both r and R2 regression scores (data\\nnot shown).\\nOverall, this indicates that our QSOR modeling approach can generalize to adjacent perceptual\\ntasks, and captures meaningful and useful structure about human olfactory perception, even when\\nmeasured in different contexts, with different methodologies.\\n5 Conclusion\\nWe assembled a novel and large dataset of expertly-labeled single-molecule odorants, and trained\\na graph neural network to predict the relationship between a molecules structure and its smell. We\\ndemonstrated state-of-the-art results on this QSOR task with respect to ﬁeld-recognized baselines.\\nFurther, we showed that the embeddings capture meaningful structure on both a local and global\\nscale. Finally, we showed that the embeddings learned by our model are useful in downstream tasks,\\nwhich is currently a rare property of modern machine learning models and data in chemistry. Thus,\\nwe believe our model and its learned embeddings might be generally useful in the rational design of\\nnew odorants. For an expanded version of this preprint, please see ( 13).\\n6 Acknowledgments\\nWe thank D. Sculley, Steven Kearnes, Jasper Snoek, Emily Reif, Carey Radebaugh, David Belanger,\\nJoel Mainland and Emily Mayhew for support, suggestions and useful discussions on the manuscript.\\nWe thank Max Bileschi and Yoni Halpern for their technical guidance. We thank Aniket Zinzuwadia\\nfor discussion and who did preliminary work on the DREAM and GoodScents datasets for his senior\\nthesis. We thank Bill Luebke of GoodScents and John Lefﬁngwell of Lefﬁngwell and Associates for\\ntheir generosity in sharing their data for research use. We thank all of our colleagues in the Google\\nBrain Cambridge ofﬁce for creating and maintaining such a supportive and stimulating environment.\\nReferences\\n[1] C S Sell. On the unpredictability of odor. Angewandte Chemie International Edition ,\\n45(38):6254–6261, 2006.\\n[2] Dagmar Stumpfe and Jürgen Bajorath. Exploring activity cliffs in medicinal chemistry. J. Med.\\nChem., 55(7):2932–2942, April 2012.\\n[3] Han Altae-Tran, Bharath Ramsundar, Aneesh S Pappu, and Vijay Pande. Low data drug dis-\\ncovery with One-Shot learning. ACS Cent Sci , 3(4):283–293, April 2017.\\n[4] Clyde Fare, Lukas Turcani, and Edward O Pyzer-Knapp. Powerful, transferable representations\\nfor molecules through intelligent task selection in deep multitask networks. September 2018,\\ndoi:1809.06334.\\n[5] Günther Ohloff, Wilhelm Pickenhagen, and Philip Kraft. Scent and Chemistry . Wiley, January\\n2012.\\n[6] The good scents company - ﬂavor, fragrance, food and cosmetics ingredients information.\\nhttp://www.thegoodscentscompany.com/. Accessed: 2019-9-4.\\n5', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='81e1279f-c119-48b3-af7d-7039c4b4a103', embedding=None, metadata={'page_label': '6', 'file_name': 'NeurIPS_ML4PS_2019_39.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\NeurIPS_ML4PS_2019_39.pdf', 'file_type': 'application/pdf', 'file_size': 702741, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='[7] John C Lefﬁngwell. Lefﬁngwell & associates, 2005.\\n[8] Justin Gilmer, Samuel S Schoenholz, Patrick F Riley, Oriol Vinyals, and George E Dahl. Neu-\\nral message passing for quantum chemistry. In Doina Precup and Yee Whye Teh, editors,\\nProceedings of the 34th International Conference on Machine Learning , volume 70 of Pro-\\nceedings of Machine Learning Research , pages 1263–1272, International Convention Centre,\\nSydney, Australia, 2017. PMLR.\\n[9] David K Duvenaud, Dougal Maclaurin, Jorge Iparraguirre, Rafael Gómez-Bombarell, Tim-\\nothy Hirzel, Alán Aspuru-Guzik, and Ryan P Adams. Convolutional Networks on Graphs\\nfor Learning Molecular Fingerprints. In Advances in Neural Information Processing Systems ,\\npages 2215–2223, 2015.\\n[10] Hirotomo Moriwaki, Yu-Shi Tian, Norihito Kawashita, and Tatsuya Takagi. Mordred: a molec-\\nular descriptor calculator. J. Cheminform., 10(1):4, February 2018.\\n[11] S J Pan and Q Yang. A survey on transfer learning. IEEE Trans. Knowl. Data Eng. ,\\n22(10):1345–1359, October 2010.\\n[12] Andreas Keller, Richard C Gerkin, Yuanfang Guan, Amit Dhurandhar, Gabor Turu, Bence\\nSzalai, Joel D Mainland, Yusuke Ihara, Chung Wen Yu, Russ Wolﬁnger, Celine Vens, Leander\\nSchietgat, Kurt De Grave, Raquel Norel, DREAM Olfaction Prediction Consortium, Gustavo\\nStolovitzky, Guillermo A Cecchi, Leslie B V osshall, and Pablo Meyer. Predicting human\\nolfactory perception from chemical features of odor molecules. Science, 355(6327):820–826,\\nFebruary 2017.\\n[13] Benjamin Sanchez-Lengeling, Jennifer N Wei, Brian K Lee, Richard C Gerkin, Alán Aspuru-\\nGuzik, and Alexander B Wiltschko. Machine learning for scent: Learning generalizable per-\\nceptual representations of small molecules. October 2019, doi:1910.10685.\\n6', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='6eb061b7-272e-48d2-9c42-dbfae9efaded', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nfrom index import create_index, create_llm\\n\\n\\nindex = create_index()\\nllm = create_llm()\\n\\nindex, llm\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='f923c48d-2138-4475-8856-c201cf38baa2', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\n# Load model directly\\nfrom transformers import AutoTokenizer, AutoModelForCausalLM\\n\\ntokenizer = AutoTokenizer.from_pretrained(\"Qwen/Qwen2-7B-Instruct\")\\nmodel = AutoModelForCausalLM.from_pretrained(\"Qwen/Qwen2-7B-Instruct\")\\n\\n\\n# #### as query engine\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='e640faa6-0ada-4ce2-af4c-646b7a0ec980', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(llm=llm)\\nresponse = query_engine.query(\"Kto prowadzi rejestr obywateli Ukrainy?\")\\nresponse\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='dbc8f384-91ca-45cd-a03b-e5b1fa7e3df7', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(f\"Typ:  {type(response)}\")\\nprint(f\"Tekst:  {response.response}\")\\nprint(f\"Źródło:  {response.get_formatted_sources()}\")\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='28a69da6-b2ed-478a-ad30-1a8918aad64e', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(llm=llm, streaming=True)\\nstreaming_response = query_engine.query(\"Kto prowadzi rejestr obywateli Ukrainy?\")\\nstreaming_response.print_response_stream()\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='6b6f8abd-8787-4a7a-8122-e9d298fe0893', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(streaming_response.response_gen)\\nprint(streaming_response.get_formatted_sources())\\nprint(streaming_response.get_response())\\nprint(streaming_response.print_response_stream())\\n\\n\\n# Response modes\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='10f917e8-cf77-46c4-9e95-b493d41cf893', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"compact\",\\n)\\nresponse = query_engine.query(\"O czym mówi ustawa?\")\\nprint(response)\\n\\n\\n# #### \\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='eb1d04d8-10c7-4613-9d6c-7a06c93c7cb9', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"refine\",\\n)\\nresponse = query_engine.query(\"O czym mówi ustawa?\")\\nprint(response)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='0c68f780-84bf-41b2-81d7-5d32ae7d1d40', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"tree_summarize\",\\n)\\nresponse = query_engine.query(\"O czym mówi ustawa?\")\\nprint(response)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='0ad92229-0c6f-4e91-a67f-0a34f30f7726', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"accumulate\",\\n)\\nresponse = query_engine.query(\"O czym mówi ustawa?\")\\nprint(response)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='0da91305-78e1-4cca-b355-3f08d546384d', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"simple_summarize\",\\n)\\nresponse = query_engine.query(\"O czym mówi ustawa?\")\\nprint(response)\\n\\n\\n# Nie widze aby sposob wyswietlania mial wieksze znaczenie niz losowosc odpowiedzi\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='0b967200-ad29-47c4-9e73-7b359f728aba', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"simple_summarize\",\\n)\\nresponse = query_engine.query(\"O czym mówi ustawa?\")\\nprint(response)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='1c072101-568c-4075-bb09-53ba057af411', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(\"\\\\nŹródła:\")\\nfor node in response.source_nodes:\\n    print(f\"  * Dokument: {node.node.ref_doc_id}\\\\n\")\\n    print(f\"    Fragment tekstu: {node.node.get_content()}\\\\n\")\\n    print(f\"    Podobieństwo: {node.score}\\\\n\")\\n    print(\"-\" * 40)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='62768b7c-13db-450f-9536-08a01e11dbba', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"tree_summarize\",\\n)\\nresponse = query_engine.query(\"Czego dotyczy ustawa o pomocy obywatelom Ukrainy?\")\\nprint(response)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='a42e60eb-8c22-4bb7-bc37-a3fe7c56bb4c', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(\"\\\\nŹródła:\")\\nfor node in response.source_nodes:\\n    print(f\"  * Dokument: {node.node.ref_doc_id}\\\\n\")\\n    print(f\"    Fragment tekstu: {node.node.get_content()}\\\\n\")\\n    print(f\"    Podobieństwo: {node.score}\\\\n\")\\n    print(\"-\" * 40)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='c98e9996-f048-49c0-a09c-39f929844aab', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nquery_engine = index.as_query_engine(\\n    llm=llm,\\n    response_mode=\"tree\",\\n)\\nresponse = query_engine.query(\"Od kogo nie można pobierać odcisków palców?\")\\nprint(response)\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='d45088c4-dab8-4892-97bf-bc032c4ae26c', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\nprint(\"\\\\nŹródła:\")\\nfor node in response.source_nodes:\\n    print(f\"  * Dokument: {node.node.ref_doc_id}\\\\n\")\\n    print(f\"    Fragment tekstu: {node.node.get_content()}\\\\n\")\\n    print(f\"    Podobieństwo: {node.score}\\\\n\")\\n    print(\"-\" * 40)\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='64b2ec3d-8074-4208-b5a9-829a63221a9d', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\n# Wyświetlanie wszystkich elementów obiektu response\\nfor attr_name in dir(response):\\n    if not attr_name.startswith(\\'__\\'):\\n        attr_value = getattr(response, attr_name)\\n        print(f\"{attr_name}: {attr_value}\\\\n\")\\n\\n\\n# In[ ]:\\n\\n\\n\\n\\n\\n# ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='cb83e862-baa6-4639-8c3d-1dc1d393f0ba', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\query_engine.ipynb', 'file_name': 'query_engine.ipynb', 'file_size': 54285, 'creation_date': '2025-04-29', 'last_modified_date': '2025-04-09'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\n\\n\\n# Wyświetlanie odpowiedzi\\nprint(\"Odpowiedź:\")\\nprint(response.response)\\n\\n# Wyświetlanie metadanych źródeł\\nprint(\"\\\\nMetadane źródeł:\")\\nfor node in response.source_nodes:\\n    print(f\"ID węzła: {node.node.id_}\")\\n    print(f\"Strona: {node.metadata[\\'page_label\\']}\")\\n    print(f\" Nazwa pliku: {node.metadata[\\'file_name\\']}\")\\n    print(f\"Ścieżka pliku: {node.metadata[\\'file_path\\']}\")\\n    print(f\"Typ pliku: {node.metadata[\\'file_type\\']}\")\\n    print(f\"Rozmiar pliku: {node.metadata[\\'file_size\\']} bajtów\")\\n    print(f\"Data utworzenia: {node.metadata[\\'creation_date\\']}\")\\n    print(f\"Data modyfikacji: {node.metadata[\\'last_modified_date\\']}\\\\n\")\\n\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
       " Document(id_='0df615c5-9104-453d-861a-9c445faa5c37', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\terms.txt', 'file_name': 'terms.txt', 'file_type': 'text/plain', 'file_size': 166, 'creation_date': '2025-04-24', 'last_modified_date': '2025-04-24'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='olfaction\\r\\nPrincipal Odor Map (POM)\\r\\nMessage Passing Neural Network (MPNN) -> specific type of graph neural network (GNN)\\r\\nGCMS (Gas Chromatography-Mass Spectrometry)', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}')]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from llama_index.core import SimpleDirectoryReader\n",
    "\n",
    "documents = SimpleDirectoryReader(r\"C:\\Users\\mikoo\\AI\\osmo\").load_data()\n",
    "documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Document(id_='ada63541-a30f-404a-995c-a2a3f90efe59', embedding=None, metadata={'page_label': '1', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='A Principal Odor Map Uni\\x00es Diverse Tasks in Human Olfactory Perception\\nBrian K. Lee1†, Emily J. Mayhew2†, Benjamin Sanchez-Lengeling1, Jennifer N. Wei1, Wesley W.\\nQian1,3, Kelsie Li\\x00le4, Ma\\x00hew Andres4, Britney B. Nguyen4, Theresa Moloy4, Jane K. Parker5,\\nRichard C. Gerkin1,6, Joel D. Mainland4,7,*, Alexander B. Wiltschko1,*\\n1Google Research, Brain Team; Cambridge, MA, USA.\\n2Depa\\x00ment of Food Science and Human Nutrition, Michigan State University; East Lansing, MI, USA.\\n3Depa\\x00ment of Computer Science, University of Illinois; Urbana-Champaign, IL, USA.\\n4Monell Chemical Senses Center; Philadelphia, PA, USA.\\n5Depa\\x00ment of Food and Nutritional Sciences, University of Reading; Reading, Berkshire, UK.\\n6School of Life Sciences, Arizona State University; Tempe, AZ, USA.\\n7Depa\\x00ment of Neuroscience, University of Pennsylvania; Philadelphia, PA, USA.\\n†These authors contributed equally.\\n*Corresponding authors. Email: jmainland@monell.org, alex.bw@googlemail.com\\nAbstract\\nMapping molecular structure to odor perception is a key challenge in olfaction. Here, we use\\ngraph neural networks (GNN) to generate a Principal Odor Map (POM) that preserves\\nperceptual relationships and enables odor quality prediction for novel odorants. The model is\\nas reliable as a human in describing odor quality: on a prospective validation set of 400 novel\\nodorants, the model-generated odor pro\\x00le more closely matched the trained panel mean\\n(n=15) than did the median panelist. Applying simple, interpretable, theoretically-rooted\\ntransformations, the POM outpe\\x00ormed chemoinformatic models on several other odor\\nprediction tasks, indicating that the POM successfully encoded a generalized map of\\nstructure-odor relationships. This approach broadly enables odor prediction and paves the\\nway toward digitizing odors.\\n(119 words)\\nOne-Sentence Summary\\nAn odor map achieves human-level odor description pe\\x00ormance and generalizes to diverse\\nodor-prediction tasks.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "documents[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:root:Removing unpickleable private attribute _chunking_tokenizer_fn\n",
      "WARNING:root:Removing unpickleable private attribute _split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _sub_sentence_split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _chunking_tokenizer_fn\n",
      "WARNING:root:Removing unpickleable private attribute _split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _sub_sentence_split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _chunking_tokenizer_fn\n",
      "WARNING:root:Removing unpickleable private attribute _split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _sub_sentence_split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _chunking_tokenizer_fn\n",
      "WARNING:root:Removing unpickleable private attribute _split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _sub_sentence_split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _chunking_tokenizer_fn\n",
      "WARNING:root:Removing unpickleable private attribute _split_fns\n",
      "WARNING:root:Removing unpickleable private attribute _sub_sentence_split_fns\n"
     ]
    }
   ],
   "source": [
    "import chromadb\n",
    "from llama_index.vector_stores.chroma import ChromaVectorStore\n",
    "from llama_index.embeddings.huggingface import HuggingFaceEmbedding\n",
    "from llama_index.core.node_parser import SentenceSplitter\n",
    "from llama_index.core.ingestion import IngestionPipeline\n",
    "\n",
    "db = chromadb.PersistentClient(path=\"./OSMO\")\n",
    "chroma_collection = db.get_or_create_collection(\"first_batch\")\n",
    "vector_store = ChromaVectorStore(chroma_collection=chroma_collection)\n",
    "\n",
    "pipeline = IngestionPipeline(\n",
    "    transformations=[\n",
    "        SentenceSplitter(),\n",
    "        HuggingFaceEmbedding(model_name=\"BAAI/bge-small-en-v1.5\"),\n",
    "        # HuggingFaceEmbedding(model_name=\"OrlikB/KartonBERT-USE-base-v1\"),\n",
    "    ],\n",
    "    vector_store=vector_store,\n",
    ")\n",
    "\n",
    "nodes = await pipeline.arun(\n",
    "    documents=documents,\n",
    "    num_workers=4,  # dopasuj do liczby rdzeni CPU\n",
    "    show_progress=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index.core import VectorStoreIndex\n",
    "\n",
    "embed_model = HuggingFaceEmbedding(model_name=\"BAAI/bge-small-en-v1.5\")\n",
    "index = VectorStoreIndex.from_vector_store(vector_store, embed_model=embed_model)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\mikoo\\miniconda3\\envs\\primo\\Lib\\site-packages\\pydantic\\_internal\\_fields.py:132: UserWarning: Field \"model_name\" in HuggingFaceInferenceAPI has conflict with protected namespace \"model_\".\n",
      "\n",
      "You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Response(response='Zakaźenie szpitalne (hospital-acquired infection) refers to infections that patients acquire while they are in a hospital or other healthcare facility. This includes both infections that occur during their stay and those that develop soon after discharge. These infections can be caused by various microorganisms such as bacteria, viruses, fungi, or parasites. They pose significant challenges for healthcare systems due to the complexity of managing multiple co-existing health conditions and the potential complications they may cause.', source_nodes=[NodeWithScore(node=TextNode(id_='163a8e43-a080-4d1e-8d94-aad60bd9d777', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='d21bdd56-6278-4f7b-b6e0-11b90dcaccc7', node_type='4', metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='ccdadbf26b1b25a09cf81d4523500bf3c6b8e13b8edef8d7cddb9b4d4a3b8e75'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='846124fc-fc0f-4fb7-a4a0-0af1e61021b3', node_type='1', metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='055b547def306db012f8faed136e7f110ee7841ffce230122a4c5cc6a1af3dda'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='0d4ddcee-eb16-498a-8364-7f240f015949', node_type='1', metadata={}, hash='f301ca0ed8b4ffc17c7873edb69931c511b732d5e7af85423cd2ccb556ad4ee5')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='GR94839_E, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(N(=O)=O)cc1\\n522, GR94839_F, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](C2)O)=O)ccc(OC)cc1\\n523, carphenazine, 1, CCC(=O)c1ccc2Sc3ccccc3N(CCCN4CCN(CCO)CC4)c2c1\\n524, cartazolate, 1, CCCCNc1c(cnc2n(CC)ncc12)C(=O)OCC\\n525, centazolone, 1, NN1C=Nc2cc3ccccc3cc2C1=O\\n526, chloralhydrate, 1, OC(O)C(Cl)(Cl)Cl\\n527, chlordiazepoxide, 1, CN=C1CN(O)C(=C2C=C(Cl)C=CC2=N1)c3ccccc3\\n528, chlorpromazine, 1, CN(C)CCCN1c2ccccc2Sc3ccc(Cl)cc13\\n529, chlorprothixene, 1, CN(C)CC\\\\C=C/1c2ccccc2Sc3ccc(Cl)cc13\\n530, chlorzoxazone, 1, Clc1ccc2OC(=O)Nc2c1\\n531, choline, 1, C[N+](C)(C)CCO\\n532, clobazam, 1, CN1C(=O)CC(=O)N(c2ccccc2)c3cc(Cl)ccc13\\n533, RO19-4603, 1, CN1Cc2n(cnc2C(=O)OC(C)(C)C)c3ccsc3C1=O\\n534, SKF101468, 1, C1C(Nc2c1c(ccc2)CCN(CCC)CCC)=O\\n535, SKF89124, 1, C1C(Nc2c1c(ccc2O)CCN(CCC)CCC)=O\\n536, Salicyluricacid, 1, OC(=O)CNC(=O)c1ccccc1O\\n537, Sulforidazine, 1, CN1CCCCC1CCN2c3ccccc3Sc4ccc(cc24)[S](C)(=O)=O\\n538, Teflurane, 1, FC(Br)C(F)(F)F\\n539, Theobromine, 1, Cn1cnc2N(C)C(=O)NC(=O)c12\\n540, Thiopental, 0, CCCC(C)C1(CC)C(=O)NC(=S)NC1=O\\n541, Toluene, 1, Cc1ccccc1\\n542, Triazolam, 1, Cc1nnc2CN=C(c3ccccc3Cl)c4cc(Cl)ccc4n12\\n543, GR94839_G, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)cc(ccc1)N(=O)=O\\n544, GR94839_H, 0, c1(ccccc1)CC(N1[C@H](CN(CC1)C(=O)C)C[N@]1CC[C@H](C1)O)=O\\n545, GR94839_I, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc([S@](=O)C)cc1\\n546, GR94839_L, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(S(=O)(=O)C)cc1\\n547, ICI197067, 1, CC(C)[C@@H](CN1CCCC1)N(C)C(=O)Cc2ccc(Cl)c(Cl)c2\\n548, ICI199441, 1,', mimetype='text/plain', start_char_idx=34804, end_char_idx=36401, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.36787948502609175), NodeWithScore(node=TextNode(id_='348f4513-2db0-47e4-b8ae-21baab136545', embedding=None, metadata={'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='f8953176-3f59-4eeb-8972-1c08db87cfa3', node_type='4', metadata={'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='6434e965d0a4ea2cd0fe7c7d8043a97c008433192254e5d76a1d2cf0a0108550')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='bounded above by panel test-retest correlation (Fig. S13). When we disaggregate by chemical\\nclasses (e.g. esters, phenols, amines), both panelist and model pe\\x00ormance is relatively\\nuniform (Fig. 3C), with sulfur-containing molecules showing strongest pe\\x00ormance from\\npanelists and the model (R = 0.52).\\nChemical materials are impure - a fact too o\\x00en unaccounted for in olfactory research(24). To\\nmeasure the contribution of impurities to the odor percept of our stimuli, we applied a gas\\nchromatography-mass spectrometry (GC-MS and gas chromatography-olfactometry\\n(GC-O) quality control (QC) procedure to 50 stimuli (Data S1). This QC procedure matches an\\nodor percept to its causal molecule, allowing us to identify stimuli for which the primary odor\\ncharacter was not due to the nominal compound. Our QC led to diverse conclusions: the\\nnominal compound caused the odor (12/50), the nominal compound and contaminants\\ncontribute to the odor (16/50), contaminants caused the odor (18/50), or the cause of the\\nodor could not be determined (4/50) (Fig. 3D). In some cases, while we purchased a novel\\nodorant, the dominant odorant was not novel; for example, the stimulus\\n4,5-dimethyl-1,3-thiazol-2-amine was described by the panel as bu\\x00ery, sweet, and dairy, but\\nthis odor percept was a\\x00ributed through QC to the contaminant diacetyl, a well-known\\nbu\\x00ery odorant. In another case, the purchased odorant, isobornyl methylacrylate, was\\ndescribed by the panel and the model as both piney and \\x00oral; however, through QC we\\ndetermined that the nominal compound was \\x00oral only and that the piney aroma was due to\\nthe closely related compound, borneol, which was detected as a contaminant in the sample.\\nBased on QC results, we removed 26 molecules known or suspected to have high degrees of\\nodorous contamination (Data S1).\\nThe prevalence of odorous contamination that we found demonstrates that it is not safe to\\nassume that the odor percept of a purchased chemical is due to the nominal compound. The\\nFlavor & Fragrance (F&F) industry is motivated to minimize odorous contaminants for\\ncommercially valued odorants, but there is no such incentive for non-F&F commodity\\nchemicals. We stress the need for caution and diligence in expanding odor stimulus space.\\nImplications of each QC result on model pe\\x00ormance are unique (Data S1). In some cases,\\nthe model pe\\x00ormed well despite the presence of odorous contaminants. We estimate that,\\nif these contaminants were removed from the rated samples, model pe\\x00ormance improves\\nin 6 of 50 scenarios, degrades in another 6 of 50 scenarios, remains neutral in 21 of 50\\nscenarios, and cannot be determined in 17 of 50 scenarios.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint', mimetype='text/plain', start_char_idx=0, end_char_idx=3001, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.36787948502609175)], metadata={'163a8e43-a080-4d1e-8d94-aad60bd9d777': {'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, '348f4513-2db0-47e4-b8ae-21baab136545': {'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}})"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "from llama_index.llms.huggingface_api import HuggingFaceInferenceAPI\n",
    "from llama_index.llms.ollama import Ollama\n",
    "\n",
    "\n",
    "llm = Ollama(model=\"qwen2:7b\")\n",
    "query_engine = index.as_query_engine(\n",
    "    llm=llm,\n",
    "    response_mode=\"tree_summarize\",\n",
    ")\n",
    "query_engine.query(\"Czym jest zakażenie szpitalne?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Response(response='Zakażenie szpitalne to każde zakażenie, które nastąpiło w związku z udzielaniem świadczeń zdrowotnych. To obowiązuje wtedy, gdy choroba nie pozostawała w momencie udzielania świadczeń zdrowotnych w okresie wylęgania albo wystąpiła po udzielaniu świadczeń zdrowotnych, w okresie nie dłuższym niż najdłuższy okres jej wylęgania.', source_nodes=[NodeWithScore(node=TextNode(id_='11804821-1602-4bcb-9caa-938bd113cd11', embedding=None, metadata={'page_label': '1', 'file_name': 'Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\Downloads\\\\index\\\\zakazenia\\\\pdf\\\\Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_type': 'application/pdf', 'file_size': 219359, 'creation_date': '2025-03-24', 'last_modified_date': '2025-03-24'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='1baa6b31-1bbf-4743-8a28-21869f6151ba', node_type='4', metadata={'page_label': '1', 'file_name': 'Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\Downloads\\\\index\\\\zakazenia\\\\pdf\\\\Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_type': 'application/pdf', 'file_size': 219359, 'creation_date': '2025-03-24', 'last_modified_date': '2025-03-24'}, hash='a9cd63b222e477b191ea0063875f648b390a18bd2947114b068ef59a8a978734')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='DEFINICJE ZAKAŻEŃ SZPITALNYCH \\nDEFINICJA OGÓLNA ZAKAŻENIA SZPITALNEGO\\nPRZYJĘTA I OBOWIĄZUJĄCA W WOJEWÓDZKIM SZPITALU ZESPOLONYM W TORUNIU\\nZakażenie szpitalne jest to każde zakażenie, które wystąpiło w związku z udzielaniem świadczeń\\nzdrowotnych, w przypadku gdy choroba:\\na) nie pozostawała w momencie udzielania świadczeń zdrowotnych w okresie wylęgania albo\\nb) wystąpiła po udzielaniu świadczeń zdrowotnych, w okresie nie dłuższym niż najdłuższy okres jej\\nwylęgania.\\n1', mimetype='text/plain', start_char_idx=0, end_char_idx=469, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.7121357094747512), NodeWithScore(node=TextNode(id_='247cfdb1-8787-4398-b948-f214c470c46b', embedding=None, metadata={'page_label': '16', 'file_name': 'Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\Downloads\\\\index\\\\zakazenia\\\\pdf\\\\Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_type': 'application/pdf', 'file_size': 219359, 'creation_date': '2025-03-24', 'last_modified_date': '2025-03-24'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='9c2f5e7a-bf75-429f-add8-7f4563e3a846', node_type='4', metadata={'page_label': '16', 'file_name': 'Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\Downloads\\\\index\\\\zakazenia\\\\pdf\\\\Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_type': 'application/pdf', 'file_size': 219359, 'creation_date': '2025-03-24', 'last_modified_date': '2025-03-24'}, hash='76b3005346569c64f8cf6d4dcca55089e24c3ab66560513727a3b99bcc7e02ca'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='3ed76e09-94e2-48fe-bb8e-c61797afe02b', node_type='1', metadata={'page_label': '16', 'file_name': 'Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\Downloads\\\\index\\\\zakazenia\\\\pdf\\\\Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_type': 'application/pdf', 'file_size': 219359, 'creation_date': '2025-03-24', 'last_modified_date': '2025-03-24'}, hash='b14e40de94eb9253f14806fb5b1c1cd8cda06df7ab6633932ed65bc655c47af8')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='Kliniczna sepsa u dorosłych i dzieci: SYS1CSEP\\n1. U pacjenta występuje co najmniej jeden z poniższych objawów bez innej uchwytnej przyczyny: \\ngorączka (>38° C), hipotensja (ciśnienie skurczowe <90 mm) lub oliguria (20 cm³ (ml)/h).\\n2. Nie wykonano posiewu krwi lub w próbce krwi nie wykryto drobnoustrojów ani ich antygenów \\n3. Brak zakażenia w innym miejscu. \\n4. Lekarz włączył leczenie sepsy.\\nWskazówki dotyczące zgłaszania: \\n1. Kod ten powinien być używany w ostateczności!\\n2. W przypadku CSEP u noworodków, należy używać definicji dla NEO CSEP (patrz niżej).\\n16', mimetype='text/plain', start_char_idx=1955, end_char_idx=2519, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.6953791581101496)], metadata={'11804821-1602-4bcb-9caa-938bd113cd11': {'page_label': '1', 'file_name': 'Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\Downloads\\\\index\\\\zakazenia\\\\pdf\\\\Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_type': 'application/pdf', 'file_size': 219359, 'creation_date': '2025-03-24', 'last_modified_date': '2025-03-24'}, '247cfdb1-8787-4398-b948-f214c470c46b': {'page_label': '16', 'file_name': 'Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\Downloads\\\\index\\\\zakazenia\\\\pdf\\\\Załącznik nr 12 do P-7.5(ZH)-18 definicje.pdf', 'file_type': 'application/pdf', 'file_size': 219359, 'creation_date': '2025-03-24', 'last_modified_date': '2025-03-24'}})"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# #### bez chromy\n",
    "# from index import *\n",
    "# from llama_index.core import Settings\n",
    "# from llama_index.embeddings.huggingface import HuggingFaceEmbedding\n",
    "# from llama_index.core.node_parser import SentenceSplitter\n",
    "\n",
    "# Settings.llm = Ollama(model=\"qwen2:7b\")\n",
    "# Settings.embed_model = HuggingFaceEmbedding(\n",
    "#     model_name=\"BAAI/bge-small-en-v1.5\"\n",
    "# )\n",
    "# documents = SimpleDirectoryReader(r\"C:\\Users\\mikoo\\Downloads\\index\\zakazenia\\pdf\").load_data()\n",
    "# index = VectorStoreIndex.from_documents(documents, transformations=[SentenceSplitter()])\n",
    "# query_engine = index.as_query_engine(response_mode=\"tree_summarize\")\n",
    "\n",
    "# query_engine.query(\"Czym jest zakażenie szpitalne?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Response(response='The document mentions a list of 74 chemically diverse mono-molecular odorants which includes various chemical compounds like drugs, chemicals, and other substances. The exact names of all these odorants are provided in the context information starting from \"GR94839_E\" through to \"ICI199441\". These represent a wide range of chemical classes including but not limited to esters, phenols, amines, etc.', source_nodes=[NodeWithScore(node=TextNode(id_='163a8e43-a080-4d1e-8d94-aad60bd9d777', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='d21bdd56-6278-4f7b-b6e0-11b90dcaccc7', node_type='4', metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='ccdadbf26b1b25a09cf81d4523500bf3c6b8e13b8edef8d7cddb9b4d4a3b8e75'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='846124fc-fc0f-4fb7-a4a0-0af1e61021b3', node_type='1', metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='055b547def306db012f8faed136e7f110ee7841ffce230122a4c5cc6a1af3dda'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='0d4ddcee-eb16-498a-8364-7f240f015949', node_type='1', metadata={}, hash='f301ca0ed8b4ffc17c7873edb69931c511b732d5e7af85423cd2ccb556ad4ee5')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='GR94839_E, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(N(=O)=O)cc1\\n522, GR94839_F, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](C2)O)=O)ccc(OC)cc1\\n523, carphenazine, 1, CCC(=O)c1ccc2Sc3ccccc3N(CCCN4CCN(CCO)CC4)c2c1\\n524, cartazolate, 1, CCCCNc1c(cnc2n(CC)ncc12)C(=O)OCC\\n525, centazolone, 1, NN1C=Nc2cc3ccccc3cc2C1=O\\n526, chloralhydrate, 1, OC(O)C(Cl)(Cl)Cl\\n527, chlordiazepoxide, 1, CN=C1CN(O)C(=C2C=C(Cl)C=CC2=N1)c3ccccc3\\n528, chlorpromazine, 1, CN(C)CCCN1c2ccccc2Sc3ccc(Cl)cc13\\n529, chlorprothixene, 1, CN(C)CC\\\\C=C/1c2ccccc2Sc3ccc(Cl)cc13\\n530, chlorzoxazone, 1, Clc1ccc2OC(=O)Nc2c1\\n531, choline, 1, C[N+](C)(C)CCO\\n532, clobazam, 1, CN1C(=O)CC(=O)N(c2ccccc2)c3cc(Cl)ccc13\\n533, RO19-4603, 1, CN1Cc2n(cnc2C(=O)OC(C)(C)C)c3ccsc3C1=O\\n534, SKF101468, 1, C1C(Nc2c1c(ccc2)CCN(CCC)CCC)=O\\n535, SKF89124, 1, C1C(Nc2c1c(ccc2O)CCN(CCC)CCC)=O\\n536, Salicyluricacid, 1, OC(=O)CNC(=O)c1ccccc1O\\n537, Sulforidazine, 1, CN1CCCCC1CCN2c3ccccc3Sc4ccc(cc24)[S](C)(=O)=O\\n538, Teflurane, 1, FC(Br)C(F)(F)F\\n539, Theobromine, 1, Cn1cnc2N(C)C(=O)NC(=O)c12\\n540, Thiopental, 0, CCCC(C)C1(CC)C(=O)NC(=S)NC1=O\\n541, Toluene, 1, Cc1ccccc1\\n542, Triazolam, 1, Cc1nnc2CN=C(c3ccccc3Cl)c4cc(Cl)ccc4n12\\n543, GR94839_G, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)cc(ccc1)N(=O)=O\\n544, GR94839_H, 0, c1(ccccc1)CC(N1[C@H](CN(CC1)C(=O)C)C[N@]1CC[C@H](C1)O)=O\\n545, GR94839_I, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc([S@](=O)C)cc1\\n546, GR94839_L, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(S(=O)(=O)C)cc1\\n547, ICI197067, 1, CC(C)[C@@H](CN1CCCC1)N(C)C(=O)Cc2ccc(Cl)c(Cl)c2\\n548, ICI199441, 1,', mimetype='text/plain', start_char_idx=34804, end_char_idx=36401, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.36787944117144233), NodeWithScore(node=TextNode(id_='348f4513-2db0-47e4-b8ae-21baab136545', embedding=None, metadata={'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='f8953176-3f59-4eeb-8972-1c08db87cfa3', node_type='4', metadata={'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='6434e965d0a4ea2cd0fe7c7d8043a97c008433192254e5d76a1d2cf0a0108550')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='bounded above by panel test-retest correlation (Fig. S13). When we disaggregate by chemical\\nclasses (e.g. esters, phenols, amines), both panelist and model pe\\x00ormance is relatively\\nuniform (Fig. 3C), with sulfur-containing molecules showing strongest pe\\x00ormance from\\npanelists and the model (R = 0.52).\\nChemical materials are impure - a fact too o\\x00en unaccounted for in olfactory research(24). To\\nmeasure the contribution of impurities to the odor percept of our stimuli, we applied a gas\\nchromatography-mass spectrometry (GC-MS and gas chromatography-olfactometry\\n(GC-O) quality control (QC) procedure to 50 stimuli (Data S1). This QC procedure matches an\\nodor percept to its causal molecule, allowing us to identify stimuli for which the primary odor\\ncharacter was not due to the nominal compound. Our QC led to diverse conclusions: the\\nnominal compound caused the odor (12/50), the nominal compound and contaminants\\ncontribute to the odor (16/50), contaminants caused the odor (18/50), or the cause of the\\nodor could not be determined (4/50) (Fig. 3D). In some cases, while we purchased a novel\\nodorant, the dominant odorant was not novel; for example, the stimulus\\n4,5-dimethyl-1,3-thiazol-2-amine was described by the panel as bu\\x00ery, sweet, and dairy, but\\nthis odor percept was a\\x00ributed through QC to the contaminant diacetyl, a well-known\\nbu\\x00ery odorant. In another case, the purchased odorant, isobornyl methylacrylate, was\\ndescribed by the panel and the model as both piney and \\x00oral; however, through QC we\\ndetermined that the nominal compound was \\x00oral only and that the piney aroma was due to\\nthe closely related compound, borneol, which was detected as a contaminant in the sample.\\nBased on QC results, we removed 26 molecules known or suspected to have high degrees of\\nodorous contamination (Data S1).\\nThe prevalence of odorous contamination that we found demonstrates that it is not safe to\\nassume that the odor percept of a purchased chemical is due to the nominal compound. The\\nFlavor & Fragrance (F&F) industry is motivated to minimize odorous contaminants for\\ncommercially valued odorants, but there is no such incentive for non-F&F commodity\\nchemicals. We stress the need for caution and diligence in expanding odor stimulus space.\\nImplications of each QC result on model pe\\x00ormance are unique (Data S1). In some cases,\\nthe model pe\\x00ormed well despite the presence of odorous contaminants. We estimate that,\\nif these contaminants were removed from the rated samples, model pe\\x00ormance improves\\nin 6 of 50 scenarios, degrades in another 6 of 50 scenarios, remains neutral in 21 of 50\\nscenarios, and cannot be determined in 17 of 50 scenarios.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint', mimetype='text/plain', start_char_idx=0, end_char_idx=3001, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.36787944117144233)], metadata={'163a8e43-a080-4d1e-8d94-aad60bd9d777': {'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, '348f4513-2db0-47e4-b8ae-21baab136545': {'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}})"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_engine.query(\"74 chemically diverse mono-molecular odorants?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Response(response='In the context provided, there is mention of a phenomenon known as \"competitive adsorption\" between two odor components in an odor mixture. This implies that when different odors are present together, they may compete for interaction with the sensing material (like APTS-functionalized graphene). The document states that the distinct binding energies and charge transfer characteristics of these odors can lead to competitive adsorption, where one odor molecule has more chances to approach the functionalized graphene and dominate the signal characteristic in the mixture. This subsequently suppresses the signal from the other odor component.\\n\\nThis highlights an effect where different odors competing for interaction with a sensing element may influence each other\\'s detection, making it challenging to accurately differentiate between them. The document further suggests that machine learning classifiers can help distinguish these odors by analyzing their unique sensing response profiles.', source_nodes=[NodeWithScore(node=TextNode(id_='9f75871c-80f4-4d7d-88dc-ebf936bcf603', embedding=None, metadata={'page_label': '8', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='05752b5b-8e58-4621-98fe-1b171da86d11', node_type='4', metadata={'page_label': '8', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='74fb00fed890caa84212dc3491eaebf36715a843868e58e76389df1bb557570a'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='95d77120-5c6f-4210-a76f-173caae4c9a6', node_type='1', metadata={'page_label': '8', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='4c27bc713b7825e4f2d32df54b0b1ec3fbd92c9b4a1195d38be87c6bf3b91e07')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='Consequently, Euge\\noccupies more space near APTS than 2Phe.\\nOdor molecules interact with APTS-functionalized graphene\\ninvolving two events: odor molecules adsorption on APTS and subse-\\nquently charge transfer between odor molecules and APTS-\\nfunctionalized graphene. Both synergistically yield a characteristic\\nsensing response.51 The charge transfer amount between odor mole-\\ncules and APTS-functionalized graphene is calculated by DFT, as\\nshown inFig. 6(i). Under the same circumstances, odor Euca transfers\\n18.4% more charge to APTS than 2Nona, while Euge transfers 83.8%\\nmore charge to APTS than 2Phe. With both odor adsorption and\\ncharge transfer in mind, odor molecules exhibiting stronger\\nFIG. 6. Odorant molecule interaction with\\nAPTS functionalized graphene. Snapshot\\nof odorant molecules interacting with\\nAPTS-functionalized graphene via molec-\\nular dynamics simulations, (a) Euca, (b)\\n2Nona, (c) Euca and 2Nona odor mixture,\\n(d) Euge, (e) 2Phe, (f) Euge and 2Phe\\nodor mixture. Color code: graphene\\n(gray), Euca (lime), 2Nona (blue), Euge\\n(orange), 2Phe (purple), carbon (cyan),\\nnitrogen (blue), sulfur (yellow), oxygen\\n(red). (g) Center of mass (COM) distance\\nbetween odor molecules and APTS mole-\\ncules in the context of individual odor and\\nbinary odor mixture, respectively. Binding\\nenergy (h) and charge transfer (i) between\\nodorant molecules and APTS from DFT\\ncalculations.\\nApplied Physics Reviews ARTICLE scitation.org/journal/are\\nAppl. Phys. Rev.10, 021406 (2023); doi: 10.1063/5.0132177 10, 021406-7\\nVC Author(s) 2023\\n 02 January 2025 11:54:10', mimetype='text/plain', start_char_idx=2750, end_char_idx=4311, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.3678794399164391), NodeWithScore(node=TextNode(id_='5bbc83ff-94dc-4e9f-937a-5c2eda3a7b82', embedding=None, metadata={'page_label': '9', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='56654426-ac98-444b-bdb4-55037e8aa49d', node_type='4', metadata={'page_label': '9', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='83f62d36a33361ea5d06db011d1a4ffff44b855c8c845471253bf21883032520'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='f1c88a92-eff5-4a6e-9b19-e4693e772f81', node_type='1', metadata={}, hash='c5c1231c3cf392b7791ca0c4398d212aa46718026dc38dffa2066c478897bc4e')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='interaction with sensing element materials embrace more chances to\\napproach the functionalized graphene and dominate the signal charac-\\nteristic in the odor mixture, which subsequently suppresses the signal\\nof the other odor. These results demonstrate that owing to the distinct\\nbinding energies and charge transfer characteristics of odors, competi-\\ntive adsorption to the sensing material occurs between two odor com-\\nponents in the mixture.\\nG. Impact of air and humidity\\nHydrogen bonding interaction induced by humidity plays an\\nimportant role in the gas sensing activities, particularly for graphene-\\nbased chemiresistive-type gas sensors, as it was veriﬁed in our previous\\nwork.24 Although the humidity and presence of air may affect the\\nintrinsic response, i.e., the interaction between the sensor and the tar-\\nget VOC, the effective response induced by such interaction is still dis-\\ntinct and, therefore, detectable by machine learning classiﬁer\\nalgorithms. This could be veriﬁed by performing experiments using air\\nas carrier gas and detecting VOCs and their binary mixture with water\\nvapor. The experiment details, response results, as well as classiﬁcation\\nresults are shown in the supplementary material.\\nFirst, ethanol (ETH) was chosen as a frequently found VOC in\\ngas sensor applications. The sensor was exposed to 48 cycles of pure\\nwater (alcohol by volume 0 or ABV0), pure ethanol (ABV100), and a\\nbinary mixture of 80% ethanol and 20% water (ABV80). The pure eth-\\nanol cluster is very well separated from the other clusters. The water\\nvapor and binary mixture have a very small degree of overlapping,\\nallowing for their classiﬁcation. The overall classiﬁcation accuracy to\\nidentify these three vapors could reach 91.7%. From these results, we\\ncan conclude that the extracted features (ﬁngerprint) from the effective\\nsensing response proﬁle are distinct. In combination with machine\\nlearning algorithms, these multiple transient features could contribute\\nto the VOC identiﬁcation at high accuracy.\\nWe further proceeded to the exposure of four additional com-\\nmon VOCs, namely, ethanol (ETH), methanol (MTH), 2-propanol\\n(PRL), and water vapor. In the PCA analysis, it could be found that\\nthe ﬁrst three components could explain around 87.7% variance. The\\nwater vapor cluster and methanol vapor cluster were well separated\\nfrom the other two, while the ethanol cluster and 2-propanol cluster\\nshowed some overlapping. With the LDA, the separability of these\\nfour clusters was better except for a small overlapping between ethanol\\nand 2-propanol. With the excellent separability in LDA space, the con-\\nfusion matrix of these gases’ classiﬁcation exhibited high accuracy (see\\nsupplementary material). The overall classiﬁcation accuracy was\\n95.8% implemented by the hold-out strategy, in which 75% of features\\nwere employed to train classiﬁer models and the rest features were\\nemployed to validate the trained classiﬁer models. These experiments\\nsuggest that the discrimination of VOCs using multiple transient fea-\\ntures is effective in ambient air and not just using nitrogen as carrier\\ngas.\\nAn aging test by exposing the sensor to various humidity levels\\nfor 36 h (720 cycles) did not evidence sensor deterioration due to water\\nvapor effects.\\nIII. CONCLUSION\\nIn this work, we introduce the development of machine learning-\\nenabled graphene-based single-channel electronic olfaction (e-\\nolfaction) sensors, along with an olfactory performance assessment\\nmethodology for characterizing their performance. The e-olfaction\\nsensor device comprises a single-channel nanosensor, which distin-\\nguishes it from conventional e-nose systems that utilize sensor arrays,\\nand operates at room temperature. This unique design shows great\\npotential for miniaturization and portability. We extract eleven kineti-\\ncally transient-state features are extracted from odor sensing response\\nproﬁles, which takes advantage of characteristic response proﬁles of\\nodor sensing. Our e-olfaction concept exhibits excellent performance\\nunder controlled lab conditions, not limited by the use of air as the\\ncarrier gas, in terms of odor threshold, odor discrimination and odor\\nidentiﬁcation. Speciﬁcally, we achieve a low odor threshold toward\\n2Phe (4.4 ppm) and an excellent odor discrimination performance\\n(83.3%).', mimetype='text/plain', start_char_idx=0, end_char_idx=4297, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.3678794399164391)], metadata={'9f75871c-80f4-4d7d-88dc-ebf936bcf603': {'page_label': '8', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, '5bbc83ff-94dc-4e9f-937a-5c2eda3a7b82': {'page_label': '9', 'file_name': 'Machine learning-enabled graphene-based electronic.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\Machine learning-enabled graphene-based electronic.pdf', 'file_type': 'application/pdf', 'file_size': 3460784, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}})"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_engine.query(\"\"\"cross-efects between diferent odors6\n",
    ", limiting the number of odor presentation per trial. Tis makes large-scale\n",
    "approaches challenging. As a result, almost all available databases originate from trained expert panels or the\n",
    "perfume industry, such as those from Dravnieks7\n",
    ", Lefngwell8\n",
    ", or Te Good Scent Company (dataset available\n",
    "from https://github.com/pyrfume, where a well-curated collection of currently available databases is stored).\n",
    "While such expert databases are a valuable resource for evaluations of olfactory quality and convince by high\n",
    "reliability, they are somewhat limited in their validity and objectivity. Most of those use predefned labels (such\n",
    "as “pleasant”, “sweet”, “bitter”, “musky”) \"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Response(response='The study overview indicates an experimental design aimed at exploring aspects of human olfactory perception. The research involved several stages:\\n\\n1. **Questionnaire Completion**: Participants filled in questionnaires using an online survey tool, LimeSurvey. This part likely covered various aspects related to the participants\\' sensory experience and preferences.\\n\\n2. **Selection of Odor Sets**: Each participant was assigned to one out of nine different odor sets. These sets contained ten specific mono-molecular odors chosen from a total pool of 74 unique compounds. \\n\\n3. **Odor Presentation**: The selected odors were presented in small plastic containers to ensure consistent and controlled exposure.\\n\\n4. **Initial Descriptions**: Participants provided initial free-form descriptions of the odors (in German).\\n\\n5. **Reevaluation**: After providing their free descriptions, participants then reevaluated the odors using a combination of eight visual analogue scales and an additional set of 16 qualitative labels that were previously identified by Keller & Vosshall.\\n\\nThis methodology allows for both subjective description and quantifiable assessment of odor perception. The inclusion of control odors (the \"anchor odors\") across different sets likely helped in standardizing the experience among participants, facilitating comparisons between groups. The study design\\'s complexity suggests an aim to identify patterns or variations in human olfactory responses under controlled conditions.\\n\\nThe mention of \"German language only\" and instructions being presented in English for better readability suggests that while all interactions might have been carried out predominantly in German (to ensure cultural relevance), researchers used English as a more universally accessible language for dissemination purposes, particularly through the publication.', source_nodes=[NodeWithScore(node=TextNode(id_='163a8e43-a080-4d1e-8d94-aad60bd9d777', embedding=None, metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='d21bdd56-6278-4f7b-b6e0-11b90dcaccc7', node_type='4', metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='ccdadbf26b1b25a09cf81d4523500bf3c6b8e13b8edef8d7cddb9b4d4a3b8e75'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='846124fc-fc0f-4fb7-a4a0-0af1e61021b3', node_type='1', metadata={'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='055b547def306db012f8faed136e7f110ee7841ffce230122a4c5cc6a1af3dda'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='0d4ddcee-eb16-498a-8364-7f240f015949', node_type='1', metadata={}, hash='f301ca0ed8b4ffc17c7873edb69931c511b732d5e7af85423cd2ccb556ad4ee5')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='GR94839_E, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(N(=O)=O)cc1\\n522, GR94839_F, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](C2)O)=O)ccc(OC)cc1\\n523, carphenazine, 1, CCC(=O)c1ccc2Sc3ccccc3N(CCCN4CCN(CCO)CC4)c2c1\\n524, cartazolate, 1, CCCCNc1c(cnc2n(CC)ncc12)C(=O)OCC\\n525, centazolone, 1, NN1C=Nc2cc3ccccc3cc2C1=O\\n526, chloralhydrate, 1, OC(O)C(Cl)(Cl)Cl\\n527, chlordiazepoxide, 1, CN=C1CN(O)C(=C2C=C(Cl)C=CC2=N1)c3ccccc3\\n528, chlorpromazine, 1, CN(C)CCCN1c2ccccc2Sc3ccc(Cl)cc13\\n529, chlorprothixene, 1, CN(C)CC\\\\C=C/1c2ccccc2Sc3ccc(Cl)cc13\\n530, chlorzoxazone, 1, Clc1ccc2OC(=O)Nc2c1\\n531, choline, 1, C[N+](C)(C)CCO\\n532, clobazam, 1, CN1C(=O)CC(=O)N(c2ccccc2)c3cc(Cl)ccc13\\n533, RO19-4603, 1, CN1Cc2n(cnc2C(=O)OC(C)(C)C)c3ccsc3C1=O\\n534, SKF101468, 1, C1C(Nc2c1c(ccc2)CCN(CCC)CCC)=O\\n535, SKF89124, 1, C1C(Nc2c1c(ccc2O)CCN(CCC)CCC)=O\\n536, Salicyluricacid, 1, OC(=O)CNC(=O)c1ccccc1O\\n537, Sulforidazine, 1, CN1CCCCC1CCN2c3ccccc3Sc4ccc(cc24)[S](C)(=O)=O\\n538, Teflurane, 1, FC(Br)C(F)(F)F\\n539, Theobromine, 1, Cn1cnc2N(C)C(=O)NC(=O)c12\\n540, Thiopental, 0, CCCC(C)C1(CC)C(=O)NC(=S)NC1=O\\n541, Toluene, 1, Cc1ccccc1\\n542, Triazolam, 1, Cc1nnc2CN=C(c3ccccc3Cl)c4cc(Cl)ccc4n12\\n543, GR94839_G, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)cc(ccc1)N(=O)=O\\n544, GR94839_H, 0, c1(ccccc1)CC(N1[C@H](CN(CC1)C(=O)C)C[N@]1CC[C@H](C1)O)=O\\n545, GR94839_I, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc([S@](=O)C)cc1\\n546, GR94839_L, 0, c1(CC(N2[C@H](CN(CC2)C(=O)C)C[N@]2CC[C@H](O)C2)=O)ccc(S(=O)(=O)C)cc1\\n547, ICI197067, 1, CC(C)[C@@H](CN1CCCC1)N(C)C(=O)Cc2ccc(Cl)c(Cl)c2\\n548, ICI199441, 1,', mimetype='text/plain', start_char_idx=34804, end_char_idx=36401, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.36787948502609175), NodeWithScore(node=TextNode(id_='348f4513-2db0-47e4-b8ae-21baab136545', embedding=None, metadata={'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='f8953176-3f59-4eeb-8972-1c08db87cfa3', node_type='4', metadata={'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, hash='6434e965d0a4ea2cd0fe7c7d8043a97c008433192254e5d76a1d2cf0a0108550')}, metadata_template='{key}: {value}', metadata_separator='\\n', text='bounded above by panel test-retest correlation (Fig. S13). When we disaggregate by chemical\\nclasses (e.g. esters, phenols, amines), both panelist and model pe\\x00ormance is relatively\\nuniform (Fig. 3C), with sulfur-containing molecules showing strongest pe\\x00ormance from\\npanelists and the model (R = 0.52).\\nChemical materials are impure - a fact too o\\x00en unaccounted for in olfactory research(24). To\\nmeasure the contribution of impurities to the odor percept of our stimuli, we applied a gas\\nchromatography-mass spectrometry (GC-MS and gas chromatography-olfactometry\\n(GC-O) quality control (QC) procedure to 50 stimuli (Data S1). This QC procedure matches an\\nodor percept to its causal molecule, allowing us to identify stimuli for which the primary odor\\ncharacter was not due to the nominal compound. Our QC led to diverse conclusions: the\\nnominal compound caused the odor (12/50), the nominal compound and contaminants\\ncontribute to the odor (16/50), contaminants caused the odor (18/50), or the cause of the\\nodor could not be determined (4/50) (Fig. 3D). In some cases, while we purchased a novel\\nodorant, the dominant odorant was not novel; for example, the stimulus\\n4,5-dimethyl-1,3-thiazol-2-amine was described by the panel as bu\\x00ery, sweet, and dairy, but\\nthis odor percept was a\\x00ributed through QC to the contaminant diacetyl, a well-known\\nbu\\x00ery odorant. In another case, the purchased odorant, isobornyl methylacrylate, was\\ndescribed by the panel and the model as both piney and \\x00oral; however, through QC we\\ndetermined that the nominal compound was \\x00oral only and that the piney aroma was due to\\nthe closely related compound, borneol, which was detected as a contaminant in the sample.\\nBased on QC results, we removed 26 molecules known or suspected to have high degrees of\\nodorous contamination (Data S1).\\nThe prevalence of odorous contamination that we found demonstrates that it is not safe to\\nassume that the odor percept of a purchased chemical is due to the nominal compound. The\\nFlavor & Fragrance (F&F) industry is motivated to minimize odorous contaminants for\\ncommercially valued odorants, but there is no such incentive for non-F&F commodity\\nchemicals. We stress the need for caution and diligence in expanding odor stimulus space.\\nImplications of each QC result on model pe\\x00ormance are unique (Data S1). In some cases,\\nthe model pe\\x00ormed well despite the presence of odorous contaminants. We estimate that,\\nif these contaminants were removed from the rated samples, model pe\\x00ormance improves\\nin 6 of 50 scenarios, degrades in another 6 of 50 scenarios, remains neutral in 21 of 50\\nscenarios, and cannot be determined in 17 of 50 scenarios.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted December 13, 2022. ; https://doi.org/10.1101/2022.09.01.504602doi: bioRxiv preprint', mimetype='text/plain', start_char_idx=0, end_char_idx=3001, metadata_seperator='\\n', text_template='{metadata_str}\\n\\n{content}'), score=0.36787948502609175)], metadata={'163a8e43-a080-4d1e-8d94-aad60bd9d777': {'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\BBBP.csv', 'file_name': 'BBBP.csv', 'file_type': 'text/csv', 'file_size': 148743, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}, '348f4513-2db0-47e4-b8ae-21baab136545': {'page_label': '7', 'file_name': 'A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_path': 'C:\\\\Users\\\\mikoo\\\\AI\\\\osmo\\\\A Principal Odor Map Unies Diverse Tasks in Human Olfactory Perception.pdf', 'file_type': 'application/pdf', 'file_size': 3198464, 'creation_date': '2025-01-02', 'last_modified_date': '2025-01-02'}})"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_engine.query(\"\"\"Overview of the study design and procedures. Te study has been conducted in Germany and in the\n",
    "German language only, instructions and scales are presented in English here for better readability. First, all\n",
    "participants flled in questionnaires using an online survey implemented via LimeSurvey. Ten, participants\n",
    "were assigned to one out of nine odor sets, containing a subset of ten out of the total 74 mono-molecular odors.\n",
    "Te odor samples were provided in small plastic containers. Two “anchor odors” were used in all odor sets\n",
    "for comparison between groups, the other eight odors difered between groups. Te participants frst gave a\n",
    "free description for each odor (in German) and then evaluated them again using eight visual analogue scales\n",
    "and 16 further qualitative labels obtained from Keller & Vosshall\"\"\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "primo",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
